<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Workflow for processing 16S rRNA samples for ASV analysis using DADA2. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables. A Microtable Object is produced to collate the data for downstream analysis.">
<title>2. DADA2 ASV Workflow – TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../workflows/ssu/otu/index.html" rel="next">
<link href="../../../workflows/ssu/sampledata/index.html" rel="prev">
<link href="../../../assets/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-listing/list.min.js"></script>
<script src="../../../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-d7d32a8442357cea8cf74a4a848b80d6.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-dada2-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-filename','listing-description',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-filename-sort'] }],
      
      searchColumns: ["listing-filename","listing-description"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-dada2-listing'] = new List('listing-dada2-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script><script src="../../../site_libs/quarto-diagram/mermaid.min.js"></script><script src="../../../site_libs/quarto-diagram/mermaid-init.js"></script><link href="../../../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo light-content"><img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo dark-content"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/portal/index.html"> 
<span class="menu-text">Data Portal</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/ssu/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/mg/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/shrimp-draft/"> <i class="bi bi-github" role="img" aria-label="Trans Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/ssu/index.html">16S rRNA</a></li><li class="breadcrumb-item"><a href="../../../workflows/ssu/dada2/index.html">2. DADA2 ASV Workflow</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../../../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/portal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Portal</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/ssu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/dada2/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. DADA2 ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/med/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/mg/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/setup/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Build Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/assembly/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/aa-results/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Assembly &amp; Annotation Results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/archive_data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting &amp; Archiving Data/Code</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link active" data-scroll-target="#workflow-input">Workflow Input</a></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a></li>
  <li>
<a href="#read-processing" id="toc-read-processing" class="nav-link" data-scroll-target="#read-processing">Read Processing</a>
  <ul class="collapse">
<li><a href="#individual-run-workflows" id="toc-individual-run-workflows" class="nav-link" data-scroll-target="#individual-run-workflows">Individual Run Workflows</a></li>
  <li>
<a href="#merged-runs-workflow" id="toc-merged-runs-workflow" class="nav-link" data-scroll-target="#merged-runs-workflow">Merged Runs Workflow</a>
  <ul class="collapse">
<li><a href="#merge-sequencing-tables" id="toc-merge-sequencing-tables" class="nav-link" data-scroll-target="#merge-sequencing-tables">1. Merge Sequencing Tables</a></li>
  <li><a href="#remove-chimeras" id="toc-remove-chimeras" class="nav-link" data-scroll-target="#remove-chimeras">2. Remove Chimeras</a></li>
  <li><a href="#track-reads-through-workflow" id="toc-track-reads-through-workflow" class="nav-link" data-scroll-target="#track-reads-through-workflow">3. Track Reads through Workflow</a></li>
  <li><a href="#assign-taxonomy" id="toc-assign-taxonomy" class="nav-link" data-scroll-target="#assign-taxonomy">4. Assign Taxonomy</a></li>
  </ul>
</li>
  <li><a href="#r-session-information-code" id="toc-r-session-information-code" class="nav-link" data-scroll-target="#r-session-information-code">R Session Information &amp; Code</a></li>
  </ul>
</li>
  <li>
<a href="#data-set-prep" id="toc-data-set-prep" class="nav-link" data-scroll-target="#data-set-prep">Data Set Prep</a>
  <ul class="collapse">
<li><a href="#read-counts-assessment" id="toc-read-counts-assessment" class="nav-link" data-scroll-target="#read-counts-assessment">Read Counts Assessment</a></li>
  <li>
<a href="#prep-data-for-microeco" id="toc-prep-data-for-microeco" class="nav-link" data-scroll-target="#prep-data-for-microeco">Prep Data for <code>microeco</code></a>
  <ul class="collapse">
<li><a href="#a.-taxonomy-table" id="toc-a.-taxonomy-table" class="nav-link" data-scroll-target="#a.-taxonomy-table">A. Taxonomy Table</a></li>
  <li><a href="#b.-sequence-table" id="toc-b.-sequence-table" class="nav-link" data-scroll-target="#b.-sequence-table">B. Sequence Table</a></li>
  <li><a href="#c.-sample-table" id="toc-c.-sample-table" class="nav-link" data-scroll-target="#c.-sample-table">C. Sample Table</a></li>
  </ul>
</li>
  <li>
<a href="#experiment-level-objects" id="toc-experiment-level-objects" class="nav-link" data-scroll-target="#experiment-level-objects">Experiment-level Objects</a>
  <ul class="collapse">
<li><a href="#add-representative-sequence" id="toc-add-representative-sequence" class="nav-link" data-scroll-target="#add-representative-sequence">Add Representative Sequence</a></li>
  </ul>
</li>
  <li>
<a href="#curate-the-data-set" id="toc-curate-the-data-set" class="nav-link" data-scroll-target="#curate-the-data-set">Curate the Data Set</a>
  <ul class="collapse">
<li><a href="#remove-any-kingdom-nas" id="toc-remove-any-kingdom-nas" class="nav-link" data-scroll-target="#remove-any-kingdom-nas">Remove any Kingdom NAs</a></li>
  <li><a href="#remove-contaminants" id="toc-remove-contaminants" class="nav-link" data-scroll-target="#remove-contaminants">Remove Contaminants</a></li>
  <li><a href="#remove-negative-controls-nc" id="toc-remove-negative-controls-nc" class="nav-link" data-scroll-target="#remove-negative-controls-nc">Remove Negative Controls (NC)</a></li>
  <li><a href="#remove-low-count-samples" id="toc-remove-low-count-samples" class="nav-link" data-scroll-target="#remove-low-count-samples">Remove Low-Count Samples</a></li>
  </ul>
</li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">Summary</a></li>
  </ul>
</li>
  <li><a href="#download-results" id="toc-download-results" class="nav-link" data-scroll-target="#download-results">Download Results</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/shrimp-draft/blob/main/workflows/ssu/dada2/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/edit/main/workflows/ssu/dada2/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/ssu/index.html">16S rRNA</a></li><li class="breadcrumb-item"><a href="../../../workflows/ssu/dada2/index.html">2. DADA2 ASV Workflow</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">2. DADA2 ASV Workflow</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
</div>

<div>
  <div class="description">
    <p>Workflow for processing 16S rRNA samples for ASV analysis using DADA2. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables. A Microtable Object is produced to collate the data for downstream analysis.</p>
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="workflow-input" class="level2"><h2 class="anchored" data-anchor-id="workflow-input">Workflow Input</h2>
<div class="callout callout-style-default callout-note no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Data &amp; Scripts
</div>
</div>
<div class="callout-body-container callout-body">
<p>Fastq sequence files, scripts, and other assets for running this workflow can be found on the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page.</p>
<p>The Data Portal page also contains a link to the curated output of this pipeline–-feature-sample count table, taxonomy table, sample metadata, &amp; representative fasta sequences. Data is available as stand-alone text files, OR bundled in microtable and phyloseq objects. Archive also includes a table tracking sample read changes.</p>
</div>
</div>
<section id="required-packages-software" class="level4"><h4 class="anchored" data-anchor-id="required-packages-software">Required Packages &amp; Software</h4>
<p>There are several R packages you need to run this workflow.</p>
<div class="cell">
<details class="code-fold"><summary>Click here for workflow library information.</summary><div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#!/usr/bin/env Rscript</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">919191</span><span class="op">)</span></span>
<span><span class="fu">pacman</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/pacman/man/p_load.html">p_load</a></span><span class="op">(</span><span class="va">tidyverse</span>, <span class="va">gridExtra</span>, <span class="va">grid</span>, <span class="va">phyloseq</span>,</span>
<span>               <span class="va">formatR</span>, <span class="va">gdata</span>, <span class="va">ff</span>, <span class="va">decontam</span>, <span class="va">dada2</span>, </span>
<span>               <span class="va">ShortRead</span>, <span class="va">Biostrings</span>, <span class="va">DECIPHER</span>, </span>
<span>               install <span class="op">=</span> <span class="cn">FALSE</span>, update <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</section></section><section id="overview" class="level2"><h2 class="anchored" data-anchor-id="overview">Overview</h2>
<p>This workflow contains the code we used to process the 16S rRNA data sets using <a href="https://benjjneb.github.io/dada2/">DADA2</a> <span class="citation" data-cites="callahan2016dada2">(<a href="#ref-callahan2016dada2" role="doc-biblioref">Callahan et al. 2016</a>)</span>. Workflow construction is based on the <a href="https://benjjneb.github.io/dada2/tutorial_1_8.html">DADA2 Pipeline Tutorial (1.8)</a> and the primer identification section of the <a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Pipeline Workflow (1.8)</a>. In the first part of the pipeline, we process the individual sequencing runs <strong>separately</strong>. Next we combine the sequence tables from each run into one merged sequences table and continue with processing.</p>
</section><section id="read-processing" class="level1"><h1>Read Processing</h1>
<p>We processed each of the six sequencing runs separately for the first part of the DADA2 workflow. While some of the outputs are slightly different (e.g.&nbsp;quality scores, filtering, ASV inference, etc.) the code is the same. For posterity, code for each run is included here.</p>
<section id="individual-run-workflows" class="level2"><h2 class="anchored" data-anchor-id="individual-run-workflows">Individual Run Workflows</h2>
<p>The first part of the workflow consists of the following steps for each of the runs:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 8%">
<col style="width: 34%">
<col style="width: 57%">
</colgroup>
<thead><tr class="header">
<th>Step</th>
<th>Command</th>
<th>What we’re doing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>multiple</td>
<td>prepare input file names &amp; paths</td>
</tr>
<tr class="even">
<td>2</td>
<td>multiple</td>
<td>Define primers (all orientations)</td>
</tr>
<tr class="odd">
<td>3</td>
<td><code>cutadapt</code></td>
<td>Remove primers</td>
</tr>
<tr class="even">
<td>4</td>
<td><code>plotQualityProfile()</code></td>
<td>Plot quality scores.</td>
</tr>
<tr class="odd">
<td>5</td>
<td><code>filterAndTrim()</code></td>
<td>Assess quality &amp; filter reads</td>
</tr>
<tr class="even">
<td>6</td>
<td><code>learnErrors()</code></td>
<td>Generate an error model for the data</td>
</tr>
<tr class="odd">
<td>7</td>
<td><code>derepFastq()</code></td>
<td>Dereplicate sequences</td>
</tr>
<tr class="even">
<td>8</td>
<td><code>dada()</code></td>
<td>Infer ASVs (forward &amp; reverse reads).</td>
</tr>
<tr class="odd">
<td>9</td>
<td>
<code>mergePairs()</code>.</td>
<td>Merge denoised forward &amp; reverse reads</td>
</tr>
<tr class="even">
<td>10</td>
<td><code>makeSequenceTable()</code></td>
<td>Generate count table for each run</td>
</tr>
<tr class="odd">
<td>11</td>
<td></td>
<td>Track reads through workflow</td>
</tr>
</tbody>
</table>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
    A(Start with raw&lt;/br&gt;sequence data)
    A --&gt; B(plotQualityProfile)
    B --&gt; C(filterAndTrim)
    C --&gt; D(plotErrors)
    C --&gt; E(learnErrors)
    E --&gt; END:::hidden
    
</pre>
</div>
<p></p>
</div>
</div>
</div>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
    BEGIN:::hidden --&gt; F(derepFastq)
    F --&gt; G(dada)
    G --&gt; I(mergePairs) 
    I --&gt; J(makeSequenceTable)
    J --&gt; K(to Merged Runs&lt;/br&gt;Workflow)
</pre>
</div>
<p></p>
</div>
</div>
</div>
<p><br></p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" href="">BCS_26</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false" href="">BCS_28</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false" href="">BCS_29</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false" href="">BCS_30</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-5-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-5" role="tab" aria-controls="tabset-1-5" aria-selected="false" href="">BCS_34</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-6-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-6" role="tab" aria-controls="tabset-1-6" aria-selected="false" href="">BCS_35</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<section id="set-working-environment" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S5, S5, S7, S8. Access the source code for <a href="include/_BCS_26.qmd">BCS_26 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_26/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_10_R1_001.trimmed.fastq" "Control_10_R2_001.trimmed.fastq"
[3] "Control_11_R1_001.trimmed.fastq" "Control_11_R2_001.trimmed.fastq"
[5] "Control_12_R1_001.trimmed.fastq" "Control_12_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_26_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering" class="level3"><h3 class="anchored" data-anchor-id="filtering">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads" class="level4"><h4 class="anchored" data-anchor-id="forward-reads">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>112559480 total bases in 511634 reads from 28 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_26_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_26_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_26_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>100501560 total bases in 558342 reads from 31 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_26_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_26_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_26_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
15 sequence variants were inferred from 813 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
13 sequence variants were inferred from 683 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_26</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_26</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>29430</strong> sequence variants from <strong>384</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_26</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>  220   221   222   223   224   225   226   227   228   229   230   231   234 
   32    30     6    13     3     2     1     9     5     2     1     3     2 
  235   236   237   238   239   240   241   242   243   244   245   246   247 
    2   479    83     3     4    45    31   218    72    20     3     3     7 
  248   249   250   251   252   253   254   255   256   257   258   259   260 
    8    10    14    47   985 25810  1025    80    27    15     6     2     3 
  261   262   263   268   270   272   273   275   276   278   279   280   286 
   10     3     2     1     1     2     1     1     4     1     2     1     1 
  289   293   294   295   296   298   300   311   316   318   319   321   323 
    1     3     3     1     2     1     1     1     2     1     1     1     1 
  324   333   334   335   337   338   339   340   341   342   343   344   345 
    1     1     7     5     9     1    10     5     3     7     1     3     2 
  346   347   348   349   350   351   352   355   356   357   359   360   361 
    3     2     5     8     5     1     7     7     2     2     4     2    22 
  362   363   364   365   366   367   368   369   370   372   373   374   377 
   28     4    30    50     5     3     3     2     1     3    10     1     3 
  378 
    2 </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>29430</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_26.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_26</span>, <span class="st">"BCS_26.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<section id="set-working-environment-1" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment-1">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S3, S4. Access the source code for <a href="include/_BCS_28.qmd">BCS_28 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_28/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_31_R1_001.trimmed.fastq" "Control_31_R2_001.trimmed.fastq"
[3] "Control_32_R1_001.trimmed.fastq" "Control_32_R2_001.trimmed.fastq"
[5] "Control_33_R1_001.trimmed.fastq" "Control_33_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores-1" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores-1">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_28_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering-1" class="level3"><h3 class="anchored" data-anchor-id="filtering-1">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates-1" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates-1">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads-1" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-1">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>100307240 total bases in 455942 reads from 31 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_28_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_28_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_28_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads-1" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-1">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>103203360 total bases in 573352 reads from 36 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_28_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_28_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_28_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads-1" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads-1">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference-1" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference-1">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
76 sequence variants were inferred from 8771 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb41"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
74 sequence variants were inferred from 6979 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads-1" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads-1">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table-1" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table-1">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb44"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_28</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_28</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>9461</strong> sequence variants from <strong>192</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_28</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code> 220  221  222  223  224  227  230  231  236  237  238  239  241  242  243  244 
   3    8    2    1    1    4    1    1    1    2    1    1    2    5    1    1 
 245  246  247  248  249  250  251  252  253  254  255  256  257  258  259  268 
   1    1    2    1    1    4    8  328 8699  252   21    9    6    1    1    1 
 270  292  318  335  336  337  338  339  341  342  344  345  348  357  359  361 
   1    1    1    1    2    2    1    1    2    3    1    1    3    1    1    5 
 362  363  364  365  366  367  368  372  373  374  376  378  385  388 
   8    3   22   13    1    1    8    1    1    1    1    1    1    1 </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>9461</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_28.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads-1" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads-1">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb47"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_28</span>, <span class="st">"BCS_28.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<section id="set-working-environment-2" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment-2">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S13, S14, S15, S16. Access the source code for <a href="include/_BCS_29.qmd">BCS_29 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_29/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_37_R1_001.trimmed.fastq" "Control_37_R2_001.trimmed.fastq"
[3] "Control_38_R1_001.trimmed.fastq" "Control_38_R2_001.trimmed.fastq"
[5] "Control_39_R1_001.trimmed.fastq" "Control_39_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores-2" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores-2">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb51"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_29_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering-2" class="level3"><h3 class="anchored" data-anchor-id="filtering-2">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb53"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates-2" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates-2">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads-2" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-2">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>104777640 total bases in 476262 reads from 36 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_29_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_29_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_29_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads-2" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-2">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>103229460 total bases in 573497 reads from 43 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_29_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_29_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_29_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads-2" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads-2">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb60"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference-2" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference-2">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb62"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
46 sequence variants were inferred from 3108 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb64"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
44 sequence variants were inferred from 2744 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads-2" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads-2">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb66"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table-2" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table-2">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_29</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_29</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>21105</strong> sequence variants from <strong>384</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb68"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_29</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>   220   221   222   223   224   225   226   227   228   229   231   234   236 
   16    22     1    10     5     5     2    10     1     2    11     3   518 
  237   238   240   241   242   243   244   245   246   247   248   249   250 
    3     7     8   155    11    81     1     2     4     4     1     2     8 
  251   252   253   254   255   256   257   258   260   262   269   270   272 
   28   690 18300   857    42    17    18     1     1     2     1     1     1 
  273   274   275   277   278   281   286   288   292   293   294   297   300 
    1     1     1     1     3     1     1     1     1     2     2     1     1 
  303   304   305   308   313   315   318   319   329   330   334   335   336 
    2     1     3     1     2     1     1     1     3     2     3     1     1 
  337   339   340   341   342   343   344   347   348   349   350   352   356 
    2     4     5    21     7     1     1     1     2     3     1     1     2 
  357   358   359   360   361   362   363   364   365   366   367   368   370 
   14     1     2     6    14    65     4    32    13     4     2     2     2 
  379   384 
    1     1  </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>21105</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_29.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads-2" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads-2">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb70"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_29</span>, <span class="st">"BCS_29.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<section id="set-working-environment-3" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment-3">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S17, S18, S19, S20. Access the source code for <a href="include/_BCS_30.qmd">BCS_30 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb71"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_30/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_49_R1_001.trimmed.fastq" "Control_49_R2_001.trimmed.fastq"
[3] "Control_50_R1_001.trimmed.fastq" "Control_50_R2_001.trimmed.fastq"
[5] "Control_51_R1_001.trimmed.fastq" "Control_51_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb73"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores-3" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores-3">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb74"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_30_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering-3" class="level3"><h3 class="anchored" data-anchor-id="filtering-3">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb75"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb76"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates-3" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates-3">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads-3" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-3">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb77"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>101887500 total bases in 463125 reads from 26 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_30_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_30_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_30_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads-3" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-3">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb80"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>105320340 total bases in 585113 reads from 30 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb82"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_30_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_30_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_30_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads-3" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads-3">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb83"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference-3" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference-3">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb84"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb85"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
9 sequence variants were inferred from 25 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb87"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
9 sequence variants were inferred from 21 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads-3" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads-3">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb89"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table-3" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table-3">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_30</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_30</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>36401</strong> sequence variants from <strong>380</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb91"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_30</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>  220   221   222   223   224   225   226   227   228   229   231   232   234 
   45    24     5    12    10     3     2    13     2     1    15     2     3 
  235   236   237   238   240   241   242   243   244   245   246   247   248 
    5   142     2    29    33   128    26   266     5     3     4    10     8 
  249   250   251   252   253   254   255   256   257   258   259   260   261 
    7    16    62  1119 31980  1708   149    56    47     7     6     2     4 
  262   263   264   267   268   269   270   271   272   273   274   276   278 
    3     1     2     1     1     1     4     7     1     2     2     1     5 
  279   282   284   285   286   288   289   291   292   293   294   295   296 
    1     1     1     3     1     1     2     1     1     5     1     1     1 
  297   298   303   305   307   308   311   313   316   317   319   320   321 
    1     1     1     2     1     1     2     2     2     1     2     1     1 
  322   323   325   326   328   332   333   334   335   336   337   338   339 
    2     1     2     1     2     1     1     2     1     6     3     3     5 
  340   341   342   343   344   345   347   348   349   350   351   352   353 
    2    21    23     4     2     4     8     6     6     3     1     2     1 
  354   355   356   357   358   359   360   361   362   363   364   365   366 
    1     1     3     6     2     8     9    76    58    18    33    22     5 
  368   371   372   373   379   380   387 
    8     1     1     1     2     1     2 </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>36401</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_30.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads-3" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads-3">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb93"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_30</span>, <span class="st">"BCS_30.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
<div id="tabset-1-5" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-5-tab">
<section id="set-working-environment-4" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment-4">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S01, S02. Access the source code for <a href="include/_BCS_34.qmd">BCS_34 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb94"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_34/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_25_R1_001.trimmed.fastq" "Control_25_R2_001.trimmed.fastq"
[3] "Control_26_R1_001.trimmed.fastq" "Control_26_R2_001.trimmed.fastq"
[5] "Control_27_R1_001.trimmed.fastq" "Control_27_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb96"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores-4" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores-4">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_34_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering-4" class="level3"><h3 class="anchored" data-anchor-id="filtering-4">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb99"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates-4" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates-4">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads-4" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-4">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>100938640 total bases in 458812 reads from 22 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb102"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_34_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_34_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_34_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads-4" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-4">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb103"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>104387760 total bases in 579932 reads from 25 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb105"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_34_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_34_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_34_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads-4" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads-4">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb106"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference-4" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference-4">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb107"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb108"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
9 sequence variants were inferred from 327 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb110"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
8 sequence variants were inferred from 255 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads-4" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads-4">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb112"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table-4" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table-4">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb113"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_34</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_34</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>18373</strong> sequence variants from <strong>190</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb114"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_34</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>  220   221   222   223   224   225   226   227   229   230   231   234   236 
   14    21     2     6     1     2     2     9     1     1     2     2   101 
  237   238   239   241   242   243   244   245   246   247   248   249   250 
   62     1     1    11    16     6     4     4     4     2     4     5     6 
  251   252   253   254   255   256   257   258   259   261   265   270   271 
   20   697 16533   584    52    15    11     5     1     2     1     2     1 
  274   278   279   285   286   290   291   295   308   309   322   325   328 
    1     1     1     1     1     1     1     1     1     1     1     1     1 
  335   336   337   338   339   340   341   342   343   344   345   347   348 
    2     1     4     1     5     1     4     8     2     3     2     1     5 
  349   356   357   358   359   360   361   362   363   364   365   366   367 
    1     1     1     3     2     1    14    15     8    25    16     2     1 
  368   371   372   373   376   377   378   385   386 
    3     2     2     4     2     1     1     1     1  </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>18373</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_34.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads-4" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads-4">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb116"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_34</span>, <span class="st">"BCS_34.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
<div id="tabset-1-6" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-6-tab">
<section id="set-working-environment-5" class="level3"><h3 class="anchored" data-anchor-id="set-working-environment-5">1. Set Working Environment</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>The following plate were sequenced in this run: ISTHMO S9, S10, S11, S12. Access the source code for <a href="include/_BCS_35.qmd">BCS_35 processing</a>.</p>
</div>
</div>
<p>First, we setup the working environment by defining a path for the working directory.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb117"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">path</span> <span class="op">&lt;-</span> <span class="st">"BCS_35/"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Control_13_R1_001.trimmed.fastq" "Control_13_R2_001.trimmed.fastq"
[3] "Control_14_R1_001.trimmed.fastq" "Control_14_R2_001.trimmed.fastq"
[5] "Control_15_R1_001.trimmed.fastq" "Control_15_R2_001.trimmed.fastq"</code></pre>
</div>
</div>
<p>Then, we generate matched lists of the forward and reverse read files. We also parse out the sample name.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb119"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R1_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="va">path</span>, pattern <span class="op">=</span> <span class="st">"_R2_001.trimmed.fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">sample.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="va">fnFs</span>, <span class="st">"_R1_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">fnFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnFs</span><span class="op">)</span></span>
<span><span class="va">fnRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="va">fnRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="plot-quality-scores-5" class="level3"><h3 class="anchored" data-anchor-id="plot-quality-scores-5">2. Plot quality scores</h3>
<p>And next we plot a visual summary of the distribution of quality scores as a function of sequence position for the input fastq files. Here we set the number of records to sample from each fastq file (<code>n</code>) to 20000 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb120"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qprofile_fwd</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnFs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile_rev</span> <span class="op">&lt;-</span> <span class="fu">plotQualityProfile</span><span class="op">(</span><span class="va">fnRs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="va">x</span><span class="op">]</span>, </span>
<span>                                   aggregate <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                   n <span class="op">=</span> <span class="fl">20000</span><span class="op">)</span></span>
<span><span class="va">qprofile</span> <span class="op">&lt;-</span> <span class="fu">grid.arrange</span><span class="op">(</span><span class="va">qprofile_fwd</span>, <span class="va">qprofile_rev</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_35_filt_plot_qscores.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Aggregated quality score plots for forward (left) &amp; reverse (right) reads.</figcaption></figure>
</div>
</div>
</div>
</section><section id="filtering-5" class="level3"><h3 class="anchored" data-anchor-id="filtering-5">3. Filtering</h3>
<p>We again make some path variables and setup a new directory of filtered reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb121"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">filtFs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_F_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="va">filtRs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">path</span>, <span class="st">"filtered"</span>, </span>
<span>                    <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">sample.names</span>, <span class="st">"_R_filt.fastq"</span><span class="op">)</span></span>
<span>                    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sample.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb122"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu">filterAndTrim</span><span class="op">(</span><span class="va">fnFs</span>, <span class="va">filtFs</span>, <span class="va">fnRs</span>, <span class="va">filtRs</span>, </span>
<span>                     truncLen<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">220</span>,<span class="fl">180</span><span class="op">)</span>, </span>
<span>                     maxN <span class="op">=</span> <span class="fl">0</span>, maxEE <span class="op">=</span> <span class="fl">2</span>, truncQ <span class="op">=</span> <span class="fl">2</span>, </span>
<span>                     rm.phix <span class="op">=</span> <span class="cn">TRUE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These parameters should be set based on the anticipated length of the amplicon and the read quality.</p>
</div>
</div>
<p>And here is a table of how the filtering step affected the number of reads in each sample.</p>
</section><section id="learn-error-rates-5" class="level3"><h3 class="anchored" data-anchor-id="learn-error-rates-5">4. Learn Error Rates</h3>
<p>Now it is time to assess the error rate of the data. The DADA2 algorithm uses a parametric error model. Every amplicon data set has a different set of error rates and the <code>learnErrors</code> method learns this error model <em>from the data</em>. It does this by alternating estimation of the error rates and inference of sample composition until they converge on a jointly consistent solution. The algorithm begins with an initial guess, for which the maximum possible error rates in the data are used.</p>
<section id="forward-reads-5" class="level4"><h4 class="anchored" data-anchor-id="forward-reads-5">Forward Reads</h4>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb123"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errF</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtFs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>101017840 total bases in 459172 reads from 48 samples 
will be used for learning the error rates.</code></pre>
<p>We can then plot the error rates for the forward reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb125"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p3</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errF</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_35_plot_errorF_1.png"</span>, <span class="va">p3</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_35_plot_errorF_2.png"</span>, <span class="va">p3</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_35_plot_errorF_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Forward reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
</section><section id="reverse-reads-5" class="level4"><h4 class="anchored" data-anchor-id="reverse-reads-5">Reverse Reads</h4>
<p>And now we can process the reverse read error rates.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb126"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">errR</span> <span class="op">&lt;-</span> <span class="fu">learnErrors</span><span class="op">(</span><span class="va">filtRs</span>, multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>100695240 total bases in 559418 reads from 59 samples 
will be used for learning the error rates.</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb128"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">p4</span> <span class="op">&lt;-</span> <span class="fu">plotErrors</span><span class="op">(</span><span class="va">errR</span>, nominalQ <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_35_plot_errorR_1.png"</span>, <span class="va">p4</span>, width <span class="op">=</span> <span class="fl">7</span>, height <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span>
<span><span class="fu">ggsave</span><span class="op">(</span><span class="st">"figures/BCS_35_plot_errorR_2.png"</span>, <span class="va">p4</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/BCS_35_plot_errorR_2.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Reverse reads: Observed frequency of each transition (e.g., T -&gt; G) as a function of the associated quality score.</figcaption></figure>
</div>
</div>
</div>
<p>The error rates for each possible transition (A to C, A to G, etc.) are shown. Points are the observed error rates for each consensus quality score. The black line shows the estimated error rates after convergence of the machine-learning algorithm. The red line shows the error rates expected under the nominal definition of the Q-score. Here the estimated error rates (black line) are a good fit to the observed rates (points), and the error rates drop with increased quality as expected.</p>
</section></section><section id="dereplicate-reads-5" class="level3"><h3 class="anchored" data-anchor-id="dereplicate-reads-5">5. Dereplicate Reads</h3>
<p>Now we can use <code>derepFastq</code> to identify the unique sequences in the forward and reverse fastq files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb129"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/basename.html">basename</a></span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span>, <span class="st">"_F_"</span><span class="op">)</span>, <span class="va">`[`</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">derepFs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtFs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepFs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span>
<span><span class="va">derepRs</span> <span class="op">&lt;-</span> <span class="fu">derepFastq</span><span class="op">(</span><span class="va">filtRs</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">derepRs</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">sam.names</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="dada2-asv-inference-5" class="level3"><h3 class="anchored" data-anchor-id="dada2-asv-inference-5">6. DADA2 &amp; ASV Inference</h3>
<p>At this point we are ready to apply the core sample inference algorithm (<code>dada</code>) to the filtered and trimmed sequence data. DADA2 offers three options for whether and how to pool samples for ASV inference.</p>
<p>If <code>pool = TRUE</code>, the algorithm will pool together all samples prior to sample inference.<br>
If <code>pool = FALSE</code>, sample inference is performed on each sample individually.<br>
If <code>pool = "pseudo"</code>, the algorithm will perform pseudo-pooling between individually processed samples.</p>
<p>For our final analysis, we chose <code>pool = pseudo</code> for this data set.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb130"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepFs</span>, err <span class="op">=</span> <span class="va">errF</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>, </span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">dadaRs</span> <span class="op">&lt;-</span> <span class="fu">dada</span><span class="op">(</span><span class="va">derepRs</span>, err <span class="op">=</span> <span class="va">errR</span>, pool <span class="op">=</span> <span class="st">"pseudo"</span>,</span>
<span>               multithread <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>As an example, we can inspect the returned <code>dada-class</code> object for the forward reads from the sample #1:</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb131"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaFs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
15 sequence variants were inferred from 432 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>And the corresponding reverse reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb133"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">dadaRs</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>dada-class: object describing DADA2 denoising results
19 sequence variants were inferred from 465 input unique sequences.
Key parameters: OMEGA_A = 1e-40, OMEGA_C = 1e-40, BAND_SIZE = 16</code></pre>
<p>These outputs tell us how many true sequence variants the DADA2 algorithm inferred from the unique sequences, in this case the sample 1.</p>
</section><section id="merge-paired-reads-5" class="level3"><h3 class="anchored" data-anchor-id="merge-paired-reads-5">7. Merge Paired Reads</h3>
<p>We now merge the forward and reverse reads together to obtain the full denoised sequence dataset. Merging is performed by aligning the denoised forward reads with the reverse-complement of the corresponding denoised reverse reads, and then constructing the merged “contig” sequences. By default, merged sequences are only output if the forward and reverse reads overlap by at least 12 bases, and are identical to each other in the overlap region (but these conditions can be changed via function arguments).</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb135"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">mergers</span> <span class="op">&lt;-</span> <span class="fu">mergePairs</span><span class="op">(</span><span class="va">dadaFs</span>, <span class="va">derepFs</span>, <span class="va">dadaRs</span>, <span class="va">derepRs</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>mergers</code> objects are lists of data.frames from each sample. Each data.frame contains the merged sequence, its abundance, and the indices of the forward and reverse sequence variants that were merged. Paired reads that did not exactly overlap were removed by <code>mergePairs</code>, further reducing spurious output.</p>
</section><section id="construct-sequence-table-5" class="level3"><h3 class="anchored" data-anchor-id="construct-sequence-table-5">8. Construct Sequence Table</h3>
<p>Now we construct an amplicon sequence variant (ASV) table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb136"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_35</span> <span class="op">&lt;-</span> <span class="fu">makeSequenceTable</span><span class="op">(</span><span class="va">mergers</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">BCS_35</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looks like we have <strong>23367</strong> sequence variants from <strong>379</strong> samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb137"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">BCS_35</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>  220   221   222   223   224   225   226   227   228   229   230   234   235 
   16    28     5     9     2     2     4     4     1     1     1     1     2 
  236   237   240   241   242   243   245   246   247   248   249   250   251 
  147     4    10    76    16    27     5     1     2     3     2     8    31 
  252   253   254   255   256   257   258   260   262   263   266   268   273 
 1004 20727   767    44     9    22     1     3     3     1     1     1     3 
  282   292   293   294   295   303   307   309   310   312   333   334   335 
    1     1     3     1     1     1     1     1     1     1     1     4     1 
  336   337   339   340   341   342   344   346   347   348   349   350   352 
    1     6     6     3    39    23     4     1    19     5     5     3     3 
  357   359   360   361   362   363   364   365   366   367   368   369   372 
   13     1     8    64    46    18    33    26     4     4    14     3     1 
  376   377 
    1     1  </code></pre>
<p>The sequence table is a matrix with rows corresponding to (and named by) the samples, and columns corresponding to (and named by) the sequence variants. We have <strong>23367</strong> sequence variants but also a range of sequence lengths. Since many of these sequence variants are singletons or doubletons, we can just select a range that corresponds to the expected amplicon length and eliminate the spurious reads. But we will do this step after merging sequence tables from all 6 runs.</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_pseudo_BCS_35.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
</section><section id="tracking-reads-5" class="level3"><h3 class="anchored" data-anchor-id="tracking-reads-5">9. Tracking Reads</h3>
<p>If you are interested how reads changed up to this point in the pipeline please see the <a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a> page for more details.</p>
<p>And save the sequence table to an <code>RDS</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb139"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">BCS_35</span>, <span class="st">"BCS_35.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section>
</div>
</div>
</div>
</section><section id="merged-runs-workflow" class="level2"><h2 class="anchored" data-anchor-id="merged-runs-workflow">Merged Runs Workflow</h2>
<p>Now it is time to combine the sequence tables from each run together into one merged sequences table.</p>
<p>We start by reading in each sequence table.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Click here to access the source code for the <a href="include/_MERGE_RUNS.qmd">Merge Runs</a> section of the workflow.</p>
</div>
</div>
<table class="caption-top table">
<thead><tr class="header">
<th>Step</th>
<th>Command</th>
<th>What we’re doing</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>10</td>
<td><code>mergeSequenceTables()</code></td>
<td>merge seqtabs from all runs.</td>
</tr>
<tr class="even">
<td>11</td>
<td><code>removeBimeraDenovo()</code></td>
<td>screen for &amp; remove chimeras</td>
</tr>
<tr class="odd">
<td>12</td>
<td></td>
<td>track reads through workflow</td>
</tr>
<tr class="even">
<td>13</td>
<td><code>assignTaxonomy()</code></td>
<td>assign taxonomy &amp; finish workflow</td>
</tr>
</tbody>
</table>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
    D(BCS_26&lt;br/&gt;BCS_28&lt;br/&gt;BCS_29&lt;br/&gt;BCS_30&lt;br/&gt;BCS_34&lt;br/&gt;BCS_35&lt;br/&gt;) --&gt; M
    D --&gt; M
    D --&gt; M
    D --&gt; M
    D --&gt; M
    D --&gt; M
    M(mergeSequenceTables) --&gt; N(removeBimeraDenovo)
    N --&gt; O(assignTaxonomy)
</pre>
</div>
<p></p>
</div>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb140"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">BCS_26</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_26.rds"</span><span class="op">)</span></span>
<span><span class="va">BCS_28</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_28.rds"</span><span class="op">)</span></span>
<span><span class="va">BCS_29</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_29.rds"</span><span class="op">)</span></span>
<span><span class="va">BCS_30</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_30.rds"</span><span class="op">)</span></span>
<span><span class="va">BCS_34</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_34.rds"</span><span class="op">)</span></span>
<span><span class="va">BCS_35</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"`BCS_35.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<section id="merge-sequencing-tables" class="level3"><h3 class="anchored" data-anchor-id="merge-sequencing-tables">1. Merge Sequencing Tables</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb141"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">seqtab.merge</span> <span class="op">&lt;-</span> <span class="fu">mergeSequenceTables</span><span class="op">(</span><span class="va">BCS_26</span>, <span class="va">BCS_28</span>, <span class="va">BCS_29</span>, </span>
<span>                                    <span class="va">BCS_30</span>, <span class="va">BCS_34</span>, <span class="va">BCS_35</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.merge</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]  1909 96680</code></pre>
</div>
</div>
<p>So our count is 96680 ASVs across 1909 samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb143"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">seqtab.merge</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>  220   221   222   223   224   225   226   227   228   229   230   231   232 
  124    67    14    36    20    13    10    25     8     6     4    30     2 
  234   235   236   237   238   239   240   241   242   243   244   245   246 
    9     8  1371   151    41     6    96   401   291   443    31    14    13 
  247   248   249   250   251   252   253   254   255   256   257   258   259 
   26    23    19    48   159  3606 83887  3756   315   121    95    20    10 
  260   261   262   263   264   265   266   267   268   269   270   271   272 
    8    16     9     4     2     1     1     1     4     2     9     8     4 
  273   274   275   276   277   278   279   280   281   282   284   285   286 
    7     3     2     5     1     7     4     1     1     2     1     4     4 
  288   289   290   291   292   293   294   295   296   297   298   300   303 
    1     3     1     2     4     8     7     2     3     2     2     2     3 
  304   305   307   308   309   310   311   312   313   315   316   317   318 
    1     5     2     3     2     1     3     1     3     1     4     1     3 
  319   320   321   322   323   324   325   326   328   329   330   332   333 
    3     1     2     3     2     1     3     1     3     3     2     1     3 
  334   335   336   337   338   339   340   341   342   343   344   345   346 
   13     6     7    17     5    25    16    70    51     8     7     8     4 
  347   348   349   350   351   352   353   354   355   356   357   358   359 
   28    17    21    10     2    11     1     1     7     6    31     6    15 
  360   361   362   363   364   365   366   367   368   369   370   371   372 
   21   161   186    43   135   107    19     9    26     5     3     3     8 
  373   374   376   377   378   379   380   384   385   386   387   388 
   11     2     3     5     2     3     1     1     1     1     2     1 </code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb145"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">read_length_all</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">seqtab.merge</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">read_length_all</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="st">"length"</span></span>
<span><span class="va">plot_all</span> <span class="op">&lt;-</span> <span class="fu">qplot</span><span class="op">(</span><span class="va">length</span>, data <span class="op">=</span> <span class="va">read_length_all</span>, geom <span class="op">=</span> <span class="st">"histogram"</span>, </span>
<span>                  binwidth <span class="op">=</span> <span class="fl">1</span>, xlab <span class="op">=</span> <span class="st">"read length"</span>, </span>
<span>                  ylab <span class="op">=</span> <span class="st">"total variants"</span>, xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">200</span>,<span class="fl">400</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><img src="include/figures/read_length_before_collapse.png" class="img-fluid figure-img" width="1050"></p>
<figcaption>Distribution of read length by total ASVs after merging &amp; before removing extreme length variants.</figcaption></figure>
</div>
</div>
</div>
<p>Then we remove length variants.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb146"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">seqtab.trim</span> <span class="op">&lt;-</span> <span class="va">seqtab.merge</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">seqtab.merge</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> </span>
<span>                              <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fl">252</span>, <span class="fl">254</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.trim</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]  1909 91249</code></pre>
</div>
</div>
<p>And now our count is 91249 ASVs across 1909 samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb148"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html">nchar</a></span><span class="op">(</span><span class="fu">getSequences</span><span class="op">(</span><span class="va">seqtab.trim</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code> 252   253   254 
 3606 83887  3756 </code></pre>
</section><section id="remove-chimeras" class="level3"><h3 class="anchored" data-anchor-id="remove-chimeras">2. Remove Chimeras</h3>
<p>Even though the <code>dada</code> method corrects substitution and indel errors, chimeric sequences remain. According to the DADA2 documentation, the accuracy of sequence variants after denoising makes identifying chimeric ASVs simpler than when dealing with fuzzy OTUs. Chimeric sequences are identified if they can be exactly reconstructed by combining a left-segment and a right-segment from two more abundant parent sequences.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb150"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">seqtab.trim.nochim.consensus</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">removeBimeraDenovo</span><span class="op">(</span><span class="va">seqtab.trim</span>, </span>
<span>                     method <span class="op">=</span> <span class="st">"consensus"</span>, </span>
<span>                     multithread <span class="op">=</span> <span class="fl">20</span>,  </span>
<span>                     verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">seqtab.trim.nochim.consensus</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>Identified 18398 bimeras out of 91249 input sequences.</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1]  1909 72851</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb153"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.nochim</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">seqtab.2</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>[1] 0.9669996</code></pre>
<p>Chimera checking removed an additional 18398 sequence variants however, when we account for the abundances of each variant, we see chimeras accounts for about 3.30004% of the merged sequence reads. Not bad.</p>
</section><section id="track-reads-through-workflow" class="level3"><h3 class="anchored" data-anchor-id="track-reads-through-workflow">3. Track Reads through Workflow</h3>
<p>At this point we can look at the number of reads that made it through each step of the workflow for every sample.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb155"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">getN</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">getUniques</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">track</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span>, </span>
<span>               <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">seqtab.trim</span><span class="op">)</span>, </span>
<span>               <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">seqtab.trim.nochim.pool</span><span class="op">)</span>, </span>
<span>               <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">seqtab.trim.nochim.consensus</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">track</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"merged"</span>, <span class="st">"trim"</span>, </span>
<span>                     <span class="st">"chimera_pool"</span>, </span>
<span>                     <span class="st">"chimera_concensus"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="assign-taxonomy" class="level3"><h3 class="anchored" data-anchor-id="assign-taxonomy">4. Assign Taxonomy</h3>
<p>The <code>assignTaxonomy</code> command implements the naive Bayesian classifier, so for reproducible results you need to set a random number seed (see issue <a href="https://github.com/benjjneb/dada2/issues/538">#538</a>). We did this at the beginning of the workflow. For taxonomic assignment, we are using the GSR database <span class="citation" data-cites="molano2024gsr">(<a href="#ref-molano2024gsr" role="doc-biblioref">Molano, Vega-Abellaneda, and Manichanh 2024</a>)</span>. The developers of DADA2 maintain <a href="https://benjjneb.github.io/dada2/training.html">formatted versions of popular databases</a>, however the GSR-DB has not been formatted by the developers yet.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Click the link to can download an appropriate version of the <a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database</a>.</p>
</div>
</div>
<p>To create a DADA2 formatted version GSR-DB<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>, we perform the following steps.</p>
<section id="download-a-data-base" class="level4"><h4 class="anchored" data-anchor-id="download-a-data-base">Download a data base</h4>
<p>Here we are using the <a href="https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz">GSR V4 database</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb156"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb156-1"><a href="#cb156-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz</span>
<span id="cb156-2"><a href="#cb156-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> GSR-DB_V4_cluster-1.tar.gz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Once you uncompress the <code>tar</code> file you should see four files, two <code>.qza</code> files (which you can ignore), a <code>_taxa.txt</code> file and a <code>_seqs.fasta</code> file. We are interested in the latter two files. These are the files we need to format for DADA2. How about we have a look at each file?</p>
<p>First the taxonomy file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb157"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> GSR-DB_V4_cluster-1_taxa.txt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>Feature ID  Taxon
AY999846    k__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales-Streptomycetales-Unknown; f__Streptomycetaceae-Unknown; g__Streptomyces-Unknown; s__Unknown
JN885187.1.1362 k__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales-Streptomycetales-Unknown; f__Streptomycetaceae-Unknown; g__Kitasatospora-Streptomyces-Unknown; s__Unknown</code></pre>
<p>And next the fasta file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb159"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb159-1"><a href="#cb159-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> GSR-DB_V4_cluster-1_seqs.fasta</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>&gt;AY999846
TACGTAGGGCGCAAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGT
CACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTA
GAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAG
GAACACCGGTGGCGAAGGCGGATCTCTGGGCCGATACTGACGCTGAGGAGCGAAAGCGTG
GGGAGCGAACAGG
&gt;JN885187.1.1362
TACGTAGGGCGCGAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGT
CACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTA
GAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAG</code></pre>
<p>DADA2 requires a very specific format for classification. For the next few step we use a tool called SeqKit <span class="citation" data-cites="shen2024seqkit2">(<a href="#ref-shen2024seqkit2" role="doc-biblioref">Shen, Sipos, and Zhao 2024</a>)</span> for fasta defline manipulation.</p>
<p>This first step replaces the original fasta defline with the correspoding lineage information.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb161"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb161-1"><a href="#cb161-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate seqkit</span>
<span id="cb161-2"><a href="#cb161-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-w</span> 0  <span class="at">-p</span> <span class="st">"(.+)"</span> <span class="at">-r</span> <span class="st">'{kv}'</span> <span class="at">-k</span> GSR-DB_V4_cluster-1_taxa.txt GSR-DB_V4_cluster-1_seqs.fasta <span class="op">&gt;</span> tmp_1.fa</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>[INFO] read key-value file: GSR-DB_V4_cluster-1_taxa.txt
[INFO] 38802 pairs of key-value loaded</code></pre>
<p>Here is what the first few entries look like.</p>
<pre><code>&gt;k__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales-Streptomycetales-Unknown; f__Streptomycetaceae-Unknown; g__Streptomyces-Unknown; s__Unknown
TACGTAGGGCGCAAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGTCACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTAGAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGGTGGCGAAGGCGGATCTCTGGGCCGATACTGACGCTGAGGAGCGAAAGCGTGGGGAGCGAACAGG
&gt;k__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales-Streptomycetales-Unknown; f__Streptomycetaceae-Unknown; g__Kitasatospora-Streptomyces-Unknown; s__Unknown
TACGTAGGGCGCGAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGTCACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTAGAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGGTGGCGAAGGCGGATCTCTGGGCCGATACTGACGCTGAGGAGCGAAAGCGTGGGGAGCGAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGGTGGGCACTAGGTGTAGG</code></pre>
<p>Next we tidy up the deflines to remove spaces and leading taxonomic rank designations.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb164"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb164-1"><a href="#cb164-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-w</span> 0  <span class="at">-p</span> <span class="st">" s__.*"</span> <span class="at">-r</span> <span class="st">''</span>  tmp_1.fa <span class="op">&gt;</span> tmp_2.fa</span>
<span id="cb164-2"><a href="#cb164-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-w</span> 0  <span class="at">-p</span> <span class="st">"\s"</span> <span class="at">-r</span> <span class="st">''</span>  tmp_2.fa <span class="op">&gt;</span> tmp_3.fa</span>
<span id="cb164-3"><a href="#cb164-3" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-w</span> 0  <span class="at">-p</span> <span class="st">"\w__"</span> <span class="at">-r</span> <span class="st">''</span>  tmp_3.fa <span class="op">&gt;</span> gsrdb_dada2.fa</span>
<span id="cb164-4"><a href="#cb164-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp_<span class="pp">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And here are the first few lines of the final formatted GSR-DB fasta file.</p>
<pre><code>&gt;Bacteria;Actinobacteria;Actinomycetia;Actinomycetales-Streptomycetales-Unknown;Streptomycetaceae-Unknown;Streptomyces-Unknown;
TACGTAGGGCGCAAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGTCACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTAGAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGGTGGCGAAGGCGGATCTCTGGGCCGATACTGACGCTGAGGAGCGAAAGCGTGGGGAGCGAACAGG
&gt;Bacteria;Actinobacteria;Actinomycetia;Actinomycetales-Streptomycetales-Unknown;Streptomycetaceae-Unknown;Kitasatospora-Streptomyces-Unknown;
TACGTAGGGCGCGAGCGTTGTCCGGAATTATTGGGCGTAAAGAGCTCGTAGGCGGCTTGTCACGTCGGTTGTGAAAGCCCGGGGCTTAACCCCGGGTCTGCAGTCGATACGGGCAGGCTAGAGTTCGGTAGGGGAGATCGGAATTCCTGGTGTAGCGGTGAAATGCGCAGATATCAGGAGGAACACCGGTGGCGAAGGCGGATCTCTGGGCCGATACTGACGCTGAGGAGCGAAAGCGTGGGGAGCGAACAGGATTAGATACCCTGGTAGTCCACGCCGTAAACGGTGGGCACTAGGTGTAGG</code></pre>
<p>Now we can run the classification step.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb166"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">seqtab.consensus</span> <span class="op">&lt;-</span> <span class="va">seqtab.trim.nochim.consensus</span></span>
<span><span class="va">tax_gsrdb.consensus</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu">assignTaxonomy</span><span class="op">(</span><span class="va">seqtab.consensus</span>, </span>
<span>                 <span class="st">"TAXONOMY_FILES/gsrdb_dada2.fa"</span>,</span>
<span>                 multithread <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                 verbose <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">tax_gsrdb.consensus</span>, <span class="st">"4.tax_gsrdb.consensus.rds"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section></section><section id="r-session-information-code" class="level2"><h2 class="anchored" data-anchor-id="r-session-information-code">R Session Information &amp; Code</h2>
<p>This workflow was run on the <a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a>, Smithsonian Institution. Below are the specific packages and versions used in this workflow using both <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code> and <code><a href="https://sessioninfo.r-lib.org/reference/session_info.html">devtools::session_info()</a></code>. Click the arrow to see the details.</p>
<details><summary>
Expand to see R Session Info
</summary><pre><code>sessionInfo()
R version 4.3.2 (2023-10-31)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Rocky Linux 8.9 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /home/scottjj/miniconda3/envs/R/lib/libopenblasp-r0.3.25.so;  LAPACK version 3.11.0

locale:
[1] C

time zone: America/New_York
tzcode source: system (glibc)

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] DECIPHER_2.30.0             RSQLite_2.3.4              
 [3] ShortRead_1.60.0            GenomicAlignments_1.38.0   
 [5] SummarizedExperiment_1.32.0 Biobase_2.62.0             
 [7] MatrixGenerics_1.14.0       matrixStats_1.2.0          
 [9] Rsamtools_2.18.0            GenomicRanges_1.54.1       
[11] Biostrings_2.70.1           GenomeInfoDb_1.38.2        
[13] XVector_0.42.0              IRanges_2.36.0             
[15] S4Vectors_0.40.2            BiocParallel_1.36.0        
[17] BiocGenerics_0.48.1         decontam_1.22.0            
[19] dplyr_1.1.4                 gridExtra_2.3              
[21] phyloseq_1.46.0             ff_4.0.9                   
[23] bit_4.0.5                   ggplot2_3.4.4              
[25] dada2_1.30.0                Rcpp_1.0.11                

loaded via a namespace (and not attached):
 [1] DBI_1.2.0               bitops_1.0-7            deldir_2.0-2           
 [4] permute_0.9-7           rlang_1.1.2             magrittr_2.0.3         
 [7] ade4_1.7-22             compiler_4.3.2          mgcv_1.9-1             
[10] systemfonts_1.0.5       png_0.1-8               vctrs_0.6.5            
[13] reshape2_1.4.4          stringr_1.5.1           pkgconfig_2.0.3        
[16] crayon_1.5.2            fastmap_1.1.1           labeling_0.4.3         
[19] utf8_1.2.4              ragg_1.2.7              zlibbioc_1.48.0        
[22] cachem_1.0.8            jsonlite_1.8.8          biomformat_1.30.0      
[25] blob_1.2.4              rhdf5filters_1.14.1     DelayedArray_0.28.0    
[28] Rhdf5lib_1.24.1         jpeg_0.1-10             cluster_2.1.6          
[31] R6_2.5.1                stringi_1.8.3           RColorBrewer_1.1-3     
[34] iterators_1.0.14        Matrix_1.6-4            splines_4.3.2          
[37] igraph_1.6.0            tidyselect_1.2.0        abind_1.4-5            
[40] vegan_2.6-4             codetools_0.2-19        hwriter_1.3.2.1        
[43] lattice_0.22-5          tibble_3.2.1            plyr_1.8.9             
[46] withr_2.5.2             survival_3.5-7          RcppParallel_5.1.7     
[49] pillar_1.9.0            foreach_1.5.2           generics_0.1.3         
[52] RCurl_1.98-1.13         munsell_0.5.0           scales_1.3.0           
[55] glue_1.6.2              tools_4.3.2             interp_1.1-5           
[58] data.table_1.14.10      rhdf5_2.46.1            ape_5.7-1              
[61] latticeExtra_0.6-30     colorspace_2.1-0        nlme_3.1-164           
[64] GenomeInfoDbData_1.2.11 cli_3.6.2               textshaping_0.3.7      
[67] fansi_1.0.6             S4Arrays_1.2.0          gtable_0.3.4           
[70] digest_0.6.33           SparseArray_1.2.2       farver_2.1.1           
[73] memoise_2.0.1           multtest_2.58.0         lifecycle_1.0.4        
[76] bit64_4.0.5             MASS_7.3-60            

devtools::session_info()
- Session info ---------------------------------------------------------------
 setting  value
 version  R version 4.3.2 (2023-10-31)
 os       Rocky Linux 8.9 (Green Obsidian)
 system   x86_64, linux-gnu
 ui       X11
 language (EN)
 collate  C
 ctype    C
 tz       America/New_York
 date     2024-08-15
 pandoc   3.1.3 @ /home/scottjj/miniconda3/envs/R/bin/pandoc

- Packages -------------------------------------------------------------------
 package              * version   date (UTC) lib source
 abind                  1.4-5     2016-07-21 [1] CRAN (R 4.3.2)
 ade4                   1.7-22    2023-02-06 [1] CRAN (R 4.3.2)
 ape                    5.7-1     2023-03-13 [1] CRAN (R 4.3.2)
 Biobase              * 2.62.0    2023-10-24 [1] Bioconductor
 BiocGenerics         * 0.48.1    2023-11-01 [1] Bioconductor
 BiocParallel         * 1.36.0    2023-10-24 [1] Bioconductor
 biomformat             1.30.0    2023-10-24 [1] Bioconductor
 Biostrings           * 2.70.1    2023-10-25 [1] Bioconductor
 bit                  * 4.0.5     2022-11-15 [1] CRAN (R 4.3.0)
 bit64                  4.0.5     2020-08-30 [1] CRAN (R 4.3.0)
 bitops                 1.0-7     2021-04-24 [1] CRAN (R 4.3.2)
 blob                   1.2.4     2023-03-17 [1] CRAN (R 4.3.0)
 cachem                 1.0.8     2023-05-01 [1] CRAN (R 4.3.0)
 cli                    3.6.2     2023-12-11 [1] CRAN (R 4.3.2)
 cluster                2.1.6     2023-12-01 [1] CRAN (R 4.3.2)
 codetools              0.2-19    2023-02-01 [1] CRAN (R 4.3.0)
 colorspace             2.1-0     2023-01-23 [1] CRAN (R 4.3.0)
 crayon                 1.5.2     2022-09-29 [1] CRAN (R 4.3.0)
 dada2                * 1.30.0    2023-10-24 [1] Bioconductor
 data.table             1.14.10   2023-12-08 [1] CRAN (R 4.3.2)
 DBI                    1.2.0     2023-12-21 [1] CRAN (R 4.3.2)
 DECIPHER             * 2.30.0    2023-10-24 [1] Bioconductor
 decontam             * 1.22.0    2023-10-24 [1] Bioconductor
 DelayedArray           0.28.0    2023-10-24 [1] Bioconductor
 deldir                 2.0-2     2023-11-23 [1] CRAN (R 4.3.2)
 devtools               2.4.5     2022-10-11 [1] CRAN (R 4.3.2)
 digest                 0.6.33    2023-07-07 [1] CRAN (R 4.3.0)
 dplyr                * 1.1.4     2023-11-17 [1] CRAN (R 4.3.2)
 ellipsis               0.3.2     2021-04-29 [1] CRAN (R 4.3.0)
 fansi                  1.0.6     2023-12-08 [1] CRAN (R 4.3.2)
 farver                 2.1.1     2022-07-06 [1] CRAN (R 4.3.0)
 fastmap                1.1.1     2023-02-24 [1] CRAN (R 4.3.0)
 ff                   * 4.0.9     2023-01-25 [1] CRAN (R 4.3.2)
 foreach                1.5.2     2022-02-02 [1] CRAN (R 4.3.0)
 fs                     1.6.3     2023-07-20 [1] CRAN (R 4.3.1)
 generics               0.1.3     2022-07-05 [1] CRAN (R 4.3.0)
 GenomeInfoDb         * 1.38.2    2023-12-13 [1] Bioconductor 3.18 (R 4.3.2)
 GenomeInfoDbData       1.2.11    2023-12-26 [1] Bioconductor
 GenomicAlignments    * 1.38.0    2023-10-24 [1] Bioconductor
 GenomicRanges        * 1.54.1    2023-10-29 [1] Bioconductor
 ggplot2              * 3.4.4     2023-10-12 [1] CRAN (R 4.3.1)
 glue                   1.6.2     2022-02-24 [1] CRAN (R 4.3.0)
 gridExtra            * 2.3       2017-09-09 [1] CRAN (R 4.3.2)
 gtable                 0.3.4     2023-08-21 [1] CRAN (R 4.3.1)
 htmltools              0.5.7     2023-11-03 [1] CRAN (R 4.3.2)
 htmlwidgets            1.6.4     2023-12-06 [1] CRAN (R 4.3.2)
 httpuv                 1.6.13    2023-12-06 [1] CRAN (R 4.3.2)
 hwriter                1.3.2.1   2022-04-08 [1] CRAN (R 4.3.2)
 igraph                 1.6.0     2023-12-11 [1] CRAN (R 4.3.2)
 interp                 1.1-5     2023-11-27 [1] CRAN (R 4.3.2)
 IRanges              * 2.36.0    2023-10-24 [1] Bioconductor
 iterators              1.0.14    2022-02-05 [1] CRAN (R 4.3.0)
 jpeg                   0.1-10    2022-11-29 [1] CRAN (R 4.3.2)
 jsonlite               1.8.8     2023-12-04 [1] CRAN (R 4.3.2)
 labeling               0.4.3     2023-08-29 [1] CRAN (R 4.3.1)
 later                  1.3.2     2023-12-06 [1] CRAN (R 4.3.2)
 lattice                0.22-5    2023-10-24 [1] CRAN (R 4.3.1)
 latticeExtra           0.6-30    2022-07-04 [1] CRAN (R 4.3.2)
 lifecycle              1.0.4     2023-11-07 [1] CRAN (R 4.3.2)
 magrittr               2.0.3     2022-03-30 [1] CRAN (R 4.3.0)
 MASS                   7.3-60    2023-05-04 [1] CRAN (R 4.3.0)
 Matrix                 1.6-4     2023-11-30 [1] CRAN (R 4.3.2)
 MatrixGenerics       * 1.14.0    2023-10-24 [1] Bioconductor
 matrixStats          * 1.2.0     2023-12-11 [1] CRAN (R 4.3.2)
 memoise                2.0.1     2021-11-26 [1] CRAN (R 4.3.0)
 mgcv                   1.9-1     2023-12-21 [1] CRAN (R 4.3.2)
 mime                   0.12      2021-09-28 [1] CRAN (R 4.3.0)
 miniUI                 0.1.1.1   2018-05-18 [1] CRAN (R 4.3.2)
 multtest               2.58.0    2023-10-24 [1] Bioconductor
 munsell                0.5.0     2018-06-12 [1] CRAN (R 4.3.0)
 nlme                   3.1-164   2023-11-27 [1] CRAN (R 4.3.2)
 permute                0.9-7     2022-01-27 [1] CRAN (R 4.3.2)
 phyloseq             * 1.46.0    2023-10-24 [1] Bioconductor
 pillar                 1.9.0     2023-03-22 [1] CRAN (R 4.3.0)
 pkgbuild               1.4.3     2023-12-10 [1] CRAN (R 4.3.2)
 pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 4.3.0)
 pkgload                1.3.3     2023-09-22 [1] CRAN (R 4.3.2)
 plyr                   1.8.9     2023-10-02 [1] CRAN (R 4.3.1)
 png                    0.1-8     2022-11-29 [1] CRAN (R 4.3.2)
 profvis                0.3.8     2023-05-02 [1] CRAN (R 4.3.2)
 promises               1.2.1     2023-08-10 [1] CRAN (R 4.3.1)
 purrr                  1.0.2     2023-08-10 [1] CRAN (R 4.3.1)
 R6                     2.5.1     2021-08-19 [1] CRAN (R 4.3.0)
 ragg                   1.2.7     2023-12-11 [1] CRAN (R 4.3.2)
 RColorBrewer           1.1-3     2022-04-03 [1] CRAN (R 4.3.0)
 Rcpp                 * 1.0.11    2023-07-06 [1] CRAN (R 4.3.0)
 RcppParallel           5.1.7     2023-02-27 [1] CRAN (R 4.3.2)
 RCurl                  1.98-1.13 2023-11-02 [1] CRAN (R 4.3.2)
 remotes                2.4.2.1   2023-07-18 [1] CRAN (R 4.3.2)
 reshape2               1.4.4     2020-04-09 [1] CRAN (R 4.3.0)
 rhdf5                  2.46.1    2023-11-29 [1] Bioconductor 3.18 (R 4.3.2)
 rhdf5filters           1.14.1    2023-11-06 [1] Bioconductor
 Rhdf5lib               1.24.1    2023-12-11 [1] Bioconductor 3.18 (R 4.3.2)
 rlang                  1.1.2     2023-11-04 [1] CRAN (R 4.3.2)
 Rsamtools            * 2.18.0    2023-10-24 [1] Bioconductor
 RSQLite              * 2.3.4     2023-12-08 [1] CRAN (R 4.3.2)
 S4Arrays               1.2.0     2023-10-24 [1] Bioconductor
 S4Vectors            * 0.40.2    2023-11-23 [1] Bioconductor 3.18 (R 4.3.2)
 scales                 1.3.0     2023-11-28 [1] CRAN (R 4.3.2)
 sessioninfo            1.2.2     2021-12-06 [1] CRAN (R 4.3.2)
 shiny                  1.8.0     2023-11-17 [1] CRAN (R 4.3.2)
 ShortRead            * 1.60.0    2023-10-24 [1] Bioconductor
 SparseArray            1.2.2     2023-11-07 [1] Bioconductor
 stringi                1.8.3     2023-12-11 [1] CRAN (R 4.3.2)
 stringr                1.5.1     2023-11-14 [1] CRAN (R 4.3.2)
 SummarizedExperiment * 1.32.0    2023-10-24 [1] Bioconductor
 survival               3.5-7     2023-08-14 [1] CRAN (R 4.3.1)
 systemfonts            1.0.5     2023-10-09 [1] CRAN (R 4.3.1)
 textshaping            0.3.7     2023-10-09 [1] CRAN (R 4.3.1)
 tibble                 3.2.1     2023-03-20 [1] CRAN (R 4.3.0)
 tidyselect             1.2.0     2022-10-10 [1] CRAN (R 4.3.0)
 urlchecker             1.0.1     2021-11-30 [1] CRAN (R 4.3.2)
 usethis                2.2.2     2023-07-06 [1] CRAN (R 4.3.0)
 utf8                   1.2.4     2023-10-22 [1] CRAN (R 4.3.1)
 vctrs                  0.6.5     2023-12-01 [1] CRAN (R 4.3.2)
 vegan                  2.6-4     2022-10-11 [1] CRAN (R 4.3.2)
 withr                  2.5.2     2023-10-30 [1] CRAN (R 4.3.1)
 xtable                 1.8-4     2019-04-21 [1] CRAN (R 4.3.0)
 XVector              * 0.42.0    2023-10-24 [1] Bioconductor
 zlibbioc               1.48.0    2023-10-24 [1] Bioconductor</code></pre>
</details><p><br></p>
<!------------------------------------------------------------------------>
<!-------------------- Use this area to save things ---------------------->
<!------------------------------------------------------------------------>
</section></section><section id="data-set-prep" class="level1"><h1>Data Set Prep</h1>
<p>In this next part of the workflow our main goal is to create a <em>microtable object</em> using the R package <a href="https://joey711.github.io/phyloseq/">microeco</a> <span class="citation" data-cites="liu2021microeco">(<a href="#ref-liu2021microeco" role="doc-biblioref">Liu et al. 2021</a>)</span>. The microtable will be used to store the ASV by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</p>
<p>We will also:</p>
<ul>
<li>Remove any ASVs without kingdom level classification.<br>
</li>
<li>Revome any contaminants (chloroplast, mitochondria, etc.).<br>
</li>
<li>Remove Negative Control (NC) samples.<br>
</li>
<li>Remove any low-count samples.</li>
</ul>
<section id="read-counts-assessment" class="level2"><h2 class="anchored" data-anchor-id="read-counts-assessment">Read Counts Assessment</h2>
<p>Before we begin, let’s create a summary table containing some basic sample metadata and the read count data from the <a href="../../../sampledata/index.html">Sample Data</a> section of the workflow. We want to inspect how total reads changed through the workflow. Table headers are as follows:</p>
<table class="caption-top table">
<colgroup>
<col style="width: 33%">
<col style="width: 66%">
</colgroup>
<thead><tr class="header">
<th>Header</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>Sample ID</code></td>
<td>New sample ID based on Ocean, species, tissue, &amp; unique ID</td>
</tr>
<tr class="even">
<td><code>input rc</code></td>
<td>No.&nbsp;of raw reads</td>
</tr>
<tr class="odd">
<td><code>final rc</code></td>
<td>Final read count after removing chimeras</td>
</tr>
<tr class="even">
<td><code>per reads retain</code></td>
<td>Percent of reads remaining from <code>input</code> to <code>final rc</code>
</td>
</tr>
<tr class="odd">
<td><code>total ASVs</code></td>
<td>No.&nbsp;of ASVs</td>
</tr>
<tr class="even">
<td><code>Ocean</code></td>
<td>Sampling ocean</td>
</tr>
<tr class="odd">
<td><code>Morphospecies</code></td>
<td>Host shrimp species</td>
</tr>
<tr class="even">
<td><code>Tissue</code></td>
<td>Shrimp tissue type</td>
</tr>
<tr class="odd">
<td><code>Habitat</code></td>
<td>Sampling habitat</td>
</tr>
<tr class="even">
<td><code>Site</code></td>
<td>Sampling site</td>
</tr>
<tr class="odd">
<td><code>Taxon</code></td>
<td>Shrimp, environmental samples, Controls</td>
</tr>
<tr class="even">
<td><code>Species_Pair</code></td>
<td>ASK MATT</td>
</tr>
<tr class="odd">
<td><code>Species_group</code></td>
<td>ASK MATT</td>
</tr>
<tr class="even">
<td><code>Species_complex</code></td>
<td>ASK MATT</td>
</tr>
<tr class="odd">
<td><code>Run</code></td>
<td>Sequencing Run</td>
</tr>
<tr class="even">
<td><code>Plate</code></td>
<td>Plate name</td>
</tr>
</tbody>
</table>
<p><br></p>
</section><section id="prep-data-for-microeco" class="level2"><h2 class="anchored" data-anchor-id="prep-data-for-microeco">Prep Data for <code>microeco</code>
</h2>
<p>Like any tool, the microeco package needs the data in a specific form. I formatted our data to match the mock data in the microeco tutorial, specifically <a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data">this section</a>.</p>
<section id="a.-taxonomy-table" class="level3"><h3 class="anchored" data-anchor-id="a.-taxonomy-table">A. Taxonomy Table</h3>
<p>Here is what the taxonomy table looks like in the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
---------------------------------------------------------------------
  &amp;nbsp;      Kingdom           Phylum                 Class         
---------- ------------- -------------------- -----------------------
  OTU_50    k__Bacteria   p__Proteobacteria    c__Betaproteobacteria 

 OTU_8058   k__Bacteria   p__Actinobacteria      c__Actinobacteria   

 OTU_7152   k__Bacteria   p__Verrucomicrobia    c__OPB35 soil group  
---------------------------------------------------------------------</code></pre>
</div>
</div>
<p>And here is the taxonomy table from the dada2 workflow. The name dada2 gives to each ASV is the actual sequence of the ASV. We need to change this for downstream analyses.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-16-contents" aria-controls="callout-16" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see the partial content of the dada2 taxonomy table
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-16" class="callout-16-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                            &amp;nbsp;                                                                                                                               Kingdom  
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ----------
 TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCTCGTAGGCGGCACGCTAAGTCGGATGTGAAATCCCAGGGCTTAACCTTGGAACTGCATTCGATACTGGCGAGCTAGAGTGTGTGAGAGGGTAGTGGAATTCCAGGTGTAGCGGTGAAATGCGTAGATATCTGGAGGAACATCAGTGGCGAAGGCGACTACCTGGCACAACACTGACGCTGAGGAGCGAAAGCGTGGGGAGCAAACAGG   Bacteria 

 TACGGAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGCATGCAGGTGGTTTGTTAAGTCAGATGTGAAAGCCCGGGGCTCAACCTCGGAATAGCATTTGAAACTGGCAGACTAGAGTACTGTAGAGGGGGGTAGAATTTCAGGTGTAGCGGTGAAATGCGTAGAGATCTGAAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAGATACTGACACTCAGATGCGAAAGCGTGGGGAGCAAACAGG   Bacteria 

 TACATAGGTGTCAAGCGTTATCCGGATTTATTGGGCGTAAAGAGTGCGCAGATGGTTTAATAAGTTTGGGGTTAAATGCTAAAGCTTAACTTTAGTATGCCTTGAAAACTGTTTTACTAGAGTGTGGTAGAAGTTGATGGAATTTCATGTGTAGCGGTGGAATGCGTAGATATATGAAGGAACACCAATGGCGAAGGCAATCAACTATGCCATTACTGACATTCATGCACGAAAGCGTGGGGAGCGAATAGG    Bacteria 
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Table: Table continues below

 
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                            &amp;nbsp;                                                                                                                                   Phylum     
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ----------------
 TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCTCGTAGGCGGCACGCTAAGTCGGATGTGAAATCCCAGGGCTTAACCTTGGAACTGCATTCGATACTGGCGAGCTAGAGTGTGTGAGAGGGTAGTGGAATTCCAGGTGTAGCGGTGAAATGCGTAGATATCTGGAGGAACATCAGTGGCGAAGGCGACTACCTGGCACAACACTGACGCTGAGGAGCGAAAGCGTGGGGAGCAAACAGG   Proteobacteria 

 TACGGAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGCATGCAGGTGGTTTGTTAAGTCAGATGTGAAAGCCCGGGGCTCAACCTCGGAATAGCATTTGAAACTGGCAGACTAGAGTACTGTAGAGGGGGGTAGAATTTCAGGTGTAGCGGTGAAATGCGTAGAGATCTGAAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAGATACTGACACTCAGATGCGAAAGCGTGGGGAGCAAACAGG   Proteobacteria 

 TACATAGGTGTCAAGCGTTATCCGGATTTATTGGGCGTAAAGAGTGCGCAGATGGTTTAATAAGTTTGGGGTTAAATGCTAAAGCTTAACTTTAGTATGCCTTGAAAACTGTTTTACTAGAGTGTGGTAGAAGTTGATGGAATTTCATGTGTAGCGGTGGAATGCGTAGATATATGAAGGAACACCAATGGCGAAGGCAATCAACTATGCCATTACTGACATTCATGCACGAAAGCGTGGGGAGCGAATAGG          NA       
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

Table: Table continues below

 
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
                                                                                                                            &amp;nbsp;                                                                                                                                      Class        
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- ---------------------
 TACGGAGGGTGCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCTCGTAGGCGGCACGCTAAGTCGGATGTGAAATCCCAGGGCTTAACCTTGGAACTGCATTCGATACTGGCGAGCTAGAGTGTGTGAGAGGGTAGTGGAATTCCAGGTGTAGCGGTGAAATGCGTAGATATCTGGAGGAACATCAGTGGCGAAGGCGACTACCTGGCACAACACTGACGCTGAGGAGCGAAAGCGTGGGGAGCAAACAGG   Gammaproteobacteria 

 TACGGAGGGTGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGCATGCAGGTGGTTTGTTAAGTCAGATGTGAAAGCCCGGGGCTCAACCTCGGAATAGCATTTGAAACTGGCAGACTAGAGTACTGTAGAGGGGGGTAGAATTTCAGGTGTAGCGGTGAAATGCGTAGAGATCTGAAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAGATACTGACACTCAGATGCGAAAGCGTGGGGAGCAAACAGG   Gammaproteobacteria 

 TACATAGGTGTCAAGCGTTATCCGGATTTATTGGGCGTAAAGAGTGCGCAGATGGTTTAATAAGTTTGGGGTTAAATGCTAAAGCTTAACTTTAGTATGCCTTGAAAACTGTTTTACTAGAGTGTGGTAGAAGTTGATGGAATTTCATGTGTAGCGGTGGAATGCGTAGATATATGAAGGAACACCAATGGCGAAGGCAATCAACTATGCCATTACTGACATTCATGCACGAAAGCGTGGGGAGCGAATAGG            NA          
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>The first step is to rename the amplicon sequence variants so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention—ASV1, ASV2, ASV3, and so on.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb170"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">tax_gsrdb</span><span class="op">)</span></span>
<span><span class="co"># adding unique ASV names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"ASV"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And this is how the taxonomy table looks after assigning new names.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
----------------------------------------------------------
 &amp;nbsp;   Kingdom        Phylum              Class        
-------- ---------- ---------------- ---------------------
  ASV1    Bacteria   Proteobacteria   Gammaproteobacteria 

  ASV2    Bacteria   Proteobacteria   Gammaproteobacteria 

  ASV3    Bacteria         NA                 NA          
----------------------------------------------------------</code></pre>
</div>
</div>
<p>Next we need to add rank definitions to each classification.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb172"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rank_prefixes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  Kingdom <span class="op">=</span> <span class="st">"k"</span>, </span>
<span>  Phylum  <span class="op">=</span> <span class="st">"p"</span>, </span>
<span>  Class   <span class="op">=</span> <span class="st">"c"</span>, </span>
<span>  Order   <span class="op">=</span> <span class="st">"o"</span>, </span>
<span>  Family  <span class="op">=</span> <span class="st">"f"</span>, </span>
<span>  Genus   <span class="op">=</span> <span class="st">"g"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, <span class="op">~</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">.</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">rank_prefixes</span><span class="op">)</span>, </span>
<span>                <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">rank_prefixes</span><span class="op">[</span><span class="fu">cur_column</span><span class="op">(</span><span class="op">)</span><span class="op">]</span>, <span class="st">"__"</span>, <span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span><span class="fu">tidy_taxonomy</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And then this. Exactly like the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
-------------------------------------------------------------------
 &amp;nbsp;     Kingdom          Phylum                 Class          
-------- ------------- ------------------- ------------------------
  ASV1    k__Bacteria   p__Proteobacteria   c__Gammaproteobacteria 

  ASV2    k__Bacteria   p__Proteobacteria   c__Gammaproteobacteria 

  ASV3    k__Bacteria          p__                   c__           
-------------------------------------------------------------------</code></pre>
</div>
</div>
</section><section id="b.-sequence-table" class="level3"><h3 class="anchored" data-anchor-id="b.-sequence-table">B. Sequence Table</h3>
<p>Here is what the sequence table looks like in the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
--------------------------------------------------------------------
  &amp;nbsp;    S1   S2   S3   S4   S5   S6   S7   S9   S10   S11   S12 
---------- ---- ---- ---- ---- ---- ---- ---- ---- ----- ----- -----
 OTU_4272   1    0    1    1    0    0    1    1     0     1     1  

 OTU_236    1    4    0    2    35   5    94   0    177   14    27  

 OTU_399    9    2    2    4    4    0    3    6     0     1     2  
--------------------------------------------------------------------</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb175"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">seqtab</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/identical.html">identical</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_st</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tax_gsrdb</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tmp_st</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_tax</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_st</span> <span class="op">&lt;-</span>  <span class="va">tmp_st</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="va">tmp_st</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"tmp"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="va">tmp_st</span>, <span class="st">"tmp"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And now the final, modified sequence table.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
--------------------------------------------------------------------
 &amp;nbsp;   EP_A_AREN_MG_8652   EP_A_AREN_MG_8654   EP_A_AREN_MG_8697 
-------- ------------------- ------------------- -------------------
  ASV1            7                   0                   0         

  ASV2          1113                1803                 31         

  ASV3            0                   0                   0         
--------------------------------------------------------------------</code></pre>
</div>
</div>
</section><section id="c.-sample-table" class="level3"><h3 class="anchored" data-anchor-id="c.-sample-table">C. Sample Table</h3>
<p>Here is what the sample table looks like in the mock data.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb177"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">pandoc.table</span><span class="op">(</span><span class="va">sample_info_16S</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span>,<span class="op">]</span>, emphasize.rownames <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="cell-output cell-output-stdout">
<pre><code>
----------------------------------------------------
 &amp;nbsp;   SampleID   Group   Type       Saline      
-------- ---------- ------- ------ -----------------
   S1        S1       IW      NE    Non-saline soil 

   S2        S2       IW      NE    Non-saline soil 

   S3        S3       IW      NE    Non-saline soil 
----------------------------------------------------</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb179"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="st">"share/ssu/sampledata/"</span>, <span class="st">"sample_data.rds"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="va">samdf</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="va">samdf</span><span class="op">$</span><span class="va">SampleID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span></span>
<span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="va">samdf</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">SampleID</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And now a partial view of the final, modified sample table.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
-------------------------------------------------------------------------
      &amp;nbsp;             SampleID        OCEAN   SPECIES   TISSUE    ID  
------------------- ------------------- ------- --------- -------- ------
 EP_A_AREN_EG_8651   EP_A_AREN_EG_8651    EP     A_AREN      EG     8651 

 EP_A_AREN_EG_8654   EP_A_AREN_EG_8654    EP     A_AREN      EG     8654 

 EP_A_AREN_EG_8698   EP_A_AREN_EG_8698    EP     A_AREN      EG     8698 
-------------------------------------------------------------------------</code></pre>
</div>
</div>
</section></section><section id="experiment-level-objects" class="level2"><h2 class="anchored" data-anchor-id="experiment-level-objects">Experiment-level Objects</h2>
<p>In the following section we create <code>microtable</code> and <code>phyloseq</code> objects. These are single, consolidated experiment-level data objects that organize various data types, such as OTU tables (taxa abundance), sample metadata, taxonomic classifications, as well as phylogenetic trees and reference sequences, into one structured format. This unified approach facilitates reproducible, interactive analysis and visualization by bringing all related information together.</p>
<p>We begin by creating the <code>microtable</code> and then use the function <code>meco2phyloseq</code> from the <a href="https://github.com/ChiLiubio/file2meco">file2meco</a> package to create the <code>phyloseq</code> object. This way all of the underlying data is identical across the two objects.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These objects contain an ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</p>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb181"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">samdf</span></span>
<span><span class="va">tax_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span></span>
<span><span class="va">otu_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_st</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb182"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_me</span> <span class="op">&lt;-</span> <span class="va">microtable</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">sample_info</span>, </span>
<span>                         otu_table <span class="op">=</span> <span class="va">otu_tab</span>, </span>
<span>                         tax_table <span class="op">=</span> <span class="va">tax_tab</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>microtable-class object:
sample_table have 1909 rows and 13 columns
otu_table have 72851 rows and 1909 columns
tax_table have 72851 rows and 6 columns</code></pre>
<section id="add-representative-sequence" class="level3"><h3 class="anchored" data-anchor-id="add-representative-sequence">Add Representative Sequence</h3>
<p>We can also add representative sequences for each OTU/ASV. For this step, we can simply grab the sequences from the row names of the DADA2 taxonomy object loaded above.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb184"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_seq</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">tax_gsrdb</span><span class="op">)</span><span class="op">)</span> <span class="op">)</span></span>
<span><span class="va">tmp_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tax_tab</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_fa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">tmp_names</span>, <span class="va">tmp_seq</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">tmp_fa</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV_ID"</span>, <span class="st">"ASV_SEQ"</span><span class="op">)</span></span>
<span><span class="va">tmp_fa</span><span class="op">$</span><span class="va">ASV_ID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">sub</a></span><span class="op">(</span><span class="st">"^"</span>, <span class="st">"&gt;"</span>, <span class="va">tmp_fa</span><span class="op">$</span><span class="va">ASV_ID</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">tmp_fa</span>, <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"rep_seq.fasta"</span><span class="op">)</span>,</span>
<span>            sep <span class="op">=</span> <span class="st">"\n"</span>, col.names <span class="op">=</span> <span class="cn">FALSE</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            quote <span class="op">=</span> <span class="cn">FALSE</span>, fileEncoding <span class="op">=</span> <span class="st">"UTF-8"</span><span class="op">)</span>       </span>
<span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readDNAStringSet</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"rep_seq.fasta"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span><span class="op">$</span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="va">rep_fasta</span></span>
<span><span class="va">tmp_me</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="curate-the-data-set" class="level2"><h2 class="anchored" data-anchor-id="curate-the-data-set">Curate the Data Set</h2>
<p>Pretty much the last thing to do is remove unwanted taxa, negative controls, and low-count samples.</p>
<section id="remove-any-kingdom-nas" class="level3"><h3 class="anchored" data-anchor-id="remove-any-kingdom-nas">Remove any Kingdom NAs</h3>
<p>Here we can just use the straight up <code>subset</code> command since we do not need to worry about any ranks above Kingdom also being removed.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb185"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_na</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">)</span></span>
<span><span class="va">tmp_no_na</span><span class="op">$</span><span class="va">tax_table</span> <span class="op">%&lt;&gt;%</span> <span class="fu">base</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Archaea"</span> <span class="op">|</span> <span class="va">Kingdom</span> <span class="op">==</span> <span class="st">"k__Bacteria"</span><span class="op">)</span></span>
<span><span class="va">tmp_no_na</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1909 rows and 14 columns
otu_table have 72827 rows and 1909 columns
tax_table have 72827 rows and 6 columns
rep_fasta have 72827 sequences</code></pre>
</div>
</div>
</section><section id="remove-contaminants" class="level3"><h3 class="anchored" data-anchor-id="remove-contaminants">Remove Contaminants</h3>
<p>Now we can remove any potential contaminants like mitochondria or chloroplasts.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb187"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_cont</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_na</span><span class="op">)</span></span>
<span><span class="va">tmp_no_cont</span><span class="op">$</span><span class="fu">filter_pollution</span><span class="op">(</span>taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"mitochondria"</span>, <span class="st">"chloroplast"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_no_cont</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_no_cont</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>Total 0 features are removed from tax_table ...</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1909 rows and 14 columns
otu_table have 72827 rows and 1909 columns
tax_table have 72827 rows and 6 columns
rep_fasta have 72827 sequences</code></pre>
</div>
</div>
</section><section id="remove-negative-controls-nc" class="level3"><h3 class="anchored" data-anchor-id="remove-negative-controls-nc">Remove Negative Controls (NC)</h3>
<p>Now we need to remove the NC samples <em>and</em> ASVs found in those sample. We first identified all ASVs that were present in at least one NC sample represented by at least 1 read. We did this by subsetting the NC samples from the new microtable object.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb190"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_cont</span><span class="op">)</span></span>
<span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">sample_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">sample_table</span>, <span class="va">TAXON</span> <span class="op">==</span> <span class="st">"Control"</span><span class="op">)</span></span>
<span><span class="va">tmp_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_nc</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>72231 taxa with 0 abundance are removed from the otu_table ...</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 60 rows and 14 columns
otu_table have 596 rows and 60 columns
tax_table have 596 rows and 6 columns
rep_fasta have 596 sequences</code></pre>
</div>
</div>
<p>Looks like there are 596 ASVs in the NC samples from a total of 205834 reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb193"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_asvs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_nc</span><span class="op">$</span><span class="va">tax_table</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "ASV1"  "ASV2"  "ASV4"  "ASV5"  "ASV6"  "ASV7"  "ASV10" "ASV11" "ASV13"
[10] "ASV14" "ASV15" "ASV16" "ASV17" "ASV18" "ASV20" "ASV22" "ASV23" "ASV26"
[19] "ASV28" "ASV31"</code></pre>
</div>
</div>
<p>There are 596 ASVs found in the NC sample. ASVs are numbered in order by total abundance in the data set so we know that many of the ASVs in the NC samples are not particularly abundant in the dataset. We can look at the abundance of these ASVs across all samples and compare it to the NC. This takes a bit of wrangling.</p>
<p>Essentially, for each ASV, the code below calculates:</p>
<ul>
<li>The total number of NC samples containing at least 1 read.<br>
</li>
<li>The total number of reads in NC samples.<br>
</li>
<li>The total number of non-NC samples containing at least 1 read.<br>
</li>
<li>The total number of reads in non-NC samples.<br>
</li>
<li>The percent of reads in the NC samples and the percent of NC samples containing reads.</li>
</ul>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb195"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_rem_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_cont</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span> </span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">nc_asvs</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb196"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#-------provide a string unique to NC samples--------------#</span></span>
<span><span class="va">nc_name</span> <span class="op">&lt;-</span> <span class="st">"Control_"</span></span>
<span><span class="co">#----------------------------------------------------------#</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span>  <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total_reads_NC <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">.</span>, <span class="fu">contains</span><span class="op">(</span><span class="va">nc_name</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>, </span>
<span>         .after <span class="op">=</span> <span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="va">nc_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>total_reads_samps <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">.</span><span class="op">[</span><span class="fl">3</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>, </span>
<span>                .after <span class="op">=</span> <span class="st">"total_reads_NC"</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span><span class="op">[</span>, <span class="fl">4</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>perc_in_neg <span class="op">=</span> <span class="fl">100</span><span class="op">*</span><span class="op">(</span></span>
<span>    <span class="va">total_reads_NC</span> <span class="op">/</span> <span class="op">(</span><span class="va">total_reads_NC</span> <span class="op">+</span> <span class="va">total_reads_samps</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                .after <span class="op">=</span> <span class="st">"total_reads_samps"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb197"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_rem_nc_df</span><span class="op">$</span><span class="va">perc_in_neg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span><span class="op">$</span><span class="va">perc_in_neg</span>, digits <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                   <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                   <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"total_samples"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">tmp_2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span>, <span class="fu">contains</span><span class="op">(</span><span class="va">nc_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_2</span><span class="op">$</span><span class="va">num_samp_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_2</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">tmp_2</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_2</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"num_samp_nc"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                      <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_3</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span>, <span class="op">-</span><span class="fu">contains</span><span class="op">(</span><span class="va">nc_name</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_3</span><span class="op">$</span><span class="va">num_samp_no_nc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">tmp_3</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">tmp_3</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">tmp_3</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"num_samp_no_nc"</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>                      <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"ASV_ID"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">tmp_rem_nc_df</span>, <span class="va">tmp_1</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_2</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>                 <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">left_join</a></span><span class="op">(</span><span class="va">.</span>, <span class="va">tmp_3</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_nc_df</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>perc_in_neg_samp <span class="op">=</span> <span class="fl">100</span><span class="op">*</span><span class="op">(</span> <span class="va">num_samp_nc</span> <span class="op">/</span> <span class="op">(</span><span class="va">num_samp_nc</span> <span class="op">+</span> <span class="va">num_samp_no_nc</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                .after <span class="op">=</span> <span class="st">"num_samp_no_nc"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb198"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_check</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc_df</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Looking at these data we can see that ASVs like ASV1, ASV2, and so on, are only represented by a really small number of NC reads and samples. On the other hand, ASVs such as ASV91, ASV121, and ASV299 are relatively abundant in NC samples. We decided to remove ASVs if:</p>
<ul>
<li>The number of reads found in NC samples accounted for more than 10% of total reads OR</li>
<li>The percent of NC samples containing the ASV was greater than 10% of total samples.</li>
</ul>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb199"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_remove</span> <span class="op">&lt;-</span> <span class="va">nc_check</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">perc_in_neg</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">|</span> <span class="va">perc_in_neg_samp</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<table class="caption-top table">
<colgroup>
<col style="width: 11%">
<col style="width: 24%">
<col style="width: 20%">
<col style="width: 21%">
<col style="width: 21%">
</colgroup>
<thead><tr class="header">
<th></th>
<th>Total ASVs</th>
<th>NC reads</th>
<th>non NC reads</th>
<th>% NC reads</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Removed</td>
<td>371</td>
<td>170482</td>
<td>233876</td>
<td>42.161</td>
</tr>
<tr class="even">
<td>Retained</td>
<td>225</td>
<td>35352</td>
<td>12394240</td>
<td>0.284</td>
</tr>
</tbody>
</table>
<p>We identified a total of 596 ASVs that were present in at least 1 NC sample by at least 1 read. We removed any ASV where more than 10% of total reads were found in NC samples OR any ASV found in more than 10% of NC samples. Based on these criteria we removed 371 ASVs from the data set, which represented 170482 total reads in NC samples and 233876 total reads in non-NC samples. Of the total reads removed 42.161% came from NC samples. Of all ASVs identified in NC samples,225 were retained because the fell below the threshhold criteria. These ASVs accounted for 35352 reads in NC samples and 12394240 reads in non-NC samples. NC samples accounted for 0.284% of these reads.</p>
<p>OK, now we can remove the NC samples and any ASVs that met our criteria described above.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb200"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_nc</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_cont</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_rem_asv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">as.factor</a></span><span class="op">(</span><span class="va">nc_remove</span><span class="op">$</span><span class="va">ASV_ID</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="va">tmp_rem_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/row.names.html">row.names</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="va">tmp_rem_asv</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">sample_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html">subset</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">sample_table</span>, </span>
<span>                                 <span class="va">TAXON</span> <span class="op">!=</span> <span class="st">"Control_"</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_no_nc</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>9 samples with 0 abundance are removed from the otu_table ...</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1900 rows and 14 columns
otu_table have 72456 rows and 1900 columns
tax_table have 72456 rows and 6 columns
rep_fasta have 72456 sequences</code></pre>
</div>
</div>
</section><section id="remove-low-count-samples" class="level3"><h3 class="anchored" data-anchor-id="remove-low-count-samples">Remove Low-Count Samples</h3>
<p>Next, we can remove samples with really low read counts—here we set the threshold to <code>1000</code> reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb203"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_no_low</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_nc</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="va">tmp_no_nc</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">%&gt;%</span></span>
<span>          <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">is.numeric</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">1000</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>26 taxa with 0 abundance are removed from the otu_table ...</code></pre>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1848 rows and 14 columns
otu_table have 72430 rows and 1848 columns
tax_table have 72430 rows and 6 columns
rep_fasta have 72430 sequences</code></pre>
</div>
</div>
<p>Giving us the final microtable object.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb206"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_final</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_low</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Lastly, we can use the package <a href="https://github.com/ChiLiubio/file2meco"><code>file2meco</code></a> to generate a <a href="https://joey711.github.io/phyloseq/">phyloseq</a> object.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb207"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ps_final</span> <span class="op">&lt;-</span> <span class="fu">file2meco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/file2meco/man/meco2phyloseq.html">meco2phyloseq</a></span><span class="op">(</span><span class="va">me_final</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="summary" class="level2"><h2 class="anchored" data-anchor-id="summary">Summary</h2>
<p>Now time to summarize the data. For this we use the R package <a href="https://microbiome.github.io">miaverse</a> <span class="citation" data-cites="felix2024mia">(<a href="#ref-felix2024mia" role="doc-biblioref">Borman et al. 2024</a>)</span>.</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">me_dataset</th>
<th style="text-align: right;">total_asvs</th>
<th style="text-align: right;">total_reads</th>
<th style="text-align: right;">total_samples</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">original</td>
<td style="text-align: right;">72851</td>
<td style="text-align: right;">35764704</td>
<td style="text-align: right;">1909</td>
</tr>
<tr class="even">
<td style="text-align: left;">no NA kingdoms</td>
<td style="text-align: right;">72827</td>
<td style="text-align: right;">35760968</td>
<td style="text-align: right;">1909</td>
</tr>
<tr class="odd">
<td style="text-align: left;">no contaminants</td>
<td style="text-align: right;">72827</td>
<td style="text-align: right;">35760968</td>
<td style="text-align: right;">1909</td>
</tr>
<tr class="even">
<td style="text-align: left;">no negative controls</td>
<td style="text-align: right;">72456</td>
<td style="text-align: right;">35356610</td>
<td style="text-align: right;">1900</td>
</tr>
<tr class="odd">
<td style="text-align: left;">no low count samps</td>
<td style="text-align: right;">72430</td>
<td style="text-align: right;">35345679</td>
<td style="text-align: right;">1848</td>
</tr>
<tr class="even">
<td style="text-align: left;">final</td>
<td style="text-align: right;">72430</td>
<td style="text-align: right;">35345679</td>
<td style="text-align: right;">1848</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell" data-tbl-cap="Dataset metrics before and after curation.">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Dataset metrics before and after curation.</caption>
<thead><tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Start</th>
<th style="text-align: left;">End</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Min. no. of reads</td>
<td style="text-align: left;">49</td>
<td style="text-align: left;">1202</td>
</tr>
<tr class="even">
<td style="text-align: left;">Max. no. of reads</td>
<td style="text-align: left;">251022</td>
<td style="text-align: left;">249532</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total no. of reads</td>
<td style="text-align: left;">35764704</td>
<td style="text-align: left;">35345679</td>
</tr>
<tr class="even">
<td style="text-align: left;">Avg. no. of reads</td>
<td style="text-align: left;">18735</td>
<td style="text-align: left;">19126</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Median no. of reads</td>
<td style="text-align: left;">14827</td>
<td style="text-align: left;">15149.500</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total ASVs</td>
<td style="text-align: left;">72851</td>
<td style="text-align: left;">72430</td>
</tr>
<tr class="odd">
<td style="text-align: left;">No.&nbsp;of singleton ASVs</td>
<td style="text-align: left;">106</td>
<td style="text-align: left;">106</td>
</tr>
<tr class="even">
<td style="text-align: left;">% of singleton ASVs</td>
<td style="text-align: left;">0.146</td>
<td style="text-align: left;">0.146</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sparsity</td>
<td style="text-align: left;">0.996</td>
<td style="text-align: left;">0.996</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We started off with 72851 ASVs and 1909 samples. After curation of the dataset, there were 72430 ASVs and 1848 samples remaining.</p>
<p>We lost a total of 61 samples after curating the dataset. This includes 50 NC samples and 11 non-NC samples.</p>
<p>Here is a list of non-NC samples that were removed.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "WA_A_BAHA_GL_7474" "WA_A_NUTT_EG_9410" "WA_A_WEBS_HP_9411"
 [4] "WA_A_CRIS_GL_9624" "WA_A_CRIS_HP_7536" "WA_A_FLOR_GL_9659"
 [7] "WA_A_PARA_GL_9474" "WA_A_PARA_MG_9364" "WA_A_PARA_ST_9477"
[10] "WA_A_PCNS_MG_9513" "WA_A_THOM_ST_9389"</code></pre>
</div>
</div>
</section></section><section id="download-results" class="level1 page-columns page-full"><h1>Download Results</h1>
<p>Quick access to read changes through the pipeline and ASVs detected in negative control samples.</p>
<div id="listing-dada2-listing" class="column-body quarto-listing quarto-listing-container-table">
<table class="quarto-listing-table table">
<thead><tr>
<th>
File Name
</th>

<th>
Description
</th>

</tr></thead>
<tbody class="list">
<tr data-index="0" data-listing-filename-sort="BCS_26_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_26_read_changes.txt" class="filename listing-filename">BCS_26_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_26 </span>
</td>

</tr>
<tr data-index="1" data-listing-filename-sort="BCS_28_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_28_read_changes.txt" class="filename listing-filename">BCS_28_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_28 </span>
</td>

</tr>
<tr data-index="2" data-listing-filename-sort="BCS_29_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_29_read_changes.txt" class="filename listing-filename">BCS_29_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_29 </span>
</td>

</tr>
<tr data-index="3" data-listing-filename-sort="BCS_30_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_30_read_changes.txt" class="filename listing-filename">BCS_30_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_30 </span>
</td>

</tr>
<tr data-index="4" data-listing-filename-sort="BCS_34_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_34_read_changes.txt" class="filename listing-filename">BCS_34_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_34 </span>
</td>

</tr>
<tr data-index="5" data-listing-filename-sort="BCS_35_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/BCS_35_read_changes.txt" class="filename listing-filename">BCS_35_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Pre merge read changes for sequencing run BCS_35 </span>
</td>

</tr>
<tr data-index="6" data-listing-filename-sort="asv_in_nc_samples.txt">
<td>
<a href="../../../share/ssu/dada2/tables/asv_in_nc_samples.txt" class="filename listing-filename">asv_in_nc_samples.txt</a>
</td>
<td>
<span class="listing-description">Abundance table of ASVs detected in Negative Control samples. </span>
</td>

</tr>
<tr data-index="7" data-listing-filename-sort="dada2_pipeline_read_changes.txt">
<td>
<a href="../../../share/ssu/dada2/tables/dada2_pipeline_read_changes.txt" class="filename listing-filename">dada2_pipeline_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Read changes for all samples through Dada2 pipeline </span>
</td>

</tr>
</tbody>
</table>
<div class="listing-no-matching d-none">No matching items</div>
</div>
<!------------------------------------------------------------------------>
<!-------------------- Use this area to save things ---------------------->
<!------------------------------------------------------------------------>
<!--------------------------------------->
<!-- These chunks are for curated data -->
<!--------------------------------------->
<!-------------------------------------------->
<!-- These chunks are for  processing data  -->
<!-------------------------------------------->
</section><div id="quarto-appendix" class="default"><section id="detailed-session-info" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Detailed Session Info</h2><div class="quarto-appendix-contents">
<details><summary>Expand to see Session Info</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Ventura 13.7.8
 system   x86_64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2025-10-03
 pandoc   3.6.3 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/x86_64/ (via rmarkdown)
 quarto   1.8.25 @ /Applications/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
======   Devtools Session info   ===================================</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.</code></pre>
</div>
</div>
</details></div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-10-03</p>



<!-- -->

</div></section><section class="level4 appendix"><h2 class="quarto-appendix-heading">Collected Links</h2><div class="quarto-appendix-contents">
<ul>
<li><a href="../../../workflows/portal/data-dada2.html">Dada2 Data Portal</a></li>
<li><a href="https://benjjneb.github.io/dada2/">DADA2</a></li>
<li><a href="https://benjjneb.github.io/dada2/tutorial_1_8.html">DADA2 Pipeline Tutorial (1.8)</a></li>
<li><a href="https://benjjneb.github.io/dada2/ITS_workflow.html">DADA2 ITS Pipeline Workflow (1.8)</a></li>
<li><a href="include/_BCS_26.qmd">BCS_26 processing</a></li>
<li><a href="include/_BCS_28.qmd">BCS_28 processing</a></li>
<li><a href="include/_BCS_29.qmd">BCS_29 processing</a></li>
<li><a href="include/_BCS_30.qmd">BCS_30 processing</a></li>
<li><a href="include/_BCS_34.qmd">BCS_34 processing</a></li>
<li><a href="include/_BCS_35.qmd">BCS_35 processing</a></li>
<li><a href="include/_MERGE_RUNS.qmd">Merge Runs</a></li>
<li><a href="https://github.com/benjjneb/dada2/issues/538">#538</a></li>
<li><a href="https://benjjneb.github.io/dada2/training.html">formatted versions of popular databases</a></li>
<li><a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database</a></li>
<li><a href="https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz">GSR V4 database</a></li>
<li><a href="https://doi.org/10.25572/SIHPC">Smithsonian High Performance Cluster (SI/HPC)</a></li>
<li><a href="https://joey711.github.io/phyloseq/">microeco</a></li>
<li><a href="../../../sampledata/index.html">Sample Data</a></li>
<li><a href="https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data">this section</a></li>
<li><a href="https://github.com/ChiLiubio/file2meco">file2meco</a></li>
<li><a href="https://joey711.github.io/phyloseq/">phyloseq</a></li>
<li><a href="https://microbiome.github.io">miaverse</a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://istmobiome.rbind.io/">The Istmobiome Project</a></li>
</ul>

</div></section><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-felix2024mia" class="csl-entry" role="listitem">
Borman, Tuomas, Felix G. M. Ernst, Sudarshan A. Shetty, and Leo Lahti. 2024. <em>Mia: Microbiome Analysis</em>. <a href="https://doi.org/10.18129/B9.bioc.mia">https://doi.org/10.18129/B9.bioc.mia</a>.
</div>
<div id="ref-callahan2016dada2" class="csl-entry" role="listitem">
Callahan, Benjamin J, Paul J McMurdie, Michael J Rosen, Andrew W Han, Amy Jo A Johnson, and Susan P Holmes. 2016. <span>“DADA2: High-Resolution Sample Inference from Illumina Amplicon Data.”</span> <em>Nature Methods</em> 13 (7): 581. <a href="https://doi.org/10.1038/nmeth.3869">https://doi.org/10.1038/nmeth.3869</a>.
</div>
<div id="ref-liu2021microeco" class="csl-entry" role="listitem">
Liu, Chi, Yaoming Cui, Xiangzhen Li, and Minjie Yao. 2021. <span>“Microeco: An r Package for Data Mining in Microbial Community Ecology.”</span> <em>FEMS Microbiology Ecology</em> 97 (2): fiaa255. <a href="https://doi.org/10.1038/s41596-025-01239-4">https://doi.org/10.1038/s41596-025-01239-4</a>.
</div>
<div id="ref-molano2024gsr" class="csl-entry" role="listitem">
Molano, Leidy-Alejandra G, Sara Vega-Abellaneda, and Chaysavanh Manichanh. 2024. <span>“GSR-DB: A Manually Curated and Optimized Taxonomical Database for 16S rRNA Amplicon Analysis.”</span> <em>Msystems</em> 9 (2): e00950–23. <a href="https://doi.org/10.1128/msystems.00950-23">https://doi.org/10.1128/msystems.00950-23</a>.
</div>
<div id="ref-shen2024seqkit2" class="csl-entry" role="listitem">
Shen, Wei, Botond Sipos, and Liuyang Zhao. 2024. <span>“SeqKit2: A Swiss Army Knife for Sequence and Alignment Processing.”</span> <em>Imeta</em> 3 (3): e191. <a href="https://doi.org/10.1002/imt2.191">https://doi.org/10.1002/imt2.191</a>.
</div>
</div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>
<ol>
<li id="fn1"><p>From the developers: GSR database (Greengenes, SILVA, and RDP database) is an integrated and manually curated database for bacterial and archaeal 16S amplicon taxonomy analysis. Unlike previous integration approaches, this database creation pipeline includes a taxonomy unification step to ensure consistency in taxonomical annotations. The database was validated with three mock communities and two real datasets and compared with existing 16S databases such as Greengenes, GTDB, ITGDB, SILVA, RDP, and MetaSquare. Results showed that the GSR database enhances taxonomical annotations of 16S sequences, outperforming current 16S databases at the species level. The GSR database is available for full-length 16S sequences and the most commonly used hypervariable regions: V4, V1-V3, V3-V4, and V3-V5.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/shrimp-draft\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../../workflows/ssu/sampledata/index.html" class="pagination-link" aria-label="1. Sample Metadata">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">1. Sample Metadata</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../workflows/ssu/otu/index.html" class="pagination-link" aria-label="4. OTU Workflow">
        <span class="nav-page-text">4. OTU Workflow</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb212" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb212-1"><a href="#cb212-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb212-2"><a href="#cb212-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "2. DADA2 ASV Workflow"</span></span>
<span id="cb212-3"><a href="#cb212-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb212-4"><a href="#cb212-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Workflow for processing 16S rRNA samples for ASV analysis using DADA2. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence and taxonomy tables. A Microtable Object is produced to collate the data for downstream analysis. </span></span>
<span id="cb212-5"><a href="#cb212-5" aria-hidden="true" tabindex="-1"></a><span class="an">listing:</span><span class="co"> </span></span>
<span id="cb212-6"><a href="#cb212-6" aria-hidden="true" tabindex="-1"></a><span class="co">    id: dada2-listing</span></span>
<span id="cb212-7"><a href="#cb212-7" aria-hidden="true" tabindex="-1"></a><span class="co">    contents: data-dada2.yml</span></span>
<span id="cb212-8"><a href="#cb212-8" aria-hidden="true" tabindex="-1"></a><span class="co">    type: table</span></span>
<span id="cb212-9"><a href="#cb212-9" aria-hidden="true" tabindex="-1"></a><span class="co">    sort-ui: false</span></span>
<span id="cb212-10"><a href="#cb212-10" aria-hidden="true" tabindex="-1"></a><span class="co">    filter-ui: false</span></span>
<span id="cb212-11"><a href="#cb212-11" aria-hidden="true" tabindex="-1"></a><span class="co">    fields: </span></span>
<span id="cb212-12"><a href="#cb212-12" aria-hidden="true" tabindex="-1"></a><span class="co">      - filename</span></span>
<span id="cb212-13"><a href="#cb212-13" aria-hidden="true" tabindex="-1"></a><span class="co">      - description</span></span>
<span id="cb212-14"><a href="#cb212-14" aria-hidden="true" tabindex="-1"></a><span class="co">    field-links: </span></span>
<span id="cb212-15"><a href="#cb212-15" aria-hidden="true" tabindex="-1"></a><span class="co">      - filename</span></span>
<span id="cb212-16"><a href="#cb212-16" aria-hidden="true" tabindex="-1"></a><span class="co">    field-display-names: </span></span>
<span id="cb212-17"><a href="#cb212-17" aria-hidden="true" tabindex="-1"></a><span class="co">      filename: File Name</span></span>
<span id="cb212-18"><a href="#cb212-18" aria-hidden="true" tabindex="-1"></a><span class="co">      description: Description</span></span>
<span id="cb212-19"><a href="#cb212-19" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb212-20"><a href="#cb212-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-21"><a href="#cb212-21" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_setup.qmd &gt;}}</span>
<span id="cb212-22"><a href="#cb212-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-25"><a href="#cb212-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-26"><a href="#cb212-26" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-27"><a href="#cb212-27" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-28"><a href="#cb212-28" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: set wd for here</span></span>
<span id="cb212-29"><a href="#cb212-29" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb212-30"><a href="#cb212-30" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"page_build"</span>, <span class="st">"dada2_part2.rdata"</span>))</span>
<span id="cb212-31"><a href="#cb212-31" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"dada2"</span> <span class="co"># e.g., med, dada2, etc</span></span>
<span id="cb212-32"><a href="#cb212-32" aria-hidden="true" tabindex="-1"></a>work_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"working_files/ssu/"</span>, workflow_name)</span>
<span id="cb212-33"><a href="#cb212-33" aria-hidden="true" tabindex="-1"></a>share_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"share/ssu/"</span>, workflow_name)</span>
<span id="cb212-34"><a href="#cb212-34" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"assets/scripts"</span>, <span class="st">"summarize_objs.R"</span>))</span>
<span id="cb212-35"><a href="#cb212-35" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"assets/scripts"</span>, <span class="st">"mia_metrics.R"</span>))</span>
<span id="cb212-36"><a href="#cb212-36" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-37"><a href="#cb212-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-38"><a href="#cb212-38" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb212-39"><a href="#cb212-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-40"><a href="#cb212-40" aria-hidden="true" tabindex="-1"></a>::: {.callout-note icon=false}</span>
<span id="cb212-41"><a href="#cb212-41" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data &amp; Scripts</span></span>
<span id="cb212-42"><a href="#cb212-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-43"><a href="#cb212-43" aria-hidden="true" tabindex="-1"></a>Fastq sequence files, scripts, and other assets for running this workflow can be found on the <span class="co">[</span><span class="ot">Dada2 Data Portal</span><span class="co">](/workflows/portal/data-dada2.qmd)</span> page. </span>
<span id="cb212-44"><a href="#cb212-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-45"><a href="#cb212-45" aria-hidden="true" tabindex="-1"></a>The Data Portal page also contains a link to the curated output of this pipeline–-feature-sample count table, taxonomy table, sample metadata, &amp; representative fasta sequences. Data is available as stand-alone text files, OR bundled in microtable and phyloseq objects. Archive also includes a table tracking sample read changes.</span>
<span id="cb212-46"><a href="#cb212-46" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb212-47"><a href="#cb212-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-48"><a href="#cb212-48" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Required Packages &amp; Software</span></span>
<span id="cb212-49"><a href="#cb212-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-50"><a href="#cb212-50" aria-hidden="true" tabindex="-1"></a>There are several R packages you need to run this workflow. </span>
<span id="cb212-51"><a href="#cb212-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-54"><a href="#cb212-54" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-55"><a href="#cb212-55" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false </span></span>
<span id="cb212-56"><a href="#cb212-56" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-57"><a href="#cb212-57" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb212-58"><a href="#cb212-58" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb212-59"><a href="#cb212-59" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb212-60"><a href="#cb212-60" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Click here for workflow library information."</span></span>
<span id="cb212-61"><a href="#cb212-61" aria-hidden="true" tabindex="-1"></a><span class="co">#!/usr/bin/env Rscript</span></span>
<span id="cb212-62"><a href="#cb212-62" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">919191</span>)</span>
<span id="cb212-63"><a href="#cb212-63" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(tidyverse, gridExtra, grid, phyloseq,</span>
<span id="cb212-64"><a href="#cb212-64" aria-hidden="true" tabindex="-1"></a>               formatR, gdata, ff, decontam, dada2, </span>
<span id="cb212-65"><a href="#cb212-65" aria-hidden="true" tabindex="-1"></a>               ShortRead, Biostrings, DECIPHER, </span>
<span id="cb212-66"><a href="#cb212-66" aria-hidden="true" tabindex="-1"></a>               <span class="at">install =</span> <span class="cn">FALSE</span>, <span class="at">update =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-67"><a href="#cb212-67" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-68"><a href="#cb212-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-69"><a href="#cb212-69" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overview</span></span>
<span id="cb212-70"><a href="#cb212-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-71"><a href="#cb212-71" aria-hidden="true" tabindex="-1"></a>This workflow contains the code we used to process the 16S rRNA data sets using <span class="co">[</span><span class="ot">DADA2</span><span class="co">](https://benjjneb.github.io/dada2/)</span> <span class="co">[</span><span class="ot">@callahan2016dada2</span><span class="co">]</span>. Workflow construction is based on the <span class="co">[</span><span class="ot">DADA2 Pipeline Tutorial (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/tutorial_1_8.html)</span> and the primer identification section of the <span class="co">[</span><span class="ot">DADA2 ITS Pipeline Workflow (1.8)</span><span class="co">](https://benjjneb.github.io/dada2/ITS_workflow.html)</span>. In the first part of the pipeline, we process the individual sequencing runs **separately**. Next we combine the sequence tables from each run  into one merged sequences table and continue with processing.</span>
<span id="cb212-72"><a href="#cb212-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-73"><a href="#cb212-73" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read Processing {#read-processing}</span></span>
<span id="cb212-74"><a href="#cb212-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-75"><a href="#cb212-75" aria-hidden="true" tabindex="-1"></a>We processed each of the six sequencing runs separately for the first part of the DADA2 workflow. While some of the outputs are slightly different (e.g. quality scores, filtering, ASV inference, etc.) the code is the same. For posterity, code for each run is included here.</span>
<span id="cb212-76"><a href="#cb212-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-77"><a href="#cb212-77" aria-hidden="true" tabindex="-1"></a><span class="fu">## Individual Run Workflows</span></span>
<span id="cb212-78"><a href="#cb212-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-79"><a href="#cb212-79" aria-hidden="true" tabindex="-1"></a>The first part of the workflow consists of the following steps for each of the runs:</span>
<span id="cb212-80"><a href="#cb212-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-81"><a href="#cb212-81" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Step <span class="pp">|</span> Command                <span class="pp">|</span> What we’re doing                       <span class="pp">|</span></span>
<span id="cb212-82"><a href="#cb212-82" aria-hidden="true" tabindex="-1"></a><span class="pp">|------|------------------------|----------------------------------------|</span></span>
<span id="cb212-83"><a href="#cb212-83" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 1    <span class="pp">|</span> multiple               <span class="pp">|</span> prepare input file names &amp; paths       <span class="pp">|</span></span>
<span id="cb212-84"><a href="#cb212-84" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 2    <span class="pp">|</span> multiple               <span class="pp">|</span> Define primers (all orientations)      <span class="pp">|</span></span>
<span id="cb212-85"><a href="#cb212-85" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 3    <span class="pp">|</span> <span class="in">`cutadapt`</span>             <span class="pp">|</span> Remove primers                         <span class="pp">|</span></span>
<span id="cb212-86"><a href="#cb212-86" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 4    <span class="pp">|</span> <span class="in">`plotQualityProfile()`</span> <span class="pp">|</span> Plot quality scores.                   <span class="pp">|</span></span>
<span id="cb212-87"><a href="#cb212-87" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 5    <span class="pp">|</span> <span class="in">`filterAndTrim()`</span>      <span class="pp">|</span> Assess quality &amp; filter reads          <span class="pp">|</span></span>
<span id="cb212-88"><a href="#cb212-88" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 6    <span class="pp">|</span> <span class="in">`learnErrors()`</span>        <span class="pp">|</span> Generate an error model for the data   <span class="pp">|</span></span>
<span id="cb212-89"><a href="#cb212-89" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 7    <span class="pp">|</span> <span class="in">`derepFastq()`</span>         <span class="pp">|</span> Dereplicate sequences                  <span class="pp">|</span></span>
<span id="cb212-90"><a href="#cb212-90" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 8    <span class="pp">|</span> <span class="in">`dada()`</span>               <span class="pp">|</span> Infer ASVs (forward &amp; reverse reads).  <span class="pp">|</span></span>
<span id="cb212-91"><a href="#cb212-91" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 9    <span class="pp">|</span> <span class="in">`mergePairs()`</span>.        <span class="pp">|</span> Merge denoised forward &amp; reverse reads <span class="pp">|</span></span>
<span id="cb212-92"><a href="#cb212-92" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 10   <span class="pp">|</span> <span class="in">`makeSequenceTable()`</span>  <span class="pp">|</span> Generate count table for each run      <span class="pp">|</span></span>
<span id="cb212-93"><a href="#cb212-93" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> 11   <span class="pp">|</span>                        <span class="pp">|</span> Track reads through workflow           <span class="pp">|</span></span>
<span id="cb212-94"><a href="#cb212-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-95"><a href="#cb212-95" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_dada2_flowchart_1.qmd &gt;}}</span>
<span id="cb212-96"><a href="#cb212-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-97"><a href="#cb212-97" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;</span><span class="kw">br</span><span class="dt">/&gt;</span></span>
<span id="cb212-98"><a href="#cb212-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-99"><a href="#cb212-99" aria-hidden="true" tabindex="-1"></a>::: panel-tabset</span>
<span id="cb212-100"><a href="#cb212-100" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_26</span></span>
<span id="cb212-101"><a href="#cb212-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-102"><a href="#cb212-102" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_26.qmd &gt;}}</span>
<span id="cb212-103"><a href="#cb212-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-104"><a href="#cb212-104" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_28</span></span>
<span id="cb212-105"><a href="#cb212-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-106"><a href="#cb212-106" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_28.qmd &gt;}}</span>
<span id="cb212-107"><a href="#cb212-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-108"><a href="#cb212-108" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_29</span></span>
<span id="cb212-109"><a href="#cb212-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-110"><a href="#cb212-110" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_29.qmd &gt;}}</span>
<span id="cb212-111"><a href="#cb212-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-112"><a href="#cb212-112" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_30</span></span>
<span id="cb212-113"><a href="#cb212-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-114"><a href="#cb212-114" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_30.qmd &gt;}}</span>
<span id="cb212-115"><a href="#cb212-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-116"><a href="#cb212-116" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_34</span></span>
<span id="cb212-117"><a href="#cb212-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-118"><a href="#cb212-118" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_34.qmd &gt;}}</span>
<span id="cb212-119"><a href="#cb212-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-120"><a href="#cb212-120" aria-hidden="true" tabindex="-1"></a><span class="fu">## BCS_35</span></span>
<span id="cb212-121"><a href="#cb212-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-122"><a href="#cb212-122" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_BCS_35.qmd &gt;}}</span>
<span id="cb212-123"><a href="#cb212-123" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb212-124"><a href="#cb212-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-125"><a href="#cb212-125" aria-hidden="true" tabindex="-1"></a><span class="fu">## Merged Runs Workflow</span></span>
<span id="cb212-126"><a href="#cb212-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-127"><a href="#cb212-127" aria-hidden="true" tabindex="-1"></a>Now it is time to combine the sequence tables from each run together into one merged sequences table.</span>
<span id="cb212-128"><a href="#cb212-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-129"><a href="#cb212-129" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_MERGE_RUNS.qmd &gt;}}</span>
<span id="cb212-130"><a href="#cb212-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-131"><a href="#cb212-131" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data Set Prep</span></span>
<span id="cb212-132"><a href="#cb212-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-135"><a href="#cb212-135" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-136"><a href="#cb212-136" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-137"><a href="#cb212-137" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-138"><a href="#cb212-138" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb212-139"><a href="#cb212-139" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb212-140"><a href="#cb212-140" aria-hidden="true" tabindex="-1"></a>seqtab <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(work_here, <span class="st">"rdata/3.seqtab.trim.nochim.consensus.rds"</span>))</span>
<span id="cb212-141"><a href="#cb212-141" aria-hidden="true" tabindex="-1"></a>tax_gsrdb <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(work_here, <span class="st">"rdata/4.tax_gsrdb.consensus.rds"</span>))</span>
<span id="cb212-142"><a href="#cb212-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-143"><a href="#cb212-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-144"><a href="#cb212-144" aria-hidden="true" tabindex="-1"></a>In this next part of the workflow our main goal is to create a *microtable object* using the R package <span class="co">[</span><span class="ot">microeco</span><span class="co">](https://joey711.github.io/phyloseq/)</span> <span class="co">[</span><span class="ot">@liu2021microeco</span><span class="co">]</span>. The microtable will be used to store the ASV by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</span>
<span id="cb212-145"><a href="#cb212-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-146"><a href="#cb212-146" aria-hidden="true" tabindex="-1"></a>We will also:</span>
<span id="cb212-147"><a href="#cb212-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-148"><a href="#cb212-148" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove any ASVs without kingdom level classification.\</span>
<span id="cb212-149"><a href="#cb212-149" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Revome any contaminants (chloroplast, mitochondria, etc.).\</span>
<span id="cb212-150"><a href="#cb212-150" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove Negative Control (NC) samples.\</span>
<span id="cb212-151"><a href="#cb212-151" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Remove any low-count samples.</span>
<span id="cb212-152"><a href="#cb212-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-153"><a href="#cb212-153" aria-hidden="true" tabindex="-1"></a><span class="fu">## Read Counts Assessment</span></span>
<span id="cb212-154"><a href="#cb212-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-155"><a href="#cb212-155" aria-hidden="true" tabindex="-1"></a>Before we begin, let's create a summary table containing some basic sample metadata and the read count data from the <span class="co">[</span><span class="ot">Sample Data</span><span class="co">](/sampledata/index.html)</span> section of the workflow. We want to inspect how total reads changed through the workflow. Table headers are as follows:</span>
<span id="cb212-156"><a href="#cb212-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-157"><a href="#cb212-157" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Header             <span class="pp">|</span> Description                                                <span class="pp">|</span></span>
<span id="cb212-158"><a href="#cb212-158" aria-hidden="true" tabindex="-1"></a><span class="pp">|------------------------|------------------------------------------------|</span></span>
<span id="cb212-159"><a href="#cb212-159" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Sample ID`</span>        <span class="pp">|</span> New sample ID based on Ocean, species, tissue, &amp; unique ID <span class="pp">|</span></span>
<span id="cb212-160"><a href="#cb212-160" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`input rc`</span>         <span class="pp">|</span> No. of raw reads                                           <span class="pp">|</span></span>
<span id="cb212-161"><a href="#cb212-161" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`final rc`</span>         <span class="pp">|</span> Final read count after removing chimeras                   <span class="pp">|</span></span>
<span id="cb212-162"><a href="#cb212-162" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`per reads retain`</span> <span class="pp">|</span> Percent of reads remaining from <span class="in">`input`</span> to <span class="in">`final rc`</span>      <span class="pp">|</span></span>
<span id="cb212-163"><a href="#cb212-163" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`total ASVs`</span>       <span class="pp">|</span> No. of ASVs                                                <span class="pp">|</span></span>
<span id="cb212-164"><a href="#cb212-164" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Ocean`</span>            <span class="pp">|</span> Sampling ocean                                             <span class="pp">|</span></span>
<span id="cb212-165"><a href="#cb212-165" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Morphospecies`</span>    <span class="pp">|</span> Host shrimp species                                        <span class="pp">|</span></span>
<span id="cb212-166"><a href="#cb212-166" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Tissue`</span>           <span class="pp">|</span> Shrimp tissue type                                         <span class="pp">|</span></span>
<span id="cb212-167"><a href="#cb212-167" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Habitat`</span>          <span class="pp">|</span> Sampling habitat                                           <span class="pp">|</span></span>
<span id="cb212-168"><a href="#cb212-168" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Site`</span>             <span class="pp">|</span> Sampling site                                              <span class="pp">|</span></span>
<span id="cb212-169"><a href="#cb212-169" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Taxon`</span>            <span class="pp">|</span> Shrimp, environmental samples, Controls                    <span class="pp">|</span></span>
<span id="cb212-170"><a href="#cb212-170" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Species_Pair`</span>     <span class="pp">|</span> ASK MATT                                                   <span class="pp">|</span></span>
<span id="cb212-171"><a href="#cb212-171" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Species_group`</span>    <span class="pp">|</span> ASK MATT                                                   <span class="pp">|</span></span>
<span id="cb212-172"><a href="#cb212-172" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Species_complex`</span>  <span class="pp">|</span> ASK MATT                                                   <span class="pp">|</span></span>
<span id="cb212-173"><a href="#cb212-173" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Run`</span>              <span class="pp">|</span> Sequencing Run                                             <span class="pp">|</span></span>
<span id="cb212-174"><a href="#cb212-174" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> <span class="in">`Plate`</span>            <span class="pp">|</span> Plate name                                                 <span class="pp">|</span></span>
<span id="cb212-175"><a href="#cb212-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-176"><a href="#cb212-176" aria-hidden="true" tabindex="-1"></a><span class="dt">&lt;</span><span class="kw">br</span><span class="dt">/&gt;</span></span>
<span id="cb212-177"><a href="#cb212-177" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-180"><a href="#cb212-180" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-181"><a href="#cb212-181" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-182"><a href="#cb212-182" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-183"><a href="#cb212-183" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"working_files/ssu"</span>, <span class="st">"sampledata/sample_data.rds"</span>))</span>
<span id="cb212-184"><a href="#cb212-184" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">here</span>(work_here,</span>
<span id="cb212-185"><a href="#cb212-185" aria-hidden="true" tabindex="-1"></a>    <span class="st">"dada2_pipeline_read_changes.txt"</span>),</span>
<span id="cb212-186"><a href="#cb212-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span></span>
<span id="cb212-187"><a href="#cb212-187" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-188"><a href="#cb212-188" aria-hidden="true" tabindex="-1"></a>tmp_tab2[<span class="dv">3</span><span class="sc">:</span><span class="dv">7</span>] <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb212-189"><a href="#cb212-189" aria-hidden="true" tabindex="-1"></a>tmp_tab1 <span class="ot">&lt;-</span> <span class="fu">arrange</span>(tmp_tab1, SampleID, <span class="at">.by_group =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-190"><a href="#cb212-190" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> <span class="fu">arrange</span>(tmp_tab2, SampleID, <span class="at">.by_group =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-191"><a href="#cb212-191" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(tmp_tab1<span class="sc">$</span>SampleID, tmp_tab2<span class="sc">$</span>SampleID)</span>
<span id="cb212-192"><a href="#cb212-192" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-193"><a href="#cb212-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-194"><a href="#cb212-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-197"><a href="#cb212-197" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-198"><a href="#cb212-198" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-199"><a href="#cb212-199" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-200"><a href="#cb212-200" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">row.names</span>(seqtab))</span>
<span id="cb212-201"><a href="#cb212-201" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_tab3) <span class="ot">&lt;-</span> <span class="st">"SampleID"</span></span>
<span id="cb212-202"><a href="#cb212-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-203"><a href="#cb212-203" aria-hidden="true" tabindex="-1"></a>tmp_seqtab <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqtab)</span>
<span id="cb212-204"><a href="#cb212-204" aria-hidden="true" tabindex="-1"></a>tmp_seqtab <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(tmp_seqtab, <span class="at">var =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb212-205"><a href="#cb212-205" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-206"><a href="#cb212-206" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(tmp_tab3<span class="sc">$</span>SampleID, tmp_seqtab<span class="sc">$</span>SampleID)</span>
<span id="cb212-207"><a href="#cb212-207" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-208"><a href="#cb212-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-211"><a href="#cb212-211" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-212"><a href="#cb212-212" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-213"><a href="#cb212-213" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-214"><a href="#cb212-214" aria-hidden="true" tabindex="-1"></a>tmp_seqtab <span class="ot">&lt;-</span> tmp_seqtab <span class="sc">%&gt;%</span></span>
<span id="cb212-215"><a href="#cb212-215" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">count =</span> <span class="fu">rowSums</span>(. <span class="sc">!=</span> <span class="dv">0</span>))</span>
<span id="cb212-216"><a href="#cb212-216" aria-hidden="true" tabindex="-1"></a>tmp_tab3<span class="sc">$</span>no_asvs <span class="ot">&lt;-</span> tmp_seqtab<span class="sc">$</span>count</span>
<span id="cb212-217"><a href="#cb212-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-218"><a href="#cb212-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-221"><a href="#cb212-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-222"><a href="#cb212-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-223"><a href="#cb212-223" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-224"><a href="#cb212-224" aria-hidden="true" tabindex="-1"></a>tmp_tab4 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">right_join</span>(tmp_tab1, tmp_tab2, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb212-225"><a href="#cb212-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-226"><a href="#cb212-226" aria-hidden="true" tabindex="-1"></a>tmp_tab2 <span class="ot">&lt;-</span> tmp_tab2[<span class="fu">order</span>(tmp_tab2<span class="sc">$</span>SampleID),]</span>
<span id="cb212-227"><a href="#cb212-227" aria-hidden="true" tabindex="-1"></a>tmp_tab4 <span class="ot">&lt;-</span> tmp_tab4[<span class="fu">order</span>(tmp_tab4<span class="sc">$</span>SampleID),]</span>
<span id="cb212-228"><a href="#cb212-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-229"><a href="#cb212-229" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(tmp_tab2<span class="sc">$</span>input, tmp_tab4<span class="sc">$</span>input)</span>
<span id="cb212-230"><a href="#cb212-230" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(tmp_tab2<span class="sc">$</span>SampleID, tmp_tab4<span class="sc">$</span>SampleID)</span>
<span id="cb212-231"><a href="#cb212-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-232"><a href="#cb212-232" aria-hidden="true" tabindex="-1"></a>tmp_tab3 <span class="ot">&lt;-</span> tmp_tab3[<span class="fu">order</span>(tmp_tab3<span class="sc">$</span>SampleID),]</span>
<span id="cb212-233"><a href="#cb212-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-234"><a href="#cb212-234" aria-hidden="true" tabindex="-1"></a>tmp_tab5 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_tab4, tmp_tab3, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb212-235"><a href="#cb212-235" aria-hidden="true" tabindex="-1"></a>tmp_tab5<span class="sc">$</span>per_reads_kept <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_tab5<span class="sc">$</span>nochim<span class="sc">/</span>tmp_tab5<span class="sc">$</span>input, <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb212-236"><a href="#cb212-236" aria-hidden="true" tabindex="-1"></a>samptab <span class="ot">&lt;-</span> tmp_tab5</span>
<span id="cb212-237"><a href="#cb212-237" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-238"><a href="#cb212-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-241"><a href="#cb212-241" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-242"><a href="#cb212-242" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-243"><a href="#cb212-243" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-244"><a href="#cb212-244" aria-hidden="true" tabindex="-1"></a>samptab <span class="ot">&lt;-</span> samptab <span class="sc">%&gt;%</span></span>
<span id="cb212-245"><a href="#cb212-245" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">relocate</span>(</span>
<span id="cb212-246"><a href="#cb212-246" aria-hidden="true" tabindex="-1"></a>        <span class="fu">c</span>(input, nochim, per_reads_kept, no_asvs),</span>
<span id="cb212-247"><a href="#cb212-247" aria-hidden="true" tabindex="-1"></a>        <span class="at">.after =</span> <span class="st">"SampleID"</span></span>
<span id="cb212-248"><a href="#cb212-248" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb212-249"><a href="#cb212-249" aria-hidden="true" tabindex="-1"></a>samptab <span class="ot">&lt;-</span> samptab <span class="sc">%&gt;%</span></span>
<span id="cb212-250"><a href="#cb212-250" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">relocate</span>(per_reads_kept, <span class="at">.after =</span> <span class="st">"nochim"</span>)</span>
<span id="cb212-251"><a href="#cb212-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-252"><a href="#cb212-252" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>(<span class="at">pattern =</span> <span class="st">"tmp_"</span>))</span>
<span id="cb212-253"><a href="#cb212-253" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-254"><a href="#cb212-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-255"><a href="#cb212-255" aria-hidden="true" tabindex="-1"></a><span class="fu">## Prep Data for `microeco`</span></span>
<span id="cb212-256"><a href="#cb212-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-257"><a href="#cb212-257" aria-hidden="true" tabindex="-1"></a>Like any tool, the microeco package needs the data in a specific form. I formatted our data to match the mock data in the microeco tutorial, specifically <span class="co">[</span><span class="ot">this section</span><span class="co">](https://chiliubio.github.io/microeco_tutorial/basic-class.html#prepare-the-example-data)</span>.</span>
<span id="cb212-258"><a href="#cb212-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-259"><a href="#cb212-259" aria-hidden="true" tabindex="-1"></a><span class="fu">### A. Taxonomy Table</span></span>
<span id="cb212-260"><a href="#cb212-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-261"><a href="#cb212-261" aria-hidden="true" tabindex="-1"></a>Here is what the taxonomy table looks like in the mock data.</span>
<span id="cb212-262"><a href="#cb212-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-265"><a href="#cb212-265" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-266"><a href="#cb212-266" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-267"><a href="#cb212-267" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-268"><a href="#cb212-268" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(taxonomy_table_16S[<span class="dv">10</span><span class="sc">:</span><span class="dv">12</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-269"><a href="#cb212-269" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-270"><a href="#cb212-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-271"><a href="#cb212-271" aria-hidden="true" tabindex="-1"></a>And here is the taxonomy table from the dada2 workflow. The name dada2 gives to each ASV is the actual sequence of the ASV. We need to change this for downstream analyses.</span>
<span id="cb212-272"><a href="#cb212-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-273"><a href="#cb212-273" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple" collapse="true"}</span>
<span id="cb212-274"><a href="#cb212-274" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-275"><a href="#cb212-275" aria-hidden="true" tabindex="-1"></a><span class="fu">### Expand to see the partial content of the dada2 taxonomy table</span></span>
<span id="cb212-276"><a href="#cb212-276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-279"><a href="#cb212-279" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-280"><a href="#cb212-280" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-281"><a href="#cb212-281" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-282"><a href="#cb212-282" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(<span class="fu">data.frame</span>(tax_gsrdb)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-283"><a href="#cb212-283" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-284"><a href="#cb212-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-285"><a href="#cb212-285" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb212-286"><a href="#cb212-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-287"><a href="#cb212-287" aria-hidden="true" tabindex="-1"></a>The first step is to rename the amplicon sequence variants so the designations are a bit more user friendly. By default, DADA2 names each ASV by its unique sequence so that data can be directly compared across studies (which is great). But this convention can get cumbersome downstream, so we rename the ASVs using a simpler convention---ASV1, ASV2, ASV3, and so on.</span>
<span id="cb212-288"><a href="#cb212-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-291"><a href="#cb212-291" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-292"><a href="#cb212-292" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-293"><a href="#cb212-293" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-294"><a href="#cb212-294" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(tax_gsrdb)</span>
<span id="cb212-295"><a href="#cb212-295" aria-hidden="true" tabindex="-1"></a><span class="co"># adding unique ASV names</span></span>
<span id="cb212-296"><a href="#cb212-296" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(tmp_tax) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV"</span>, <span class="fu">seq</span>(<span class="fu">nrow</span>(tmp_tax)))</span>
<span id="cb212-297"><a href="#cb212-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-298"><a href="#cb212-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-299"><a href="#cb212-299" aria-hidden="true" tabindex="-1"></a>And this is how the taxonomy table looks after assigning new names.</span>
<span id="cb212-300"><a href="#cb212-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-303"><a href="#cb212-303" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-304"><a href="#cb212-304" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-305"><a href="#cb212-305" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-306"><a href="#cb212-306" aria-hidden="true" tabindex="-1"></a>tax.head1 <span class="ot">&lt;-</span> tmp_tax[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb212-307"><a href="#cb212-307" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-308"><a href="#cb212-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-311"><a href="#cb212-311" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-312"><a href="#cb212-312" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-313"><a href="#cb212-313" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-314"><a href="#cb212-314" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(tax.head1, <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-315"><a href="#cb212-315" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-316"><a href="#cb212-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-317"><a href="#cb212-317" aria-hidden="true" tabindex="-1"></a>Next we need to add rank definitions to each classification.</span>
<span id="cb212-318"><a href="#cb212-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-321"><a href="#cb212-321" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-322"><a href="#cb212-322" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-323"><a href="#cb212-323" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-324"><a href="#cb212-324" aria-hidden="true" tabindex="-1"></a>rank_prefixes <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb212-325"><a href="#cb212-325" aria-hidden="true" tabindex="-1"></a>  <span class="at">Kingdom =</span> <span class="st">"k"</span>, </span>
<span id="cb212-326"><a href="#cb212-326" aria-hidden="true" tabindex="-1"></a>  <span class="at">Phylum  =</span> <span class="st">"p"</span>, </span>
<span id="cb212-327"><a href="#cb212-327" aria-hidden="true" tabindex="-1"></a>  <span class="at">Class   =</span> <span class="st">"c"</span>, </span>
<span id="cb212-328"><a href="#cb212-328" aria-hidden="true" tabindex="-1"></a>  <span class="at">Order   =</span> <span class="st">"o"</span>, </span>
<span id="cb212-329"><a href="#cb212-329" aria-hidden="true" tabindex="-1"></a>  <span class="at">Family  =</span> <span class="st">"f"</span>, </span>
<span id="cb212-330"><a href="#cb212-330" aria-hidden="true" tabindex="-1"></a>  <span class="at">Genus   =</span> <span class="st">"g"</span></span>
<span id="cb212-331"><a href="#cb212-331" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-332"><a href="#cb212-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-333"><a href="#cb212-333" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span></span>
<span id="cb212-334"><a href="#cb212-334" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="st">""</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb212-335"><a href="#cb212-335" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">names</span>(rank_prefixes), </span>
<span id="cb212-336"><a href="#cb212-336" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span> <span class="fu">paste0</span>(rank_prefixes[<span class="fu">cur_column</span>()], <span class="st">"__"</span>, .))) <span class="sc">%&gt;%</span></span>
<span id="cb212-337"><a href="#cb212-337" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy_taxonomy</span>()</span>
<span id="cb212-338"><a href="#cb212-338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-339"><a href="#cb212-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-340"><a href="#cb212-340" aria-hidden="true" tabindex="-1"></a>And then this. Exactly like the mock data. </span>
<span id="cb212-341"><a href="#cb212-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-344"><a href="#cb212-344" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-345"><a href="#cb212-345" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-346"><a href="#cb212-346" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-347"><a href="#cb212-347" aria-hidden="true" tabindex="-1"></a>tax.head2 <span class="ot">&lt;-</span> tmp_tax[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb212-348"><a href="#cb212-348" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-349"><a href="#cb212-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-352"><a href="#cb212-352" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-353"><a href="#cb212-353" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-354"><a href="#cb212-354" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-355"><a href="#cb212-355" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(tax.head2, <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-356"><a href="#cb212-356" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-357"><a href="#cb212-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-358"><a href="#cb212-358" aria-hidden="true" tabindex="-1"></a><span class="fu">### B. Sequence Table</span></span>
<span id="cb212-359"><a href="#cb212-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-360"><a href="#cb212-360" aria-hidden="true" tabindex="-1"></a>Here is what the sequence table looks like in the mock data.</span>
<span id="cb212-361"><a href="#cb212-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-364"><a href="#cb212-364" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-365"><a href="#cb212-365" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-366"><a href="#cb212-366" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-367"><a href="#cb212-367" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(otu_table_16S[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-368"><a href="#cb212-368" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-369"><a href="#cb212-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-372"><a href="#cb212-372" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-373"><a href="#cb212-373" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-374"><a href="#cb212-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-375"><a href="#cb212-375" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(seqtab)</span>
<span id="cb212-376"><a href="#cb212-376" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">colnames</span>(tmp_st), <span class="fu">row.names</span>(tax_gsrdb))</span>
<span id="cb212-377"><a href="#cb212-377" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tmp_st) <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tmp_tax)</span>
<span id="cb212-378"><a href="#cb212-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-379"><a href="#cb212-379" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span>  tmp_st <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>()</span>
<span id="cb212-380"><a href="#cb212-380" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-381"><a href="#cb212-381" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> tmp_st <span class="sc">%&gt;%</span></span>
<span id="cb212-382"><a href="#cb212-382" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>), <span class="at">names_to =</span> <span class="st">"tmp"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-383"><a href="#cb212-383" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="fu">c</span>(<span class="dv">1</span>))</span>
<span id="cb212-384"><a href="#cb212-384" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(tmp_st, <span class="st">"tmp"</span>)</span>
<span id="cb212-385"><a href="#cb212-385" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-386"><a href="#cb212-386" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-389"><a href="#cb212-389" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-390"><a href="#cb212-390" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-391"><a href="#cb212-391" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-392"><a href="#cb212-392" aria-hidden="true" tabindex="-1"></a>st.head <span class="ot">&lt;-</span> tmp_st[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">61</span><span class="sc">:</span><span class="dv">63</span>]</span>
<span id="cb212-393"><a href="#cb212-393" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-394"><a href="#cb212-394" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-395"><a href="#cb212-395" aria-hidden="true" tabindex="-1"></a>And now the final, modified sequence table.</span>
<span id="cb212-396"><a href="#cb212-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-399"><a href="#cb212-399" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-400"><a href="#cb212-400" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-401"><a href="#cb212-401" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-402"><a href="#cb212-402" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(st.head, <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-403"><a href="#cb212-403" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-404"><a href="#cb212-404" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-405"><a href="#cb212-405" aria-hidden="true" tabindex="-1"></a><span class="fu">### C. Sample Table</span></span>
<span id="cb212-406"><a href="#cb212-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-407"><a href="#cb212-407" aria-hidden="true" tabindex="-1"></a>Here is what the sample table looks like in the mock data.</span>
<span id="cb212-408"><a href="#cb212-408" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-411"><a href="#cb212-411" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-412"><a href="#cb212-412" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-413"><a href="#cb212-413" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-414"><a href="#cb212-414" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(sample_info_16S[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-415"><a href="#cb212-415" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-416"><a href="#cb212-416" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-419"><a href="#cb212-419" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-420"><a href="#cb212-420" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-421"><a href="#cb212-421" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-422"><a href="#cb212-422" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"share/ssu/sampledata/"</span>, <span class="st">"sample_data.rds"</span>))</span>
<span id="cb212-423"><a href="#cb212-423" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb212-424"><a href="#cb212-424" aria-hidden="true" tabindex="-1"></a>samdf<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(samdf)</span>
<span id="cb212-425"><a href="#cb212-425" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> <span class="fu">relocate</span>(SampleID)</span>
<span id="cb212-426"><a href="#cb212-426" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-427"><a href="#cb212-427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-428"><a href="#cb212-428" aria-hidden="true" tabindex="-1"></a>And now a partial view of the final, modified sample table.</span>
<span id="cb212-429"><a href="#cb212-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-432"><a href="#cb212-432" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-433"><a href="#cb212-433" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-434"><a href="#cb212-434" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-435"><a href="#cb212-435" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(samdf[<span class="dv">61</span><span class="sc">:</span><span class="dv">63</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-436"><a href="#cb212-436" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-437"><a href="#cb212-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-438"><a href="#cb212-438" aria-hidden="true" tabindex="-1"></a><span class="fu">## Experiment-level Objects</span></span>
<span id="cb212-439"><a href="#cb212-439" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-440"><a href="#cb212-440" aria-hidden="true" tabindex="-1"></a>In the following section we create <span class="in">`microtable`</span> and <span class="in">`phyloseq`</span> objects. These are single, consolidated experiment-level data objects that organize various data types, such as OTU tables (taxa abundance), sample metadata, taxonomic classifications, as well as phylogenetic trees and reference sequences, into one structured format. This unified approach facilitates reproducible, interactive analysis and visualization by bringing all related information together. </span>
<span id="cb212-441"><a href="#cb212-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-442"><a href="#cb212-442" aria-hidden="true" tabindex="-1"></a>We begin by creating the <span class="in">`microtable`</span> and then use the function <span class="in">`meco2phyloseq`</span> from the <span class="co">[</span><span class="ot">file2meco</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span> package to create the <span class="in">`phyloseq`</span> object. This way all of the underlying data is identical across the two objects. </span>
<span id="cb212-443"><a href="#cb212-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-444"><a href="#cb212-444" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb212-445"><a href="#cb212-445" aria-hidden="true" tabindex="-1"></a>These objects contain an ASV table (taxa abundances), sample metadata, and taxonomy table (mapping between ASVs and higher-level taxonomic classifications).</span>
<span id="cb212-446"><a href="#cb212-446" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb212-447"><a href="#cb212-447" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-450"><a href="#cb212-450" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-451"><a href="#cb212-451" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-452"><a href="#cb212-452" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-453"><a href="#cb212-453" aria-hidden="true" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> samdf</span>
<span id="cb212-454"><a href="#cb212-454" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> tmp_tax</span>
<span id="cb212-455"><a href="#cb212-455" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> tmp_st</span>
<span id="cb212-456"><a href="#cb212-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-457"><a href="#cb212-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-460"><a href="#cb212-460" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-461"><a href="#cb212-461" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-462"><a href="#cb212-462" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-463"><a href="#cb212-463" aria-hidden="true" tabindex="-1"></a>tmp_me <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> sample_info, </span>
<span id="cb212-464"><a href="#cb212-464" aria-hidden="true" tabindex="-1"></a>                         <span class="at">otu_table =</span> otu_tab, </span>
<span id="cb212-465"><a href="#cb212-465" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tax_table =</span> tax_tab)</span>
<span id="cb212-466"><a href="#cb212-466" aria-hidden="true" tabindex="-1"></a>tmp_me</span>
<span id="cb212-467"><a href="#cb212-467" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-468"><a href="#cb212-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-469"><a href="#cb212-469" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb212-470"><a href="#cb212-470" aria-hidden="true" tabindex="-1"></a><span class="in">microtable-class object:</span></span>
<span id="cb212-471"><a href="#cb212-471" aria-hidden="true" tabindex="-1"></a><span class="in">sample_table have 1909 rows and 13 columns</span></span>
<span id="cb212-472"><a href="#cb212-472" aria-hidden="true" tabindex="-1"></a><span class="in">otu_table have 72851 rows and 1909 columns</span></span>
<span id="cb212-473"><a href="#cb212-473" aria-hidden="true" tabindex="-1"></a><span class="in">tax_table have 72851 rows and 6 columns</span></span>
<span id="cb212-474"><a href="#cb212-474" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-475"><a href="#cb212-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-476"><a href="#cb212-476" aria-hidden="true" tabindex="-1"></a><span class="fu">### Add Representative Sequence</span></span>
<span id="cb212-477"><a href="#cb212-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-478"><a href="#cb212-478" aria-hidden="true" tabindex="-1"></a>We can also add representative sequences for each OTU/ASV. For this step, we can simply grab the sequences from the row names of the DADA2 taxonomy object loaded above.</span>
<span id="cb212-479"><a href="#cb212-479" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-482"><a href="#cb212-482" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-483"><a href="#cb212-483" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-484"><a href="#cb212-484" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-485"><a href="#cb212-485" aria-hidden="true" tabindex="-1"></a>tmp_seq <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">row.names</span>(<span class="fu">data.frame</span>(tax_gsrdb)) )</span>
<span id="cb212-486"><a href="#cb212-486" aria-hidden="true" tabindex="-1"></a>tmp_names <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">row.names</span>(tax_tab))</span>
<span id="cb212-487"><a href="#cb212-487" aria-hidden="true" tabindex="-1"></a>tmp_fa <span class="ot">&lt;-</span> <span class="fu">cbind</span>(tmp_names, tmp_seq)</span>
<span id="cb212-488"><a href="#cb212-488" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(tmp_fa) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"ASV_ID"</span>, <span class="st">"ASV_SEQ"</span>)</span>
<span id="cb212-489"><a href="#cb212-489" aria-hidden="true" tabindex="-1"></a>tmp_fa<span class="sc">$</span>ASV_ID <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^"</span>, <span class="st">"&gt;"</span>, tmp_fa<span class="sc">$</span>ASV_ID)</span>
<span id="cb212-490"><a href="#cb212-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-491"><a href="#cb212-491" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp_fa, <span class="fu">here</span>(work_here, <span class="st">"rep_seq.fasta"</span>),</span>
<span id="cb212-492"><a href="#cb212-492" aria-hidden="true" tabindex="-1"></a>            <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="at">col.names =</span> <span class="cn">FALSE</span>, <span class="at">row.names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb212-493"><a href="#cb212-493" aria-hidden="true" tabindex="-1"></a>            <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">fileEncoding =</span> <span class="st">"UTF-8"</span>)       </span>
<span id="cb212-494"><a href="#cb212-494" aria-hidden="true" tabindex="-1"></a>rep_fasta <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(<span class="fu">here</span>(work_here, <span class="st">"rep_seq.fasta"</span>))</span>
<span id="cb212-495"><a href="#cb212-495" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span>rep_fasta <span class="ot">&lt;-</span> rep_fasta</span>
<span id="cb212-496"><a href="#cb212-496" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-497"><a href="#cb212-497" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-498"><a href="#cb212-498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-501"><a href="#cb212-501" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-502"><a href="#cb212-502" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-503"><a href="#cb212-503" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-504"><a href="#cb212-504" aria-hidden="true" tabindex="-1"></a>me_raw <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_me)</span>
<span id="cb212-505"><a href="#cb212-505" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-506"><a href="#cb212-506" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-507"><a href="#cb212-507" aria-hidden="true" tabindex="-1"></a><span class="fu">## Curate the Data Set</span></span>
<span id="cb212-508"><a href="#cb212-508" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-509"><a href="#cb212-509" aria-hidden="true" tabindex="-1"></a>Pretty much the last thing to do is remove unwanted taxa, negative controls, and low-count samples.</span>
<span id="cb212-510"><a href="#cb212-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-511"><a href="#cb212-511" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove any Kingdom NAs</span></span>
<span id="cb212-512"><a href="#cb212-512" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-513"><a href="#cb212-513" aria-hidden="true" tabindex="-1"></a>Here we can just use the straight up <span class="in">`subset`</span> command since we do not need to worry about any ranks above Kingdom also being removed.</span>
<span id="cb212-514"><a href="#cb212-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-517"><a href="#cb212-517" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-518"><a href="#cb212-518" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-519"><a href="#cb212-519" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-520"><a href="#cb212-520" aria-hidden="true" tabindex="-1"></a>tmp_no_na <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_me)</span>
<span id="cb212-521"><a href="#cb212-521" aria-hidden="true" tabindex="-1"></a>tmp_no_na<span class="sc">$</span>tax_table <span class="sc">%&lt;&gt;%</span> base<span class="sc">::</span><span class="fu">subset</span>(Kingdom <span class="sc">==</span> <span class="st">"k__Archaea"</span> <span class="sc">|</span> Kingdom <span class="sc">==</span> <span class="st">"k__Bacteria"</span>)</span>
<span id="cb212-522"><a href="#cb212-522" aria-hidden="true" tabindex="-1"></a>tmp_no_na<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-523"><a href="#cb212-523" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-524"><a href="#cb212-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-527"><a href="#cb212-527" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-528"><a href="#cb212-528" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-529"><a href="#cb212-529" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-530"><a href="#cb212-530" aria-hidden="true" tabindex="-1"></a>me_no_na <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb212-531"><a href="#cb212-531" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-532"><a href="#cb212-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-535"><a href="#cb212-535" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-536"><a href="#cb212-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-537"><a href="#cb212-537" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-538"><a href="#cb212-538" aria-hidden="true" tabindex="-1"></a>me_no_na</span>
<span id="cb212-539"><a href="#cb212-539" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-540"><a href="#cb212-540" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-541"><a href="#cb212-541" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Contaminants</span></span>
<span id="cb212-542"><a href="#cb212-542" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-543"><a href="#cb212-543" aria-hidden="true" tabindex="-1"></a>Now we can remove any potential contaminants like mitochondria or chloroplasts.</span>
<span id="cb212-544"><a href="#cb212-544" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-547"><a href="#cb212-547" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-548"><a href="#cb212-548" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-549"><a href="#cb212-549" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-550"><a href="#cb212-550" aria-hidden="true" tabindex="-1"></a>tmp_no_cont <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_na)</span>
<span id="cb212-551"><a href="#cb212-551" aria-hidden="true" tabindex="-1"></a>tmp_no_cont<span class="sc">$</span><span class="fu">filter_pollution</span>(<span class="at">taxa =</span> <span class="fu">c</span>(<span class="st">"mitochondria"</span>, <span class="st">"chloroplast"</span>))</span>
<span id="cb212-552"><a href="#cb212-552" aria-hidden="true" tabindex="-1"></a>tmp_no_cont<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-553"><a href="#cb212-553" aria-hidden="true" tabindex="-1"></a>tmp_no_cont</span>
<span id="cb212-554"><a href="#cb212-554" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-555"><a href="#cb212-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-556"><a href="#cb212-556" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb212-557"><a href="#cb212-557" aria-hidden="true" tabindex="-1"></a><span class="in">Total 0 features are removed from tax_table ...</span></span>
<span id="cb212-558"><a href="#cb212-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-559"><a href="#cb212-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-562"><a href="#cb212-562" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-563"><a href="#cb212-563" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-564"><a href="#cb212-564" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-565"><a href="#cb212-565" aria-hidden="true" tabindex="-1"></a>me_no_cont <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_cont)</span>
<span id="cb212-566"><a href="#cb212-566" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-567"><a href="#cb212-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-570"><a href="#cb212-570" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-571"><a href="#cb212-571" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-572"><a href="#cb212-572" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-573"><a href="#cb212-573" aria-hidden="true" tabindex="-1"></a>me_no_cont</span>
<span id="cb212-574"><a href="#cb212-574" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-575"><a href="#cb212-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-576"><a href="#cb212-576" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Negative Controls (NC)</span></span>
<span id="cb212-577"><a href="#cb212-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-578"><a href="#cb212-578" aria-hidden="true" tabindex="-1"></a>Now we need to remove the NC samples *and* ASVs found in those sample. We first identified all ASVs that were present in at least one NC sample represented by at least 1 read. We did this by subsetting the NC samples from the new microtable object.</span>
<span id="cb212-579"><a href="#cb212-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-582"><a href="#cb212-582" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-583"><a href="#cb212-583" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-584"><a href="#cb212-584" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-585"><a href="#cb212-585" aria-hidden="true" tabindex="-1"></a>tmp_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_cont)</span>
<span id="cb212-586"><a href="#cb212-586" aria-hidden="true" tabindex="-1"></a>tmp_nc<span class="sc">$</span>sample_table <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_nc<span class="sc">$</span>sample_table, TAXON <span class="sc">==</span> <span class="st">"Control"</span>)</span>
<span id="cb212-587"><a href="#cb212-587" aria-hidden="true" tabindex="-1"></a>tmp_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-588"><a href="#cb212-588" aria-hidden="true" tabindex="-1"></a>tmp_nc</span>
<span id="cb212-589"><a href="#cb212-589" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-590"><a href="#cb212-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-591"><a href="#cb212-591" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-592"><a href="#cb212-592" aria-hidden="true" tabindex="-1"></a><span class="in">72231 taxa with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb212-593"><a href="#cb212-593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-594"><a href="#cb212-594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-597"><a href="#cb212-597" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-598"><a href="#cb212-598" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-599"><a href="#cb212-599" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-600"><a href="#cb212-600" aria-hidden="true" tabindex="-1"></a>me_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_nc)</span>
<span id="cb212-601"><a href="#cb212-601" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-602"><a href="#cb212-602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-605"><a href="#cb212-605" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-606"><a href="#cb212-606" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-607"><a href="#cb212-607" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-608"><a href="#cb212-608" aria-hidden="true" tabindex="-1"></a>me_nc</span>
<span id="cb212-609"><a href="#cb212-609" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-610"><a href="#cb212-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-611"><a href="#cb212-611" aria-hidden="true" tabindex="-1"></a>Looks like there are <span class="in">`r nrow(me_nc$tax_table)`</span> ASVs in the NC samples from a total of <span class="in">`r sum(me_nc$taxa_sums())`</span> reads.</span>
<span id="cb212-612"><a href="#cb212-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-615"><a href="#cb212-615" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-616"><a href="#cb212-616" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-617"><a href="#cb212-617" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-618"><a href="#cb212-618" aria-hidden="true" tabindex="-1"></a>nc_asvs <span class="ot">&lt;-</span> <span class="fu">row.names</span>(tmp_nc<span class="sc">$</span>tax_table)</span>
<span id="cb212-619"><a href="#cb212-619" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-620"><a href="#cb212-620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-623"><a href="#cb212-623" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-624"><a href="#cb212-624" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-625"><a href="#cb212-625" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-626"><a href="#cb212-626" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(nc_asvs, <span class="at">n =</span> <span class="dv">20</span>)</span>
<span id="cb212-627"><a href="#cb212-627" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-628"><a href="#cb212-628" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-629"><a href="#cb212-629" aria-hidden="true" tabindex="-1"></a>There are <span class="in">`r length(nc_asvs)`</span> ASVs found in the NC sample. ASVs are numbered in order by total abundance in the data set so we know that many of the ASVs in the NC samples are not particularly abundant in the dataset. We can look at the abundance of these ASVs across all samples and compare it to the NC. This takes a bit of wrangling.</span>
<span id="cb212-630"><a href="#cb212-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-631"><a href="#cb212-631" aria-hidden="true" tabindex="-1"></a>Essentially, for each ASV, the code below calculates:</span>
<span id="cb212-632"><a href="#cb212-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-633"><a href="#cb212-633" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of NC samples containing at least 1 read.\</span>
<span id="cb212-634"><a href="#cb212-634" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of reads in NC samples.\</span>
<span id="cb212-635"><a href="#cb212-635" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of non-NC samples containing at least 1 read.\</span>
<span id="cb212-636"><a href="#cb212-636" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The total number of reads in non-NC samples.\</span>
<span id="cb212-637"><a href="#cb212-637" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The percent of reads in the NC samples and the percent of NC samples containing reads.</span>
<span id="cb212-638"><a href="#cb212-638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-641"><a href="#cb212-641" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-642"><a href="#cb212-642" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-643"><a href="#cb212-643" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-644"><a href="#cb212-644" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_cont)</span>
<span id="cb212-645"><a href="#cb212-645" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc<span class="sc">$</span>otu_table</span>
<span id="cb212-646"><a href="#cb212-646" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span> </span>
<span id="cb212-647"><a href="#cb212-647" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">row.names</span>(tmp_rem_nc_df) <span class="sc">%in%</span> nc_asvs)</span>
<span id="cb212-648"><a href="#cb212-648" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb212-649"><a href="#cb212-649" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-650"><a href="#cb212-650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-653"><a href="#cb212-653" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-654"><a href="#cb212-654" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-655"><a href="#cb212-655" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-656"><a href="#cb212-656" aria-hidden="true" tabindex="-1"></a><span class="co">#-------provide a string unique to NC samples--------------#</span></span>
<span id="cb212-657"><a href="#cb212-657" aria-hidden="true" tabindex="-1"></a>nc_name <span class="ot">&lt;-</span> <span class="st">"Control_"</span></span>
<span id="cb212-658"><a href="#cb212-658" aria-hidden="true" tabindex="-1"></a><span class="co">#----------------------------------------------------------#</span></span>
<span id="cb212-659"><a href="#cb212-659" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df  <span class="sc">%&gt;%</span> </span>
<span id="cb212-660"><a href="#cb212-660" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">total_reads_NC =</span> <span class="fu">rowSums</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(., <span class="fu">contains</span>(nc_name))), </span>
<span id="cb212-661"><a href="#cb212-661" aria-hidden="true" tabindex="-1"></a>         <span class="at">.after =</span> <span class="st">"ASV_ID"</span>)</span>
<span id="cb212-662"><a href="#cb212-662" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc_df, <span class="sc">-</span><span class="fu">contains</span>(nc_name))</span>
<span id="cb212-663"><a href="#cb212-663" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb212-664"><a href="#cb212-664" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">total_reads_samps =</span> <span class="fu">rowSums</span>(.[<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_rem_nc_df)]), </span>
<span id="cb212-665"><a href="#cb212-665" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">"total_reads_NC"</span>)</span>
<span id="cb212-666"><a href="#cb212-666" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df[, <span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(tmp_rem_nc_df)] <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="cn">NULL</span>)</span>
<span id="cb212-667"><a href="#cb212-667" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb212-668"><a href="#cb212-668" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">perc_in_neg =</span> <span class="dv">100</span><span class="sc">*</span>(</span>
<span id="cb212-669"><a href="#cb212-669" aria-hidden="true" tabindex="-1"></a>    total_reads_NC <span class="sc">/</span> (total_reads_NC <span class="sc">+</span> total_reads_samps)),</span>
<span id="cb212-670"><a href="#cb212-670" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">"total_reads_samps"</span>)</span>
<span id="cb212-671"><a href="#cb212-671" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-672"><a href="#cb212-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-675"><a href="#cb212-675" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-676"><a href="#cb212-676" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-677"><a href="#cb212-677" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-678"><a href="#cb212-678" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df<span class="sc">$</span>perc_in_neg <span class="ot">&lt;-</span> <span class="fu">round</span>(tmp_rem_nc_df<span class="sc">$</span>perc_in_neg, <span class="at">digits =</span> <span class="dv">6</span>)</span>
<span id="cb212-679"><a href="#cb212-679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-680"><a href="#cb212-680" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rowSums</span>(tmp_rem_nc<span class="sc">$</span>otu_table <span class="sc">!=</span> <span class="dv">0</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-681"><a href="#cb212-681" aria-hidden="true" tabindex="-1"></a>                   tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb212-682"><a href="#cb212-682" aria-hidden="true" tabindex="-1"></a>                   dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"total_samples"</span> <span class="ot">=</span> <span class="dv">2</span>) </span>
<span id="cb212-683"><a href="#cb212-683" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-684"><a href="#cb212-684" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc<span class="sc">$</span>otu_table, <span class="fu">contains</span>(nc_name))</span>
<span id="cb212-685"><a href="#cb212-685" aria-hidden="true" tabindex="-1"></a>tmp_2<span class="sc">$</span>num_samp_nc <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp_2 <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb212-686"><a href="#cb212-686" aria-hidden="true" tabindex="-1"></a>tmp_2 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_2, <span class="fu">contains</span>(<span class="st">"num_samp_nc"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-687"><a href="#cb212-687" aria-hidden="true" tabindex="-1"></a>                      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb212-688"><a href="#cb212-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-689"><a href="#cb212-689" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_rem_nc<span class="sc">$</span>otu_table, <span class="sc">-</span><span class="fu">contains</span>(nc_name))</span>
<span id="cb212-690"><a href="#cb212-690" aria-hidden="true" tabindex="-1"></a>tmp_3<span class="sc">$</span>num_samp_no_nc <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(tmp_3 <span class="sc">!=</span> <span class="dv">0</span>)</span>
<span id="cb212-691"><a href="#cb212-691" aria-hidden="true" tabindex="-1"></a>tmp_3 <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(tmp_3, <span class="fu">contains</span>(<span class="st">"num_samp_no_nc"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb212-692"><a href="#cb212-692" aria-hidden="true" tabindex="-1"></a>                      tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb212-693"><a href="#cb212-693" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-694"><a href="#cb212-694" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(tmp_rem_nc_df, tmp_1) <span class="sc">%&gt;%</span></span>
<span id="cb212-695"><a href="#cb212-695" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_2) <span class="sc">%&gt;%</span></span>
<span id="cb212-696"><a href="#cb212-696" aria-hidden="true" tabindex="-1"></a>                 dplyr<span class="sc">::</span><span class="fu">left_join</span>(., tmp_3)</span>
<span id="cb212-697"><a href="#cb212-697" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-698"><a href="#cb212-698" aria-hidden="true" tabindex="-1"></a>tmp_rem_nc_df <span class="ot">&lt;-</span> tmp_rem_nc_df <span class="sc">%&gt;%</span></span>
<span id="cb212-699"><a href="#cb212-699" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">perc_in_neg_samp =</span> <span class="dv">100</span><span class="sc">*</span>( num_samp_nc <span class="sc">/</span> (num_samp_nc <span class="sc">+</span> num_samp_no_nc)),</span>
<span id="cb212-700"><a href="#cb212-700" aria-hidden="true" tabindex="-1"></a>                <span class="at">.after =</span> <span class="st">"num_samp_no_nc"</span>)</span>
<span id="cb212-701"><a href="#cb212-701" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-702"><a href="#cb212-702" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-705"><a href="#cb212-705" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-706"><a href="#cb212-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-707"><a href="#cb212-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-708"><a href="#cb212-708" aria-hidden="true" tabindex="-1"></a>nc_check <span class="ot">&lt;-</span> tmp_rem_nc_df</span>
<span id="cb212-709"><a href="#cb212-709" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-710"><a href="#cb212-710" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-711"><a href="#cb212-711" aria-hidden="true" tabindex="-1"></a>Looking at these data we can see that ASVs like ASV1, ASV2, and so on, are only represented by a really small number of NC reads and samples. On the other hand, ASVs such as ASV91, ASV121, and ASV299 are relatively abundant in NC samples. We decided to remove ASVs if:</span>
<span id="cb212-712"><a href="#cb212-712" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-713"><a href="#cb212-713" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The number of reads found in NC samples accounted for more than 10% of total reads OR</span>
<span id="cb212-714"><a href="#cb212-714" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The percent of NC samples containing the ASV was greater than 10% of total samples.</span>
<span id="cb212-715"><a href="#cb212-715" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-718"><a href="#cb212-718" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-719"><a href="#cb212-719" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-720"><a href="#cb212-720" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-721"><a href="#cb212-721" aria-hidden="true" tabindex="-1"></a>nc_remove <span class="ot">&lt;-</span> nc_check <span class="sc">%&gt;%</span> </span>
<span id="cb212-722"><a href="#cb212-722" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(perc_in_neg <span class="sc">&gt;</span> <span class="dv">10</span> <span class="sc">|</span> perc_in_neg_samp <span class="sc">&gt;</span> <span class="dv">10</span>)</span>
<span id="cb212-723"><a href="#cb212-723" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-724"><a href="#cb212-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-727"><a href="#cb212-727" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-728"><a href="#cb212-728" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-729"><a href="#cb212-729" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-730"><a href="#cb212-730" aria-hidden="true" tabindex="-1"></a>nc_remain <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">anti_join</span>(nc_check, nc_remove)</span>
<span id="cb212-731"><a href="#cb212-731" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-732"><a href="#cb212-732" aria-hidden="true" tabindex="-1"></a>rem_nc_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remove<span class="sc">$</span>total_reads_NC)</span>
<span id="cb212-733"><a href="#cb212-733" aria-hidden="true" tabindex="-1"></a>rem_sam_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remove<span class="sc">$</span>total_reads_samps)</span>
<span id="cb212-734"><a href="#cb212-734" aria-hidden="true" tabindex="-1"></a>per_reads_rem <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>( rem_nc_reads <span class="sc">/</span> (rem_nc_reads <span class="sc">+</span> rem_sam_reads)), </span>
<span id="cb212-735"><a href="#cb212-735" aria-hidden="true" tabindex="-1"></a>                       <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb212-736"><a href="#cb212-736" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-737"><a href="#cb212-737" aria-hidden="true" tabindex="-1"></a>ret_nc_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remain<span class="sc">$</span>total_reads_NC)</span>
<span id="cb212-738"><a href="#cb212-738" aria-hidden="true" tabindex="-1"></a>ret_sam_reads <span class="ot">&lt;-</span> <span class="fu">sum</span>(nc_remain<span class="sc">$</span>total_reads_samps)</span>
<span id="cb212-739"><a href="#cb212-739" aria-hidden="true" tabindex="-1"></a>per_reads_ret <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>( ret_nc_reads <span class="sc">/</span> (ret_nc_reads <span class="sc">+</span> ret_sam_reads)), </span>
<span id="cb212-740"><a href="#cb212-740" aria-hidden="true" tabindex="-1"></a>                       <span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb212-741"><a href="#cb212-741" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-742"><a href="#cb212-742" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-743"><a href="#cb212-743" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span>          <span class="pp">|</span> Total ASVs          <span class="pp">|</span> NC reads         <span class="pp">|</span> non NC reads      <span class="pp">|</span> \% NC reads       <span class="pp">|</span></span>
<span id="cb212-744"><a href="#cb212-744" aria-hidden="true" tabindex="-1"></a><span class="pp">|----------|---------------------|------------------|-------------------|-------------------|</span></span>
<span id="cb212-745"><a href="#cb212-745" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Removed  <span class="pp">|</span> <span class="in">`r nrow(nc_remove)`</span> <span class="pp">|</span> <span class="in">`r rem_nc_reads`</span> <span class="pp">|</span> <span class="in">`r rem_sam_reads`</span> <span class="pp">|</span> <span class="in">`r per_reads_rem`</span> <span class="pp">|</span></span>
<span id="cb212-746"><a href="#cb212-746" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> Retained <span class="pp">|</span> <span class="in">`r nrow(nc_remain)`</span> <span class="pp">|</span> <span class="in">`r ret_nc_reads`</span> <span class="pp">|</span> <span class="in">`r ret_sam_reads`</span> <span class="pp">|</span> <span class="in">`r per_reads_ret`</span> <span class="pp">|</span></span>
<span id="cb212-747"><a href="#cb212-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-748"><a href="#cb212-748" aria-hidden="true" tabindex="-1"></a>We identified a total of <span class="in">`r nrow(nc_check)`</span> ASVs that were present in at least 1 NC sample by at least 1 read. We removed any ASV where more than 10% of total reads were found in NC samples OR any ASV found in more than 10% of NC samples. Based on these criteria we removed <span class="in">`r nrow(nc_remove)`</span> ASVs from the data set, which represented <span class="in">`r rem_nc_reads`</span> total reads in NC samples and <span class="in">`r rem_sam_reads`</span> total reads in non-NC samples. Of the total reads removed <span class="in">`r per_reads_rem`</span>% came from NC samples. Of all ASVs identified in NC samples,<span class="in">`r nrow(nc_remain)`</span> were retained because the fell below the threshhold criteria. These ASVs accounted for <span class="in">`r ret_nc_reads`</span> reads in NC samples and <span class="in">`r ret_sam_reads`</span> reads in non-NC samples. NC samples accounted for <span class="in">`r per_reads_ret`</span>% of these reads.</span>
<span id="cb212-749"><a href="#cb212-749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-750"><a href="#cb212-750" aria-hidden="true" tabindex="-1"></a>OK, now we can remove the NC samples and any ASVs that met our criteria described above.</span>
<span id="cb212-751"><a href="#cb212-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-752"><a href="#cb212-752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-755"><a href="#cb212-755" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-756"><a href="#cb212-756" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-757"><a href="#cb212-757" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-758"><a href="#cb212-758" aria-hidden="true" tabindex="-1"></a>tmp_no_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_cont)</span>
<span id="cb212-759"><a href="#cb212-759" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-760"><a href="#cb212-760" aria-hidden="true" tabindex="-1"></a>tmp_rem_asv <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(nc_remove<span class="sc">$</span>ASV_ID)</span>
<span id="cb212-761"><a href="#cb212-761" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span>otu_table <span class="ot">&lt;-</span> tmp_rem_nc<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span> </span>
<span id="cb212-762"><a href="#cb212-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">row.names</span>(tmp_no_nc<span class="sc">$</span>otu_table) <span class="sc">%in%</span> tmp_rem_asv)</span>
<span id="cb212-763"><a href="#cb212-763" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-764"><a href="#cb212-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-765"><a href="#cb212-765" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span>sample_table <span class="ot">&lt;-</span> <span class="fu">subset</span>(tmp_no_nc<span class="sc">$</span>sample_table, </span>
<span id="cb212-766"><a href="#cb212-766" aria-hidden="true" tabindex="-1"></a>                                 TAXON <span class="sc">!=</span> <span class="st">"Control_"</span>)</span>
<span id="cb212-767"><a href="#cb212-767" aria-hidden="true" tabindex="-1"></a>tmp_no_nc<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-768"><a href="#cb212-768" aria-hidden="true" tabindex="-1"></a>tmp_no_nc</span>
<span id="cb212-769"><a href="#cb212-769" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-770"><a href="#cb212-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-771"><a href="#cb212-771" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb212-772"><a href="#cb212-772" aria-hidden="true" tabindex="-1"></a><span class="in">9 samples with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb212-773"><a href="#cb212-773" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-774"><a href="#cb212-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-777"><a href="#cb212-777" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-778"><a href="#cb212-778" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-779"><a href="#cb212-779" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-780"><a href="#cb212-780" aria-hidden="true" tabindex="-1"></a>me_no_nc <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_nc)</span>
<span id="cb212-781"><a href="#cb212-781" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-782"><a href="#cb212-782" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-785"><a href="#cb212-785" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-786"><a href="#cb212-786" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-787"><a href="#cb212-787" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-788"><a href="#cb212-788" aria-hidden="true" tabindex="-1"></a>me_no_nc</span>
<span id="cb212-789"><a href="#cb212-789" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-790"><a href="#cb212-790" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-791"><a href="#cb212-791" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Low-Count Samples</span></span>
<span id="cb212-792"><a href="#cb212-792" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-793"><a href="#cb212-793" aria-hidden="true" tabindex="-1"></a>Next, we can remove samples with really low read counts---here we set the threshold to <span class="in">`1000`</span> reads.</span>
<span id="cb212-794"><a href="#cb212-794" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-797"><a href="#cb212-797" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-798"><a href="#cb212-798" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-799"><a href="#cb212-799" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-800"><a href="#cb212-800" aria-hidden="true" tabindex="-1"></a>tmp_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_nc)</span>
<span id="cb212-801"><a href="#cb212-801" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span>otu_table <span class="ot">&lt;-</span> tmp_no_nc<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span></span>
<span id="cb212-802"><a href="#cb212-802" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">is.numeric</span>(.) <span class="sc">&amp;&amp;</span> <span class="fu">sum</span>(.) <span class="sc">&gt;=</span> <span class="dv">1000</span>))</span>
<span id="cb212-803"><a href="#cb212-803" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb212-804"><a href="#cb212-804" aria-hidden="true" tabindex="-1"></a>tmp_no_low</span>
<span id="cb212-805"><a href="#cb212-805" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-806"><a href="#cb212-806" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-807"><a href="#cb212-807" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb212-808"><a href="#cb212-808" aria-hidden="true" tabindex="-1"></a><span class="in">26 taxa with 0 abundance are removed from the otu_table ...</span></span>
<span id="cb212-809"><a href="#cb212-809" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-810"><a href="#cb212-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-813"><a href="#cb212-813" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-814"><a href="#cb212-814" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-815"><a href="#cb212-815" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-816"><a href="#cb212-816" aria-hidden="true" tabindex="-1"></a>me_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_low)</span>
<span id="cb212-817"><a href="#cb212-817" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-818"><a href="#cb212-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-821"><a href="#cb212-821" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-822"><a href="#cb212-822" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-823"><a href="#cb212-823" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-824"><a href="#cb212-824" aria-hidden="true" tabindex="-1"></a>me_no_low</span>
<span id="cb212-825"><a href="#cb212-825" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-826"><a href="#cb212-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-827"><a href="#cb212-827" aria-hidden="true" tabindex="-1"></a>Giving us the final microtable object.</span>
<span id="cb212-828"><a href="#cb212-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-831"><a href="#cb212-831" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-832"><a href="#cb212-832" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-833"><a href="#cb212-833" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-834"><a href="#cb212-834" aria-hidden="true" tabindex="-1"></a>me_final <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_low)</span>
<span id="cb212-835"><a href="#cb212-835" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-836"><a href="#cb212-836" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-837"><a href="#cb212-837" aria-hidden="true" tabindex="-1"></a>Lastly, we can use the package <span class="co">[</span><span class="ot">`file2meco`</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span> to generate a <span class="co">[</span><span class="ot">phyloseq</span><span class="co">](https://joey711.github.io/phyloseq/)</span> object. </span>
<span id="cb212-838"><a href="#cb212-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-841"><a href="#cb212-841" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-842"><a href="#cb212-842" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb212-843"><a href="#cb212-843" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-844"><a href="#cb212-844" aria-hidden="true" tabindex="-1"></a>ps_final <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb212-845"><a href="#cb212-845" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-846"><a href="#cb212-846" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-847"><a href="#cb212-847" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb212-848"><a href="#cb212-848" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-849"><a href="#cb212-849" aria-hidden="true" tabindex="-1"></a>Now time to summarize the data. For this we use the R package <span class="co">[</span><span class="ot">miaverse</span><span class="co">](https://microbiome.github.io)</span> <span class="co">[</span><span class="ot">@felix2024mia</span><span class="co">]</span>.</span>
<span id="cb212-850"><a href="#cb212-850" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-853"><a href="#cb212-853" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-854"><a href="#cb212-854" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-855"><a href="#cb212-855" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-856"><a href="#cb212-856" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_raw<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_raw<span class="sc">$</span>otu_table))</span>
<span id="cb212-857"><a href="#cb212-857" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_final<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_final<span class="sc">$</span>otu_table))</span>
<span id="cb212-858"><a href="#cb212-858" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-859"><a href="#cb212-859" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-862"><a href="#cb212-862" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-863"><a href="#cb212-863" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-864"><a href="#cb212-864" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-865"><a href="#cb212-865" aria-hidden="true" tabindex="-1"></a>objs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"me_raw"</span>, <span class="st">"me_no_na"</span>, </span>
<span id="cb212-866"><a href="#cb212-866" aria-hidden="true" tabindex="-1"></a>          <span class="st">"me_no_cont"</span>, <span class="st">"me_no_nc"</span>, </span>
<span id="cb212-867"><a href="#cb212-867" aria-hidden="true" tabindex="-1"></a>          <span class="st">"me_no_low"</span>, <span class="st">"me_final"</span>)</span>
<span id="cb212-868"><a href="#cb212-868" aria-hidden="true" tabindex="-1"></a>pipe_summary <span class="ot">&lt;-</span> <span class="fu">summarize_objs</span>(objs)</span>
<span id="cb212-869"><a href="#cb212-869" aria-hidden="true" tabindex="-1"></a>pipe_summary<span class="sc">$</span>me_dataset <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb212-870"><a href="#cb212-870" aria-hidden="true" tabindex="-1"></a>  <span class="st">"original"</span>, <span class="st">"no NA kingdoms"</span>, </span>
<span id="cb212-871"><a href="#cb212-871" aria-hidden="true" tabindex="-1"></a>  <span class="st">"no contaminants"</span>, <span class="st">"no negative controls"</span>, </span>
<span id="cb212-872"><a href="#cb212-872" aria-hidden="true" tabindex="-1"></a>  <span class="st">"no low count samps"</span>, <span class="st">"final"</span>)</span>
<span id="cb212-873"><a href="#cb212-873" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pipe_summary)</span>
<span id="cb212-874"><a href="#cb212-874" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-875"><a href="#cb212-875" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-878"><a href="#cb212-878" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-879"><a href="#cb212-879" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-880"><a href="#cb212-880" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-881"><a href="#cb212-881" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pipe_summary)</span>
<span id="cb212-882"><a href="#cb212-882" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-883"><a href="#cb212-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-886"><a href="#cb212-886" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-887"><a href="#cb212-887" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-888"><a href="#cb212-888" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-889"><a href="#cb212-889" aria-hidden="true" tabindex="-1"></a>tmp_raw_ps <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_raw)</span>
<span id="cb212-890"><a href="#cb212-890" aria-hidden="true" tabindex="-1"></a>tmp_final_ps <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb212-891"><a href="#cb212-891" aria-hidden="true" tabindex="-1"></a>tmp_mia_raw <span class="ot">&lt;-</span> <span class="fu">mia_metrics</span>(tmp_raw_ps)</span>
<span id="cb212-892"><a href="#cb212-892" aria-hidden="true" tabindex="-1"></a>tmp_mia_final <span class="ot">&lt;-</span> <span class="fu">mia_metrics</span>(tmp_final_ps)</span>
<span id="cb212-893"><a href="#cb212-893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-894"><a href="#cb212-894" aria-hidden="true" tabindex="-1"></a>tmp_metrics_final <span class="ot">&lt;-</span> <span class="fu">rbind</span>(tmp_mia_raw, tmp_mia_final)</span>
<span id="cb212-895"><a href="#cb212-895" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_metrics_final) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Start"</span>, <span class="st">"End"</span>)</span>
<span id="cb212-896"><a href="#cb212-896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-897"><a href="#cb212-897" aria-hidden="true" tabindex="-1"></a><span class="co"># suppose ds_metrics_final has rownames "Start" and "End"</span></span>
<span id="cb212-898"><a href="#cb212-898" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl <span class="ot">&lt;-</span> tmp_metrics_final <span class="sc">%&gt;%</span></span>
<span id="cb212-899"><a href="#cb212-899" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span>                         <span class="co"># transpose: metrics in rows</span></span>
<span id="cb212-900"><a href="#cb212-900" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-901"><a href="#cb212-901" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Metric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-902"><a href="#cb212-902" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Start =</span> Start, <span class="at">End =</span> End)</span>
<span id="cb212-903"><a href="#cb212-903" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-904"><a href="#cb212-904" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl <span class="ot">&lt;-</span> ds_metrics_tbl <span class="sc">%&gt;%</span></span>
<span id="cb212-905"><a href="#cb212-905" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb212-906"><a href="#cb212-906" aria-hidden="true" tabindex="-1"></a>    <span class="at">Start =</span> <span class="fu">ifelse</span>(Start <span class="sc">%%</span> <span class="dv">1</span> <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb212-907"><a href="#cb212-907" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(Start, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">0</span>),</span>
<span id="cb212-908"><a href="#cb212-908" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(Start, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">3</span>)),</span>
<span id="cb212-909"><a href="#cb212-909" aria-hidden="true" tabindex="-1"></a>    <span class="at">End   =</span> <span class="fu">ifelse</span>(End <span class="sc">%%</span> <span class="dv">1</span> <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb212-910"><a href="#cb212-910" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(End, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">0</span>),</span>
<span id="cb212-911"><a href="#cb212-911" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(End, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">3</span>))</span>
<span id="cb212-912"><a href="#cb212-912" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb212-913"><a href="#cb212-913" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-914"><a href="#cb212-914" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl<span class="sc">$</span>Metric <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb212-915"><a href="#cb212-915" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Min. no. of reads"</span>,</span>
<span id="cb212-916"><a href="#cb212-916" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Max. no. of reads"</span>,</span>
<span id="cb212-917"><a href="#cb212-917" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Total no. of reads"</span>,</span>
<span id="cb212-918"><a href="#cb212-918" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Avg. no. of reads"</span>,</span>
<span id="cb212-919"><a href="#cb212-919" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Median no. of reads"</span>,</span>
<span id="cb212-920"><a href="#cb212-920" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Total ASVs"</span>,</span>
<span id="cb212-921"><a href="#cb212-921" aria-hidden="true" tabindex="-1"></a>  <span class="st">"No. of singleton ASVs"</span>,</span>
<span id="cb212-922"><a href="#cb212-922" aria-hidden="true" tabindex="-1"></a>  <span class="st">"% of singleton ASVs"</span>,</span>
<span id="cb212-923"><a href="#cb212-923" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Sparsity"</span></span>
<span id="cb212-924"><a href="#cb212-924" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-925"><a href="#cb212-925" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-926"><a href="#cb212-926" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-929"><a href="#cb212-929" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-930"><a href="#cb212-930" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-931"><a href="#cb212-931" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-932"><a href="#cb212-932" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Dataset metrics before and after curation."</span></span>
<span id="cb212-933"><a href="#cb212-933" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(ds_metrics_tbl, <span class="at">format =</span> <span class="st">"markdown"</span>)</span>
<span id="cb212-934"><a href="#cb212-934" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-935"><a href="#cb212-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-936"><a href="#cb212-936" aria-hidden="true" tabindex="-1"></a>We started off with <span class="in">`r nrow(me_raw$tax_table)`</span> ASVs and <span class="in">`r nrow(me_raw$sample_table)`</span> samples. After curation of the dataset, there were <span class="in">`r nrow(me_final$tax_table)`</span> ASVs and <span class="in">`r nrow(me_final$sample_table)`</span> samples remaining.</span>
<span id="cb212-937"><a href="#cb212-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-940"><a href="#cb212-940" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-941"><a href="#cb212-941" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-942"><a href="#cb212-942" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-943"><a href="#cb212-943" aria-hidden="true" tabindex="-1"></a><span class="co"># helper functions</span></span>
<span id="cb212-944"><a href="#cb212-944" aria-hidden="true" tabindex="-1"></a>make_tables <span class="ot">&lt;-</span> <span class="cf">function</span>(obj, prefix) {</span>
<span id="cb212-945"><a href="#cb212-945" aria-hidden="true" tabindex="-1"></a>  rc <span class="ot">&lt;-</span> obj<span class="sc">$</span><span class="fu">sample_sums</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-946"><a href="#cb212-946" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-947"><a href="#cb212-947" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-948"><a href="#cb212-948" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="sc">!!</span><span class="fu">paste0</span>(prefix, <span class="st">"_rc"</span>) <span class="sc">:</span><span class="er">=</span> <span class="dv">2</span>)</span>
<span id="cb212-949"><a href="#cb212-949" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-950"><a href="#cb212-950" aria-hidden="true" tabindex="-1"></a>  asv <span class="ot">&lt;-</span> obj<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span></span>
<span id="cb212-951"><a href="#cb212-951" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-952"><a href="#cb212-952" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-953"><a href="#cb212-953" aria-hidden="true" tabindex="-1"></a>    { <span class="fu">rowSums</span>(. <span class="sc">&gt;</span> <span class="dv">0</span>) } <span class="sc">%&gt;%</span></span>
<span id="cb212-954"><a href="#cb212-954" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb212-955"><a href="#cb212-955" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-956"><a href="#cb212-956" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="sc">!!</span><span class="fu">paste0</span>(prefix, <span class="st">"_asv"</span>) <span class="sc">:</span><span class="er">=</span> <span class="dv">2</span>)</span>
<span id="cb212-957"><a href="#cb212-957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-958"><a href="#cb212-958" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">rc =</span> rc, <span class="at">asv =</span> asv)</span>
<span id="cb212-959"><a href="#cb212-959" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb212-960"><a href="#cb212-960" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-961"><a href="#cb212-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-964"><a href="#cb212-964" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-965"><a href="#cb212-965" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-966"><a href="#cb212-966" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-967"><a href="#cb212-967" aria-hidden="true" tabindex="-1"></a><span class="co"># --- input rc file ---</span></span>
<span id="cb212-968"><a href="#cb212-968" aria-hidden="true" tabindex="-1"></a>tmp_rc <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="fu">here</span>(work_here, <span class="st">"dada2_pipeline_read_changes.txt"</span>)) </span>
<span id="cb212-969"><a href="#cb212-969" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-970"><a href="#cb212-970" aria-hidden="true" tabindex="-1"></a><span class="co"># --- generate tables ---</span></span>
<span id="cb212-971"><a href="#cb212-971" aria-hidden="true" tabindex="-1"></a>start_tbls <span class="ot">&lt;-</span> <span class="fu">make_tables</span>(microeco<span class="sc">::</span><span class="fu">clone</span>(me_raw), <span class="st">"start"</span>)</span>
<span id="cb212-972"><a href="#cb212-972" aria-hidden="true" tabindex="-1"></a>final_tbls <span class="ot">&lt;-</span> <span class="fu">make_tables</span>(microeco<span class="sc">::</span><span class="fu">clone</span>(me_final), <span class="st">"final"</span>)</span>
<span id="cb212-973"><a href="#cb212-973" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-974"><a href="#cb212-974" aria-hidden="true" tabindex="-1"></a><span class="co"># --- summary ---</span></span>
<span id="cb212-975"><a href="#cb212-975" aria-hidden="true" tabindex="-1"></a>curate_summary <span class="ot">&lt;-</span> tmp_rc <span class="sc">%&gt;%</span></span>
<span id="cb212-976"><a href="#cb212-976" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">full_join</span>(start_tbls<span class="sc">$</span>rc,  <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-977"><a href="#cb212-977" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(start_tbls<span class="sc">$</span>asv, <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-978"><a href="#cb212-978" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(final_tbls<span class="sc">$</span>rc,  <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb212-979"><a href="#cb212-979" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(final_tbls<span class="sc">$</span>asv, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb212-980"><a href="#cb212-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-981"><a href="#cb212-981" aria-hidden="true" tabindex="-1"></a>removed_samples <span class="ot">&lt;-</span> curate_summary <span class="sc">%&gt;%</span></span>
<span id="cb212-982"><a href="#cb212-982" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(final_rc)) <span class="sc">%&gt;%</span></span>
<span id="cb212-983"><a href="#cb212-983" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(SampleID)</span>
<span id="cb212-984"><a href="#cb212-984" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-985"><a href="#cb212-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-986"><a href="#cb212-986" aria-hidden="true" tabindex="-1"></a>We lost a total of <span class="in">`r length(removed_samples)`</span> samples after curating the dataset. This includes <span class="in">`r length(removed_samples[grepl("^Control_", removed_samples)])`</span> NC samples and <span class="in">`r length(removed_samples[!grepl("^Control_", removed_samples)])`</span> non-NC samples.</span>
<span id="cb212-987"><a href="#cb212-987" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-988"><a href="#cb212-988" aria-hidden="true" tabindex="-1"></a>Here is a list of non-NC samples that were removed. </span>
<span id="cb212-989"><a href="#cb212-989" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-992"><a href="#cb212-992" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-993"><a href="#cb212-993" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb212-994"><a href="#cb212-994" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-995"><a href="#cb212-995" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-996"><a href="#cb212-996" aria-hidden="true" tabindex="-1"></a>removed_samples[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^Control_"</span>, removed_samples)]</span>
<span id="cb212-997"><a href="#cb212-997" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-998"><a href="#cb212-998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-999"><a href="#cb212-999" aria-hidden="true" tabindex="-1"></a><span class="fu"># Download Results {#download-results}</span></span>
<span id="cb212-1000"><a href="#cb212-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1001"><a href="#cb212-1001" aria-hidden="true" tabindex="-1"></a>Quick access to read changes through the pipeline and ASVs detected in negative control samples.</span>
<span id="cb212-1002"><a href="#cb212-1002" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1003"><a href="#cb212-1003" aria-hidden="true" tabindex="-1"></a>::: {#dada2-listing .column-body}</span>
<span id="cb212-1004"><a href="#cb212-1004" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb212-1005"><a href="#cb212-1005" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1006"><a href="#cb212-1006" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!------------------------------------------------------------------------&gt;</span></span>
<span id="cb212-1007"><a href="#cb212-1007" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-------------------- Use this area to save things ----------------------&gt;</span></span>
<span id="cb212-1008"><a href="#cb212-1008" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!------------------------------------------------------------------------&gt;</span></span>
<span id="cb212-1009"><a href="#cb212-1009" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1010"><a href="#cb212-1010" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!---------------------------------------&gt;</span></span>
<span id="cb212-1011"><a href="#cb212-1011" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- These chunks are for curated data --&gt;</span></span>
<span id="cb212-1012"><a href="#cb212-1012" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!---------------------------------------&gt;</span></span>
<span id="cb212-1013"><a href="#cb212-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1016"><a href="#cb212-1016" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1017"><a href="#cb212-1017" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1018"><a href="#cb212-1018" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1019"><a href="#cb212-1019" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"dada2"</span></span>
<span id="cb212-1020"><a href="#cb212-1020" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data"</span>)), </span>
<span id="cb212-1021"><a href="#cb212-1021" aria-hidden="true" tabindex="-1"></a>           <span class="at">recursive =</span> <span class="cn">FALSE</span>, <span class="at">showWarnings =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1022"><a href="#cb212-1022" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1023"><a href="#cb212-1023" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1026"><a href="#cb212-1026" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1027"><a href="#cb212-1027" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1028"><a href="#cb212-1028" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1029"><a href="#cb212-1029" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(nc_check, <span class="fu">here</span>(share_here, <span class="st">"asv_in_nc_samples.txt"</span>),</span>
<span id="cb212-1030"><a href="#cb212-1030" aria-hidden="true" tabindex="-1"></a>    <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb212-1031"><a href="#cb212-1031" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1032"><a href="#cb212-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1035"><a href="#cb212-1035" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1036"><a href="#cb212-1036" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1037"><a href="#cb212-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1038"><a href="#cb212-1038" aria-hidden="true" tabindex="-1"></a>tmp_path <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data/"</span>))</span>
<span id="cb212-1039"><a href="#cb212-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1040"><a href="#cb212-1040" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb212-1041"><a href="#cb212-1041" aria-hidden="true" tabindex="-1"></a>  <span class="at">otu =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_otu_table.txt"</span>),</span>
<span id="cb212-1042"><a href="#cb212-1042" aria-hidden="true" tabindex="-1"></a>  <span class="at">tax =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_tax_table.txt"</span>),</span>
<span id="cb212-1043"><a href="#cb212-1043" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_sample_table.txt"</span>),</span>
<span id="cb212-1044"><a href="#cb212-1044" aria-hidden="true" tabindex="-1"></a>  <span class="at">rep =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_rep.fasta"</span>),</span>
<span id="cb212-1045"><a href="#cb212-1045" aria-hidden="true" tabindex="-1"></a>  <span class="at">me =</span> <span class="fu">paste0</span>(<span class="st">"me_"</span>, workflow_name, <span class="st">".rds"</span>),</span>
<span id="cb212-1046"><a href="#cb212-1046" aria-hidden="true" tabindex="-1"></a>  <span class="at">ps =</span> <span class="fu">paste0</span>(<span class="st">"ps_"</span>, workflow_name, <span class="st">".rds"</span>),</span>
<span id="cb212-1047"><a href="#cb212-1047" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_track_read_counts.txt"</span>)</span>
<span id="cb212-1048"><a href="#cb212-1048" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-1049"><a href="#cb212-1049" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1050"><a href="#cb212-1050" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1053"><a href="#cb212-1053" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1054"><a href="#cb212-1054" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1055"><a href="#cb212-1055" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1056"><a href="#cb212-1056" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------OTU Table--------------#</span></span>
<span id="cb212-1057"><a href="#cb212-1057" aria-hidden="true" tabindex="-1"></a>tmp_otu <span class="ot">&lt;-</span> me_final<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span> </span>
<span id="cb212-1058"><a href="#cb212-1058" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb212-1059"><a href="#cb212-1059" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_otu, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>otu), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb212-1060"><a href="#cb212-1060" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------TAX Table--------------#</span></span>
<span id="cb212-1061"><a href="#cb212-1061" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> me_final<span class="sc">$</span>tax_table <span class="sc">%&gt;%</span> </span>
<span id="cb212-1062"><a href="#cb212-1062" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"ASV_ID"</span>)</span>
<span id="cb212-1063"><a href="#cb212-1063" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_tax, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>tax), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb212-1064"><a href="#cb212-1064" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------SAMP Table--------------#</span></span>
<span id="cb212-1065"><a href="#cb212-1065" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(me_final<span class="sc">$</span>sample_table, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>sample), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb212-1066"><a href="#cb212-1066" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------REP FASTA--------------#</span></span>
<span id="cb212-1067"><a href="#cb212-1067" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(</span>
<span id="cb212-1068"><a href="#cb212-1068" aria-hidden="true" tabindex="-1"></a>  <span class="at">sequences =</span> <span class="fu">as.list</span>(<span class="fu">as.character</span>(me_final<span class="sc">$</span>rep_fasta)),</span>
<span id="cb212-1069"><a href="#cb212-1069" aria-hidden="true" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">names</span>(me_final<span class="sc">$</span>rep_fasta),</span>
<span id="cb212-1070"><a href="#cb212-1070" aria-hidden="true" tabindex="-1"></a>  <span class="at">file.out =</span> <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>rep)</span>
<span id="cb212-1071"><a href="#cb212-1071" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-1072"><a href="#cb212-1072" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------        --------------#</span></span>
<span id="cb212-1073"><a href="#cb212-1073" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(me_final, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>me))</span>
<span id="cb212-1074"><a href="#cb212-1074" aria-hidden="true" tabindex="-1"></a>ps_final <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb212-1075"><a href="#cb212-1075" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_final, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>ps))</span>
<span id="cb212-1076"><a href="#cb212-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------read count changes--------------#</span></span>
<span id="cb212-1077"><a href="#cb212-1077" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(curate_summary, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>counts), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb212-1078"><a href="#cb212-1078" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(<span class="st">"working_files/ssu/sampledata"</span>,  <span class="st">"all_metadata.txt"</span>), </span>
<span id="cb212-1079"><a href="#cb212-1079" aria-hidden="true" tabindex="-1"></a>          tmp_path, </span>
<span id="cb212-1080"><a href="#cb212-1080" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb212-1081"><a href="#cb212-1081" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-1082"><a href="#cb212-1082" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1083"><a href="#cb212-1083" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1086"><a href="#cb212-1086" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1087"><a href="#cb212-1087" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1088"><a href="#cb212-1088" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1089"><a href="#cb212-1089" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data.zip"</span>)), </span>
<span id="cb212-1090"><a href="#cb212-1090" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_curated_data"</span>)), </span>
<span id="cb212-1091"><a href="#cb212-1091" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb212-1092"><a href="#cb212-1092" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1093"><a href="#cb212-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1096"><a href="#cb212-1096" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1097"><a href="#cb212-1097" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1098"><a href="#cb212-1098" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1099"><a href="#cb212-1099" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(work_here,  <span class="st">"dada2_pipeline_read_changes.txt"</span>), </span>
<span id="cb212-1100"><a href="#cb212-1100" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(share_here), </span>
<span id="cb212-1101"><a href="#cb212-1101" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb212-1102"><a href="#cb212-1102" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-1103"><a href="#cb212-1103" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1104"><a href="#cb212-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1105"><a href="#cb212-1105" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!--------------------------------------------&gt;</span></span>
<span id="cb212-1106"><a href="#cb212-1106" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- These chunks are for  processing data  --&gt;</span></span>
<span id="cb212-1107"><a href="#cb212-1107" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!--------------------------------------------&gt;</span></span>
<span id="cb212-1108"><a href="#cb212-1108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1111"><a href="#cb212-1111" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1112"><a href="#cb212-1112" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1113"><a href="#cb212-1113" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1114"><a href="#cb212-1114" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing"</span>)), </span>
<span id="cb212-1115"><a href="#cb212-1115" aria-hidden="true" tabindex="-1"></a>           <span class="at">recursive =</span> <span class="cn">FALSE</span>, <span class="at">showWarnings =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1116"><a href="#cb212-1116" aria-hidden="true" tabindex="-1"></a>copy_here <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing/"</span>))</span>
<span id="cb212-1117"><a href="#cb212-1117" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1118"><a href="#cb212-1118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1121"><a href="#cb212-1121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1122"><a href="#cb212-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1123"><a href="#cb212-1123" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1124"><a href="#cb212-1124" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: from sampledata (these data are common to all 16S wf)</span></span>
<span id="cb212-1125"><a href="#cb212-1125" aria-hidden="true" tabindex="-1"></a>tmp_sampdata_path <span class="ot">&lt;-</span> <span class="st">"working_files/ssu/sampledata"</span></span>
<span id="cb212-1126"><a href="#cb212-1126" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"fastq_rename_results/"</span>), </span>
<span id="cb212-1127"><a href="#cb212-1127" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"fastq_rename_results/"</span>), </span>
<span id="cb212-1128"><a href="#cb212-1128" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1129"><a href="#cb212-1129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1130"><a href="#cb212-1130" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"fastq_rename_lookup/"</span>), </span>
<span id="cb212-1131"><a href="#cb212-1131" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"fastq_rename_lookup/"</span>), </span>
<span id="cb212-1132"><a href="#cb212-1132" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1133"><a href="#cb212-1133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1134"><a href="#cb212-1134" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"rename.sh"</span>), </span>
<span id="cb212-1135"><a href="#cb212-1135" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb212-1136"><a href="#cb212-1136" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb212-1137"><a href="#cb212-1137" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-1138"><a href="#cb212-1138" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1139"><a href="#cb212-1139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1142"><a href="#cb212-1142" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1143"><a href="#cb212-1143" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1144"><a href="#cb212-1144" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1145"><a href="#cb212-1145" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_hydra_scripts/"</span>)), </span>
<span id="cb212-1146"><a href="#cb212-1146" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_hydra_scripts/"</span>)), </span>
<span id="cb212-1147"><a href="#cb212-1147" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1148"><a href="#cb212-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1149"><a href="#cb212-1149" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_scripts/"</span>)), </span>
<span id="cb212-1150"><a href="#cb212-1150" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_scripts/"</span>)), </span>
<span id="cb212-1151"><a href="#cb212-1151" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1152"><a href="#cb212-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1153"><a href="#cb212-1153" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_read_changes/"</span>)), </span>
<span id="cb212-1154"><a href="#cb212-1154" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_read_changes/"</span>)), </span>
<span id="cb212-1155"><a href="#cb212-1155" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1156"><a href="#cb212-1156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1157"><a href="#cb212-1157" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="st">"dada_results/"</span>), </span>
<span id="cb212-1158"><a href="#cb212-1158" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"dada_results/"</span>), </span>
<span id="cb212-1159"><a href="#cb212-1159" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1160"><a href="#cb212-1160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1161"><a href="#cb212-1161" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(share_here,  <span class="st">"dada2_pipeline_read_changes.txt"</span>), </span>
<span id="cb212-1162"><a href="#cb212-1162" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb212-1163"><a href="#cb212-1163" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb212-1164"><a href="#cb212-1164" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-1165"><a href="#cb212-1165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1166"><a href="#cb212-1166" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"all_metadata.txt"</span>), </span>
<span id="cb212-1167"><a href="#cb212-1167" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb212-1168"><a href="#cb212-1168" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb212-1169"><a href="#cb212-1169" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb212-1170"><a href="#cb212-1170" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1171"><a href="#cb212-1171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1174"><a href="#cb212-1174" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1175"><a href="#cb212-1175" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1176"><a href="#cb212-1176" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1177"><a href="#cb212-1177" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing.zip"</span>)), </span>
<span id="cb212-1178"><a href="#cb212-1178" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_processing"</span>)), </span>
<span id="cb212-1179"><a href="#cb212-1179" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb212-1180"><a href="#cb212-1180" aria-hidden="true" tabindex="-1"></a><span class="fu">dir_delete</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_processing"</span>)))</span>
<span id="cb212-1181"><a href="#cb212-1181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1182"><a href="#cb212-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1185"><a href="#cb212-1185" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1186"><a href="#cb212-1186" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1187"><a href="#cb212-1187" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb212-1188"><a href="#cb212-1188" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb212-1189"><a href="#cb212-1189" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb212-1190"><a href="#cb212-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: save R script</span></span>
<span id="cb212-1191"><a href="#cb212-1191" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"dada2"</span></span>
<span id="cb212-1192"><a href="#cb212-1192" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">knitr.duplicate.label =</span> <span class="st">"allow"</span>)</span>
<span id="cb212-1193"><a href="#cb212-1193" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sources and numbered outputs</span></span>
<span id="cb212-1194"><a href="#cb212-1194" aria-hidden="true" tabindex="-1"></a>sources <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb212-1195"><a href="#cb212-1195" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_26.qmd"</span>,</span>
<span id="cb212-1196"><a href="#cb212-1196" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_28.qmd"</span>,</span>
<span id="cb212-1197"><a href="#cb212-1197" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_29.qmd"</span>,</span>
<span id="cb212-1198"><a href="#cb212-1198" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_30.qmd"</span>,</span>
<span id="cb212-1199"><a href="#cb212-1199" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_34.qmd"</span>,</span>
<span id="cb212-1200"><a href="#cb212-1200" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_BCS_35.qmd"</span>,</span>
<span id="cb212-1201"><a href="#cb212-1201" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_MERGE_RUNS.qmd"</span>,</span>
<span id="cb212-1202"><a href="#cb212-1202" aria-hidden="true" tabindex="-1"></a>  <span class="st">"index.qmd"</span></span>
<span id="cb212-1203"><a href="#cb212-1203" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-1204"><a href="#cb212-1204" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate corresponding temporary output files</span></span>
<span id="cb212-1205"><a href="#cb212-1205" aria-hidden="true" tabindex="-1"></a>tmp_outputs <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_workflow"</span>, <span class="fu">seq_along</span>(sources), <span class="st">".R"</span>))</span>
<span id="cb212-1206"><a href="#cb212-1206" aria-hidden="true" tabindex="-1"></a><span class="co"># Final combined output</span></span>
<span id="cb212-1207"><a href="#cb212-1207" aria-hidden="true" tabindex="-1"></a>final_output <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_workflow.R"</span>))</span>
<span id="cb212-1208"><a href="#cb212-1208" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Extract code from qmd to temporary R scripts</span></span>
<span id="cb212-1209"><a href="#cb212-1209" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">walk2</span>(sources, tmp_outputs, <span class="sc">~</span></span>
<span id="cb212-1210"><a href="#cb212-1210" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">purl</span>(.x, <span class="at">output =</span> .y, <span class="at">documentation =</span> <span class="dv">0</span>)</span>
<span id="cb212-1211"><a href="#cb212-1211" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb212-1212"><a href="#cb212-1212" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Concatenate all temporary R scripts into one, with separators</span></span>
<span id="cb212-1213"><a href="#cb212-1213" aria-hidden="true" tabindex="-1"></a><span class="fu">sink</span>(final_output)</span>
<span id="cb212-1214"><a href="#cb212-1214" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">walk2</span>(tmp_outputs, <span class="fu">seq_along</span>(tmp_outputs), <span class="sc">~</span>{</span>
<span id="cb212-1215"><a href="#cb212-1215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">readLines</span>(.x), <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb212-1216"><a href="#cb212-1216" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>,</span>
<span id="cb212-1217"><a href="#cb212-1217" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">"# -------------------- END OF SCRIPT "</span>, .y, <span class="st">" --------------------</span><span class="sc">\n\n</span><span class="st">"</span>),</span>
<span id="cb212-1218"><a href="#cb212-1218" aria-hidden="true" tabindex="-1"></a>      <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb212-1219"><a href="#cb212-1219" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb212-1220"><a href="#cb212-1220" aria-hidden="true" tabindex="-1"></a><span class="fu">sink</span>()</span>
<span id="cb212-1221"><a href="#cb212-1221" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Clean up temporary files</span></span>
<span id="cb212-1222"><a href="#cb212-1222" aria-hidden="true" tabindex="-1"></a><span class="fu">file.remove</span>(tmp_outputs)</span>
<span id="cb212-1223"><a href="#cb212-1223" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1224"><a href="#cb212-1224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1227"><a href="#cb212-1227" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb212-1228"><a href="#cb212-1228" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb212-1229"><a href="#cb212-1229" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb212-1230"><a href="#cb212-1230" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb212-1231"><a href="#cb212-1231" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(curate_summary, ds_metrics_tbl, me_final, me_nc, </span>
<span id="cb212-1232"><a href="#cb212-1232" aria-hidden="true" tabindex="-1"></a>            me_no_cont, me_no_low, me_no_na, me_no_nc, </span>
<span id="cb212-1233"><a href="#cb212-1233" aria-hidden="true" tabindex="-1"></a>            me_raw, nc_asvs, nc_check, nc_name, nc_remain, nc_remove, </span>
<span id="cb212-1234"><a href="#cb212-1234" aria-hidden="true" tabindex="-1"></a>            per_reads_rem, per_reads_ret, pipe_summary, rem_nc_reads, </span>
<span id="cb212-1235"><a href="#cb212-1235" aria-hidden="true" tabindex="-1"></a>            rem_sam_reads, removed_samples, ret_nc_reads, ret_sam_reads, </span>
<span id="cb212-1236"><a href="#cb212-1236" aria-hidden="true" tabindex="-1"></a>            samdf, sample_info, samptab, seqtab, share_here, st.head, </span>
<span id="cb212-1237"><a href="#cb212-1237" aria-hidden="true" tabindex="-1"></a>            tax.head1, tax.head2, seqtab, tax_gsrdb, </span>
<span id="cb212-1238"><a href="#cb212-1238" aria-hidden="true" tabindex="-1"></a>            <span class="at">sure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb212-1239"><a href="#cb212-1239" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="fu">here</span>(<span class="st">"page_build"</span>, <span class="st">"dada2_part2.rdata"</span>))</span>
<span id="cb212-1240"><a href="#cb212-1240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb212-1241"><a href="#cb212-1241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb212-1242"><a href="#cb212-1242" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_footer.qmd &gt;}}</span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../../../site_info.html">
<p>Site Information</p>
</a>
  </li>  
</ul>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/istmobiome/shrimp-draft/blob/main/workflows/ssu/dada2/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/edit/main/workflows/ssu/dada2/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>.</p>
</div>
  </div>
</footer>


</body></html>