<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Constructing taxonomic &amp; functional annotation databases.">
<title>1. Build Databases – TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../workflows/mg/assembly/index.html" rel="next">
<link href="../../../workflows/mg/index.html" rel="prev">
<link href="../../../assets/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-d7d32a8442357cea8cf74a4a848b80d6.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><link rel="stylesheet" href="../../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo light-content"><img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo dark-content"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/portal/index.html"> 
<span class="menu-text">Data Portal</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/ssu/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/mg/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/trans-shrimp/"> <i class="bi bi-github" role="img" aria-label="Trans Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/mg/index.html">Metagenomics</a></li><li class="breadcrumb-item"><a href="../../../workflows/mg/setup/index.html">1. Build Databases</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/mg/index.html">Metagenomics</a></li><li class="breadcrumb-item"><a href="../../../workflows/mg/setup/index.html">1. Build Databases</a></li></ol></nav>
      <div class="quarto-title-block"><div><h1 class="title">1. Build Databases</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>Constructing taxonomic &amp; functional annotation databases.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../../../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/portal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Portal</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/ssu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/dada2/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. DADA2 ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/med/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/mg/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/setup/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">1. Build Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/assembly/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/aa-results/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Assembly &amp; Annotation Results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/archive_data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting &amp; Archiving Data/Code</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li>
<a href="#annotation-databases" id="toc-annotation-databases" class="nav-link active" data-scroll-target="#annotation-databases">Annotation Databases</a>
  <ul class="collapse">
<li>
<a href="#taxonomic-classification" id="toc-taxonomic-classification" class="nav-link" data-scroll-target="#taxonomic-classification">Taxonomic Classification</a>
  <ul class="collapse">
<li><a href="#kaiju" id="toc-kaiju" class="nav-link" data-scroll-target="#kaiju">Kaiju</a></li>
  <li><a href="#kraken2" id="toc-kraken2" class="nav-link" data-scroll-target="#kraken2">Kraken2</a></li>
  <li><a href="#virsorter2" id="toc-virsorter2" class="nav-link" data-scroll-target="#virsorter2">VirSorter2</a></li>
  <li><a href="#single-copy-gene-scg-taxonomy" id="toc-single-copy-gene-scg-taxonomy" class="nav-link" data-scroll-target="#single-copy-gene-scg-taxonomy">Single Copy Gene (SCG) taxonomy</a></li>
  </ul>
</li>
  <li><a href="#functional-annotations" id="toc-functional-annotations" class="nav-link" data-scroll-target="#functional-annotations">Functional Annotations</a></li>
  </ul>
</li>
  <li><a href="#concluding-remarks" id="toc-concluding-remarks" class="nav-link" data-scroll-target="#concluding-remarks">Concluding remarks</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/trans-shrimp/blob/main/workflows/mg/setup/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/edit/main/workflows/mg/setup/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content"><section id="annotation-databases" class="level1"><h1>Annotation Databases</h1>
<p>There are two main types of annotations we are interested in for this metagenomic project—<strong>taxonomic</strong> and <strong>functional</strong>—and there are many, many ways to accomplish both of these goals. This next section involves building the databases and installing any additional tools we need for annotation.</p>
<p>Let’s start with the tools and databases for <strong>taxonomic</strong> classification.</p>
<section id="taxonomic-classification" class="level2"><h2 class="anchored" data-anchor-id="taxonomic-classification">Taxonomic Classification</h2>
<p>There are many algorithms and databases for taxonomic classification. We will use <a href="https://github.com/DerrickWood/kraken2">Kraken2</a> to classify short reads. We will also use <a href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a> and <a href="https://github.com/jiarong/VirSorter2">VirSorter2</a> for contigs. Anvi’o has methods of importing data from each of these approaches but if you have a have a favorite tool/database there are workarounds to get most results into the appropriate anvio database.</p>
<section id="kaiju" class="level3"><h3 class="anchored" data-anchor-id="kaiju">Kaiju</h3>
<p><em><a href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a> <span class="citation" data-cites="menzel2016kaiju">(<a href="#ref-menzel2016kaiju" role="doc-biblioref">Menzel, Ng, and Krogh 2016</a>)</span>… finds maximum (in-)exact matches on the protein-level using the Burrows–Wheeler transform.</em> <em>Kaiju is a program for the taxonomic classification… of metagenomic DNA. Reads are directly assigned to taxa using the NCBI taxonomy and a reference database of protein sequences from microbial and viral genomes.</em></p>
<p>First we need to install Kaiju. Again, I will run kaiju in a separate conda environment. Now I can either install kaiju from the source code or as conda package. The latter is easier but often conda packages may lag behind the source code versions. I usually compare the release dates of the <a href="https://anaconda.org/bioconda/kaiju">conda package</a> with the <a href="https://github.com/bioinformatics-centre/kaiju">source code</a> and look at the number of downloads. In this case, the conda version of kaiju looks fine.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create generic environment</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> kaiju</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiji</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install <span class="at">-c</span> bioconda kaiju</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>After kaiju is installed, the next thing to do is generate the database. You can find a description of kaiju databases in the section on <a href="https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index">dreating the Kaiju index</a>. I downloaded and formatted the <code>progenomes</code> database, whic contains representative sets of genomes from the <a href="http://progenomes.embl.de/">proGenomes</a> database and viruses from the NCBI RefSeq database.</p>
<p>Simply run the <code>kaiju-makedb</code> command and specify a database.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate kaiju</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> kaiju</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> kaiju</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-makedb</span> <span class="at">-s</span> progenomes</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The <code>progenomes</code> database is 95GB.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand for the KAIJU_DB_BUILD Hydra script
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<pre><code># /bin/sh
# ----------------Parameters---------------------- #
#$ -S /bin/sh
#$ -pe mthread 15
#$ -q sThM.q
#$ -l mres=225G,h_data=15G,h_vmem=15G,himem
#$ -cwd
#$ -j y
#$ -N job_build_anvio_dbs
#$ -o hydra_logs/job_build_kaiju.log
#
# ----------------Modules------------------------- #
module load gcc/4.9.2
#
# ----------------Your Commands------------------- #
#
echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
# ----------------CALLING ANVIO------------------- #
export PATH=/home/scottjj/miniconda3:$PATH
export PATH=/home/scottjj/miniconda3/bin:$PATH
#
source activate kaiju
mkdir kaiju
cd kaiju
kaiju-makedb -s progenomes</code></pre>
</div>
</div>
</div>
</section><section id="kraken2" class="level3"><h3 class="anchored" data-anchor-id="kraken2">Kraken2</h3>
<p><em><a href="https://github.com/DerrickWood/kraken2">Kraken2</a> <span class="citation" data-cites="wood2019kraken2">(<a href="#ref-wood2019kraken2" role="doc-biblioref">Wood, Lu, and Langmead 2019</a>)</span> is a novel metagenomics classifier that combines the fast k-mer-based classification of Kraken with an efficient algorithm for assessing the coverage of unique k-mers found in each species in a dataset.</em></p>
<p>Installed in separate conda environment</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> kraken2</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install kraken2</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kraken2</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The standard way of installing a Kraken2 database is to run <code>kraken2-build</code> and calling the <code>--standard</code> flag, which will construct a database containing Refeq archaea, bacteria, viral, plasmid, human data plus data from UniVec_Core. Here is the command.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--standard</span> <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The first problem arises because of issues with the NCBI servers and <code>kraken2-build</code> use of <code>rsync</code>. So we add the flag <code>--use-ftp</code>. Like so…</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--standard</span> <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>But this did not work either–it failed repeatedly. If you look at the Kraken2 GitHub <a href="https://github.com/DerrickWood/kraken2/issues">issues page</a> you will see that this is a common issue. So instead we tried using one of the <a href="https://benlangmead.github.io/aws-indexes/k2">prebuilt database</a>, however none of these contained files we needed for downstream analysis. Strike 2. So we tried a different approach.</p>
<p>First, we followed <a href="https://github.com/DerrickWood/kraken2/issues/515#issuecomment-949354093">this suggestion</a> to change line 16 of the file <code>PATH_to_KRAKEN2/download_taxonomy.sh</code> from this…</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="va">FTP_SERVER</span><span class="op">=</span><span class="st">"ftp://</span><span class="va">$NCBI_SERVER</span><span class="st">"</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>…to this</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">TO:</span> FTP_SERVER=<span class="st">"https://</span><span class="va">$NCBI_SERVER</span><span class="st">"</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Supposedly this is to help with the timeout issues from the NCBI servers. Next, we tried building our own database. The first step was to download the NCBI taxonomy.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-taxonomy</span> <span class="at">--db</span> kraken2_db <span class="at">--use-ftp</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Once this step was complete, we downloaded individual libraries in order to build a custom Kraken2 database.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> archaea <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> bacteria <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> plasmid <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> viral <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> fungi <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> protozoa <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> UniVec_Core <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And finally constructed the database.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--build</span> <span class="at">--db</span> kraken2_db</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And it worked. The whole process took about 8 hours and the final DB is ~90GB.</p>
<div class="copy callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand for the KRAKEN2_DB_BUILD Hydra script
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<pre><code># /bin/sh
# ----------------Parameters---------------------- #
#$ -S /bin/sh
#$ -pe mthread 30
#$ -q mThM.q
#$ -l mres=900G,h_data=30G,h_vmem=30G,himem
#$ -cwd
#$ -j y
#$ -N job_00_kraken2-build
#$ -o job_00_kraken2-build4.job
#
# ----------------Modules------------------------- #
module load bioinformatics/blast
#
# ----------------Load Envs------------------- #
#
echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
export PATH=/home/scottjj/miniconda3/bin:$PATH
export PATH=/home/scottjj/miniconda3/envs/kraken2/bin:$PATH

######################################################################
# Changed line 16 of this file:
# FROM: FTP_SERVER="ftp://$NCBI_SERVER"
# TO: FTP_SERVER="https://$NCBI_SERVER"
# /home/scottjj/miniconda3/envs/kraken2/share/kraken2-2.1.3-1/libexec/download_taxonomy.sh
# per https://github.com/DerrickWood/kraken2/issues/515#issuecomment-949354093
######################################################################

source activate kraken2

######################################################################
## STANDARD BUILD-FAILED
######################################################################
###kraken2-build --standard --db kraken2_db --threads $NSLOTS --use-ftp

######################################################################
## CUSTOM BUILD
######################################################################

## TAXONOMY
kraken2-build --download-taxonomy --db kraken2_db --use-ftp

## DATABASES
kraken2-build --download-library archaea --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library bacteria --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library plasmid --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library viral --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library fungi --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library protozoa --db kraken2_db --threads $NSLOTS --use-ftp 
kraken2-build --download-library UniVec_Core --db kraken2_db --threads $NSLOTS --use-ftp 

## BUILD
kraken2-build --build --db kraken2_db

conda deactivate

echo = `date` job $JOB_NAME done</code></pre>
</div>
</div>
</div>
</section><section id="virsorter2" class="level3"><h3 class="anchored" data-anchor-id="virsorter2">VirSorter2</h3>
<p><em><a href="https://github.com/jiarong/VirSorter2">VirSorter</a> <span class="citation" data-cites="guo2021virsorter2">(<a href="#ref-guo2021virsorter2" role="doc-biblioref">Guo et al. 2021</a>)</span>, applies a multi-classifier, expert-guided approach to detect diverse DNA and RNA virus genomes. It has made major updates to its previous version.</em></p>
<p>I followed <a href="https://github.com/jiarong/VirSorter2?tab=readme-ov-file#option-1-bioconda-version">this recipe</a> for installing VirSorter2. Piece of cake.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> virsorter2 <span class="at">-c</span> conda-forge <span class="at">-c</span> bioconda virsorter=2</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate virsorter2</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Good to go? Now time to build the VirSorter2 database. Pretty straightforward actually. The general instructions can be found <a href="https://github.com/jiarong/VirSorter2?tab=readme-ov-file#download-database-and-dependencies">here</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate virsorter2</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> setup <span class="at">--db-dir</span> /pool/genomics/stri_istmobiome/dbs/virsorter2/ <span class="at">-j</span> <span class="va">$NSLOTS</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And then a quick test to make sure the database is ok.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> TEST_virsorter2</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> TEST_virsorter2</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> test.fa https://raw.githubusercontent.com/jiarong/VirSorter2/master/test/8seq.fa</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">-w</span> test.out <span class="at">-i</span> test.fa <span class="at">-j</span> <span class="va">$NSLOTS</span> <span class="at">--db-dir</span> dbs/virsorter2/ <span class="at">--tmpdir</span> TEST_virsorter2/tmp_vir <span class="at">--rm-tmpdir</span> <span class="at">--min-length</span> 1500 all</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The uncompressed database is a little over 10GB.</p>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand for the VIRSORTER2_DB_BUILD Hydra script
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<pre><code># /bin/sh
# ----------------Parameters---------------------- #
#$ -S /bin/sh
#$ -pe mthread 15
#$ -q sThM.q
#$ -l mres=225G,h_data=15G,h_vmem=15G,himem
#$ -cwd
#$ -j y
#$ -N job_build_anvio_dbs
#$ -o hydra_logs/job_build_virsorter2.log
#
# ----------------Modules------------------------- #
module load gcc/4.9.2
#
# ----------------Your Commands------------------- #
#
echo + `date` job $JOB_NAME started in $QUEUE with jobID=$JOB_ID on $HOSTNAME
echo + NSLOTS = $NSLOTS
#
# ----------------CALLING ANVIO------------------- #
export PATH=/home/scottjj/miniconda3:$PATH
export PATH=/home/scottjj/miniconda3/bin:$PATH
#

source activate virsorter2
virsorter setup --db-dir /pool/genomics/stri_istmobiome/dbs/virsorter2/ -j $NSLOTS

############### TEST VIRSORTER 2 ###############

mkdir TEST_virsorter2
cd TEST_virsorter2
wget -O test.fa https://raw.githubusercontent.com/jiarong/VirSorter2/master/test/8seq.fa
virsorter run -w test.out -i test.fa -j $NSLOTS --db-dir /pool/genomics/stri_istmobiome/dbs/virsorter2/ --tmpdir TEST_virsorter2/tmp_vir --rm-tmpdir --min-length 1500 all

conda deactivate

echo = `date` job $JOB_NAME done</code></pre>
</div>
</div>
</div>
</section><section id="single-copy-gene-scg-taxonomy" class="level3"><h3 class="anchored" data-anchor-id="single-copy-gene-scg-taxonomy">Single Copy Gene (SCG) taxonomy</h3>
<p>Anvi’o has native support for building tRNA and Single Copy Gene (SCG) taxonomy from the <a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-scg-taxonomy</span> <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-trna-taxonomy</span> <span class="at">-T</span> <span class="va">$NSLOTS</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="functional-annotations" class="level2"><h2 class="anchored" data-anchor-id="functional-annotations">Functional Annotations</h2>
<p>Now we can install databases for <strong>functional</strong> annotation. Anvi’o has native support for building <a href="https://pfam.xfam.org/">Pfam</a>, <a href="https://www.ncbi.nlm.nih.gov/COG/">COG</a>, <a href="https://www.genome.jp/tools/kofamkoala/">KEGG</a>, and <a href="http://www.cazy.org/">CAZymes</a> databases.</p>
<p>Anvi’o makes this super simple.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-pfams</span> <span class="at">--pfam-data-dir</span> dbs/pfam_db</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-kegg-data</span> <span class="at">--mode</span> all <span class="at">--kegg-data-dir</span> dbs/kegg_kofam  <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-ncbi-cogs</span> <span class="at">--cog-data-dir</span> dbs/cog_db <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-cazymes</span> <span class="at">--cazyme-data-dir</span> dbs/cazymes</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And that’s it. We can add more databases as we need them.</p>
</section></section><section id="concluding-remarks" class="level1"><h1>Concluding remarks</h1>
<p>At this point we should be good to go with the main setup. We may need to install other tools along the way—we can add those instructions here. Or you may decide to use other tools instead. For example, we are using <a href="https://github.com/voutcn/megahit">megahit</a> for the assemblies but you may want to use <a href="https://github.com/ablab/spades">metaspades</a> or <a href="https://github.com/loneknightpy/idba">idba_ud</a>. These packages need to be installed separately.</p>
</section><div id="quarto-appendix" class="default"><section id="detailed-session-info" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Detailed Session Info</h2><div class="quarto-appendix-contents">
<details><summary>Expand to see Session Info</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Ventura 13.7.8
 system   x86_64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2025-10-02
 pandoc   3.6.3 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/x86_64/ (via rmarkdown)
 quarto   1.8.25 @ /Applications/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
======   Devtools Session info   ===================================</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.</code></pre>
</div>
</div>
</details></div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-10-02</p>


<!-- -->

</div></section><section class="level4 appendix"><h2 class="quarto-appendix-heading">Collected Links</h2><div class="quarto-appendix-contents">
<ul>
<li><a href="https://github.com/DerrickWood/kraken2">Kraken2</a></li>
<li><a href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a></li>
<li><a href="https://github.com/jiarong/VirSorter2">VirSorter2</a></li>
<li><a href="https://anaconda.org/bioconda/kaiju">conda package</a></li>
<li><a href="https://github.com/bioinformatics-centre/kaiju">source code</a></li>
<li><a href="https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index">dreating the Kaiju index</a></li>
<li><a href="http://progenomes.embl.de/">proGenomes</a></li>
<li><a href="https://github.com/DerrickWood/kraken2/issues">issues page</a></li>
<li><a href="https://benlangmead.github.io/aws-indexes/k2">prebuilt database</a></li>
<li><a href="https://github.com/DerrickWood/kraken2/issues/515#issuecomment-949354093">this suggestion</a></li>
<li><a href="https://github.com/jiarong/VirSorter2">VirSorter</a></li>
<li><a href="https://github.com/jiarong/VirSorter2?tab=readme-ov-file#option-1-bioconda-version">this recipe</a></li>
<li><a href="https://github.com/jiarong/VirSorter2?tab=readme-ov-file#download-database-and-dependencies">here</a></li>
<li><a href="https://gtdb.ecogenomic.org/">Genome Taxonomy Database (GTDB)</a></li>
<li><a href="https://pfam.xfam.org/">Pfam</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/COG/">COG</a></li>
<li><a href="https://www.genome.jp/tools/kofamkoala/">KEGG</a></li>
<li><a href="http://www.cazy.org/">CAZymes</a></li>
<li><a href="https://github.com/voutcn/megahit">megahit</a></li>
<li><a href="https://github.com/ablab/spades">metaspades</a></li>
<li><a href="https://github.com/loneknightpy/idba">idba_ud</a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://istmobiome.rbind.io/">The Istmobiome Project</a></li>
</ul>

</div></section><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-guo2021virsorter2" class="csl-entry" role="listitem">
Guo, Jiarong, Ben Bolduc, Ahmed A Zayed, Arvind Varsani, Guillermo Dominguez-Huerta, Tom O Delmont, Akbar Adjie Pratama, et al. 2021. <span>“VirSorter2: A Multi-Classifier, Expert-Guided Approach to Detect Diverse DNA and RNA Viruses.”</span> <em>Microbiome</em> 9: 113. <a href="https://doi.org/10.1186/s40168-020-00990-y">https://doi.org/10.1186/s40168-020-00990-y</a>.
</div>
<div id="ref-menzel2016kaiju" class="csl-entry" role="listitem">
Menzel, Peter, Kim Lee Ng, and Anders Krogh. 2016. <span>“Fast and Sensitive Taxonomic Classification for Metagenomics with Kaiju.”</span> <em>Nature Communications</em> 7: 11257. <a href="https://doi.org/10.1038/ncomms11257">https://doi.org/10.1038/ncomms11257</a>.
</div>
<div id="ref-wood2019kraken2" class="csl-entry" role="listitem">
Wood, Derrick E, Jennifer Lu, and Ben Langmead. 2019. <span>“Improved Metagenomic Analysis with Kraken 2.”</span> <em>Genome Biology</em> 20: 113. <a href="https://doi.org/10.1186/s13059-019-1891-0">https://doi.org/10.1186/s13059-019-1891-0</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/trans-shrimp\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../../workflows/mg/index.html" class="pagination-link" aria-label="Metagenomics">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Metagenomics</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../workflows/mg/assembly/index.html" class="pagination-link" aria-label="2. Co-Assembly &amp; Annotations">
        <span class="nav-page-text">2. Co-Assembly &amp; Annotations</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb22" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "1. Build Databases"</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Constructing taxonomic &amp; functional annotation databases. </span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: set wd for here</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"setup"</span> <span class="co"># e.g., med, dada2, etc</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a>work_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"working_files/mg/"</span>, workflow_name)</span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>share_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"share/mg/"</span>, workflow_name)</span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="fu"># Annotation Databases</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_setup.qmd &gt;}}</span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a>There are two main types of annotations we are interested in for this metagenomic project---**taxonomic** and **functional**---and there are many, many ways to accomplish both of these goals. This next section involves building the databases and installing any additional tools we need for annotation.</span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a>Let’s start with the tools and databases for **taxonomic** classification.</span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic Classification</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a>There are many algorithms and databases for taxonomic classification. We will use <span class="co">[</span><span class="ot">Kraken2</span><span class="co">](https://github.com/DerrickWood/kraken2)</span> to classify short reads. We will also use <span class="co">[</span><span class="ot">Kaiju</span><span class="co">](https://github.com/bioinformatics-centre/kaiju)</span> and <span class="co">[</span><span class="ot">VirSorter2</span><span class="co">](https://github.com/jiarong/VirSorter2)</span> for contigs. Anvi'o has methods of importing data from each of these approaches but if you have a have a favorite tool/database there are workarounds to get most results into the appropriate anvio database.</span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="fu">### Kaiju</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a>*[Kaiju](https://github.com/bioinformatics-centre/kaiju) [@menzel2016kaiju]... finds maximum (in-)exact matches on the protein-level using the Burrows–Wheeler transform.* *Kaiju is a program for the taxonomic classification... of metagenomic DNA. Reads are directly assigned to taxa using the NCBI taxonomy and a reference database of protein sequences from microbial and viral genomes.*</span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>First we need to install Kaiju. Again, I will run kaiju in a separate conda environment. Now I can either install kaiju from the source code or as conda package. The latter is easier but often conda packages may lag behind the source code versions. I usually compare the release dates of the <span class="co">[</span><span class="ot">conda package</span><span class="co">](https://anaconda.org/bioconda/kaiju)</span> with the <span class="co">[</span><span class="ot">source code</span><span class="co">](https://github.com/bioinformatics-centre/kaiju)</span> and look at the number of downloads. In this case, the conda version of kaiju looks fine.</span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a><span class="co"># create generic environment</span></span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> kaiju</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiji</span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install <span class="at">-c</span> bioconda kaiju</span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>After kaiju is installed, the next thing to do is generate the database. You can find a description of kaiju databases in the section on  <span class="co">[</span><span class="ot">dreating the Kaiju index</span><span class="co">](https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index)</span>. I downloaded and formatted the <span class="in">`progenomes`</span>  database, whic contains representative sets of genomes from the <span class="co">[</span><span class="ot">proGenomes</span><span class="co">](http://progenomes.embl.de/)</span> database and viruses from the NCBI RefSeq database.</span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>Simply run the <span class="in">`kaiju-makedb`</span> command and specify a database.</span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-53"><a href="#cb22-53" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate kaiju</span>
<span id="cb22-54"><a href="#cb22-54" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> kaiju</span>
<span id="cb22-55"><a href="#cb22-55" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> kaiju</span>
<span id="cb22-56"><a href="#cb22-56" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-makedb</span> <span class="at">-s</span> progenomes</span>
<span id="cb22-57"><a href="#cb22-57" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-58"><a href="#cb22-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-59"><a href="#cb22-59" aria-hidden="true" tabindex="-1"></a>The <span class="in">`progenomes`</span> database is 95GB.</span>
<span id="cb22-60"><a href="#cb22-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-61"><a href="#cb22-61" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple" collapse="true"}</span>
<span id="cb22-62"><a href="#cb22-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-63"><a href="#cb22-63" aria-hidden="true" tabindex="-1"></a><span class="fu">### Expand for the KAIJU_DB_BUILD Hydra script</span></span>
<span id="cb22-64"><a href="#cb22-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-65"><a href="#cb22-65" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_JOB_KAIJU_DB_BUILD.qmd &gt;}}</span>
<span id="cb22-66"><a href="#cb22-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-67"><a href="#cb22-67" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb22-68"><a href="#cb22-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-69"><a href="#cb22-69" aria-hidden="true" tabindex="-1"></a><span class="fu">### Kraken2</span></span>
<span id="cb22-70"><a href="#cb22-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-71"><a href="#cb22-71" aria-hidden="true" tabindex="-1"></a>*[Kraken2](https://github.com/DerrickWood/kraken2) [@wood2019kraken2] is a novel metagenomics classifier that combines the fast k-mer-based classification of Kraken with an efficient algorithm for assessing the coverage of unique k-mers found in each species in a dataset.*</span>
<span id="cb22-72"><a href="#cb22-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-73"><a href="#cb22-73" aria-hidden="true" tabindex="-1"></a>Installed in separate conda environment</span>
<span id="cb22-74"><a href="#cb22-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-77"><a href="#cb22-77" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-78"><a href="#cb22-78" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> create <span class="at">-n</span> kraken2</span>
<span id="cb22-79"><a href="#cb22-79" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> install kraken2</span>
<span id="cb22-80"><a href="#cb22-80" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kraken2</span>
<span id="cb22-81"><a href="#cb22-81" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-82"><a href="#cb22-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-83"><a href="#cb22-83" aria-hidden="true" tabindex="-1"></a>The standard way of installing a Kraken2 database is to run <span class="in">`kraken2-build`</span> and calling the <span class="in">`--standard`</span> flag, which will construct a database containing Refeq archaea, bacteria, viral, plasmid, human data plus data from UniVec_Core. Here is the command. </span>
<span id="cb22-84"><a href="#cb22-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-87"><a href="#cb22-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-88"><a href="#cb22-88" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--standard</span> <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-89"><a href="#cb22-89" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-90"><a href="#cb22-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-91"><a href="#cb22-91" aria-hidden="true" tabindex="-1"></a>The first problem arises because of issues with the NCBI servers and <span class="in">`kraken2-build`</span> use of <span class="in">`rsync`</span>. So we add the flag <span class="in">`--use-ftp`</span>. Like so...</span>
<span id="cb22-92"><a href="#cb22-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-95"><a href="#cb22-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-96"><a href="#cb22-96" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--standard</span> <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span></span>
<span id="cb22-97"><a href="#cb22-97" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-98"><a href="#cb22-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-99"><a href="#cb22-99" aria-hidden="true" tabindex="-1"></a>But this did not work either--it failed repeatedly. If you look at the Kraken2 GitHub <span class="co">[</span><span class="ot">issues page</span><span class="co">](https://github.com/DerrickWood/kraken2/issues)</span> you will see that this is a common issue.  So instead we tried using one of the <span class="co">[</span><span class="ot">prebuilt database</span><span class="co">](https://benlangmead.github.io/aws-indexes/k2)</span>, however none of these contained files we needed for downstream analysis. Strike 2. So we tried a different approach. </span>
<span id="cb22-100"><a href="#cb22-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-101"><a href="#cb22-101" aria-hidden="true" tabindex="-1"></a>First, we followed <span class="co">[</span><span class="ot">this suggestion</span><span class="co">](https://github.com/DerrickWood/kraken2/issues/515#issuecomment-949354093)</span> to change line 16 of the file <span class="in">`PATH_to_KRAKEN2/download_taxonomy.sh`</span> from this...</span>
<span id="cb22-102"><a href="#cb22-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-105"><a href="#cb22-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-106"><a href="#cb22-106" aria-hidden="true" tabindex="-1"></a><span class="va">FTP_SERVER</span><span class="op">=</span><span class="st">"ftp://</span><span class="va">$NCBI_SERVER</span><span class="st">"</span></span>
<span id="cb22-107"><a href="#cb22-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-108"><a href="#cb22-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-109"><a href="#cb22-109" aria-hidden="true" tabindex="-1"></a>...to this</span>
<span id="cb22-110"><a href="#cb22-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-113"><a href="#cb22-113" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-114"><a href="#cb22-114" aria-hidden="true" tabindex="-1"></a><span class="ex">TO:</span> FTP_SERVER=<span class="st">"https://</span><span class="va">$NCBI_SERVER</span><span class="st">"</span></span>
<span id="cb22-115"><a href="#cb22-115" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-116"><a href="#cb22-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-117"><a href="#cb22-117" aria-hidden="true" tabindex="-1"></a>Supposedly this is to help with the timeout issues from the NCBI servers. Next, we tried building our own database. The first step was to download the NCBI taxonomy. </span>
<span id="cb22-118"><a href="#cb22-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-121"><a href="#cb22-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-122"><a href="#cb22-122" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-taxonomy</span> <span class="at">--db</span> kraken2_db <span class="at">--use-ftp</span></span>
<span id="cb22-123"><a href="#cb22-123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-124"><a href="#cb22-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-125"><a href="#cb22-125" aria-hidden="true" tabindex="-1"></a>Once this step was complete, we downloaded individual libraries in order to build a custom Kraken2 database. </span>
<span id="cb22-126"><a href="#cb22-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-129"><a href="#cb22-129" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-130"><a href="#cb22-130" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> archaea <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-131"><a href="#cb22-131" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> bacteria <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-132"><a href="#cb22-132" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> plasmid <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-133"><a href="#cb22-133" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> viral <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-134"><a href="#cb22-134" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> fungi <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-135"><a href="#cb22-135" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> protozoa <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-136"><a href="#cb22-136" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--download-library</span> UniVec_Core <span class="at">--db</span> kraken2_db <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="at">--use-ftp</span> </span>
<span id="cb22-137"><a href="#cb22-137" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-138"><a href="#cb22-138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-139"><a href="#cb22-139" aria-hidden="true" tabindex="-1"></a>And finally constructed the database. </span>
<span id="cb22-140"><a href="#cb22-140" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-143"><a href="#cb22-143" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-144"><a href="#cb22-144" aria-hidden="true" tabindex="-1"></a><span class="ex">kraken2-build</span> <span class="at">--build</span> <span class="at">--db</span> kraken2_db</span>
<span id="cb22-145"><a href="#cb22-145" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-146"><a href="#cb22-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-147"><a href="#cb22-147" aria-hidden="true" tabindex="-1"></a>And it worked. The whole process took about 8 hours and the final DB is ~90GB. </span>
<span id="cb22-148"><a href="#cb22-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-149"><a href="#cb22-149" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple" collapse="true" .copy}</span>
<span id="cb22-150"><a href="#cb22-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-151"><a href="#cb22-151" aria-hidden="true" tabindex="-1"></a><span class="fu">### Expand for the KRAKEN2_DB_BUILD Hydra script</span></span>
<span id="cb22-152"><a href="#cb22-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-153"><a href="#cb22-153" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_JOB_KRAKEN2_DB_BUILD.qmd &gt;}}</span>
<span id="cb22-154"><a href="#cb22-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-155"><a href="#cb22-155" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb22-156"><a href="#cb22-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-157"><a href="#cb22-157" aria-hidden="true" tabindex="-1"></a><span class="fu">### VirSorter2</span></span>
<span id="cb22-158"><a href="#cb22-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-159"><a href="#cb22-159" aria-hidden="true" tabindex="-1"></a>*[VirSorter](https://github.com/jiarong/VirSorter2) [@guo2021virsorter2], applies a multi-classifier, expert-guided approach to detect diverse DNA and RNA virus genomes. It has made major updates to its previous version.*</span>
<span id="cb22-160"><a href="#cb22-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-161"><a href="#cb22-161" aria-hidden="true" tabindex="-1"></a>I followed <span class="co">[</span><span class="ot">this recipe</span><span class="co">](https://github.com/jiarong/VirSorter2?tab=readme-ov-file#option-1-bioconda-version)</span> for installing VirSorter2. Piece of cake.</span>
<span id="cb22-162"><a href="#cb22-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-165"><a href="#cb22-165" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-166"><a href="#cb22-166" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> create <span class="at">-n</span> virsorter2 <span class="at">-c</span> conda-forge <span class="at">-c</span> bioconda virsorter=2</span>
<span id="cb22-167"><a href="#cb22-167" aria-hidden="true" tabindex="-1"></a><span class="ex">mamba</span> activate virsorter2</span>
<span id="cb22-168"><a href="#cb22-168" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-169"><a href="#cb22-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-170"><a href="#cb22-170" aria-hidden="true" tabindex="-1"></a>Good to go? Now time to build the VirSorter2 database. Pretty straightforward actually. The general instructions can be found <span class="co">[</span><span class="ot">here</span><span class="co">](https://github.com/jiarong/VirSorter2?tab=readme-ov-file#download-database-and-dependencies)</span>. </span>
<span id="cb22-171"><a href="#cb22-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-174"><a href="#cb22-174" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-175"><a href="#cb22-175" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> activate virsorter2</span>
<span id="cb22-176"><a href="#cb22-176" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> setup <span class="at">--db-dir</span> /pool/genomics/stri_istmobiome/dbs/virsorter2/ <span class="at">-j</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-177"><a href="#cb22-177" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-178"><a href="#cb22-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-179"><a href="#cb22-179" aria-hidden="true" tabindex="-1"></a>And then a quick test to make sure the database is ok.</span>
<span id="cb22-180"><a href="#cb22-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-183"><a href="#cb22-183" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-184"><a href="#cb22-184" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> TEST_virsorter2</span>
<span id="cb22-185"><a href="#cb22-185" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> TEST_virsorter2</span>
<span id="cb22-186"><a href="#cb22-186" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> test.fa https://raw.githubusercontent.com/jiarong/VirSorter2/master/test/8seq.fa</span>
<span id="cb22-187"><a href="#cb22-187" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">-w</span> test.out <span class="at">-i</span> test.fa <span class="at">-j</span> <span class="va">$NSLOTS</span> <span class="at">--db-dir</span> dbs/virsorter2/ <span class="at">--tmpdir</span> TEST_virsorter2/tmp_vir <span class="at">--rm-tmpdir</span> <span class="at">--min-length</span> 1500 all</span>
<span id="cb22-188"><a href="#cb22-188" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-189"><a href="#cb22-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-190"><a href="#cb22-190" aria-hidden="true" tabindex="-1"></a>The uncompressed database is a little over 10GB.</span>
<span id="cb22-191"><a href="#cb22-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-192"><a href="#cb22-192" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple" collapse="true"}</span>
<span id="cb22-193"><a href="#cb22-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-194"><a href="#cb22-194" aria-hidden="true" tabindex="-1"></a><span class="fu">### Expand for the VIRSORTER2_DB_BUILD Hydra script</span></span>
<span id="cb22-195"><a href="#cb22-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-196"><a href="#cb22-196" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_JOB_VIRSORTER2_DB_BUILD.qmd &gt;}}</span>
<span id="cb22-197"><a href="#cb22-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-198"><a href="#cb22-198" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb22-199"><a href="#cb22-199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-200"><a href="#cb22-200" aria-hidden="true" tabindex="-1"></a><span class="fu">### Single Copy Gene (SCG) taxonomy</span></span>
<span id="cb22-201"><a href="#cb22-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-202"><a href="#cb22-202" aria-hidden="true" tabindex="-1"></a>Anvi'o has native support for building tRNA and Single Copy Gene (SCG) taxonomy from the <span class="co">[</span><span class="ot">Genome Taxonomy Database (GTDB)</span><span class="co">](https://gtdb.ecogenomic.org/)</span>. </span>
<span id="cb22-203"><a href="#cb22-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-206"><a href="#cb22-206" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-207"><a href="#cb22-207" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-scg-taxonomy</span> <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-208"><a href="#cb22-208" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-trna-taxonomy</span> <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-209"><a href="#cb22-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-210"><a href="#cb22-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-211"><a href="#cb22-211" aria-hidden="true" tabindex="-1"></a><span class="fu">## Functional Annotations</span></span>
<span id="cb22-212"><a href="#cb22-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-213"><a href="#cb22-213" aria-hidden="true" tabindex="-1"></a>Now we can install databases for **functional** annotation. Anvi'o has native support for building <span class="co">[</span><span class="ot">Pfam</span><span class="co">](https://pfam.xfam.org/)</span>, <span class="co">[</span><span class="ot">COG</span><span class="co">](https://www.ncbi.nlm.nih.gov/COG/)</span>,  <span class="co">[</span><span class="ot">KEGG</span><span class="co">](https://www.genome.jp/tools/kofamkoala/)</span>, and <span class="co">[</span><span class="ot">CAZymes</span><span class="co">](http://www.cazy.org/)</span> databases.</span>
<span id="cb22-214"><a href="#cb22-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-215"><a href="#cb22-215" aria-hidden="true" tabindex="-1"></a>Anvi'o makes this super simple.</span>
<span id="cb22-216"><a href="#cb22-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-219"><a href="#cb22-219" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb22-220"><a href="#cb22-220" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-pfams</span> <span class="at">--pfam-data-dir</span> dbs/pfam_db</span>
<span id="cb22-221"><a href="#cb22-221" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-kegg-data</span> <span class="at">--mode</span> all <span class="at">--kegg-data-dir</span> dbs/kegg_kofam  <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-222"><a href="#cb22-222" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-ncbi-cogs</span> <span class="at">--cog-data-dir</span> dbs/cog_db <span class="at">-T</span> <span class="va">$NSLOTS</span></span>
<span id="cb22-223"><a href="#cb22-223" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-setup-cazymes</span> <span class="at">--cazyme-data-dir</span> dbs/cazymes</span>
<span id="cb22-224"><a href="#cb22-224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb22-225"><a href="#cb22-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-226"><a href="#cb22-226" aria-hidden="true" tabindex="-1"></a>And that’s it. We can add more databases as we need them.</span>
<span id="cb22-227"><a href="#cb22-227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-228"><a href="#cb22-228" aria-hidden="true" tabindex="-1"></a><span class="fu"># Concluding remarks</span></span>
<span id="cb22-229"><a href="#cb22-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-230"><a href="#cb22-230" aria-hidden="true" tabindex="-1"></a>At this point we should be good to go with the main setup. We may need to install other tools along the way---we can add those instructions here. Or you may decide to use other tools instead. For example, we are using <span class="co">[</span><span class="ot">megahit</span><span class="co">](https://github.com/voutcn/megahit)</span> for the assemblies but you may want to use <span class="co">[</span><span class="ot">metaspades</span><span class="co">](https://github.com/ablab/spades)</span> or <span class="co">[</span><span class="ot">idba_ud</span><span class="co">](https://github.com/loneknightpy/idba)</span>. These packages need to be installed separately.</span>
<span id="cb22-231"><a href="#cb22-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-232"><a href="#cb22-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-233"><a href="#cb22-233" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_footer.qmd &gt;}}</span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../../../site_info.html">
<p>Site Information</p>
</a>
  </li>  
</ul>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/istmobiome/trans-shrimp/blob/main/workflows/mg/setup/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/edit/main/workflows/mg/setup/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>.</p>
</div>
  </div>
</footer>


</body></html>