<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Data Portal – TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<link href="../../assets/favicon.ico" rel="icon">
<script src="../../site_libs/quarto-listing/list.min.js"></script>
<script src="../../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark-d7d32a8442357cea8cf74a4a848b80d6.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-med-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-name','listing-filename','listing-description',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-filename-sort'] }],
      
      searchColumns: ["listing-name","listing-filename","listing-description"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-med-listing'] = new List('listing-med-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script><script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../images/icon.png" alt="Istmobiome logo." class="navbar-logo light-content"><img src="../../images/icon.png" alt="Istmobiome logo." class="navbar-logo dark-content"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../workflows/portal/index.html"> 
<span class="menu-text">Data Portal</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../workflows/ssu/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../workflows/mg/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../workflows/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/shrimp-draft/"> <i class="bi bi-github" role="img" aria-label="Trans Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <h1 class="quarto-secondary-nav-title no-breadcrumbs">Data Portal</h1>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/portal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Portal</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../workflows/ssu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/ssu/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/ssu/dada2/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. DADA2 ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/ssu/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/ssu/med/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../workflows/mg/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/mg/setup/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Build Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/mg/assembly/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/mg/aa-results/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Assembly &amp; Annotation Results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../workflows/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../workflows/pub/archive_data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting &amp; Archiving Data/Code</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
        
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title d-none d-lg-block">Data Portal</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><div class="column-body-outset-right">
<p>Quick access to fastq sequence files, processing scripts, and curated datasets. If you want to process the data yourself, select a pipeline, download the sequence data and processing scripts, and run the workflow.</p>
</div>
<div class="column-body-outset-right">

<style type="text/css">
/* keep global nav-tabs tweaks */
.nav-tabs {
  margin-top: 0.5rem;
  border-bottom: none;
}

.callout {
  margin-top: 0;
}

/* --------------------- */
/* Scoped only to #index-chooser */
#index-chooser .nav-link {
  text-align: center;
  margin-right: 15px;
  margin-top: 10px;
  width: 155px;
  font-size: 1em; /* changed from 0.9 to 1.0 by jjs */
  font-weight: 600;
  display: flex;
  flex-direction: column;
  justify-content: center;  /* vertically center content */
  align-items: center;      /* horizontally center content */
  min-height: 6rem;   
}

#index-chooser .nav-link,
#index-chooser .nav-link.active,
#index-chooser .nav-item.show .nav-link {
  border: 1px solid rgb(222, 226, 230);
  border-radius: 10px;
  color: rgb(80,146,221);
}

#index-chooser .nav-link:hover {
   border-color: rgb(80,146,221);
   border-width: 1px;
} 

#index-chooser .nav-link.active,
#index-chooser .nav-item.show .nav-link {
  border-color: rgb(80,146,221);
  border-width: 2px;
}

#index-chooser .nav-link i {
  display: block;
  font-size: 3rem;
  color: rgb(80,146,221);
  margin-bottom: 5px;
}

#index-chooser .nav-link img {
  display: block;
  max-width: 100%;    /* don’t exceed box width */
  max-height: 4rem;   /* don’t exceed 4rem height */
  width: auto;        /* keep aspect ratio */
  height: auto;       /* keep aspect ratio */
  margin: 0 auto 5px auto; /* center + bottom margin */
}
/* --------------------- */

/* keep global listing tweaks */
.quarto-listing {
  margin-top: 2em;
}

.quarto-listing .listing-name,
.quarto-listing .listing-author {
  white-space: nowrap;
}

.quarto-listing .listing-actions-group h3 {
  margin-top: 0;
}
</style>
<ul id="index-chooser" class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation">
    <a class="nav-link" href="data-dada2.html">
      <img src="../../images/dada2_icon.png" alt="dada2 icon">
      16S rRNA Dada2 processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-otu.html">
      <img src="../../images/mothur_icon.png" alt="mothur icon">
      16S rRNA OTU processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-med.html">
      <img src="../../images/med_icon.png" alt="med icon">
      16S rRNA MED processing
    </a>
  </li>
  <li class="nav-item" role="presentation">
    <a class="nav-link" href="data-mg.html">
      <img src="../../images/anvio_icon.png" alt="anvio icon">
      Metagenomic processing
    </a>
  </li>
</ul>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  // get file name
  const filename = window.location.pathname.split("/").slice(-1)[0];

  // latch active
  const toolLinks = window.document.querySelectorAll("#index-chooser a");
  for (const tool of toolLinks) {
    if (filename && filename !== "index.html") {
      if (tool.href.endsWith(filename)) {
        tool.classList.add("active");
      } 
    } else {
      if (tool.href.endsWith("listing-filters.html")) {
        tool.classList.add("active");
      }
    }
  }
  
  // move heading into table
  document.querySelector(".listing-actions-group").prepend(document.querySelector("h3.unlisted"));
});

</script>
</div>
<p><br></p>
<section id="data-scripts" class="level2 unlisted page-columns page-full"><h2 class="unlisted anchored" data-anchor-id="data-scripts">Data &amp; Scripts</h2>
<div class="callout callout-style-default callout-note no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>A note about curated data
</div>
</div>
<div class="callout-body-container callout-body">
<p>Curated means that after processing, the data has been cleaned of unwanted taxa (e.g., NA kingdoms, potential contaminants, etc). Negative control &amp; low-count samples have also been removed. These data are ready for downstream analysis.</p>
</div>
</div>
<p>Here we provide two options for processing sequence data–the detailed pipeline (listed in the table) and a <a href="#med-processing">quick set of commands linked below</a>.</p>
<div id="listing-med-listing" class="column-body-outset-right quarto-listing quarto-listing-container-table">
<table class="quarto-listing-table table">
<thead><tr>
<th>
Asset
</th>

<th>
File Name
</th>

<th>
Description
</th>

</tr></thead>
<tbody class="list">
<tr data-index="0" data-listing-filename-sort="med_curated_data.zip">
<td>
<span class="listing-name">Curated Data</span>
</td>
<td>
<a href="../../share/ssu/med/med_curated_data.zip" class="filename listing-filename">med_curated_data.zip</a>
</td>
<td>
<span class="listing-description">Archive containing the curated output of MED pipeline–-OTU table, taxonomy table, sample metadata, &amp; representative fasta sequences. Data is available as a) stand-alone text files, b) bundled in a microtable object, and c) bundled in a phyloseq object. Archive also includes a table tracking sample read changes. </span>
</td>

</tr>
<tr data-index="1" data-listing-filename-sort="PRJEB36632">
<td>
<span class="listing-name">Sequence Data</span>
</td>
<td>
<a href="https://www.ebi.ac.uk/ena/browser/view/PRJEB36632" class="filename listing-filename">PRJEB36632</a>
</td>
<td>
<span class="listing-description">Trimmed (primers removed) 16S rRNA data from 1909 samples. </span>
</td>

</tr>
<tr data-index="2" data-listing-filename-sort="med_processing.zip">
<td>
<span class="listing-name">Processing Data &amp; Scripts</span>
</td>
<td>
<a href="../../share/ssu/med/med_processing.zip" class="filename listing-filename">med_processing.zip</a>
</td>
<td>
<span class="listing-description">Archived directory containing fastq rename tables, fastq rename results, sample data, mothur batchfile script, MED prep scripts, MED mapping file, pipeline read changes, &amp; Hydra job scripts. </span>
</td>

</tr>
<tr data-index="3" data-listing-filename-sort="MED workflow">
<td>
<span class="listing-name">Detailed Pipeline</span>
</td>
<td>
<a href="../../workflows/ssu/med/" class="filename listing-filename">MED workflow</a>
</td>
<td>
<span class="listing-description">Detailed workflow for 16S rRNA analysis using MED. Workflow uses paired end reads, beginning with raw fastq files, ending with sequence &amp; taxonomy tables. Microtable &amp; phyloseq objects are produced to collate the data for downstream analysis. </span>
</td>

</tr>
</tbody>
</table>
<div class="listing-no-matching d-none">No matching items</div>
</div>
</section><section id="med-processing" class="level2 column-body-outset-right"><h2 class="anchored" data-anchor-id="med-processing">MED Processing</h2>
<p>Processing scripts for <a href="https://merenlab.org/2014/11/04/med/">Minimum Entropy Decomposition (MED)</a> analysis. The pipeline begins with the output fasta and count files from the <code>align.seqs</code> portion of the mothur OTU pipeline. From there we use R and mothur to remove negative control samples, check for chimera, and run taxonomic classifications. It is important to note that the MED workflow does not <code>precluster</code> sequences (as in the mothur pipeline) because MED relies on every sequence (including redundant reads) for the analysis. This pipeline has four main steps:</p>
<ol type="1">
<li>run the mothur workflow.<br>
</li>
<li>modify and run the <code>mothur2oligo.sh</code> script. This script transforms the mothur output to appropriate format for MED. It must be run in the mothur environment because the script uses mothur. We need access to the following mothur files to run this script.
<ol type="i">
<li>
<strong>taxonomy file</strong>: <code>final_med.taxonomy</code><br>
</li>
<li>
<strong>count file</strong>: <code>final_med.count_table</code><br>
</li>
<li>
<strong>fasta file</strong>: <code>final_med.fasta</code><br>
</li>
</ol>
</li>
<li>trim uninformative columns from alignment (in the MED environment)<br>
</li>
<li>run the MED command</li>
</ol>
<div class="cell">
<details class="code-fold"><summary>mothur processing script for MED analysis</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export DATA=01_TRIMMED_DATA/</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export TYPE=gz</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export PROC=30</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export REF_LOC=reference_dbs</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export TAXREF_FASTA=gsrdb.fasta</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export TAXREF_TAX=gsrdb.tax</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export ALIGNREF=silva.v4.fasta</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="co">#$ export CONTAMINENTS=Chloroplast-Mitochondria-unknown-Eukaryota</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="co">#### THIS PIPELINE BEGINS AFTER ALIGN.SEQS (AND ASSOCIATED SCREENING STEP)</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="co">#### IN THE MOTHUR OTU PIPLINE. IF YOU HAVE NO USE FOR STRICT OTU ANAYSIS</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="co">#### SIMPLY UNCOMMENT THE MOTHUR COMMANDS BELOW TO RUN THE FULL PIPELINE HERE</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="ex">set.dir</span><span class="er">(</span><span class="va">output</span><span class="op">=</span>pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># UNCOMMENT FROM HERE....</span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co">#make.file(inputdir=$DATA/, type=$TYPE, prefix=shrimp)</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co">#### had to fix shimp.files bc mothur splits name at first underscore (_)</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co">#make.contigs(file=shrimp.files, processors=$PROC)</span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=shrimp.trim.contigs.fasta, count=shrimp.contigs.count_table, processors=$PROC)</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co">#count.groups(count=shrimp.contigs.count_table)</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a><span class="co">#screen.seqs(fasta=current, count=current, maxambig=0, minlength=252, maxlength=254, maxhomop=6, processors=$PROC)</span></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=current, count=current, processors=$PROC)</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co">#count.groups(count=current)</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co">#unique.seqs(fasta=current, count=current)</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(count=current, processors=$PROC)</span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co">#### Aligning reads</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a><span class="co"># https://mothur.org/wiki/silva_reference_files/</span></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="co">#### Prep reference file</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="co">#pcr.seqs(fasta=$REF_LOC/silva.seed_v138_2.align, start=13862, end=23445, keepdots=F, processors=$PROC)</span></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="co">#rename.file(input=pipelineFiles/silva.seed_v138_2.pcr.align, new=pipelineFiles/$ALIGNREF)</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=pipelineFiles/$ALIGNREF, processors=$PROC)</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co">#### Align reads</span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="co">#align.seqs(fasta=shrimp.trim.contigs.good.unique.fasta, reference=pipelineFiles/$ALIGNREF, processors=$PROC)</span></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=shrimp.trim.contigs.good.unique.align, count=shrimp.trim.contigs.good.count_table, processors=$PROC)</span></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="co">#### Further processing</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a><span class="co">#screen.seqs(fasta=current, count=current, start=1, end=9583, processors=$PROC)</span></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=current, count=current, processors=$PROC)</span></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="co">#count.groups(count=current)</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a><span class="co">#filter.seqs(fasta=current, vertical=T, trump=., processors=$PROC)</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a><span class="co">#unique.seqs(fasta=current, count=current)</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="co">#summary.seqs(fasta=current, count=current, processors=$PROC)</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="co"># ....TO HERE</span></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co">###############################################################################</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a><span class="co">#### Copy mothur output files</span></span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="ex">system</span><span class="er">(</span><span class="fu">cp</span> pipelineFiles/shrimp.trim.contigs.good.unique.good.filter.unique.fasta pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="ex">system</span><span class="er">(</span><span class="fu">cp</span> pipelineFiles/shrimp.trim.contigs.good.unique.good.filter.count_table pipelineFiles_med/<span class="kw">)</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="co">#### Remove Negative Control samples (files generated in R)</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="co">########## IN R ########################################################</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_accnos &lt;- readr::read_delim(here(work_here, "nc_screen/shrimp.files"), delim = "\t", col_names = FALSE)</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_accnos[, 2:3] &lt;- NULL</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_accnos &lt;- tmp_accnos[grepl("Control_", tmp_accnos$X1), ]</span></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_delim(tmp_accnos, file = here(work_here, "nc_screen/nc_samples.accnos"), col_names = FALSE)</span></span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="ex">get.groups</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.count_table, <span class="va">accnos</span><span class="op">=</span>nc_samples.accnos<span class="kw">)</span></span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, <span class="va">new</span><span class="op">=</span>nc.fasta<span class="kw">)</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table, <span class="va">new</span><span class="op">=</span>nc.count_table<span class="kw">)</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>nc.fasta, <span class="va">count</span><span class="op">=</span>nc.count_table, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a><span class="ex">list.seqs</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>nc.count_table<span class="kw">)</span></span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a><span class="ex">count.seqs</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>nc.count_table, <span class="va">compress</span><span class="op">=</span>f<span class="kw">)</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a><span class="ex">get.seqs</span><span class="er">(</span><span class="va">accnos</span><span class="op">=</span>nc.accnos, <span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.count_table<span class="kw">)</span></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, <span class="va">new</span><span class="op">=</span>subset.fasta<span class="kw">)</span></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="ex">rename.file</span><span class="er">(</span><span class="va">input</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table, <span class="va">new</span><span class="op">=</span>subset.count_table<span class="kw">)</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a><span class="ex">count.seqs</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>subset.count_table, <span class="va">compress</span><span class="op">=</span>f<span class="kw">)</span></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a><span class="co">########## IN R ########################################################</span></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a><span class="co">#full_count_tab &lt;- readr::read_delim(here(work_here, "nc_screen/subset.full.count_table"), delim = "\t", col_names = TRUE)</span></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a><span class="co">#control_cols     &lt;- grep("^Control_", names(full_count_tab), value = TRUE)</span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="co">#noncontrol_cols  &lt;- setdiff(names(full_count_tab)[-(1:2)], control_cols)</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="co">#read_totals &lt;- full_count_tab %&gt;%</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a><span class="co">#  rowwise() %&gt;%</span></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(</span></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="co">#    total_reads_nc   = sum(c_across(all_of(control_cols)), na.rm = TRUE),</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="co">#    total_reads_non_nc = sum(c_across(all_of(noncontrol_cols)), na.rm = TRUE)</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a><span class="co">#  ) %&gt;%</span></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="co">#  ungroup() %&gt;%</span></span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(1, 2, total_reads_nc, total_reads_non_nc)</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a><span class="co">#read_totals &lt;- read_totals %&gt;% dplyr::rename("total_reads" = 2)</span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_read_totals &lt;- read_totals %&gt;% dplyr::mutate(perc_reads_in_nc = 100*(total_reads_nc / (total_reads_nc + total_reads_non_nc)), .after = "total_reads_non_nc")</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_read_totals$perc_reads_in_nc &lt;- round(tmp_read_totals$perc_reads_in_nc, digits = 6)</span></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="co">#control_cols     &lt;- grep("^Control_", names(full_count_tab), value = TRUE)</span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="co">#noncontrol_cols  &lt;- setdiff(names(full_count_tab)[-(1:2)], control_cols)</span></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="co"># rowwise tally of non-zero columns</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a><span class="co">#samp_totals &lt;- full_count_tab %&gt;%</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="co">#  rowwise() %&gt;%</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="co">#  mutate(</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="co">#    num_nc_samp     = sum(c_across(all_of(control_cols)) != 0, na.rm = TRUE),</span></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a><span class="co">#    num_non_nc_samp = sum(c_across(all_of(noncontrol_cols)) != 0, na.rm = TRUE)</span></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a><span class="co">#  ) %&gt;%</span></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a><span class="co">#  ungroup() %&gt;%</span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a><span class="co">#  select(1, num_nc_samp, num_non_nc_samp)</span></span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a><span class="co">#samp_totals$total_samp &lt;- samp_totals$num_nc_samp + samp_totals$num_non_nc_samp</span></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a><span class="co">#samp_totals &lt;- samp_totals %&gt;%  dplyr::relocate("total_samp", .after = "Representative_Sequence")</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a><span class="co">#samp_totals &lt;- samp_totals %&gt;% dplyr::mutate(perc_nc_samp = 100*( num_nc_samp / (num_nc_samp + num_non_nc_samp)), .after = "num_non_nc_samp")</span></span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a><span class="co">#nc_check &lt;- dplyr::left_join(tmp_read_totals, samp_totals, by = "Representative_Sequence")</span></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="co">#write_delim(nc_check, here(work_here, "nc_screen/reads_in_nc_samples.txt"), delim = "\t")</span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a><span class="co">#nc_remove &lt;- nc_check %&gt;% filter(perc_reads_in_nc &gt; 10 | perc_nc_samp &gt; 10)</span></span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a><span class="co">#nc_remain &lt;- dplyr::anti_join(nc_check, nc_remove)</span></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a><span class="co">#rem_nc_reads &lt;- sum(nc_remove$total_reads_nc)</span></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a><span class="co">#rem_sam_reads &lt;- sum(nc_remove$total_reads_non_nc)</span></span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a><span class="co">#per_reads_rem &lt;- round(100*( rem_nc_reads / (rem_nc_reads + rem_sam_reads)), digits = 3)</span></span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a><span class="co">#ret_nc_reads &lt;- sum(nc_remain$total_reads_nc)</span></span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a><span class="co">#ret_sam_reads &lt;- sum(nc_remain$total_reads_non_nc)</span></span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="co">#per_reads_ret &lt;- round(100*( ret_nc_reads / (ret_nc_reads + ret_sam_reads)), digits = 3)</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.seqs</span><span class="er">(</span><span class="va">accnos</span><span class="op">=</span>nc_repseq_remove.accnos, <span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.count_table<span class="kw">)</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table<span class="kw">)</span></span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a><span class="co">########## IN R ########################################################</span></span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_before &lt;- read_tsv(here(work_here, "nc_screen/shrimp.trim.contigs.good.unique.good.filter.count.summary"), col_names = FALSE, col_select = 1)</span></span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_after &lt;- read_tsv(here(work_here, "nc_screen/shrimp.trim.contigs.good.unique.good.filter.pick.count.summary"), col_names = FALSE, col_select = 1)</span></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_nc_lost &lt;- anti_join(tmp_before, tmp_after)</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_nc_lost$X1</span></span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a><span class="co">#nc_to_remove &lt;- semi_join(tmp_before, tmp_after)</span></span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a><span class="co">#nc_to_remove &lt;- nc_to_remove %&gt;% dplyr::filter(stringr::str_starts(X1, "Control")) </span></span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a><span class="co">#readr::write_delim(nc_to_remove, file = here(work_here, "nc_screen/nc_samples_remove.accnos"), col_names = FALSE)</span></span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a><span class="co">########################################################################</span></span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.count_table, <span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, <span class="va">accnos</span><span class="op">=</span>nc_samples_remove.accnos<span class="kw">)</span></span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a><span class="co">### NEGATIVE CONTROLS Should be GONE </span><span class="al">###</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a><span class="co">########################################</span></span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table<span class="kw">)</span></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="ex">chimera.vsearch</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table, <span class="va">dereplicate</span><span class="op">=</span>t, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table<span class="kw">)</span></span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a><span class="ex">classify.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, <span class="va">reference</span><span class="op">=</span>reference_dbs/gsrdb.fasta, <span class="va">taxonomy</span><span class="op">=</span>reference_dbs/gsrdb.tax, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a><span class="ex">remove.lineage</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, <span class="va">taxonomy</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.taxonomy, <span class="va">taxon</span><span class="op">=</span>Chloroplast-Mitochondria-unknown-Eukaryota<span class="kw">)</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.seqs</span><span class="er">(</span><span class="va">fasta</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.pick.fasta, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table, <span class="va">processors</span><span class="op">=</span><span class="va">$PROC</span><span class="kw">)</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a><span class="ex">summary.tax</span><span class="er">(</span><span class="va">taxonomy</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.taxonomy, <span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table<span class="kw">)</span></span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a><span class="ex">count.groups</span><span class="er">(</span><span class="va">count</span><span class="op">=</span>shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table<span class="kw">)</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a><span class="co">############################################</span></span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a><span class="co">### PROCEED To  mothur2oligo.sh pipeline </span><span class="al">###</span></span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a><span class="co">### MUST BE RUN IN MOTHUR ENVIRONMENT  #####</span></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a><span class="co">############################################</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Once you have the script and data you simply run the pipeline like so.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb2"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="ex">mothur</span> med_batchfile_processing</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Once the mothur portion of the workflow is complete, the script <a href="https://github.com/DenefLab/MicrobeMiseq/tree/master/mothur2oligo"><code>mothur2oligo.sh</code></a> needs to be run in the <code>mothur</code> environment and modified for your specific purposes. You should not need to modify the associated <code>renamer.pl</code> script but it does need to be in the same location as <code>mothur2oligo.sh</code>. You can download the modified script we used in the table above.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> mothur2oligo.sh</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist">
<li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" href="">mothur2oligo</a></li>
<li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false" href="">renamer</a></li>
</ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="cell">
<details class="code-fold"><summary>Expand for the <code>mothur2oligo.sh</code> script</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># USAGE: sh mothur2oligo.sh</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># This is a shell script for transforming mothur output to appropriate format for </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># A. murat Eren's oligotyping pipeline </span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a><span class="co">## Set variables</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjust the file names to your own study - these are the files from the mothur SOP</span></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="va">taxonomy</span><span class="op">=</span><span class="st">"final_med.taxonomy"</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="va">fasta</span><span class="op">=</span><span class="st">"final_med.fasta"</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a><span class="va">count</span><span class="op">=</span><span class="st">"final_med.count_table"</span></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="va">processors</span><span class="op">=</span>3</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the taxon you want to select, separate taxonomic levels with ";" </span></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Do not touch inner and outer quotes</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="va">taxon</span><span class="op">=</span><span class="st">"'Bacteria;-Archaea;'"</span></span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a><span class="co">########## Script  #############</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a><span class="co">################################</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a><span class="va">redundantFasta</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${fasta}</span>.pick.redundant.fasta <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.fasta//'</span><span class="va">)</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="va">groups</span><span class="op">=</span><span class="va">$(</span><span class="bu">echo</span> <span class="va">${count}</span>.pick.redundant.groups <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/.count_table//'</span><span class="va">)</span> </span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Call mothur commands for generating deuniqued fasta file for a specific lineage</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a><span class="ex">mothur</span> <span class="st">"#set.current(processors=</span><span class="va">$processors</span><span class="st">); get.lineage(taxonomy=</span><span class="va">$taxonomy</span><span class="st">, taxon=</span><span class="va">$taxon</span><span class="st">, count=</span><span class="va">$count</span><span class="st">); list.seqs(count=current); get.seqs(accnos=current, fasta=</span><span class="va">$fasta</span><span class="st">); deunique.seqs(fasta=current, count=current)"</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace all "_" in fasta header with a ":"</span></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="va">$groups</span> <span class="kw">|</span> <span class="fu">sed</span> <span class="st">'s/_/:/g'</span> <span class="op">&gt;</span> intermediate1</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Make a file which maps sample names to sequence headers</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="fu">paste</span> <span class="va">$groups</span> intermediate1 <span class="kw">|</span> <span class="fu">awk</span> <span class="st">'BEGIN{FS="\t"}{print $1"\t"$2"_"$3}'</span> <span class="op">&gt;</span> intermediate2</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Perl script to rename the headers of the fasta to include the sample name at the beginning followed by a "_"</span></span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> renamer.pl <span class="va">$redundantFasta</span> intermediate2 </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="cell">
<details class="code-fold"><summary>and the companion <code>renamer.pl</code> script</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co">#! /usr/bin/perl</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#from http://www.perlmonks.org/?node_id=975419</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a><span class="ex">use</span> strict<span class="kw">;</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="ex">use</span> warnings<span class="kw">;</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="ex">@ARGV</span> == 2 or die <span class="st">"usage: </span><span class="va">$0</span><span class="st"> &lt;multifasta file&gt; &lt;header replacement fil</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a><span class="st">+e&gt;\n"</span><span class="kw">;</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> <span class="er">(</span> <span class="va">$fasta_file</span><span class="ex">,</span> <span class="va">$header_file</span> <span class="kw">)</span> <span class="ex">=</span> @ARGV<span class="kw">;</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> <span class="va">$destination</span> = <span class="va">$fasta_file</span> . <span class="st">'_headers-replaced.fasta'</span><span class="kw">;</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> IN2, <span class="st">'&lt;'</span>, <span class="va">$header_file</span> or die <span class="st">"Can't read from tab-delimited head</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a><span class="st">+er replacement file </span><span class="va">$header_file</span><span class="st">: </span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="ex">my</span> %head_seqs<span class="kw">;</span></span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="kw">(</span> <span class="op">&lt;</span>IN2<span class="op">&gt;</span> <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>    <span class="ex">chomp</span><span class="kw">;</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>    <span class="ex">my</span> <span class="er">(</span> <span class="va">$old</span><span class="ex">,</span> <span class="va">$new</span> <span class="kw">)</span> <span class="ex">=</span> split /\t/<span class="kw">;</span></span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>    <span class="va">$head_seqs</span><span class="ex">{</span> <span class="va">$old</span> } = <span class="va">$new</span><span class="kw">;</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>    <span class="kw">}</span></span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> IN2<span class="kw">;</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> IN1, <span class="st">'&lt;'</span>, <span class="va">$fasta_file</span> or die <span class="st">"Can't read from multifasta file wit</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="st">+h alternating lines of headers and sequences </span><span class="va">$fasta_file</span><span class="st">: </span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="ex">open</span> OUT, <span class="st">'&gt;'</span>, <span class="va">$destination</span> or die <span class="st">"Can't write to file </span><span class="va">$destination</span><span class="st">: </span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a><span class="st">+</span><span class="va">$!</span><span class="st">\n"</span><span class="kw">;</span>    </span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> <span class="kw">(</span> <span class="op">&lt;</span>IN1<span class="op">&gt;</span> <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-32"><a href="#cb5-32" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> <span class="kw">(</span> <span class="ex">/^</span><span class="op">&gt;(</span><span class="bu">.</span>+<span class="op">)</span>$/ <span class="kw">&amp;&amp;</span> <span class="ex">exists</span> <span class="va">$head_seqs</span>{ <span class="va">$1</span> } <span class="kw">)</span> <span class="kw">{</span></span>
<span id="cb5-33"><a href="#cb5-33" aria-hidden="true" tabindex="-1"></a>        <span class="va">$_</span> = <span class="st">"&gt;</span><span class="va">$head_seqs</span><span class="st">{ </span><span class="va">$1</span><span class="st"> }\n"</span><span class="kw">;</span></span>
<span id="cb5-34"><a href="#cb5-34" aria-hidden="true" tabindex="-1"></a>        <span class="kw">}</span></span>
<span id="cb5-35"><a href="#cb5-35" aria-hidden="true" tabindex="-1"></a>    <span class="ex">print</span> OUT <span class="va">$_</span><span class="kw">;</span></span>
<span id="cb5-36"><a href="#cb5-36" aria-hidden="true" tabindex="-1"></a>    <span class="kw">}</span></span>
<span id="cb5-37"><a href="#cb5-37" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> IN1<span class="kw">;</span></span>
<span id="cb5-38"><a href="#cb5-38" aria-hidden="true" tabindex="-1"></a><span class="ex">close</span> OUT<span class="kw">;</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
</div>
</div>
</div>
<p>Great. Now within the oligotype/MED environment run the following commands for the MED analysis. You will need the <code>mapping.txt</code> file linked above for this step.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb6"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate oligotyping</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="ex">o-trim-uninformative-columns-from-alignment</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>        final_med.pick.redundant.fasta_headers-replaced.fasta</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">decompose</span> final_med.pick.redundant.fasta_headers-replaced.fasta-TRIMMED <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">-E</span> mapping.txt <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output-directory</span> MED <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">--number-of-threads</span> 24 <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">--skip-gen-figures</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>In the resources listed above, we include a table that summarizes read changes for each sample through the pipeline.</p>
</section><div class="column-body-outset-right">

</div>
<div id="quarto-appendix" class="default"><section id="detailed-session-info" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Detailed Session Info</h2><div class="quarto-appendix-contents">
<details><summary>Expand to see Session Info</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Ventura 13.7.8
 system   x86_64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2025-10-03
 pandoc   3.6.3 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/x86_64/ (via rmarkdown)
 quarto   1.8.25 @ /Applications/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
======   Devtools Session info   ===================================</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.</code></pre>
</div>
</div>
</details></div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-10-03</p>



<!-- -->

</div></section><section class="level4 appendix"><h2 class="quarto-appendix-heading">Collected Links</h2><div class="quarto-appendix-contents">
<ul>
<li><a href="#med-processing">quick set of commands linked below</a></li>
<li><a href="https://merenlab.org/2014/11/04/med/">Minimum Entropy Decomposition (MED)</a></li>
<li><a href="https://github.com/DenefLab/MicrobeMiseq/tree/master/mothur2oligo"><code>mothur2oligo.sh</code></a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://istmobiome.rbind.io/">The Istmobiome Project</a></li>
</ul></div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/shrimp-draft\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb10" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Data Portal"</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="an">listing:</span><span class="co"> </span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co">  id: med-listing</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co">  contents: listings/data-med.yml</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="an">metadata-files:</span><span class="co"> </span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">  - listings/_metadata.yml</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_listing-preamble.qmd &gt;}}</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>Here we provide two options for processing sequence data--the detailed pipeline (listed in the table) and a <span class="co">[</span><span class="ot">quick set of commands linked below</span><span class="co">](#med-processing)</span>. </span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>::: {#med-listing .column-body-outset-right}</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>::: column-body-outset-right</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="fu">## MED Processing </span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>Processing scripts for <span class="co">[</span><span class="ot">Minimum Entropy Decomposition (MED)</span><span class="co">](https://merenlab.org/2014/11/04/med/)</span> analysis. The pipeline begins with the output fasta and count files from the <span class="in">`align.seqs`</span> portion of the mothur OTU pipeline. From there we use R and mothur to remove negative control samples, check for chimera, and run taxonomic classifications. It is important to note that the MED workflow does not <span class="in">`precluster`</span> sequences (as in the mothur pipeline) because MED relies on every sequence (including redundant reads) for the analysis. This pipeline has four main steps:</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>run the mothur workflow.   </span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>modify and run the <span class="in">`mothur2oligo.sh`</span> script. This script transforms the mothur output to appropriate format for MED. It must be run in the mothur environment because the script uses mothur. We need access to the following mothur files to run this script. </span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>    i. **taxonomy file**: <span class="in">`final_med.taxonomy`</span>   </span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>    ii. **count file**: <span class="in">`final_med.count_table`</span>   </span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>    iii. **fasta file**: <span class="in">`final_med.fasta`</span>   </span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>trim uninformative columns from alignment (in the MED environment)     </span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a><span class="ss">4. </span>run the MED command</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "mothur processing script for MED analysis"</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a><span class="ex">{{</span><span class="op">&lt;</span> include <span class="ex">include/scripts/med/_med_batchfile_processing</span> <span class="op">&gt;</span>}}</span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a>Once you have the script and data you simply run the pipeline like so. </span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a><span class="ex">mothur</span> med_batchfile_processing</span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a>Once the mothur portion of the workflow is complete, the script <span class="co">[</span><span class="ot">`mothur2oligo.sh`</span><span class="co">](https://github.com/DenefLab/MicrobeMiseq/tree/master/mothur2oligo)</span> needs to be run in the <span class="in">`mothur`</span> environment and modified for your specific purposes. You should not need to modify the associated <span class="in">`renamer.pl`</span> script but it does need to be in the same location as <span class="in">`mothur2oligo.sh`</span>. You can download the modified script we used in the table above. </span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> mothur2oligo.sh</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a>::: panel-tabset</span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a><span class="fu">## mothur2oligo</span></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Expand for the `mothur2oligo.sh` script"</span></span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a><span class="ex">{{</span><span class="op">&lt;</span> include <span class="ex">include/scripts/med/_mothur2oligo.sh</span> <span class="op">&gt;</span>}}</span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a><span class="fu">## renamer</span></span>
<span id="cb10-77"><a href="#cb10-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-80"><a href="#cb10-80" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb10-81"><a href="#cb10-81" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb10-82"><a href="#cb10-82" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-83"><a href="#cb10-83" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb10-84"><a href="#cb10-84" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "and the companion `renamer.pl` script"</span></span>
<span id="cb10-85"><a href="#cb10-85" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate mothur</span>
<span id="cb10-86"><a href="#cb10-86" aria-hidden="true" tabindex="-1"></a><span class="ex">{{</span><span class="op">&lt;</span> include <span class="ex">include/scripts/med/_renamer.pl</span> <span class="op">&gt;</span>}}</span>
<span id="cb10-87"><a href="#cb10-87" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-88"><a href="#cb10-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-89"><a href="#cb10-89" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb10-90"><a href="#cb10-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-91"><a href="#cb10-91" aria-hidden="true" tabindex="-1"></a>Great. Now within the oligotype/MED environment run the following commands for the MED analysis. You will need the <span class="in">`mapping.txt`</span> file  linked above for this step. </span>
<span id="cb10-92"><a href="#cb10-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-95"><a href="#cb10-95" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb10-96"><a href="#cb10-96" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb10-97"><a href="#cb10-97" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb10-98"><a href="#cb10-98" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate oligotyping</span>
<span id="cb10-99"><a href="#cb10-99" aria-hidden="true" tabindex="-1"></a><span class="ex">o-trim-uninformative-columns-from-alignment</span> <span class="dt">\</span></span>
<span id="cb10-100"><a href="#cb10-100" aria-hidden="true" tabindex="-1"></a>        final_med.pick.redundant.fasta_headers-replaced.fasta</span>
<span id="cb10-101"><a href="#cb10-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-102"><a href="#cb10-102" aria-hidden="true" tabindex="-1"></a><span class="ex">decompose</span> final_med.pick.redundant.fasta_headers-replaced.fasta-TRIMMED <span class="dt">\</span></span>
<span id="cb10-103"><a href="#cb10-103" aria-hidden="true" tabindex="-1"></a>        <span class="at">-E</span> mapping.txt <span class="dt">\</span></span>
<span id="cb10-104"><a href="#cb10-104" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output-directory</span> MED <span class="dt">\</span></span>
<span id="cb10-105"><a href="#cb10-105" aria-hidden="true" tabindex="-1"></a>        <span class="at">--number-of-threads</span> 24 <span class="dt">\</span></span>
<span id="cb10-106"><a href="#cb10-106" aria-hidden="true" tabindex="-1"></a>        <span class="at">--skip-gen-figures</span></span>
<span id="cb10-107"><a href="#cb10-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb10-108"><a href="#cb10-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-109"><a href="#cb10-109" aria-hidden="true" tabindex="-1"></a>In the resources listed above, we include a table that summarizes read changes for each sample through the pipeline.</span>
<span id="cb10-110"><a href="#cb10-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-111"><a href="#cb10-111" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb10-112"><a href="#cb10-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-113"><a href="#cb10-113" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_listing-footer.qmd &gt;}}</span>
<span id="cb10-114"><a href="#cb10-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-115"><a href="#cb10-115" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_footer.qmd &gt;}}</span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../../site_info.html">
<p>Site Information</p>
</a>
  </li>  
</ul>
<div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/shrimp-draft/blob/main/workflows/portal/data-med.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/edit/main/workflows/portal/data-med.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>.</p>
</div>
  </div>
</footer>


</body></html>