<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="This section describes the steps we took to co-assemble the metagenomic samples, classify taxonomy, and assign functions.">
<title>2. Co-Assembly &amp; Annotations – TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../workflows/mg/aa-results/index.html" rel="next">
<link href="../../../workflows/mg/setup/index.html" rel="prev">
<link href="../../../assets/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-listing/list.min.js"></script>
<script src="../../../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-d7d32a8442357cea8cf74a4a848b80d6.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script src="../../../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../../../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../../../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-snaklemake-workflow .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-name','listing-tool','listing-description','listing-input','listing-reference',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-tool-sort'] }],
      
      searchColumns: ["listing-name","listing-tool","listing-description","listing-input","listing-reference"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-snaklemake-workflow'] = new List('listing-snaklemake-workflow', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script><script src="../../../site_libs/quarto-diagram/mermaid.min.js"></script><script src="../../../site_libs/quarto-diagram/mermaid-init.js"></script><link href="../../../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo light-content"><img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo dark-content"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/portal/index.html"> 
<span class="menu-text">Data Portal</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/ssu/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/mg/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/trans-shrimp/"> <i class="bi bi-github" role="img" aria-label="Trans Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/mg/index.html">Metagenomics</a></li><li class="breadcrumb-item"><a href="../../../workflows/mg/assembly/index.html">2. Co-Assembly &amp; Annotations</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><header id="title-block-header" class="quarto-title-block default page-columns page-full"><div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/mg/index.html">Metagenomics</a></li><li class="breadcrumb-item"><a href="../../../workflows/mg/assembly/index.html">2. Co-Assembly &amp; Annotations</a></li></ol></nav>
      <div class="quarto-title-block"><div><h1 class="title">2. Co-Assembly &amp; Annotations</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
                  <div>
        <div class="description">
          <p>This section describes the steps we took to co-assemble the metagenomic samples, classify taxonomy, and assign functions.</p>
        </div>
      </div>
                </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../../../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/portal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Portal</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/ssu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/dada2/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. DADA2 ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/med/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">5. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/mg/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/setup/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Build Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/assembly/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/aa-results/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Assembly &amp; Annotation Results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/archive_data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting &amp; Archiving Data/Code</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#data-availability" id="toc-data-availability" class="nav-link active" data-scroll-target="#data-availability">Data Availability</a></li>
  <li><a href="#the-pipeline" id="toc-the-pipeline" class="nav-link" data-scroll-target="#the-pipeline">The Pipeline</a></li>
  <li>
<a href="#snakemake-workflow" id="toc-snakemake-workflow" class="nav-link" data-scroll-target="#snakemake-workflow">Snakemake Workflow</a>
  <ul class="collapse">
<li><a href="#snakemake-citations" id="toc-snakemake-citations" class="nav-link" data-scroll-target="#snakemake-citations">Snakemake Citations</a></li>
  </ul>
</li>
  <li><a href="#pipeline-output" id="toc-pipeline-output" class="nav-link" data-scroll-target="#pipeline-output">Pipeline Output</a></li>
  <li>
<a href="#annotations" id="toc-annotations" class="nav-link" data-scroll-target="#annotations">Annotations</a>
  <ul class="collapse">
<li>
<a href="#taxonomic" id="toc-taxonomic" class="nav-link" data-scroll-target="#taxonomic">Taxonomic</a>
  <ul class="collapse">
<li><a href="#short-reads-with-kraken2" id="toc-short-reads-with-kraken2" class="nav-link" data-scroll-target="#short-reads-with-kraken2">Short-reads with Kraken2</a></li>
  <li><a href="#virsorter-annotation" id="toc-virsorter-annotation" class="nav-link" data-scroll-target="#virsorter-annotation">VirSorter Annotation</a></li>
  </ul>
</li>
  <li><a href="#kaiju-annotation" id="toc-kaiju-annotation" class="nav-link" data-scroll-target="#kaiju-annotation">Kaiju Annotation</a></li>
  </ul>
</li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/trans-shrimp/blob/main/workflows/mg/assembly/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/edit/main/workflows/mg/assembly/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block page-columns page-full" id="quarto-document-content"><section id="data-availability" class="level1"><h1>Data Availability</h1>
<p>PENDING</p>
</section><section id="the-pipeline" class="level1 page-columns page-full"><h1>The Pipeline</h1>
<p>In this section of the workflow we begin with raw, paired-end Illumina data. We will use a Snakemake <span class="citation" data-cites="koster2012snakemake">(<a href="#ref-koster2012snakemake" role="doc-biblioref">Köster and Rahmann 2012</a>)</span> workflow within the anvi’o <span class="citation" data-cites="eren2021community">(<a href="#ref-eren2021community" role="doc-biblioref">Eren et al. 2021</a>)</span> environment for most of the steps, though we will at times call on additional tools for specific steps. The overall structure of our workflow was modeled after the one described by Delmont &amp; Eren on <a href="http://merenlab.org/data/tara-oceans-mags/">Recovering Microbial Genomes from TARA Oceans Metagenomes</a> <span class="citation" data-cites="delmont2018nitrogen">(<a href="#ref-delmont2018nitrogen" role="doc-biblioref">Delmont et al. 2018</a>)</span>.</p>
<p>The main steps of this workflow (in order) are:</p>
<div id="listing-snaklemake-workflow" class="column-body quarto-listing quarto-listing-container-table">
<table class="quarto-listing-table table">
<thead><tr>
<th>
Task
</th>

<th>
Tool
</th>

<th>
Description
</th>

<th>
Input
</th>

<th>
reference
</th>

</tr></thead>
<tbody class="list">
<tr data-index="0" data-listing-tool-sort="illumina-utils">
<td>
<span class="listing-name">Quality filtering</span>
</td>
<td>
<a href="https://github.com/merenlab/illumina-utils" class="tool listing-tool">illumina-utils</a>
</td>
<td>
<span class="listing-description">Minoche QC trimmed reads</span>
</td>
<td>
<span class="listing-input">trimmed reads</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="eren2013filtering minoche2011evaluation">(<a href="#ref-eren2013filtering" role="doc-biblioref">Eren et al. 2013</a>; <a href="#ref-minoche2011evaluation" role="doc-biblioref">Minoche, Dohm, and Himmelbauer 2011</a>)</span></span>
</td>

</tr>
<tr data-index="1" data-listing-tool-sort="MEGAHIT">
<td>
<span class="listing-name">Co-Assembly</span>
</td>
<td>
<a href="https://github.com/voutcn/megahit" class="tool listing-tool">MEGAHIT</a>
</td>
<td>
<span class="listing-description">co-assemble samples</span>
</td>
<td>
<span class="listing-input">QC reads</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="li2015megahit">(<a href="#ref-li2015megahit" role="doc-biblioref">D. Li et al. 2015</a>)</span></span>
</td>

</tr>
<tr data-index="2" data-listing-tool-sort="PRODIGAL">
<td>
<span class="listing-name">Gene Calling</span>
</td>
<td>
<a href="https://github.com/hyattpd/Prodigal" class="tool listing-tool">PRODIGAL</a>
</td>
<td>
<span class="listing-description">gene calling</span>
</td>
<td>
<span class="listing-input">assembled contigs</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="hyatt2010prodigal">(<a href="#ref-hyatt2010prodigal" role="doc-biblioref">Hyatt et al. 2010</a>)</span></span>
</td>

</tr>
<tr data-index="3" data-listing-tool-sort="Kraken2">
<td>
<span class="listing-name">Taxonomic Annotation</span>
</td>
<td>
<a href="https://github.com/DerrickWood/kraken2" class="tool listing-tool">Kraken2</a>
</td>
<td>
<span class="listing-description">short read classification</span>
</td>
<td>
<span class="listing-input">QC reads</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="wood2019kraken2">(<a href="#ref-wood2019kraken2" role="doc-biblioref">Wood, Lu, and Langmead 2019</a>)</span></span>
</td>

</tr>
<tr data-index="4" data-listing-tool-sort="BOWTIE2">
<td>
<span class="listing-name">Recruitment</span>
</td>
<td>
<a href="https://github.com/BenLangmead/bowtie2" class="tool listing-tool">BOWTIE2</a>
</td>
<td>
<span class="listing-description">map short reads to assembly</span>
</td>
<td>
<span class="listing-input">QC reads, contigs db</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="langmead2012bowtie">(<a href="#ref-langmead2012bowtie" role="doc-biblioref">Langmead and Salzberg 2012</a>)</span></span>
</td>

</tr>
<tr data-index="5" data-listing-tool-sort="SAMtools">
<td>
<span class="listing-name">Recruitment</span>
</td>
<td>
<a href="https://github.com/samtools/samtools" class="tool listing-tool">SAMtools</a>
</td>
<td>
<span class="listing-description">map short reads to assembly</span>
</td>
<td>
<span class="listing-input">QC reads, contigs db</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="li2009samtools">(<a href="#ref-li2009samtools" role="doc-biblioref">H. Li et al. 2009</a>)</span></span>
</td>

</tr>
<tr data-index="6" data-listing-tool-sort="anvi-profile">
<td>
<span class="listing-name">Profiling</span>
</td>
<td>
<a href="https://anvio.org/help/main/programs/anvi-profile/" class="tool listing-tool">anvi-profile</a>
</td>
<td>
<span class="listing-description">Profile mapping results</span>
</td>
<td>
<span class="listing-input">BAM file</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="eren2021community">(<a href="#ref-eren2021community" role="doc-biblioref">Eren et al. 2021</a>)</span></span>
</td>

</tr>
<tr data-index="7" data-listing-tool-sort="Kaiju">
<td>
<span class="listing-name">Taxonomic Annotation</span>
</td>
<td>
<a href="https://github.com/bioinformaticscentre/kaiju" class="tool listing-tool">Kaiju</a>
</td>
<td>
<span class="listing-description">contig classification</span>
</td>
<td>
<span class="listing-input">contigs (dna format)</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="menzel2016kaiju">(<a href="#ref-menzel2016kaiju" role="doc-biblioref">Menzel, Ng, and Krogh 2016</a>)</span></span>
</td>

</tr>
<tr data-index="8" data-listing-tool-sort="Annotation DBs">
<td>
<span class="listing-name">Functional Annotation</span>
</td>
<td>
<a href="https://anvio.org/help/main/artifacts/functions/" class="tool listing-tool">Annotation DBs</a>
</td>
<td>
<span class="listing-description">gene annotations</span>
</td>
<td>
<span class="listing-input">genes (aa format)</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="mistry2021pfam drula2022cazyme aramaki2020kegg galperin2021cog">(<a href="#ref-mistry2021pfam" role="doc-biblioref">Mistry et al. 2021</a>; <a href="#ref-drula2022cazyme" role="doc-biblioref">Drula et al. 2022</a>; <a href="#ref-aramaki2020kegg" role="doc-biblioref">Aramaki et al. 2020</a>; <a href="#ref-galperin2021cog" role="doc-biblioref">Galperin et al. 2021</a>)</span></span>
</td>

</tr>
<tr data-index="9" data-listing-tool-sort="anvi-merge">
<td>
<span class="listing-name">Merging</span>
</td>
<td>
<a href="https://anvio.org/help/main/programs/anvi-merge/" class="tool listing-tool">anvi-merge</a>
</td>
<td>
<span class="listing-description">create merged profile db</span>
</td>
<td>
<span class="listing-input">individual profile dbs</span>
</td>
<td>
<span class="listing-reference"><span class="citation" data-cites="eren2021community">(<a href="#ref-eren2021community" role="doc-biblioref">Eren et al. 2021</a>)</span></span>
</td>

</tr>
</tbody>
</table>
<div class="listing-no-matching d-none">No matching items</div>
</div>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
  A(QC &lt;br/&gt;metagenomic &lt;br/&gt;reads) --&gt; B("Classify &lt;br/&gt;short reads &lt;br/&gt; (Kraken2)")
  A --&gt; C("Co-assemble &lt;br/&gt; EP &amp; WA&lt;br/&gt; (e.g. Megahit)")
  C --&gt; D(Map Reads&lt;br/&gt;Bowtie &amp; &lt;br/&gt;Samtools)
  C --&gt; E(fa:fa-database Contigs DB &lt;br/&gt; EP &amp; WA)
  D --&gt; F(fa:fa-database Profile DBs&lt;br/&gt; all samples)
  E --&gt; G("Gene Annotations &lt;br/&gt; (e.g. HMMS, PFam,&lt;br/&gt;COG, etc.)")
  E --&gt; H("Classify contigs &lt;br/&gt; (Kaiju)")
  F --&gt; I(fa:fa-database Merge  &lt;br/&gt; Profile DBs &lt;br/&gt; EP &amp; WA)
  G --&gt; J(fa:fa-database Annotated &lt;br/&gt;Contigs DBs &lt;br/&gt; EP &amp; WA)
  H --&gt; J
  I --&gt; K
  J --&gt; K("Analysis &lt;br/&gt; (e.g., binning,&lt;br/&gt;MAGS,&lt;br/&gt;Phylogenomics,&lt;br/&gt;etc.)")
</pre>
</div>
<p></p>
</div>
</div>
</div>
<p>We have a total of 58 samples …DETAILS</p>
</section><section id="snakemake-workflow" class="level1"><h1>Snakemake Workflow</h1>
<p>Thanks to the anvi’o implementation of Snakemake we can run many commands sequentially and/or simultaneously. There is plenty of good documentation on the anvi’o website about <a href="http://merenlab.org/2018/07/09/anvio-snakemake-workflows/">setting up Snakemake workflows</a> so we will refrain from any lengthy explanations.</p>
<p>You can see all the settings we used and <a href="include/default_mg.json">download our configs file</a> or grab it from the folded code below. Some commands we chose not to run but they remain in the workflow for posterity.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb1"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--list-workflows</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>WARNING
===============================================
If you publish results from this workflow, please do not forget to cite
snakemake (doi:10.1093/bioinformatics/bts480)

HMM profiles .................................: 9 sources have been loaded: 
Archaea_76 (76 genes, domain: archaea), 
Bacteria_71 (71 genes, domain: bacteria), 
Protista_83 (83 genes, domain: eukarya), 
Ribosomal_RNA_12S (1 genes, domain: None), 
Ribosomal_RNA_16S (3 genes, domain: None), 
Ribosomal_RNA_18S (1 genes, domain: None), 
Ribosomal_RNA_23S (2 genes, domain: None), 
Ribosomal_RNA_28S (1 genes, domain: None),
Ribosomal_RNA_5S (5 genes, domain: None)
Available workflows ..........................: contigs, metagenomics, pangenomics, phylogenomics, trnaseq, ecophylo, sra_download</code></pre>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb3"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--workflow</span> metagenomics <span class="dt">\</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--get-default-config</span> default_mg.json  <span class="dt">\</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--list-dependencies</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>You can find a description of the metagenomic workflow <a href="https://anvio.org/help/main/workflows/metagenomics/">on the anvi’o website</a>.</p>
<div class="cell">
<details class="code-fold"><summary>Show JSON-formatted configuration file.</summary><div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="kw">{</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>    <span class="st">"fasta_txt"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_gen_contigs_database"</span><span class="ex">:</span> {</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--project-name"</span><span class="ex">:</span> <span class="st">"{group}"</span>,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--description"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-gene-calling"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--ignore-internal-stop-codons"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-mindful-splitting"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--contigs-fasta"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--split-length"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--kmer-size"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-predict-frame"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--prodigal-translation-table"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>    <span class="st">"centrifuge"</span><span class="ex">:</span> {</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>        <span class="st">"db"</span><span class="ex">:</span> <span class="st">"/pool/genomics/stri_istmobiome/dbs/centrifuge_dbs/p+h+v"</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_hmms"</span><span class="ex">:</span> {</span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--also-scan-trnas"</span><span class="ex">:</span> true,</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--installed-hmm-profile"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--hmm-profile-dir"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--add-to-functions-table"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_kegg_kofams"</span><span class="ex">:</span> {</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--kegg-data-dir"</span><span class="ex">:</span> <span class="st">"/pool/genomics/stri_istmobiome/dbs/kegg_kofam/"</span>,</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--hmmer-program"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--keep-all-hits"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--log-bitscores"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--just-do-it"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_ncbi_cogs"</span><span class="ex">:</span> {</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--cog-data-dir"</span><span class="ex">:</span> <span class="st">"/pool/genomics/stri_istmobiome/dbs/cog_db/"</span>,</span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--temporary-dir-path"</span><span class="ex">:</span> <span class="st">"/pool/genomics/stri_istmobiome/dbs/cog_db/tmp_mega/"</span>,</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--search-with"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_scg_taxonomy"</span><span class="ex">:</span> {</span>
<span id="cb4-46"><a href="#cb4-46" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-47"><a href="#cb4-47" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-48"><a href="#cb4-48" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--scgs-taxonomy-data-dir"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-49"><a href="#cb4-49" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-50"><a href="#cb4-50" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_trna_scan"</span><span class="ex">:</span> {</span>
<span id="cb4-51"><a href="#cb4-51" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-52"><a href="#cb4-52" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-53"><a href="#cb4-53" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--trna-cutoff-score"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-54"><a href="#cb4-54" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-55"><a href="#cb4-55" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_script_reformat_fasta"</span><span class="ex">:</span> {</span>
<span id="cb4-56"><a href="#cb4-56" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-57"><a href="#cb4-57" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--prefix"</span><span class="ex">:</span> <span class="st">"{group}"</span>,</span>
<span id="cb4-58"><a href="#cb4-58" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--simplify-names"</span><span class="ex">:</span> true,</span>
<span id="cb4-59"><a href="#cb4-59" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--keep-ids"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-60"><a href="#cb4-60" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--exclude-ids"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-61"><a href="#cb4-61" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-len"</span><span class="ex">:</span> <span class="st">"1000"</span>,</span>
<span id="cb4-62"><a href="#cb4-62" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--seq-type"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-63"><a href="#cb4-63" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-64"><a href="#cb4-64" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-65"><a href="#cb4-65" aria-hidden="true" tabindex="-1"></a>    <span class="st">"emapper"</span><span class="ex">:</span> {</span>
<span id="cb4-66"><a href="#cb4-66" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--database"</span><span class="ex">:</span> <span class="st">"bact"</span>,</span>
<span id="cb4-67"><a href="#cb4-67" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--usemem"</span><span class="ex">:</span> true,</span>
<span id="cb4-68"><a href="#cb4-68" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--override"</span><span class="ex">:</span> true,</span>
<span id="cb4-69"><a href="#cb4-69" aria-hidden="true" tabindex="-1"></a>        <span class="st">"path_to_emapper_dir"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-70"><a href="#cb4-70" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-71"><a href="#cb4-71" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-72"><a href="#cb4-72" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_script_run_eggnog_mapper"</span><span class="ex">:</span> {</span>
<span id="cb4-73"><a href="#cb4-73" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--use-version"</span><span class="ex">:</span> <span class="st">"0.12.6"</span>,</span>
<span id="cb4-74"><a href="#cb4-74" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-75"><a href="#cb4-75" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--cog-data-dir"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-76"><a href="#cb4-76" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--drop-previous-annotations"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-77"><a href="#cb4-77" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-78"><a href="#cb4-78" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-79"><a href="#cb4-79" aria-hidden="true" tabindex="-1"></a>    <span class="st">"samples_txt"</span><span class="ex">:</span> <span class="st">"samples.txt"</span>,</span>
<span id="cb4-80"><a href="#cb4-80" aria-hidden="true" tabindex="-1"></a>    <span class="st">"metaspades"</span><span class="ex">:</span> {</span>
<span id="cb4-81"><a href="#cb4-81" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">"--only-assembler"</span>,</span>
<span id="cb4-82"><a href="#cb4-82" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7,</span>
<span id="cb4-83"><a href="#cb4-83" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-84"><a href="#cb4-84" aria-hidden="true" tabindex="-1"></a>        <span class="st">"use_scaffolds"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-85"><a href="#cb4-85" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-86"><a href="#cb4-86" aria-hidden="true" tabindex="-1"></a>    <span class="st">"megahit"</span><span class="ex">:</span> {</span>
<span id="cb4-87"><a href="#cb4-87" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-contig-len"</span><span class="ex">:</span> 1000,</span>
<span id="cb4-88"><a href="#cb4-88" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--memory"</span><span class="ex">:</span> 0.9,</span>
<span id="cb4-89"><a href="#cb4-89" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14,</span>
<span id="cb4-90"><a href="#cb4-90" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-91"><a href="#cb4-91" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-count"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-92"><a href="#cb4-92" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--k-min"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-93"><a href="#cb4-93" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--k-max"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-94"><a href="#cb4-94" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--k-step"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-95"><a href="#cb4-95" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--k-list"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-96"><a href="#cb4-96" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no-mercy"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-97"><a href="#cb4-97" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no-bubble"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-98"><a href="#cb4-98" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--merge-level"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-99"><a href="#cb4-99" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--prune-level"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-100"><a href="#cb4-100" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--prune-depth"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-101"><a href="#cb4-101" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--low-local-ratio"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-102"><a href="#cb4-102" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--max-tip-len"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-103"><a href="#cb4-103" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no-local"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-104"><a href="#cb4-104" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--kmin-1pass"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-105"><a href="#cb4-105" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--presets"</span><span class="ex">:</span> <span class="st">"meta-sensitive"</span>,</span>
<span id="cb4-106"><a href="#cb4-106" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--mem-flag"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-107"><a href="#cb4-107" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--use-gpu"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-108"><a href="#cb4-108" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--gpu-mem"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-109"><a href="#cb4-109" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--keep-tmp-files"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-110"><a href="#cb4-110" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--tmp-dir"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-111"><a href="#cb4-111" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--continue"</span><span class="ex">:</span> true,</span>
<span id="cb4-112"><a href="#cb4-112" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--verbose"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-113"><a href="#cb4-113" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-114"><a href="#cb4-114" aria-hidden="true" tabindex="-1"></a>    <span class="st">"idba_ud"</span><span class="ex">:</span> {</span>
<span id="cb4-115"><a href="#cb4-115" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min_contig"</span><span class="ex">:</span> 1000,</span>
<span id="cb4-116"><a href="#cb4-116" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7,</span>
<span id="cb4-117"><a href="#cb4-117" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-118"><a href="#cb4-118" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--mink"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-119"><a href="#cb4-119" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--maxk"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-120"><a href="#cb4-120" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--step"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-121"><a href="#cb4-121" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--inner_mink"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-122"><a href="#cb4-122" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--inner_step"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-123"><a href="#cb4-123" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--prefix"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-124"><a href="#cb4-124" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min_count"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-125"><a href="#cb4-125" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min_support"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-126"><a href="#cb4-126" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--seed_kmer"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-127"><a href="#cb4-127" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--similar"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-128"><a href="#cb4-128" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--max_mismatch"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-129"><a href="#cb4-129" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min_pairs"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-130"><a href="#cb4-130" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no_bubble"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-131"><a href="#cb4-131" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no_local"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-132"><a href="#cb4-132" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no_coverage"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-133"><a href="#cb4-133" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--no_correct"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-134"><a href="#cb4-134" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--pre_correction"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-135"><a href="#cb4-135" aria-hidden="true" tabindex="-1"></a>        <span class="st">"use_scaffolds"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-136"><a href="#cb4-136" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-137"><a href="#cb4-137" aria-hidden="true" tabindex="-1"></a>    <span class="st">"iu_filter_quality_minoche"</span><span class="ex">:</span> {</span>
<span id="cb4-138"><a href="#cb4-138" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-139"><a href="#cb4-139" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--ignore-deflines"</span><span class="ex">:</span> true,</span>
<span id="cb4-140"><a href="#cb4-140" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--visualize-quality-curves"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-141"><a href="#cb4-141" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--limit-num-pairs"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-142"><a href="#cb4-142" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--print-qual-scores"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-143"><a href="#cb4-143" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--store-read-fate"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-144"><a href="#cb4-144" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-145"><a href="#cb4-145" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-146"><a href="#cb4-146" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gzip_fastqs"</span><span class="ex">:</span> {</span>
<span id="cb4-147"><a href="#cb4-147" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-148"><a href="#cb4-148" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-149"><a href="#cb4-149" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-150"><a href="#cb4-150" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bowtie"</span><span class="ex">:</span> {</span>
<span id="cb4-151"><a href="#cb4-151" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">"--no-unal"</span>,</span>
<span id="cb4-152"><a href="#cb4-152" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-153"><a href="#cb4-153" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-154"><a href="#cb4-154" aria-hidden="true" tabindex="-1"></a>    <span class="st">"samtools_view"</span><span class="ex">:</span> {</span>
<span id="cb4-155"><a href="#cb4-155" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">"-F 4"</span>,</span>
<span id="cb4-156"><a href="#cb4-156" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-157"><a href="#cb4-157" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-158"><a href="#cb4-158" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_profile"</span><span class="ex">:</span> {</span>
<span id="cb4-159"><a href="#cb4-159" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7,</span>
<span id="cb4-160"><a href="#cb4-160" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--sample-name"</span><span class="ex">:</span> <span class="st">"{sample}"</span>,</span>
<span id="cb4-161"><a href="#cb4-161" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--overwrite-output-destinations"</span><span class="ex">:</span> true,</span>
<span id="cb4-162"><a href="#cb4-162" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--report-variability-full"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-163"><a href="#cb4-163" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-SNV-profiling"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-164"><a href="#cb4-164" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--profile-SCVs"</span><span class="ex">:</span> true,</span>
<span id="cb4-165"><a href="#cb4-165" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--description"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-166"><a href="#cb4-166" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-hierarchical-clustering"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-167"><a href="#cb4-167" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--distance"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-168"><a href="#cb4-168" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--linkage"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-169"><a href="#cb4-169" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-contig-length"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-170"><a href="#cb4-170" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-mean-coverage"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-171"><a href="#cb4-171" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-coverage-for-variability"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-172"><a href="#cb4-172" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--cluster-contigs"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-173"><a href="#cb4-173" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--contigs-of-interest"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-174"><a href="#cb4-174" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--queue-size"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-175"><a href="#cb4-175" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--write-buffer-size-per-thread"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-176"><a href="#cb4-176" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--fetch-filter"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-177"><a href="#cb4-177" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-percent-identity"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-178"><a href="#cb4-178" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--max-contig-length"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-179"><a href="#cb4-179" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-180"><a href="#cb4-180" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_merge"</span><span class="ex">:</span> {</span>
<span id="cb4-181"><a href="#cb4-181" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--sample-name"</span><span class="ex">:</span> <span class="st">"{group}"</span>,</span>
<span id="cb4-182"><a href="#cb4-182" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--overwrite-output-destinations"</span><span class="ex">:</span> true,</span>
<span id="cb4-183"><a href="#cb4-183" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--description"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-184"><a href="#cb4-184" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--skip-hierarchical-clustering"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-185"><a href="#cb4-185" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--enforce-hierarchical-clustering"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-186"><a href="#cb4-186" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--distance"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-187"><a href="#cb4-187" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--linkage"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-188"><a href="#cb4-188" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14</span>
<span id="cb4-189"><a href="#cb4-189" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-190"><a href="#cb4-190" aria-hidden="true" tabindex="-1"></a>    <span class="st">"import_percent_of_reads_mapped"</span><span class="ex">:</span> {</span>
<span id="cb4-191"><a href="#cb4-191" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-192"><a href="#cb4-192" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 7</span>
<span id="cb4-193"><a href="#cb4-193" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-194"><a href="#cb4-194" aria-hidden="true" tabindex="-1"></a>    <span class="st">"krakenuniq"</span><span class="ex">:</span> {</span>
<span id="cb4-195"><a href="#cb4-195" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 3,</span>
<span id="cb4-196"><a href="#cb4-196" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--gzip-compressed"</span><span class="ex">:</span> true,</span>
<span id="cb4-197"><a href="#cb4-197" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-198"><a href="#cb4-198" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-199"><a href="#cb4-199" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--db"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-200"><a href="#cb4-200" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-201"><a href="#cb4-201" aria-hidden="true" tabindex="-1"></a>    <span class="st">"remove_short_reads_based_on_references"</span><span class="ex">:</span> {</span>
<span id="cb4-202"><a href="#cb4-202" aria-hidden="true" tabindex="-1"></a>        <span class="st">"delimiter-for-iu-remove-ids-from-fastq"</span><span class="ex">:</span> <span class="st">" "</span>,</span>
<span id="cb4-203"><a href="#cb4-203" aria-hidden="true" tabindex="-1"></a>        <span class="st">"dont_remove_just_map"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-204"><a href="#cb4-204" aria-hidden="true" tabindex="-1"></a>        <span class="st">"references_for_removal_txt"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-205"><a href="#cb4-205" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-206"><a href="#cb4-206" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-207"><a href="#cb4-207" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_cluster_contigs"</span><span class="ex">:</span> {</span>
<span id="cb4-208"><a href="#cb4-208" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--collection-name"</span><span class="ex">:</span> <span class="st">"{driver}"</span>,</span>
<span id="cb4-209"><a href="#cb4-209" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-210"><a href="#cb4-210" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--driver"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-211"><a href="#cb4-211" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--just-do-it"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-212"><a href="#cb4-212" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--additional-params-concoct"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-213"><a href="#cb4-213" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--additional-params-metabat2"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-214"><a href="#cb4-214" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--additional-params-maxbin2"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-215"><a href="#cb4-215" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--additional-params-dastool"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-216"><a href="#cb4-216" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--additional-params-binsanity"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-217"><a href="#cb4-217" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-218"><a href="#cb4-218" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-219"><a href="#cb4-219" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gen_external_genome_file"</span><span class="ex">:</span> {</span>
<span id="cb4-220"><a href="#cb4-220" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-221"><a href="#cb4-221" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-222"><a href="#cb4-222" aria-hidden="true" tabindex="-1"></a>    <span class="st">"export_gene_calls_for_centrifuge"</span><span class="ex">:</span> {</span>
<span id="cb4-223"><a href="#cb4-223" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-224"><a href="#cb4-224" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-225"><a href="#cb4-225" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_import_taxonomy_for_genes"</span><span class="ex">:</span> {</span>
<span id="cb4-226"><a href="#cb4-226" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-227"><a href="#cb4-227" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-228"><a href="#cb4-228" aria-hidden="true" tabindex="-1"></a>    <span class="st">"annotate_contigs_database"</span><span class="ex">:</span> {</span>
<span id="cb4-229"><a href="#cb4-229" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-230"><a href="#cb4-230" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-231"><a href="#cb4-231" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_get_sequences_for_gene_calls"</span><span class="ex">:</span> {</span>
<span id="cb4-232"><a href="#cb4-232" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-233"><a href="#cb4-233" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-234"><a href="#cb4-234" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gunzip_fasta"</span><span class="ex">:</span> {</span>
<span id="cb4-235"><a href="#cb4-235" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-236"><a href="#cb4-236" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-237"><a href="#cb4-237" aria-hidden="true" tabindex="-1"></a>    <span class="st">"reformat_external_gene_calls_table"</span><span class="ex">:</span> {</span>
<span id="cb4-238"><a href="#cb4-238" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-239"><a href="#cb4-239" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-240"><a href="#cb4-240" aria-hidden="true" tabindex="-1"></a>    <span class="st">"reformat_external_functions"</span><span class="ex">:</span> {</span>
<span id="cb4-241"><a href="#cb4-241" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-242"><a href="#cb4-242" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-243"><a href="#cb4-243" aria-hidden="true" tabindex="-1"></a>    <span class="st">"import_external_functions"</span><span class="ex">:</span> {</span>
<span id="cb4-244"><a href="#cb4-244" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-245"><a href="#cb4-245" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-246"><a href="#cb4-246" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_run_pfams"</span><span class="ex">:</span> {</span>
<span id="cb4-247"><a href="#cb4-247" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> true,</span>
<span id="cb4-248"><a href="#cb4-248" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--pfam-data-dir"</span><span class="ex">:</span> <span class="st">"/pool/genomics/stri_istmobiome/dbs/pfam_db"</span>,</span>
<span id="cb4-249"><a href="#cb4-249" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14</span>
<span id="cb4-250"><a href="#cb4-250" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-251"><a href="#cb4-251" aria-hidden="true" tabindex="-1"></a>    <span class="st">"iu_gen_configs"</span><span class="ex">:</span> {</span>
<span id="cb4-252"><a href="#cb4-252" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--r1-prefix"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-253"><a href="#cb4-253" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--r2-prefix"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-254"><a href="#cb4-254" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> 14</span>
<span id="cb4-255"><a href="#cb4-255" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-256"><a href="#cb4-256" aria-hidden="true" tabindex="-1"></a>    <span class="st">"gen_qc_report"</span><span class="ex">:</span> {</span>
<span id="cb4-257"><a href="#cb4-257" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-258"><a href="#cb4-258" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-259"><a href="#cb4-259" aria-hidden="true" tabindex="-1"></a>    <span class="st">"merge_fastqs_for_co_assembly"</span><span class="ex">:</span> {</span>
<span id="cb4-260"><a href="#cb4-260" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-261"><a href="#cb4-261" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-262"><a href="#cb4-262" aria-hidden="true" tabindex="-1"></a>    <span class="st">"merge_fastas_for_co_assembly"</span><span class="ex">:</span> {</span>
<span id="cb4-263"><a href="#cb4-263" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-264"><a href="#cb4-264" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-265"><a href="#cb4-265" aria-hidden="true" tabindex="-1"></a>    <span class="st">"bowtie_build"</span><span class="ex">:</span> {</span>
<span id="cb4-266"><a href="#cb4-266" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-267"><a href="#cb4-267" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-268"><a href="#cb4-268" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-269"><a href="#cb4-269" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_init_bam"</span><span class="ex">:</span> {</span>
<span id="cb4-270"><a href="#cb4-270" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-271"><a href="#cb4-271" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-272"><a href="#cb4-272" aria-hidden="true" tabindex="-1"></a>    <span class="st">"krakenuniq_mpa_report"</span><span class="ex">:</span> {</span>
<span id="cb4-273"><a href="#cb4-273" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-274"><a href="#cb4-274" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-275"><a href="#cb4-275" aria-hidden="true" tabindex="-1"></a>    <span class="st">"import_krakenuniq_taxonomy"</span><span class="ex">:</span> {</span>
<span id="cb4-276"><a href="#cb4-276" aria-hidden="true" tabindex="-1"></a>        <span class="st">"--min-abundance"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-277"><a href="#cb4-277" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-278"><a href="#cb4-278" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-279"><a href="#cb4-279" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_summarize"</span><span class="ex">:</span> {</span>
<span id="cb4-280"><a href="#cb4-280" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-281"><a href="#cb4-281" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-282"><a href="#cb4-282" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-283"><a href="#cb4-283" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-284"><a href="#cb4-284" aria-hidden="true" tabindex="-1"></a>    <span class="st">"anvi_split"</span><span class="ex">:</span> {</span>
<span id="cb4-285"><a href="#cb4-285" aria-hidden="true" tabindex="-1"></a>        <span class="st">"additional_params"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-286"><a href="#cb4-286" aria-hidden="true" tabindex="-1"></a>        <span class="st">"run"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-287"><a href="#cb4-287" aria-hidden="true" tabindex="-1"></a>        <span class="st">"threads"</span><span class="ex">:</span> <span class="st">""</span></span>
<span id="cb4-288"><a href="#cb4-288" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-289"><a href="#cb4-289" aria-hidden="true" tabindex="-1"></a>    <span class="st">"references_mode"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-290"><a href="#cb4-290" aria-hidden="true" tabindex="-1"></a>    <span class="st">"all_against_all"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-291"><a href="#cb4-291" aria-hidden="true" tabindex="-1"></a>    <span class="st">"kraken_txt"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-292"><a href="#cb4-292" aria-hidden="true" tabindex="-1"></a>    <span class="st">"collections_txt"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-293"><a href="#cb4-293" aria-hidden="true" tabindex="-1"></a>    <span class="st">"output_dirs"</span><span class="ex">:</span> {</span>
<span id="cb4-294"><a href="#cb4-294" aria-hidden="true" tabindex="-1"></a>        <span class="st">"FASTA_DIR"</span><span class="ex">:</span> <span class="st">"02_FASTA"</span>,</span>
<span id="cb4-295"><a href="#cb4-295" aria-hidden="true" tabindex="-1"></a>        <span class="st">"CONTIGS_DIR"</span><span class="ex">:</span> <span class="st">"03_CONTIGS"</span>,</span>
<span id="cb4-296"><a href="#cb4-296" aria-hidden="true" tabindex="-1"></a>        <span class="st">"QC_DIR"</span><span class="ex">:</span> <span class="st">"01_QC"</span>,</span>
<span id="cb4-297"><a href="#cb4-297" aria-hidden="true" tabindex="-1"></a>        <span class="st">"MAPPING_DIR"</span><span class="ex">:</span> <span class="st">"04_MAPPING"</span>,</span>
<span id="cb4-298"><a href="#cb4-298" aria-hidden="true" tabindex="-1"></a>        <span class="st">"PROFILE_DIR"</span><span class="ex">:</span> <span class="st">"05_ANVIO_PROFILE"</span>,</span>
<span id="cb4-299"><a href="#cb4-299" aria-hidden="true" tabindex="-1"></a>        <span class="st">"MERGE_DIR"</span><span class="ex">:</span> <span class="st">"06_MERGED"</span>,</span>
<span id="cb4-300"><a href="#cb4-300" aria-hidden="true" tabindex="-1"></a>        <span class="st">"TAXONOMY_DIR"</span><span class="ex">:</span> <span class="st">"07_TAXONOMY"</span>,</span>
<span id="cb4-301"><a href="#cb4-301" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SUMMARY_DIR"</span><span class="ex">:</span> <span class="st">"08_SUMMARY"</span>,</span>
<span id="cb4-302"><a href="#cb4-302" aria-hidden="true" tabindex="-1"></a>        <span class="st">"SPLIT_PROFILES_DIR"</span><span class="ex">:</span> <span class="st">"09_SPLIT_PROFILES"</span>,</span>
<span id="cb4-303"><a href="#cb4-303" aria-hidden="true" tabindex="-1"></a>        <span class="st">"LOGS_DIR"</span><span class="ex">:</span> <span class="st">"00_LOGS"</span></span>
<span id="cb4-304"><a href="#cb4-304" aria-hidden="true" tabindex="-1"></a>    <span class="ex">},</span></span>
<span id="cb4-305"><a href="#cb4-305" aria-hidden="true" tabindex="-1"></a>    <span class="st">"max_threads"</span><span class="ex">:</span> <span class="st">""</span>,</span>
<span id="cb4-306"><a href="#cb4-306" aria-hidden="true" tabindex="-1"></a>    <span class="st">"config_version"</span><span class="ex">:</span> <span class="st">"3"</span>,</span>
<span id="cb4-307"><a href="#cb4-307" aria-hidden="true" tabindex="-1"></a>    <span class="st">"workflow_name"</span><span class="ex">:</span> <span class="st">"metagenomics"</span></span>
<span id="cb4-308"><a href="#cb4-308" aria-hidden="true" tabindex="-1"></a><span class="kw">}</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<p>Since we are doing a co-assembly, we need a separate file called <a href="include/samples.txt"><code>samples.txt</code></a>. This file tells anvi’o where to find the <strong>raw</strong> fastq files for each sample and what group the sample belongs to. The file is a four-column, tab-delimited file.</p>
<table class="caption-top table">
<thead><tr class="header">
<th>sample</th>
<th>group</th>
<th>r1</th>
<th>r2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>EP_ALPH_AREN_GL_P01</td>
<td>EP</td>
<td>forward reads</td>
<td>reverse reads</td>
</tr>
<tr class="even">
<td>EP_ALPH_AREN_HP_P01</td>
<td>EP</td>
<td>forward reads</td>
<td>reverse reads</td>
</tr>
<tr class="odd">
<td>WA_ALPH_BOUV_GL_P02</td>
<td>WA</td>
<td>forward reads</td>
<td>reverse reads</td>
</tr>
<tr class="even">
<td>WA_ALPH_BOUV_HP_P02</td>
<td>WA</td>
<td>forward reads</td>
<td>reverse reads</td>
</tr>
</tbody>
</table>
<p>Remember, there are 4 fastq files <em>per</em> direction (forward or reverse) <em>per</em> sample. So each comma separated list in the <code>r1</code> column needs four file names and the same with the <code>r2</code> column. And you must include <em>relative path names</em>.</p>
<p>To see what will happen with the <code>config</code> file we can visualize the Snakemake workflow using a Directed acyclic graph (DAG) where edge connections represent dependencies and nodes represent commands. The workflow begins with trimmed reads and continues up to and including automatic binning of contigs. At the end of this workflow we then proceed with manual binning and MAG generation. We added nodes for VirSorter and Kaiju annotations since these are not part of the normal anvi’o workflow.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb5"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--workflow</span> metagenomics <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--save-workflow-graph</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">-c</span> default_mg_for_dag.json</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<details class="code-fold"><summary>Show DAG R Code script.</summary><div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rich-iannone.github.io/DiagrammeR/">DiagrammeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/rich-iannone/DiagrammeRsvg">DiagrammeRsvg</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://docs.ropensci.org/rsvg/">rsvg</a></span><span class="op">)</span></span>
<span><span class="va">dag</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rich-iannone.github.io/DiagrammeR/reference/grViz.html">grViz</a></span> <span class="op">(</span><span class="st">"</span></span>
<span><span class="st">digraph boxes_and_circles {</span></span>
<span><span class="st">  graph [layout = dot, align=center]</span></span>
<span><span class="st"></span></span>
<span><span class="st">  node [shape = rectangle, style = 'rounded,filled' fontname=sans, fontsize=12, penwidth=4]</span></span>
<span><span class="st">  edge[penwidth=4, color=grey];</span></span>
<span><span class="st"></span></span>
<span><span class="st">0[label = 'metagenomics_workflow_target_rule', color = 'grey'];</span></span>
<span><span class="st">1[label = 'anvi_merge', color = '#CC79A7'];</span></span>
<span><span class="st">2[label = 'anvi_merge', color = '#CC79A7'];</span></span>
<span><span class="st">3[label = 'annotate_contigs_database', color = '#E69F00'];</span></span>
<span><span class="st">4[label = 'annotate_contigs_database', color = '#E69F00'];</span></span>
<span><span class="st">5[label = 'gen_qc_report', color = '#56B4E9'];</span></span>
<span><span class="st">6[label = 'anvi_cluster_contigs\ndriver: concoct', color = '#F0E442'];</span></span>
<span><span class="st">7[label = 'anvi_cluster_contigs\ndriver: concoct', color = '#F0E442'];</span></span>
<span><span class="st">8[label = 'anvi_gen_contigs_database', color = '#CC79A7'];</span></span>
<span><span class="st">9[label = 'anvi_profile', color = '#CC79A7'];</span></span>
<span><span class="st">10[label = 'anvi_profile', color = '#CC79A7'];</span></span>
<span><span class="st">11[label = 'import_percent_of_reads_mapped', color = '#009E73'];</span></span>
<span><span class="st">12[label = 'import_percent_of_reads_mapped', color = '#009E73'];</span></span>
<span><span class="st">13[label = 'import_krakenuniq_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">14[label = 'import_krakenuniq_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">15[label = 'anvi_gen_contigs_database', color = '#CC79A7'];</span></span>
<span><span class="st">16[label = 'anvi_profile', color = '#CC79A7'];</span></span>
<span><span class="st">17[label = 'anvi_profile', color = '#CC79A7'];</span></span>
<span><span class="st">18[label = 'import_percent_of_reads_mapped', color = '#009E73'];</span></span>
<span><span class="st">19[label = 'import_percent_of_reads_mapped', color = '#009E73'];</span></span>
<span><span class="st">20[label = 'import_krakenuniq_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">21[label = 'import_krakenuniq_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">22[label = 'anvi_import_taxonomy_for_genes', color = '#E69F00'];</span></span>
<span><span class="st">23[label = 'anvi_run_hmms', color = '#E69F00'];</span></span>
<span><span class="st">24[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];</span></span>
<span><span class="st">25[label = 'anvi_run_scg_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">26[label = 'anvi_run_pfams', color = '#E69F00'];</span></span>
<span><span class="st">27[label = 'anvi_import_taxonomy_for_genes', color = '#E69F00'];</span></span>
<span><span class="st">28[label = 'anvi_run_hmms', color = '#E69F00'];</span></span>
<span><span class="st">29[label = 'anvi_run_ncbi_cogs', color = '#E69F00'];</span></span>
<span><span class="st">30[label = 'anvi_run_scg_taxonomy', color = '#E69F00'];</span></span>
<span><span class="st">31[label = 'anvi_run_pfams', color = '#E69F00'];</span></span>
<span><span class="st">32[label = 'iu_filter_quality_minoche\nsample: EP_ALPH_PANA_HP_P07', color = '#56B4E9'];</span></span>
<span><span class="st">33[label = 'iu_filter_quality_minoche\nsample: EP_ALPH_PANA_MG_P07', color = '#56B4E9'];</span></span>
<span><span class="st">34[label = 'iu_filter_quality_minoche\nsample: WA_ALPH_FORM_HP_P07', color = '#56B4E9'];</span></span>
<span><span class="st">35[label = 'iu_filter_quality_minoche\nsample: WA_ALPH_FORM_MG_P07', color = '#56B4E9'];</span></span>
<span><span class="st">36[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];</span></span>
<span><span class="st">37[label = 'anvi_init_bam', color = '#009E73'];</span></span>
<span><span class="st">38[label = 'anvi_init_bam', color = '#009E73'];</span></span>
<span><span class="st">39[label = 'krakenuniq_mpa_report', color = '#E69F00'];</span></span>
<span><span class="st">40[label = 'krakenuniq_mpa_report', color = '#E69F00'];</span></span>
<span><span class="st">41[label = 'anvi_script_reformat_fasta', color = '#56B4E9'];</span></span>
<span><span class="st">42[label = 'anvi_init_bam', color = '#009E73'];</span></span>
<span><span class="st">43[label = 'anvi_init_bam', color = '#009E73'];</span></span>
<span><span class="st">44[label = 'krakenuniq_mpa_report', color = '#E69F00'];</span></span>
<span><span class="st">45[label = 'krakenuniq_mpa_report', color = '#E69F00'];</span></span>
<span><span class="st">46[label = 'centrifuge', color = '#E69F00'];</span></span>
<span><span class="st">47[label = 'centrifuge', color = '#E69F00'];</span></span>
<span><span class="st">48[label = 'iu_gen_configs', color = '#56B4E9'];</span></span>
<span><span class="st">49[label = 'anvi_script_reformat_fasta_prefix_only', color = '#56B4E9'];</span></span>
<span><span class="st">50[label = 'samtools_view', color = '#009E73'];</span></span>
<span><span class="st">51[label = 'samtools_view', color = '#009E73'];</span></span>
<span><span class="st">52[label = 'krakenuniq', color = '#E69F00'];</span></span>
<span><span class="st">53[label = 'krakenuniq', color = '#E69F00'];</span></span>
<span><span class="st">54[label = 'anvi_script_reformat_fasta_prefix_only', color = '#56B4E9'];</span></span>
<span><span class="st">55[label = 'samtools_view', color = '#009E73'];</span></span>
<span><span class="st">56[label = 'samtools_view', color = '#009E73'];</span></span>
<span><span class="st">57[label = 'krakenuniq', color = '#E69F00'];</span></span>
<span><span class="st">58[label = 'krakenuniq', color = '#E69F00'];</span></span>
<span><span class="st">59[label = 'export_gene_calls_for_centrifuge', color = '#E69F00'];</span></span>
<span><span class="st">60[label = 'export_gene_calls_for_centrifuge', color = '#E69F00'];</span></span>
<span><span class="st">61[label = 'megahit\ngroup: EP', color = '#56B4E9'];</span></span>
<span><span class="st">62[label = 'bowtie', color = '#009E73'];</span></span>
<span><span class="st">63[label = 'bowtie', color = '#009E73'];</span></span>
<span><span class="st">64[label = 'gzip_fastqs\nR: R1', color = '#56B4E9'];</span></span>
<span><span class="st">65[label = 'gzip_fastqs\nR: R2', color = '#56B4E9'];</span></span>
<span><span class="st">66[label = 'gzip_fastqs\nR: R1', color = '#56B4E9'];</span></span>
<span><span class="st">67[label = 'gzip_fastqs\nR: R2', color = '#56B4E9'];</span></span>
<span><span class="st">68[label = 'megahit\ngroup: WA', color = '#56B4E9'];</span></span>
<span><span class="st">69[label = 'bowtie', color = '#009E73'];</span></span>
<span><span class="st">70[label = 'bowtie', color = '#009E73'];</span></span>
<span><span class="st">71[label = 'gzip_fastqs\nR: R1', color = '#56B4E9'];</span></span>
<span><span class="st">72[label = 'gzip_fastqs\nR: R2', color = '#56B4E9'];</span></span>
<span><span class="st">73[label = 'gzip_fastqs\nR: R1', color = '#56B4E9'];</span></span>
<span><span class="st">74[label = 'gzip_fastqs\nR: R2', color = '#56B4E9'];</span></span>
<span><span class="st">75[label = 'bowtie_build', color = '#009E73'];</span></span>
<span><span class="st">76[label = 'bowtie_build', color = '#009E73'];</span></span>
<span><span class="st">77[label = 'virsorter2', color = '#E69F00', style = 'dashed'];</span></span>
<span><span class="st">78[label = 'virsorter2', color = '#E69F00', style = 'dashed'];</span></span>
<span><span class="st">79[label = 'kaiju', color = '#E69F00', style = 'dashed'];</span></span>
<span><span class="st">80[label = 'kaiju', color = '#E69F00', style = 'dashed'];</span></span>
<span><span class="st"></span></span>
<span><span class="st">1-&gt;0; 2-&gt;0; 3-&gt;0; 4-&gt;0; 5-&gt;0; 6-&gt;0; 7-&gt;0; 8-&gt;1; 9-&gt;1;</span></span>
<span><span class="st">10-&gt;1; 11-&gt;1; 12-&gt;1; 13-&gt;1; 14-&gt;1; 15-&gt;2; 16-&gt;2; 17-&gt;2;</span></span>
<span><span class="st">18-&gt;2; 19-&gt;2; 20-&gt;2; 21-&gt;2; 8-&gt;3; 22-&gt;3; 23-&gt;3; 24-&gt;3;</span></span>
<span><span class="st">25-&gt;3; 26-&gt;3; 15-&gt;4; 27-&gt;4; 28-&gt;4; 29-&gt;4; 30-&gt;4; 31-&gt;4;</span></span>
<span><span class="st">{32 33 34 35}-&gt;5;</span></span>
<span><span class="st">48-&gt;{32 33 34 35};</span></span>
<span><span class="st">32-&gt;{64 65};</span></span>
<span><span class="st">33-&gt;{66 67};</span></span>
<span><span class="st">{64 65 66 67}-&gt;61;</span></span>
<span><span class="st">{64 65}-&gt;52;</span></span>
<span><span class="st">{66 67}-&gt;53;</span></span>
<span><span class="st">34-&gt;{71 72};</span></span>
<span><span class="st">35-&gt;{73 74};</span></span>
<span><span class="st">{71 72 73 74}-&gt;68</span></span>
<span><span class="st">{71 72}-&gt;57;</span></span>
<span><span class="st">{73 74}-&gt;58;</span></span>
<span><span class="st"></span></span>
<span><span class="st">8-&gt;6; 1-&gt;6; 15-&gt;7; 2-&gt;7;</span></span>
<span><span class="st">36-&gt;8; 37-&gt;9; 8-&gt;9; 38-&gt;10; 8-&gt;10; 9-&gt;11; 10-&gt;12; 39-&gt;13;</span></span>
<span><span class="st">11-&gt;13; 9-&gt;13; 40-&gt;14; 12-&gt;14; 10-&gt;14; 41-&gt;15; 42-&gt;16; 15-&gt;16;</span></span>
<span><span class="st">43-&gt;17; 15-&gt;17; 16-&gt;18; 17-&gt;19; 44-&gt;20; 18-&gt;20; 16-&gt;20; 45-&gt;21;</span></span>
<span><span class="st">19-&gt;21; 17-&gt;21; 46-&gt;22; 8-&gt;22; 8-&gt;23; 8-&gt;24; 23-&gt;25; 8-&gt;25;</span></span>
<span><span class="st">8-&gt;26; 47-&gt;27; 15-&gt;27; 15-&gt;28; 15-&gt;29; 28-&gt;30; 15-&gt;30; 15-&gt;31;</span></span>
<span><span class="st">49-&gt;36; 50-&gt;37; 51-&gt;38; 52-&gt;39;</span></span>
<span><span class="st">53-&gt;40; 54-&gt;41; 55-&gt;42; 56-&gt;43; 57-&gt;44; 58-&gt;45; 59-&gt;46; 60-&gt;47;</span></span>
<span><span class="st">61-&gt;49; 62-&gt;50; 63-&gt;51; 68-&gt;54;</span></span>
<span><span class="st">69-&gt;55; 70-&gt;56; 8-&gt;59; 15-&gt;60;</span></span>
<span><span class="st">75-&gt;62; 64-&gt;62; 65-&gt;62; 75-&gt;63;</span></span>
<span><span class="st">66-&gt;63; 67-&gt;63; 76-&gt;69; 71-&gt;69; 72-&gt;69; 76-&gt;70; 73-&gt;70; 74-&gt;70;</span></span>
<span><span class="st">36-&gt;75; 41-&gt;76; 79-&gt;22; 80-&gt;27;</span></span>
<span><span class="st">15-&gt;77; 8-&gt;78; 77-&gt;4; 78-&gt;3; 59-&gt;79; 60-&gt;80;</span></span>
<span><span class="st"></span></span>
<span><span class="st">    graph [nodesep = 0.1]</span></span>
<span><span class="st">{ rank=same; 13, 14, 20, 21 }</span></span>
<span><span class="st">{ rank=same; 39, 40, 44, 45 }</span></span>
<span><span class="st">{ rank=same; 32, 33, 34, 35 }</span></span>
<span><span class="st">{ rank=same; 64, 65, 66, 67, 71, 72, 73, 74 }</span></span>
<span><span class="st">{ rank=same; 5, 49, 54 }</span></span>
<span><span class="st"></span></span>
<span><span class="st">}</span></span>
<span><span class="st">"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">export_svg</span><span class="op">(</span><span class="va">dag</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/rawConversion.html">charToRaw</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">rsvg</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">png</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/png/man/writePNG.html">writePNG</a></span><span class="op">(</span><span class="st">"include/dag.png"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</details>
</div>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure"><p><a href="include/dag.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="DAG of the metagenomic workflow for a small subset of samples."><img src="include/dag.png" class="img-fluid figure-img" width="1008" alt="DAG of the metagenomic workflow for a small subset of samples."></a></p>
<figcaption>DAG of the metagenomic workflow for a small subset of samples.</figcaption></figure>
</div>
</div>
<p>Colors indicate broad divisions of workflow: sky blue, short-read prep &amp; co-assembly; blueish green, short-read mapping to assembly; orange, taxonomic or functional classification; yellow, automatic binning; reddish purple, databases construction.</p>
</div>
<p>And here are the commands we used to run the workflow.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb7"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">-w</span> metagenomics <span class="at">-c</span> default_mg.json <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--additional-params</span> <span class="at">--jobs</span> 28 <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--resources</span> nodes=28 <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--keep-going</span> <span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--rerun-incomplete</span> <span class="at">--unlock</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">-w</span> metagenomics <span class="at">-c</span> default_mg.json <span class="dt">\</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--additional-params</span> <span class="at">--jobs</span> 28 <span class="dt">\</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--resources</span> nodes=28 <span class="at">--keep-going</span> <span class="at">--rerun-incomplete</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>If a Snakemake job fails the default behavior is to lock the workflow and, because large jobs can fail for a lot of reasons, we decided to always include an identical command first with the <code>--unlock</code> flag. This will unlock the workflow and then the second command will execute. You can find an explanation of this post on <a href="http://merenlab.org/2018/07/09/anvio-snakemake-workflows/#how-can-i-restart-a-failed-job">anvi’o snakemake workflows</a>.</p>
<section id="snakemake-citations" class="level2"><h2 class="anchored" data-anchor-id="snakemake-citations">Snakemake Citations</h2>
<p>There are many tools used in the workflow that need to be cited. See the <a href="../../../workflows/pub/references/">Workflow References</a> page for details.</p>
<p>There are also a few tools that we ran outside of the Snakemake workflow. Results from these steps need to be added to the individual <code>PROFILE.db</code>’s, merged <code>PROFILE.db</code>, or <code>CONTIGS.db</code>. Therefore, before the <code>anvi-merge</code> portion of the Snakemake workflow finished, we killed the job, ran the accessory analyses described below, and then restarted the workflow to finish the missing step. Cumbersome, yes, but it got the job done.</p>
</section></section><section id="pipeline-output" class="level1"><h1>Pipeline Output</h1>
<p>When the workflow is complete, you should see the following directories within your project folder.</p>
<p>Directory names and descriptions:</p>
<ul>
<li>
<strong>00_LOGS/</strong>: Individual log files for each step of the snakemake workflow.</li>
<li>
<strong>01_QC/</strong>: Merged forward (R1) &amp; reverse (R2) QC’ed fastq files. QC STATS file for each sample. <code>qc-report.txt</code> file, a summary table of all QC results.</li>
<li>
<strong>02_FASTA/</strong>: Contig fasta files and reformat reports for each assembly.</li>
<li>
<strong>03_CONTIGS/</strong>: An annotated contig database for each assembly.</li>
<li>
<strong>04_MAPPING/</strong>: Short read BAM files.</li>
<li>
<strong>05_ANVIO_PROFILE/</strong>: Profile database for each sample.</li>
<li>
<strong>06_MERGED/</strong>: Single merged profile database.</li>
</ul>
<p>The <code>contigs.db</code> contains all taxonomic and functional annotations. The <code>profile.db</code> contains information on coverage, detection, etc.</p>
</section><section id="annotations" class="level1"><h1>Annotations</h1>
<p>This part of the workflow contains information for ancillary annotations (i.e., outside the Snakmake workflow). We will mainly cover taxonomic annotations as most functional annotations (e.g., KEGG, COG, Pfam, etc.) are run in the workflow.</p>
<section id="taxonomic" class="level2"><h2 class="anchored" data-anchor-id="taxonomic">Taxonomic</h2>
<p>In this section we discuss taxonomic classification of short reads, contigs, and gene calls. We go through the steps of analyzing the data and getting the results into anvi’o databases.</p>
<section id="short-reads-with-kraken2" class="level3"><h3 class="anchored" data-anchor-id="short-reads-with-kraken2">Short-reads with Kraken2</h3>
<p>In this section we use Kraken2 <span class="citation" data-cites="wood2019kraken2">(<a href="#ref-wood2019kraken2" role="doc-biblioref">Wood, Lu, and Langmead 2019</a>)</span> to classify the <strong>short reads</strong>. Our goal is to classify short-reads, generate an input file for anvi’o, and create <a href="https://github.com/marbl/Krona/wiki">Krona plots</a> for data visualization. Brace yourself.</p>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Since Kraken2 annotation is performed on individual samples and the results are imported into the individual <code>profile.db</code>’s we will need to re-merge the all <code>profile.db</code> after these steps are completed. The merging step is basically the last step of the Snakemake workflow.</p>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb8"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kraken2</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken2</span> <span class="at">--paired</span> 01_QC/<span class="va">$sample</span>-QUALITY_PASSED_R1.fastq.gz  <span class="dt">\</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>                     01_QC/<span class="va">$sample</span>-QUALITY_PASSED_R2.fastq.gz <span class="dt">\</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--db</span> kraken2_db/ <span class="dt">\</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--use-names</span> <span class="dt">\</span></span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="dt">\</span></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--output</span> <span class="va">$sample</span>-kraken.out <span class="dt">\</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--report</span> <span class="va">$sample</span>-kraken-report.txt</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>After this is finished we should have two files for each sample–a <code>.out</code> file containing the results of the Kraken2 annotation and a <code>.report.txt</code> file that summarizes the results.</p>
<p>First, we generate the file that anvi’o needs–the format is very specific. For this task we use <a href="https://github.com/jenniferlu717/KrakenTools">KrakenTool</a>–a suite of very handy scripts to Kraken 2 data. We will use a tool called <a href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file#kreport2mpapy"><code>kreport2mpa.py</code></a>, which takes a Kraken report file and prints out a MPA (MetaPhlAn)-style TEXT file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb9"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krakentools</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kreport2mpa.py</span> <span class="at">-r</span> <span class="va">$sample</span>-kraken-report.txt <span class="at">-o</span> <span class="va">$sample</span>-kraken-mpa.txt    </span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Easy as that. Now we can import all MPA files into their respective contig databases. Here we had to split the sample list by co-assembly because I could not figure out an easier way.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb10"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> EP_list.txt<span class="kw">`</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>    <span class="ex">anvi-import-taxonomy-for-layers</span> <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-p</span> 05_ANVIO_PROFILE/EP/<span class="va">$sample</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">--parse</span> krakenuniq <span class="dt">\</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-i</span> 07_TAXONOMY/KRAKEN_TAXONOMY/<span class="va">$sample</span>-kraken-mpa.txt</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> WA_list.txt<span class="kw">`</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>    <span class="ex">anvi-import-taxonomy-for-layers</span> <span class="dt">\</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-p</span> 05_ANVIO_PROFILE/WA/<span class="va">$sample</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>                   <span class="at">--parse</span> krakenuniq <span class="at">-i</span> <span class="dt">\</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>                   07_TAXONOMY/KRAKEN_TAXONOMY/<span class="va">$sample</span>-kraken-mpa.txt</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Rad. With this done we can re-merge the profile databases.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb11"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-merge</span> 05_ANVIO_PROFILE/WA/<span class="pp">*</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>          <span class="at">-o</span> 06_MERGED/WA</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-merge</span> 05_ANVIO_PROFILE/EP/<span class="pp">*</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>          <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">-o</span> 06_MERGED/EP</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Alright, time to make some Krona plots. This is a two-steo process. First, we use two scripts from <a href="https://github.com/bxlab/metaWRAP">metaWRAP</a>. The first, called <code>kraken2_translate.py</code>, is used to generate full taxonomic lineages from a taxid. The input file for this is the output of the Kraken2 annotation. The next script is called <code>kraken_to_krona.py</code> which takes the output of the first script (translated Kraken file) and parses it into a format that Krona can use to produce plots.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb12"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate metawrap</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken2_translate.py</span> kraken2_db <span class="va">$sample</span>-kraken.out <span class="va">$sample</span>-kraken.trans</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken_to_krona.py</span> <span class="va">$sample</span>-kraken.trans <span class="op">&gt;</span> <span class="va">$KRAKEN</span>/<span class="va">$sample</span>-kraken.krona</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Once that is complete, we can use the output files and a script called <code>ktImportText</code> from the Krona package to produce HTML Krona plots for each sample.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb13"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krona</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>    <span class="ex">ktImportText</span> <span class="va">$sample</span>-kraken.krona <span class="at">-o</span> <span class="va">$sample</span>-kraken.krona.html</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>A plot for every sample. How great is that?</p>
</section><section id="virsorter-annotation" class="level3"><h3 class="anchored" data-anchor-id="virsorter-annotation">VirSorter Annotation</h3>
<p>To classify any viral sequences, we ran <a href="https://github.com/jiarong/VirSorter2">VirSorter2</a> <span class="citation" data-cites="guo2021virsorter2">(<a href="#ref-guo2021virsorter2" role="doc-biblioref">Guo et al. 2021</a>)</span> on contigs from the co-assembly using our newly created <code>contig.db</code>. First, we need something for VirSorter2 to classify. For that we export fasta files from each anvi’o co-assembly.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb14"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-contigs</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> 03_CONTIGS/WA-splits.fa <span class="dt">\ </span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">--splits-mode</span> <span class="at">--no-wrap</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-contigs</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> 03_CONTIGS/EP-splits.fa <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--splits-mode</span> <span class="at">--no-wrap</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And the code we used to run VirSorter2.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb15"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate virsorter2</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">--seqfile</span> 03_CONTIGS/WA-splits.fa <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">--working-dir</span> WA/ <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">--keep-original-seq</span> <span class="dt">\</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">--prep-for-dramv</span> <span class="dt">\</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>              <span class="at">--hallmark-required-on-short</span> <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>              <span class="at">--db-dir</span> virsorter2_db </span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">--seqfile</span> 03_CONTIGS/EP-splits.fa <span class="dt">\</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>              <span class="at">--working-dir</span> EP/ <span class="dt">\</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>              <span class="at">--keep-original-seq</span> <span class="dt">\</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>              <span class="at">--prep-for-dramv</span> <span class="dt">\</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>              <span class="at">--hallmark-required-on-short</span> <span class="dt">\</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>              <span class="at">--db-dir</span> virsorter2_db</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Now, to get Virsorter2 annotations into the anvi’o contig databases there are a few special steps that need to be taken. Please <a href="https://github.com/simroux/VirSorter2_to_Anvio">see this post</a> and associated <code>virsorter_to_anvio.py</code> script for more details. Here we will only include the code with minimal explanation. Too long…</p>
<p>First we export two tables that the <code>virsorter_to_anvio.py</code> script needs for import.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb16"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/WA-contigs.db  <span class="at">--table</span> splits_basic_info <span class="dt">\</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--output-file</span> WA_splits_basic_info.txt</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/EP-contigs.db  <span class="at">--table</span> splits_basic_info <span class="dt">\</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--output-file</span> EP_splits_basic_info.txt</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> WA_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--gene-caller</span> prodigal</span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> EP_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--gene-caller</span> prodigal</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Time to get messy. At the time of this writing, the <code>gene_calls</code> file exported from anvi’o is a 10-column tab-delimited text file. The <code>virsorter_to_anvio.py</code> script needs only 8 of these, and they need to be in a specific order. No problem, we can use <a href="https://www.gnu.org/software/gawk/manual/gawk.html"><code>awk</code></a>.</p>
<div class="callout callout-style-default callout-note callout-titled" title="irSorter2 column values">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>irSorter2 column values
</div>
</div>
<div class="callout-body-container callout-body">
<p>These are the column values needed by the <code>virsorter_to_anvio.py</code> script:<code>gene_callers_id</code>, <code>contig</code>, <code>start</code>, <code>stop</code>, <code>direction</code>, <code>partial</code>, <code>source</code>, <code>version</code></p>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb17"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'BEGIN {FS="\t"; OFS="\t"} {print $1, $2, $3, $4, $5, $6, $8, $9}'</span> WA_all_gene_calls.txt</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'BEGIN {FS="\t"; OFS="\t"} {print $1, $2, $3, $4, $5, $6, $8, $9}'</span> EP_all_gene_calls.txt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>After this we run the parsing script.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb18"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter_to_anvio.py</span> <span class="at">-i</span> WA/ <span class="at">-s</span> WA/WA_splits_basic_info.txt <span class="dt">\</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-n</span> WA/WA_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-d</span> virsorter2_db <span class="dt">\</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-A</span> WA/WA_virsorter_additional_info.txt <span class="dt">\</span></span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-C</span> WA/WA_virsorter_collection.txt <span class="dt">\</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-F</span> WA/WA_virsorter_annotations.txt</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter_to_anvio.py</span> <span class="at">-i</span> EP/ <span class="at">-s</span> EP/EP_splits_basic_info.txt <span class="dt">\</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-n</span> EP/EP_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-d</span> virsorter2_db <span class="dt">\</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-A</span> EP/EP_virsorter_additional_info.txt <span class="dt">\</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-C</span> EP/EP_virsorter_collection.txt <span class="dt">\</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-F</span> EP/EP_virsorter_annotations.txt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And import the resulting files to anvi’o</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb19"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-misc-data</span> WA-virsorter_additional_info.txt </span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/WA/PROFILE.db  </span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">--target-data-table</span> items</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-misc-data</span> EP-virsorter_additional_info.txt </span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/EP/PROFILE.db  </span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">--target-data-table</span> items</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-collection</span> WA-virsorter_collection.txt </span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-c</span> 03_CONTIGS/WA-contigs.db </span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/WA/PROFILE.db </span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-C</span> VIRSORTER2</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-collection</span> EP-virsorter_collection.txt </span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-c</span> 03_CONTIGS/EP-contigs.db </span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/EP/PROFILE.db</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-C</span> VIRSORTER2</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>VirSorter2 annotation complete.</p>
</section></section><section id="kaiju-annotation" class="level2"><h2 class="anchored" data-anchor-id="kaiju-annotation">Kaiju Annotation</h2>
<p>Here we use <a href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a> <span class="citation" data-cites="menzel2016kaiju">(<a href="#ref-menzel2016kaiju" role="doc-biblioref">Menzel, Ng, and Krogh 2016</a>)</span> to classify gene calls. We do this against the <a href="https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index">progenomes databases</a>, a r epresentative set of genomes from the <a href="http://progenomes.embl.de/">proGenomes</a> database and viruses from the NCBI RefSeq database. We describe the contruction of this database <a href="../setup/index.html#kaiju">here</a>.</p>
<p>Start by grabbing gene call fasta files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb20"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="at">-o</span> EP_gene_calls.fna</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="at">-o</span> WA_gene_calls.fna</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb21"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">-f</span> kaiju_db/kaiju_db_progenomes.fmi <span class="dt">\</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>      <span class="at">-i</span> EP_gene_calls.fna <span class="dt">\ </span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>      <span class="ex">-o</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>      <span class="at">-z</span> <span class="va">$NSLOTS</span> <span class="at">-v</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>      <span class="at">-f</span> kaiju_db/kaiju_db_progenomes.fmi <span class="dt">\</span></span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>      <span class="at">-i</span> WA_gene_calls.fna <span class="dt">\</span></span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>      <span class="at">-o</span> WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>      <span class="at">-z</span> <span class="va">$NSLOTS</span> <span class="at">-v</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-addTaxonNames</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-i</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> EP_kaiju_nr.names <span class="dt">\</span></span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-r</span> superkingdom,phylum,order,class,family,genus,species</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-addTaxonNames</span> <span class="at">-t</span> kaiju_db/nodes.dmp </span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-i</span> WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> WA_kaiju_nr.names <span class="dt">\ </span></span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">-r</span> superkingdom,phylum,order,class,family,genus,species</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Import the output to anvi’o.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb22"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-taxonomy-for-genes</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-p</span> kaiju <span class="dt">\</span></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-i</span> EP_kaiju_nr.names </span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-taxonomy-for-genes</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-p</span> kaiju <span class="dt">\</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-i</span> WA_kaiju_nr.names</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And generate Krona plots of the data. A little dance between the Kaiju and Krona environments.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb23"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2krona</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">-i</span> <span class="va">$KAIJU</span>/WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> WA_kaiju_nr.out.krona</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2krona</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">-i</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> EP_kaiju_nr.out.krona</span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb24"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krona</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="ex">ktImportText</span> <span class="at">-o</span> WA_kaiju_nr.out.html WA_kaiju_nr.out.krona</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="ex">ktImportText</span> <span class="at">-o</span> EP_kaiju_nr.out.html EP_kaiju_nr.out.krona</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb25"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2table</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">-r</span> class <span class="dt">\</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">-l</span> phylum,class,order,family <span class="dt">\</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> WA_kaiju_nr.out.summary WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>kaiju2table <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">-r</span> class <span class="dt">\</span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">-l</span> phylum,class,order,family <span class="dt">\</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> EP_kaiju_nr.out.summary EP_kaiju_nr.out</span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Kaiju done.</p>
</section></section><div id="quarto-appendix" class="default"><section id="detailed-session-info" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Detailed Session Info</h2><div class="quarto-appendix-contents">
<details><summary>Expand to see Session Info</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Ventura 13.7.8
 system   x86_64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2025-10-02
 pandoc   3.6.3 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/x86_64/ (via rmarkdown)
 quarto   1.8.25 @ /Applications/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
======   Devtools Session info   ===================================</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.</code></pre>
</div>
</div>
</details></div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-10-02</p>



<!-- -->

</div></section><section class="level4 appendix"><h2 class="quarto-appendix-heading">Collected Links</h2><div class="quarto-appendix-contents">
<ul>
<li><a href="http://merenlab.org/data/tara-oceans-mags/">Recovering Microbial Genomes from TARA Oceans Metagenomes</a></li>
<li><a href="http://merenlab.org/2018/07/09/anvio-snakemake-workflows/">setting up Snakemake workflows</a></li>
<li><a href="include/default_mg.json">download our configs file</a></li>
<li><a href="https://anvio.org/help/main/workflows/metagenomics/">on the anvi’o website</a></li>
<li><a href="include/samples.txt"><code>samples.txt</code></a></li>
<li><a href="http://merenlab.org/2018/07/09/anvio-snakemake-workflows/#how-can-i-restart-a-failed-job">anvi’o snakemake workflows</a></li>
<li><a href="../../../workflows/pub/references/">Workflow References</a></li>
<li><a href="https://github.com/marbl/Krona/wiki">Krona plots</a></li>
<li><a href="https://github.com/jenniferlu717/KrakenTools">KrakenTool</a></li>
<li><a href="https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file#kreport2mpapy"><code>kreport2mpa.py</code></a></li>
<li><a href="https://github.com/bxlab/metaWRAP">metaWRAP</a></li>
<li><a href="https://github.com/jiarong/VirSorter2">VirSorter2</a></li>
<li><a href="https://github.com/simroux/VirSorter2_to_Anvio">see this post</a></li>
<li><a href="https://www.gnu.org/software/gawk/manual/gawk.html"><code>awk</code></a></li>
<li><a href="https://github.com/bioinformatics-centre/kaiju">Kaiju</a></li>
<li><a href="https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index">progenomes databases</a></li>
<li><a href="http://progenomes.embl.de/">proGenomes</a></li>
<li><a href="../setup/index.html#kaiju">here</a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://istmobiome.rbind.io/">The Istmobiome Project</a></li>
</ul>

</div></section><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-aramaki2020kegg" class="csl-entry" role="listitem">
Aramaki, Takuya, Romain Blanc-Mathieu, Hisashi Endo, Koichi Ohkubo, Minoru Kanehisa, Susumu Goto, and Hiroyuki Ogata. 2020. <span>“KofamKOALA: KEGG Ortholog Assignment Based on Profile HMM and Adaptive Score Threshold.”</span> <em>Bioinformatics</em> 36 (7): 22512252. <a href="https://doi.org/10.1093/bioinformatics/btz859">https://doi.org/10.1093/bioinformatics/btz859</a>.
</div>
<div id="ref-delmont2018nitrogen" class="csl-entry" role="listitem">
Delmont, Tom O, Christopher Quince, Alon Shaiber, Özcan C Esen, Sonny TM Lee, Michael S Rappé, Sandra L McLellan, Sebastian Lücker, and A Murat Eren. 2018. <span>“Nitrogen-Fixing Populations of Planctomycetes and Proteobacteria Are Abundant in Surface Ocean Metagenomes.”</span> <em>Nature Microbiology</em> 3 (7): 804–13. <a href="https://doi.org/10.1038/s41564-018-0176-9">https://doi.org/10.1038/s41564-018-0176-9</a>.
</div>
<div id="ref-drula2022cazyme" class="csl-entry" role="listitem">
Drula, Elodie, Marie-Line Garron, Suzan Dogan, Vincent Lombard, Bernard Henrissat, and Nicolas Terrapon. 2022. <span>“The Carbohydrate-Active Enzyme Database: Functions and Literature.”</span> <em>Nucleic Acids Research</em> 50 (D1): D571D577. <a href="https://doi.org/10.1093/nar/gkab1045">https://doi.org/10.1093/nar/gkab1045</a>.
</div>
<div id="ref-eren2021community" class="csl-entry" role="listitem">
Eren, A Murat, E Kiefl, A Shaiber, I Veseli, SE Miller, MS Schechter, I Fink, et al. 2021. <span>“Community-Led, Integrated, Reproducible Multi-Omics with Anvi’o. 6: 3–6.”</span> <em>Nature Microbiology</em>. <a href="https://doi.org/10.1038/s41564-020-00834-3">https://doi.org/10.1038/s41564-020-00834-3</a>.
</div>
<div id="ref-eren2013filtering" class="csl-entry" role="listitem">
Eren, A Murat, Joseph H Vineis, Hilary G Morrison, and Mitchell L Sogin. 2013. <span>“A Filtering Method to Generate High Quality Short Reads Using Illumina Paired-End Technology.”</span> <em>PLoS One</em> 8 (6). <a href="https://doi.org/10.1371/journal.pone.0066643">https://doi.org/10.1371/journal.pone.0066643</a>.
</div>
<div id="ref-galperin2021cog" class="csl-entry" role="listitem">
Galperin, Michael Y, Yuri I Wolf, Kira S Makarova, Roberto Vera Alvarez, David Landsman, and Eugene V Koonin. 2021. <span>“COG Database Update: Focus on Microbial Diversity, Model Organisms, and Widespread Pathogens.”</span> <em>Nucleic Acids Research</em> 49 (D1): D274D281. <a href="https://doi.org/10.1093/nar/gkaa1018">https://doi.org/10.1093/nar/gkaa1018</a>.
</div>
<div id="ref-guo2021virsorter2" class="csl-entry" role="listitem">
Guo, Jiarong, Ben Bolduc, Ahmed A Zayed, Arvind Varsani, Guillermo Dominguez-Huerta, Tom O Delmont, Akbar Adjie Pratama, et al. 2021. <span>“VirSorter2: A Multi-Classifier, Expert-Guided Approach to Detect Diverse DNA and RNA Viruses.”</span> <em>Microbiome</em> 9: 113. <a href="https://doi.org/10.1186/s40168-020-00990-y">https://doi.org/10.1186/s40168-020-00990-y</a>.
</div>
<div id="ref-hyatt2010prodigal" class="csl-entry" role="listitem">
Hyatt, Doug, Gwo-Liang Chen, Philip F LoCascio, Miriam L Land, Frank W Larimer, and Loren J Hauser. 2010. <span>“Prodigal: Prokaryotic Gene Recognition and Translation Initiation Site Identification.”</span> <em>BMC Bioinformatics</em> 11 (1): 119. <a href="https://doi.org/10.1186/1471-2105-11-119">https://doi.org/10.1186/1471-2105-11-119</a>.
</div>
<div id="ref-koster2012snakemake" class="csl-entry" role="listitem">
Köster, Johannes, and Sven Rahmann. 2012. <span>“Snakemake—a Scalable Bioinformatics Workflow Engine.”</span> <em>Bioinformatics</em> 28 (19): 2520–22. <a href="https://doi.org/10.1093/bioinformatics/bts480">https://doi.org/10.1093/bioinformatics/bts480</a>.
</div>
<div id="ref-langmead2012bowtie" class="csl-entry" role="listitem">
Langmead, Ben, and Steven L Salzberg. 2012. <span>“Fast Gapped-Read Alignment with Bowtie 2.”</span> <em>Nature Methods</em> 9 (4): 357. <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.
</div>
<div id="ref-li2015megahit" class="csl-entry" role="listitem">
Li, Dinghua, Chi-Man Liu, Ruibang Luo, Kunihiko Sadakane, and Tak-Wah Lam. 2015. <span>“MEGAHIT: An Ultra-Fast Single-Node Solution for Large and Complex Metagenomics Assembly via Succinct de Bruijn Graph.”</span> <em>Bioinformatics</em> 31 (10): 1674–76. <a href="https://doi.org/10.1093/bioinformatics/btv033">https://doi.org/10.1093/bioinformatics/btv033</a>.
</div>
<div id="ref-li2009samtools" class="csl-entry" role="listitem">
Li, Heng, Bob Handsaker, Alec Wysoker, Tim Fennell, Jue Ruan, Nils Homer, Gabor Marth, Goncalo Abecasis, and Richard Durbin. 2009. <span>“The Sequence Alignment/Map Format and SAMtools.”</span> <em>Bioinformatics</em> 25 (16): 2078–79. <a href="https://doi.org/10.1093/bioinformatics/btp352">https://doi.org/10.1093/bioinformatics/btp352</a>.
</div>
<div id="ref-menzel2016kaiju" class="csl-entry" role="listitem">
Menzel, Peter, Kim Lee Ng, and Anders Krogh. 2016. <span>“Fast and Sensitive Taxonomic Classification for Metagenomics with Kaiju.”</span> <em>Nature Communications</em> 7: 11257. <a href="https://doi.org/10.1038/ncomms11257">https://doi.org/10.1038/ncomms11257</a>.
</div>
<div id="ref-minoche2011evaluation" class="csl-entry" role="listitem">
Minoche, André E, Juliane C Dohm, and Heinz Himmelbauer. 2011. <span>“Evaluation of Genomic High-Throughput Sequencing Data Generated on Illumina HiSeq and Genome Analyzer Systems.”</span> <em>Genome Biology</em> 12 (11): R112. <a href="https://doi.org/10.1186/gb-2011-12-11-r112">https://doi.org/10.1186/gb-2011-12-11-r112</a>.
</div>
<div id="ref-mistry2021pfam" class="csl-entry" role="listitem">
Mistry, Jaina, Sara Chuguransky, Lowri Williams, Matloob Qureshi, Gustavo A Salazar, Erik LL Sonnhammer, Silvio CE Tosatto, et al. 2021. <span>“Pfam: The Protein Families Database in 2021.”</span> <em>Nucleic Acids Research</em> 49 (D1): D412D419. <a href="https://doi.org/10.1093/nar/gkaa913">https://doi.org/10.1093/nar/gkaa913</a>.
</div>
<div id="ref-wood2019kraken2" class="csl-entry" role="listitem">
Wood, Derrick E, Jennifer Lu, and Ben Langmead. 2019. <span>“Improved Metagenomic Analysis with Kraken 2.”</span> <em>Genome Biology</em> 20: 113. <a href="https://doi.org/10.1186/s13059-019-1891-0">https://doi.org/10.1186/s13059-019-1891-0</a>.
</div>
</div></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/trans-shrimp\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../../workflows/mg/setup/index.html" class="pagination-link" aria-label="1. Build Databases">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">1. Build Databases</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../../../workflows/mg/aa-results/index.html" class="pagination-link" aria-label="3. Assembly &amp; Annotation Results">
        <span class="nav-page-text">3. Assembly &amp; Annotation Results</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb29" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "2. Co-Assembly &amp; Annotations"</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co">  This section describes the steps we took to co-assemble the metagenomic samples, classify taxonomy, and assign functions.</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="an">lightbox:</span><span class="co"> true</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a><span class="an">listing:</span><span class="co"> </span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="co">  id: snaklemake-workflow</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co">  contents: include/snaklemake-workflow.yaml</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="co">  type: table</span></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a><span class="co">  sort-ui: false</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="co">  filter-ui: false</span></span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a><span class="co">  fields: [name, tool, description, input, reference]</span></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co">  field-links: </span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="co">    - tool</span></span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a><span class="co">  field-display-names: </span></span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a><span class="co">    name: Task</span></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a><span class="co">    tool: Tool</span></span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a><span class="co">    description: Description</span></span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a><span class="co">    input: Input</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_setup.qmd &gt;}}</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data Availability</span></span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>PENDING</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a><span class="fu"># The Pipeline</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>In this section of the workflow we begin with raw, paired-end Illumina data. We will use a Snakemake <span class="co">[</span><span class="ot">@koster2012snakemake</span><span class="co">]</span> workflow within the anvi'o <span class="co">[</span><span class="ot">@eren2021community</span><span class="co">]</span> environment for most of the steps, though we will at times call on additional tools for specific steps. The overall structure of our workflow was modeled after the one described by Delmont &amp; Eren on <span class="co">[</span><span class="ot">Recovering Microbial Genomes from TARA Oceans Metagenomes</span><span class="co">](http://merenlab.org/data/tara-oceans-mags/)</span> <span class="co">[</span><span class="ot">@delmont2018nitrogen</span><span class="co">]</span>.</span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>The main steps of this workflow (in order) are:</span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-34"><a href="#cb29-34" aria-hidden="true" tabindex="-1"></a>::: {#snaklemake-workflow .column-body}</span>
<span id="cb29-35"><a href="#cb29-35" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-36"><a href="#cb29-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-37"><a href="#cb29-37" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_mg_flowchart_1.qmd &gt;}}</span>
<span id="cb29-38"><a href="#cb29-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-39"><a href="#cb29-39" aria-hidden="true" tabindex="-1"></a>We have a total of 58 samples ...DETAILS</span>
<span id="cb29-40"><a href="#cb29-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-41"><a href="#cb29-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># Snakemake Workflow</span></span>
<span id="cb29-42"><a href="#cb29-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-43"><a href="#cb29-43" aria-hidden="true" tabindex="-1"></a>Thanks to the anvi'o implementation of Snakemake we can run many commands sequentially and/or simultaneously. There is plenty of good documentation on the anvi'o website about <span class="co">[</span><span class="ot">setting up Snakemake workflows</span><span class="co">](http://merenlab.org/2018/07/09/anvio-snakemake-workflows/)</span> so we will refrain from any lengthy explanations.</span>
<span id="cb29-44"><a href="#cb29-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-45"><a href="#cb29-45" aria-hidden="true" tabindex="-1"></a>You can see all the settings we used and <span class="co">[</span><span class="ot">download our configs file</span><span class="co">](include/default_mg.json)</span> or grab it from the folded code below. Some commands we chose not to run but they remain in the workflow for posterity.</span>
<span id="cb29-46"><a href="#cb29-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-49"><a href="#cb29-49" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-50"><a href="#cb29-50" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--list-workflows</span></span>
<span id="cb29-51"><a href="#cb29-51" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-52"><a href="#cb29-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-53"><a href="#cb29-53" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb29-54"><a href="#cb29-54" aria-hidden="true" tabindex="-1"></a><span class="in">WARNING</span></span>
<span id="cb29-55"><a href="#cb29-55" aria-hidden="true" tabindex="-1"></a><span class="in">===============================================</span></span>
<span id="cb29-56"><a href="#cb29-56" aria-hidden="true" tabindex="-1"></a><span class="in">If you publish results from this workflow, please do not forget to cite</span></span>
<span id="cb29-57"><a href="#cb29-57" aria-hidden="true" tabindex="-1"></a><span class="in">snakemake (doi:10.1093/bioinformatics/bts480)</span></span>
<span id="cb29-58"><a href="#cb29-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-59"><a href="#cb29-59" aria-hidden="true" tabindex="-1"></a><span class="in">HMM profiles .................................: 9 sources have been loaded: </span></span>
<span id="cb29-60"><a href="#cb29-60" aria-hidden="true" tabindex="-1"></a><span class="in">Archaea_76 (76 genes, domain: archaea), </span></span>
<span id="cb29-61"><a href="#cb29-61" aria-hidden="true" tabindex="-1"></a><span class="in">Bacteria_71 (71 genes, domain: bacteria), </span></span>
<span id="cb29-62"><a href="#cb29-62" aria-hidden="true" tabindex="-1"></a><span class="in">Protista_83 (83 genes, domain: eukarya), </span></span>
<span id="cb29-63"><a href="#cb29-63" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_12S (1 genes, domain: None), </span></span>
<span id="cb29-64"><a href="#cb29-64" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_16S (3 genes, domain: None), </span></span>
<span id="cb29-65"><a href="#cb29-65" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_18S (1 genes, domain: None), </span></span>
<span id="cb29-66"><a href="#cb29-66" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_23S (2 genes, domain: None), </span></span>
<span id="cb29-67"><a href="#cb29-67" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_28S (1 genes, domain: None),</span></span>
<span id="cb29-68"><a href="#cb29-68" aria-hidden="true" tabindex="-1"></a><span class="in">Ribosomal_RNA_5S (5 genes, domain: None)</span></span>
<span id="cb29-69"><a href="#cb29-69" aria-hidden="true" tabindex="-1"></a><span class="in">Available workflows ..........................: contigs, metagenomics, pangenomics, phylogenomics, trnaseq, ecophylo, sra_download</span></span>
<span id="cb29-70"><a href="#cb29-70" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-71"><a href="#cb29-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-74"><a href="#cb29-74" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-75"><a href="#cb29-75" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--workflow</span> metagenomics <span class="dt">\</span></span>
<span id="cb29-76"><a href="#cb29-76" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--get-default-config</span> default_mg.json  <span class="dt">\</span></span>
<span id="cb29-77"><a href="#cb29-77" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--list-dependencies</span></span>
<span id="cb29-78"><a href="#cb29-78" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-79"><a href="#cb29-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-80"><a href="#cb29-80" aria-hidden="true" tabindex="-1"></a>You can find a description of the metagenomic workflow <span class="co">[</span><span class="ot">on the anvi'o website</span><span class="co">](https://anvio.org/help/main/workflows/metagenomics/)</span>.</span>
<span id="cb29-81"><a href="#cb29-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-84"><a href="#cb29-84" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb29-85"><a href="#cb29-85" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-fold: true</span></span>
<span id="cb29-86"><a href="#cb29-86" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb29-87"><a href="#cb29-87" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb29-88"><a href="#cb29-88" aria-hidden="true" tabindex="-1"></a><span class="co">#| code-summary: "Show JSON-formatted configuration file."</span></span>
<span id="cb29-89"><a href="#cb29-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-90"><a href="#cb29-90" aria-hidden="true" tabindex="-1"></a><span class="ex">{{</span><span class="op">&lt;</span> include <span class="ex">include/_default_mg.qmd</span> <span class="op">&gt;</span>}}</span>
<span id="cb29-91"><a href="#cb29-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-92"><a href="#cb29-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-93"><a href="#cb29-93" aria-hidden="true" tabindex="-1"></a>Since we are doing a co-assembly, we need a separate file called <span class="co">[</span><span class="ot">`samples.txt`</span><span class="co">](include/samples.txt)</span>. This file tells anvi'o where to find the **raw** fastq files for each sample and what group the sample belongs to. The file is a four-column, tab-delimited file.</span>
<span id="cb29-94"><a href="#cb29-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-95"><a href="#cb29-95" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> sample              <span class="pp">|</span> group <span class="pp">|</span> r1            <span class="pp">|</span> r2            <span class="pp">|</span></span>
<span id="cb29-96"><a href="#cb29-96" aria-hidden="true" tabindex="-1"></a><span class="pp">|---------------------|-------|---------------|---------------|</span></span>
<span id="cb29-97"><a href="#cb29-97" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> EP_ALPH_AREN_GL_P01 <span class="pp">|</span> EP    <span class="pp">|</span> forward reads <span class="pp">|</span> reverse reads <span class="pp">|</span></span>
<span id="cb29-98"><a href="#cb29-98" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> EP_ALPH_AREN_HP_P01 <span class="pp">|</span> EP    <span class="pp">|</span> forward reads <span class="pp">|</span> reverse reads <span class="pp">|</span></span>
<span id="cb29-99"><a href="#cb29-99" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> WA_ALPH_BOUV_GL_P02 <span class="pp">|</span> WA    <span class="pp">|</span> forward reads <span class="pp">|</span> reverse reads <span class="pp">|</span></span>
<span id="cb29-100"><a href="#cb29-100" aria-hidden="true" tabindex="-1"></a><span class="pp">|</span> WA_ALPH_BOUV_HP_P02 <span class="pp">|</span> WA    <span class="pp">|</span> forward reads <span class="pp">|</span> reverse reads <span class="pp">|</span></span>
<span id="cb29-101"><a href="#cb29-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-102"><a href="#cb29-102" aria-hidden="true" tabindex="-1"></a>Remember, there are 4 fastq files *per* direction (forward or reverse) *per* sample. So each comma separated list in the `r1` column needs four file names and the same with the `r2` column. And you must include *relative path names*.</span>
<span id="cb29-103"><a href="#cb29-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-104"><a href="#cb29-104" aria-hidden="true" tabindex="-1"></a>To see what will happen with the <span class="in">`config`</span> file we can visualize the Snakemake workflow using a Directed acyclic graph (DAG) where edge connections represent dependencies and nodes represent commands. The workflow begins with trimmed reads and continues up to and including automatic binning of contigs. At the end of this workflow we then proceed with manual binning and MAG generation. We added nodes for VirSorter and Kaiju annotations since these are not part of the normal anvi'o workflow.</span>
<span id="cb29-105"><a href="#cb29-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-108"><a href="#cb29-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-109"><a href="#cb29-109" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">--workflow</span> metagenomics <span class="dt">\</span></span>
<span id="cb29-110"><a href="#cb29-110" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--save-workflow-graph</span> <span class="dt">\</span></span>
<span id="cb29-111"><a href="#cb29-111" aria-hidden="true" tabindex="-1"></a>                  <span class="at">-c</span> default_mg_for_dag.json</span>
<span id="cb29-112"><a href="#cb29-112" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-113"><a href="#cb29-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-114"><a href="#cb29-114" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_DAG_CODE.qmd &gt;}}</span>
<span id="cb29-115"><a href="#cb29-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-118"><a href="#cb29-118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb29-119"><a href="#cb29-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb29-120"><a href="#cb29-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb29-121"><a href="#cb29-121" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: |</span></span>
<span id="cb29-122"><a href="#cb29-122" aria-hidden="true" tabindex="-1"></a><span class="co">#|   Colors indicate broad divisions of workflow: </span></span>
<span id="cb29-123"><a href="#cb29-123" aria-hidden="true" tabindex="-1"></a><span class="co">#|   sky blue, short-read prep &amp; co-assembly; </span></span>
<span id="cb29-124"><a href="#cb29-124" aria-hidden="true" tabindex="-1"></a><span class="co">#|   blueish green, short-read mapping to assembly; </span></span>
<span id="cb29-125"><a href="#cb29-125" aria-hidden="true" tabindex="-1"></a><span class="co">#|   orange, taxonomic or functional classification; </span></span>
<span id="cb29-126"><a href="#cb29-126" aria-hidden="true" tabindex="-1"></a><span class="co">#|   yellow, automatic binning; </span></span>
<span id="cb29-127"><a href="#cb29-127" aria-hidden="true" tabindex="-1"></a><span class="co">#|   reddish purple, databases construction.</span></span>
<span id="cb29-128"><a href="#cb29-128" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-subcap:</span></span>
<span id="cb29-129"><a href="#cb29-129" aria-hidden="true" tabindex="-1"></a><span class="co">#|   - "DAG of the metagenomic workflow for a small subset of samples."</span></span>
<span id="cb29-130"><a href="#cb29-130" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">"include/dag.png"</span>)</span>
<span id="cb29-131"><a href="#cb29-131" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-132"><a href="#cb29-132" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-133"><a href="#cb29-133" aria-hidden="true" tabindex="-1"></a>And here are the commands we used to run the workflow.</span>
<span id="cb29-134"><a href="#cb29-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-137"><a href="#cb29-137" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-138"><a href="#cb29-138" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">-w</span> metagenomics <span class="at">-c</span> default_mg.json <span class="dt">\</span></span>
<span id="cb29-139"><a href="#cb29-139" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--additional-params</span> <span class="at">--jobs</span> 28 <span class="dt">\</span></span>
<span id="cb29-140"><a href="#cb29-140" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--resources</span> nodes=28 <span class="dt">\</span></span>
<span id="cb29-141"><a href="#cb29-141" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--keep-going</span> <span class="dt">\</span></span>
<span id="cb29-142"><a href="#cb29-142" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--rerun-incomplete</span> <span class="at">--unlock</span></span>
<span id="cb29-143"><a href="#cb29-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-144"><a href="#cb29-144" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-run-workflow</span> <span class="at">-w</span> metagenomics <span class="at">-c</span> default_mg.json <span class="dt">\</span></span>
<span id="cb29-145"><a href="#cb29-145" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--additional-params</span> <span class="at">--jobs</span> 28 <span class="dt">\</span></span>
<span id="cb29-146"><a href="#cb29-146" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--resources</span> nodes=28 <span class="at">--keep-going</span> <span class="at">--rerun-incomplete</span></span>
<span id="cb29-147"><a href="#cb29-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-148"><a href="#cb29-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-149"><a href="#cb29-149" aria-hidden="true" tabindex="-1"></a>If a Snakemake job fails the default behavior is to lock the workflow and, because large jobs can fail for a lot of reasons, we decided to always include an identical command first with the <span class="in">`--unlock`</span> flag. This will unlock the workflow and then the second command will execute. You can find an explanation of this post on <span class="co">[</span><span class="ot">anvi'o snakemake workflows</span><span class="co">](http://merenlab.org/2018/07/09/anvio-snakemake-workflows/#how-can-i-restart-a-failed-job)</span>.</span>
<span id="cb29-150"><a href="#cb29-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-151"><a href="#cb29-151" aria-hidden="true" tabindex="-1"></a><span class="fu">## Snakemake Citations</span></span>
<span id="cb29-152"><a href="#cb29-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-153"><a href="#cb29-153" aria-hidden="true" tabindex="-1"></a>There are many tools used in the workflow that need to be cited. See the <span class="co">[</span><span class="ot">Workflow References</span><span class="co">](/workflows/pub/references/)</span> page for details. </span>
<span id="cb29-154"><a href="#cb29-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-155"><a href="#cb29-155" aria-hidden="true" tabindex="-1"></a>There are also a few tools that we ran outside of the Snakemake workflow. Results from these steps need to be added to the individual <span class="in">`PROFILE.db`</span>'s, merged <span class="in">`PROFILE.db`</span>, or <span class="in">`CONTIGS.db`</span>. Therefore, before the <span class="in">`anvi-merge`</span> portion of the Snakemake workflow finished, we killed the job, ran the accessory analyses described below, and then restarted the workflow to finish the missing step. Cumbersome, yes, but it got the job done.</span>
<span id="cb29-156"><a href="#cb29-156" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-157"><a href="#cb29-157" aria-hidden="true" tabindex="-1"></a><span class="fu"># Pipeline Output</span></span>
<span id="cb29-158"><a href="#cb29-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-159"><a href="#cb29-159" aria-hidden="true" tabindex="-1"></a>When the workflow is complete, you should see the following directories within your project folder. </span>
<span id="cb29-160"><a href="#cb29-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-161"><a href="#cb29-161" aria-hidden="true" tabindex="-1"></a>Directory names and descriptions:</span>
<span id="cb29-162"><a href="#cb29-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-163"><a href="#cb29-163" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**00_LOGS/**: Individual log files for each step of the snakemake workflow.</span>
<span id="cb29-164"><a href="#cb29-164" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**01_QC/**: Merged forward (R1) &amp; reverse (R2) QC'ed fastq files. QC STATS file for each sample. <span class="in">`qc-report.txt`</span> file, a summary table of all QC results.</span>
<span id="cb29-165"><a href="#cb29-165" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**02_FASTA/**: Contig fasta files and reformat reports for each assembly.</span>
<span id="cb29-166"><a href="#cb29-166" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**03_CONTIGS/**: An annotated contig database for each assembly.</span>
<span id="cb29-167"><a href="#cb29-167" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**04_MAPPING/**: Short read BAM files.</span>
<span id="cb29-168"><a href="#cb29-168" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**05_ANVIO_PROFILE/**: Profile database for each sample.</span>
<span id="cb29-169"><a href="#cb29-169" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>**06_MERGED/**: Single merged profile database.</span>
<span id="cb29-170"><a href="#cb29-170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-171"><a href="#cb29-171" aria-hidden="true" tabindex="-1"></a>The <span class="in">`contigs.db`</span> contains all taxonomic and functional annotations. The <span class="in">`profile.db`</span> contains information on coverage, detection, etc.</span>
<span id="cb29-172"><a href="#cb29-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-173"><a href="#cb29-173" aria-hidden="true" tabindex="-1"></a><span class="fu"># Annotations</span></span>
<span id="cb29-174"><a href="#cb29-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-175"><a href="#cb29-175" aria-hidden="true" tabindex="-1"></a>This part of the workflow contains information for ancillary annotations (i.e., outside the Snakmake workflow). We will mainly cover taxonomic annotations as most functional annotations (e.g., KEGG, COG, Pfam, etc.) are run in the workflow.</span>
<span id="cb29-176"><a href="#cb29-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-177"><a href="#cb29-177" aria-hidden="true" tabindex="-1"></a><span class="fu">## Taxonomic </span></span>
<span id="cb29-178"><a href="#cb29-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-179"><a href="#cb29-179" aria-hidden="true" tabindex="-1"></a>In this section we discuss taxonomic classification of short reads, contigs, and gene calls. We go through the steps of analyzing the data and getting the results into anvi'o databases. </span>
<span id="cb29-180"><a href="#cb29-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-181"><a href="#cb29-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-182"><a href="#cb29-182" aria-hidden="true" tabindex="-1"></a><span class="fu">### Short-reads with Kraken2</span></span>
<span id="cb29-183"><a href="#cb29-183" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-184"><a href="#cb29-184" aria-hidden="true" tabindex="-1"></a>In this section we use Kraken2 <span class="co">[</span><span class="ot">@wood2019kraken2</span><span class="co">]</span> to classify the **short reads**. Our goal is to classify short-reads, generate an input file for anvi'o, and create <span class="co">[</span><span class="ot">Krona plots</span><span class="co">](https://github.com/marbl/Krona/wiki)</span> for data visualization. Brace yourself.</span>
<span id="cb29-185"><a href="#cb29-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-186"><a href="#cb29-186" aria-hidden="true" tabindex="-1"></a>::: {.callout-important}</span>
<span id="cb29-187"><a href="#cb29-187" aria-hidden="true" tabindex="-1"></a>Since Kraken2 annotation is performed on individual samples and the results are imported into the individual <span class="in">`profile.db`</span>'s we will need to re-merge the all <span class="in">`profile.db`</span> after these steps are completed. The merging step is basically the last step of the Snakemake workflow. </span>
<span id="cb29-188"><a href="#cb29-188" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-189"><a href="#cb29-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-192"><a href="#cb29-192" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-193"><a href="#cb29-193" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kraken2</span>
<span id="cb29-194"><a href="#cb29-194" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb29-195"><a href="#cb29-195" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb29-196"><a href="#cb29-196" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken2</span> <span class="at">--paired</span> 01_QC/<span class="va">$sample</span>-QUALITY_PASSED_R1.fastq.gz  <span class="dt">\</span></span>
<span id="cb29-197"><a href="#cb29-197" aria-hidden="true" tabindex="-1"></a>                     01_QC/<span class="va">$sample</span>-QUALITY_PASSED_R2.fastq.gz <span class="dt">\</span></span>
<span id="cb29-198"><a href="#cb29-198" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--db</span> kraken2_db/ <span class="dt">\</span></span>
<span id="cb29-199"><a href="#cb29-199" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--use-names</span> <span class="dt">\</span></span>
<span id="cb29-200"><a href="#cb29-200" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--threads</span> <span class="va">$NSLOTS</span> <span class="dt">\</span></span>
<span id="cb29-201"><a href="#cb29-201" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--output</span> <span class="va">$sample</span>-kraken.out <span class="dt">\</span></span>
<span id="cb29-202"><a href="#cb29-202" aria-hidden="true" tabindex="-1"></a>                     <span class="at">--report</span> <span class="va">$sample</span>-kraken-report.txt</span>
<span id="cb29-203"><a href="#cb29-203" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-204"><a href="#cb29-204" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span>
<span id="cb29-205"><a href="#cb29-205" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-206"><a href="#cb29-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-207"><a href="#cb29-207" aria-hidden="true" tabindex="-1"></a>After this is finished we should have two files for each sample--a <span class="in">`.out`</span> file containing the results of the Kraken2 annotation and a <span class="in">`.report.txt`</span> file that summarizes the results. </span>
<span id="cb29-208"><a href="#cb29-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-209"><a href="#cb29-209" aria-hidden="true" tabindex="-1"></a>First, we generate the file that anvi'o needs--the format is very specific. For this task we use <span class="co">[</span><span class="ot">KrakenTool</span><span class="co">](https://github.com/jenniferlu717/KrakenTools)</span>--a suite of very handy scripts to Kraken 2 data. We will use a tool called <span class="co">[</span><span class="ot">`kreport2mpa.py`</span><span class="co">](https://github.com/jenniferlu717/KrakenTools?tab=readme-ov-file#kreport2mpapy)</span>, which takes a Kraken report file and prints out a MPA (MetaPhlAn)-style TEXT file. </span>
<span id="cb29-210"><a href="#cb29-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-213"><a href="#cb29-213" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-214"><a href="#cb29-214" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krakentools</span>
<span id="cb29-215"><a href="#cb29-215" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb29-216"><a href="#cb29-216" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb29-217"><a href="#cb29-217" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kreport2mpa.py</span> <span class="at">-r</span> <span class="va">$sample</span>-kraken-report.txt <span class="at">-o</span> <span class="va">$sample</span>-kraken-mpa.txt    </span>
<span id="cb29-218"><a href="#cb29-218" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-219"><a href="#cb29-219" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span>
<span id="cb29-220"><a href="#cb29-220" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-221"><a href="#cb29-221" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-222"><a href="#cb29-222" aria-hidden="true" tabindex="-1"></a>Easy as that. Now we can import all MPA files into their respective contig databases. Here we had to split the sample list by co-assembly because I could not figure out an easier way. </span>
<span id="cb29-223"><a href="#cb29-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-226"><a href="#cb29-226" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-227"><a href="#cb29-227" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> EP_list.txt<span class="kw">`</span></span>
<span id="cb29-228"><a href="#cb29-228" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb29-229"><a href="#cb29-229" aria-hidden="true" tabindex="-1"></a>    <span class="ex">anvi-import-taxonomy-for-layers</span> <span class="dt">\</span></span>
<span id="cb29-230"><a href="#cb29-230" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-p</span> 05_ANVIO_PROFILE/EP/<span class="va">$sample</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb29-231"><a href="#cb29-231" aria-hidden="true" tabindex="-1"></a>                   <span class="at">--parse</span> krakenuniq <span class="dt">\</span></span>
<span id="cb29-232"><a href="#cb29-232" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-i</span> 07_TAXONOMY/KRAKEN_TAXONOMY/<span class="va">$sample</span>-kraken-mpa.txt</span>
<span id="cb29-233"><a href="#cb29-233" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-234"><a href="#cb29-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-235"><a href="#cb29-235" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> WA_list.txt<span class="kw">`</span></span>
<span id="cb29-236"><a href="#cb29-236" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span> </span>
<span id="cb29-237"><a href="#cb29-237" aria-hidden="true" tabindex="-1"></a>    <span class="ex">anvi-import-taxonomy-for-layers</span> <span class="dt">\</span></span>
<span id="cb29-238"><a href="#cb29-238" aria-hidden="true" tabindex="-1"></a>                   <span class="at">-p</span> 05_ANVIO_PROFILE/WA/<span class="va">$sample</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb29-239"><a href="#cb29-239" aria-hidden="true" tabindex="-1"></a>                   <span class="at">--parse</span> krakenuniq <span class="at">-i</span> <span class="dt">\</span></span>
<span id="cb29-240"><a href="#cb29-240" aria-hidden="true" tabindex="-1"></a>                   07_TAXONOMY/KRAKEN_TAXONOMY/<span class="va">$sample</span>-kraken-mpa.txt</span>
<span id="cb29-241"><a href="#cb29-241" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-242"><a href="#cb29-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-243"><a href="#cb29-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-244"><a href="#cb29-244" aria-hidden="true" tabindex="-1"></a>Rad. With this done we can re-merge the profile databases. </span>
<span id="cb29-245"><a href="#cb29-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-248"><a href="#cb29-248" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-249"><a href="#cb29-249" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-merge</span> 05_ANVIO_PROFILE/WA/<span class="pp">*</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb29-250"><a href="#cb29-250" aria-hidden="true" tabindex="-1"></a>          <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb29-251"><a href="#cb29-251" aria-hidden="true" tabindex="-1"></a>          <span class="at">-o</span> 06_MERGED/WA</span>
<span id="cb29-252"><a href="#cb29-252" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-merge</span> 05_ANVIO_PROFILE/EP/<span class="pp">*</span>/PROFILE.db <span class="dt">\</span></span>
<span id="cb29-253"><a href="#cb29-253" aria-hidden="true" tabindex="-1"></a>          <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb29-254"><a href="#cb29-254" aria-hidden="true" tabindex="-1"></a>          <span class="at">-o</span> 06_MERGED/EP</span>
<span id="cb29-255"><a href="#cb29-255" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-256"><a href="#cb29-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-257"><a href="#cb29-257" aria-hidden="true" tabindex="-1"></a>Alright, time to make some Krona plots. This is a two-steo process. First, we use two scripts from  <span class="co">[</span><span class="ot">metaWRAP</span><span class="co">](https://github.com/bxlab/metaWRAP)</span>. The first, called <span class="in">`kraken2_translate.py`</span>, is used to generate full taxonomic lineages from a taxid. The input file for this is the output of the Kraken2 annotation. The next script is called <span class="in">`kraken_to_krona.py`</span> which takes the output of the first script (translated Kraken file) and parses it into a format that Krona can use to produce plots.</span>
<span id="cb29-258"><a href="#cb29-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-261"><a href="#cb29-261" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-262"><a href="#cb29-262" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate metawrap</span>
<span id="cb29-263"><a href="#cb29-263" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb29-264"><a href="#cb29-264" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb29-265"><a href="#cb29-265" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken2_translate.py</span> kraken2_db <span class="va">$sample</span>-kraken.out <span class="va">$sample</span>-kraken.trans</span>
<span id="cb29-266"><a href="#cb29-266" aria-hidden="true" tabindex="-1"></a>    <span class="ex">kraken_to_krona.py</span> <span class="va">$sample</span>-kraken.trans <span class="op">&gt;</span> <span class="va">$KRAKEN</span>/<span class="va">$sample</span>-kraken.krona</span>
<span id="cb29-267"><a href="#cb29-267" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-268"><a href="#cb29-268" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span>
<span id="cb29-269"><a href="#cb29-269" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-270"><a href="#cb29-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-271"><a href="#cb29-271" aria-hidden="true" tabindex="-1"></a>Once that is complete, we can use the output files and a script called <span class="in">`ktImportText`</span> from the Krona package to produce HTML Krona plots for each sample. </span>
<span id="cb29-272"><a href="#cb29-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-275"><a href="#cb29-275" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-276"><a href="#cb29-276" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krona</span>
<span id="cb29-277"><a href="#cb29-277" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> sample <span class="kw">in</span> <span class="kw">`</span><span class="fu">cat</span> sampleskraken2.txt<span class="kw">`</span></span>
<span id="cb29-278"><a href="#cb29-278" aria-hidden="true" tabindex="-1"></a><span class="cf">do</span></span>
<span id="cb29-279"><a href="#cb29-279" aria-hidden="true" tabindex="-1"></a>    <span class="ex">ktImportText</span> <span class="va">$sample</span>-kraken.krona <span class="at">-o</span> <span class="va">$sample</span>-kraken.krona.html</span>
<span id="cb29-280"><a href="#cb29-280" aria-hidden="true" tabindex="-1"></a><span class="cf">done</span></span>
<span id="cb29-281"><a href="#cb29-281" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span>
<span id="cb29-282"><a href="#cb29-282" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-283"><a href="#cb29-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-284"><a href="#cb29-284" aria-hidden="true" tabindex="-1"></a>A plot for every sample. How great is that?</span>
<span id="cb29-285"><a href="#cb29-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-286"><a href="#cb29-286" aria-hidden="true" tabindex="-1"></a><span class="fu">### VirSorter Annotation</span></span>
<span id="cb29-287"><a href="#cb29-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-288"><a href="#cb29-288" aria-hidden="true" tabindex="-1"></a>To classify any viral sequences, we ran <span class="co">[</span><span class="ot">VirSorter2</span><span class="co">](https://github.com/jiarong/VirSorter2)</span> <span class="co">[</span><span class="ot">@guo2021virsorter2</span><span class="co">]</span> on contigs from the co-assembly using our newly created <span class="in">`contig.db`</span>. First, we need something for VirSorter2 to classify. For that we export fasta files from each anvi'o co-assembly. </span>
<span id="cb29-289"><a href="#cb29-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-292"><a href="#cb29-292" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-293"><a href="#cb29-293" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-contigs</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb29-294"><a href="#cb29-294" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> 03_CONTIGS/WA-splits.fa <span class="dt">\ </span></span>
<span id="cb29-295"><a href="#cb29-295" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">--splits-mode</span> <span class="at">--no-wrap</span></span>
<span id="cb29-296"><a href="#cb29-296" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-contigs</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb29-297"><a href="#cb29-297" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> 03_CONTIGS/EP-splits.fa <span class="dt">\</span></span>
<span id="cb29-298"><a href="#cb29-298" aria-hidden="true" tabindex="-1"></a>                    <span class="at">--splits-mode</span> <span class="at">--no-wrap</span></span>
<span id="cb29-299"><a href="#cb29-299" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-300"><a href="#cb29-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-301"><a href="#cb29-301" aria-hidden="true" tabindex="-1"></a>And the code we used to run VirSorter2.</span>
<span id="cb29-302"><a href="#cb29-302" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-305"><a href="#cb29-305" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-306"><a href="#cb29-306" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate virsorter2</span>
<span id="cb29-307"><a href="#cb29-307" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">--seqfile</span> 03_CONTIGS/WA-splits.fa <span class="dt">\</span></span>
<span id="cb29-308"><a href="#cb29-308" aria-hidden="true" tabindex="-1"></a>              <span class="at">--working-dir</span> WA/ <span class="dt">\</span></span>
<span id="cb29-309"><a href="#cb29-309" aria-hidden="true" tabindex="-1"></a>              <span class="at">--keep-original-seq</span> <span class="dt">\</span></span>
<span id="cb29-310"><a href="#cb29-310" aria-hidden="true" tabindex="-1"></a>              <span class="at">--prep-for-dramv</span> <span class="dt">\</span></span>
<span id="cb29-311"><a href="#cb29-311" aria-hidden="true" tabindex="-1"></a>              <span class="at">--hallmark-required-on-short</span> <span class="dt">\</span></span>
<span id="cb29-312"><a href="#cb29-312" aria-hidden="true" tabindex="-1"></a>              <span class="at">--db-dir</span> virsorter2_db </span>
<span id="cb29-313"><a href="#cb29-313" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter</span> run <span class="at">--seqfile</span> 03_CONTIGS/EP-splits.fa <span class="dt">\</span></span>
<span id="cb29-314"><a href="#cb29-314" aria-hidden="true" tabindex="-1"></a>              <span class="at">--working-dir</span> EP/ <span class="dt">\</span></span>
<span id="cb29-315"><a href="#cb29-315" aria-hidden="true" tabindex="-1"></a>              <span class="at">--keep-original-seq</span> <span class="dt">\</span></span>
<span id="cb29-316"><a href="#cb29-316" aria-hidden="true" tabindex="-1"></a>              <span class="at">--prep-for-dramv</span> <span class="dt">\</span></span>
<span id="cb29-317"><a href="#cb29-317" aria-hidden="true" tabindex="-1"></a>              <span class="at">--hallmark-required-on-short</span> <span class="dt">\</span></span>
<span id="cb29-318"><a href="#cb29-318" aria-hidden="true" tabindex="-1"></a>              <span class="at">--db-dir</span> virsorter2_db</span>
<span id="cb29-319"><a href="#cb29-319" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span>
<span id="cb29-320"><a href="#cb29-320" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-321"><a href="#cb29-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-322"><a href="#cb29-322" aria-hidden="true" tabindex="-1"></a>Now, to get Virsorter2 annotations into the anvi'o contig databases there are a few special steps that need to be taken. Please <span class="co">[</span><span class="ot">see this post</span><span class="co">](https://github.com/simroux/VirSorter2_to_Anvio)</span> and associated <span class="in">`virsorter_to_anvio.py`</span> script for more details. Here we will only include the code with minimal explanation. Too long...</span>
<span id="cb29-323"><a href="#cb29-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-324"><a href="#cb29-324" aria-hidden="true" tabindex="-1"></a>First we export two tables that the <span class="in">`virsorter_to_anvio.py`</span> script needs for import. </span>
<span id="cb29-325"><a href="#cb29-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-328"><a href="#cb29-328" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-329"><a href="#cb29-329" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/WA-contigs.db  <span class="at">--table</span> splits_basic_info <span class="dt">\</span></span>
<span id="cb29-330"><a href="#cb29-330" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--output-file</span> WA_splits_basic_info.txt</span>
<span id="cb29-331"><a href="#cb29-331" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-table</span> 03_CONTIGS/EP-contigs.db  <span class="at">--table</span> splits_basic_info <span class="dt">\</span></span>
<span id="cb29-332"><a href="#cb29-332" aria-hidden="true" tabindex="-1"></a>                  <span class="at">--output-file</span> EP_splits_basic_info.txt</span>
<span id="cb29-333"><a href="#cb29-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-334"><a href="#cb29-334" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb29-335"><a href="#cb29-335" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> WA_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb29-336"><a href="#cb29-336" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--gene-caller</span> prodigal</span>
<span id="cb29-337"><a href="#cb29-337" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-export-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb29-338"><a href="#cb29-338" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--output-file</span> EP_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb29-339"><a href="#cb29-339" aria-hidden="true" tabindex="-1"></a>                       <span class="at">--gene-caller</span> prodigal</span>
<span id="cb29-340"><a href="#cb29-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-341"><a href="#cb29-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-342"><a href="#cb29-342" aria-hidden="true" tabindex="-1"></a>Time to get messy. At the time of this writing,  the <span class="in">`gene_calls`</span> file exported from anvi'o is a 10-column tab-delimited text file. The <span class="in">`virsorter_to_anvio.py`</span> script needs only 8 of these, and they need to be in a specific order. No problem, we can use <span class="co">[</span><span class="ot">`awk`</span><span class="co">](https://www.gnu.org/software/gawk/manual/gawk.html)</span>. </span>
<span id="cb29-343"><a href="#cb29-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-344"><a href="#cb29-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-345"><a href="#cb29-345" aria-hidden="true" tabindex="-1"></a>::: {.callout-note title="irSorter2 column values"}</span>
<span id="cb29-346"><a href="#cb29-346" aria-hidden="true" tabindex="-1"></a>These are the column values needed by the <span class="in">`virsorter_to_anvio.py`</span> script:<span class="in">`gene_callers_id`</span>, <span class="in">`contig`</span>, <span class="in">`start`</span>, <span class="in">`stop`</span>, <span class="in">`direction`</span>, <span class="in">`partial`</span>, <span class="in">`source`</span>, <span class="in">`version`</span></span>
<span id="cb29-347"><a href="#cb29-347" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb29-348"><a href="#cb29-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-351"><a href="#cb29-351" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-352"><a href="#cb29-352" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'BEGIN {FS="\t"; OFS="\t"} {print $1, $2, $3, $4, $5, $6, $8, $9}'</span> WA_all_gene_calls.txt</span>
<span id="cb29-353"><a href="#cb29-353" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'BEGIN {FS="\t"; OFS="\t"} {print $1, $2, $3, $4, $5, $6, $8, $9}'</span> EP_all_gene_calls.txt</span>
<span id="cb29-354"><a href="#cb29-354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-355"><a href="#cb29-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-356"><a href="#cb29-356" aria-hidden="true" tabindex="-1"></a>After this we run the parsing script. </span>
<span id="cb29-357"><a href="#cb29-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-360"><a href="#cb29-360" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-361"><a href="#cb29-361" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter_to_anvio.py</span> <span class="at">-i</span> WA/ <span class="at">-s</span> WA/WA_splits_basic_info.txt <span class="dt">\</span></span>
<span id="cb29-362"><a href="#cb29-362" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-n</span> WA/WA_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb29-363"><a href="#cb29-363" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-d</span> virsorter2_db <span class="dt">\</span></span>
<span id="cb29-364"><a href="#cb29-364" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-A</span> WA/WA_virsorter_additional_info.txt <span class="dt">\</span></span>
<span id="cb29-365"><a href="#cb29-365" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-C</span> WA/WA_virsorter_collection.txt <span class="dt">\</span></span>
<span id="cb29-366"><a href="#cb29-366" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-F</span> WA/WA_virsorter_annotations.txt</span>
<span id="cb29-367"><a href="#cb29-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-368"><a href="#cb29-368" aria-hidden="true" tabindex="-1"></a><span class="ex">virsorter_to_anvio.py</span> <span class="at">-i</span> EP/ <span class="at">-s</span> EP/EP_splits_basic_info.txt <span class="dt">\</span></span>
<span id="cb29-369"><a href="#cb29-369" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-n</span> EP/EP_all_gene_calls.txt <span class="dt">\</span></span>
<span id="cb29-370"><a href="#cb29-370" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-d</span> virsorter2_db <span class="dt">\</span></span>
<span id="cb29-371"><a href="#cb29-371" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-A</span> EP/EP_virsorter_additional_info.txt <span class="dt">\</span></span>
<span id="cb29-372"><a href="#cb29-372" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-C</span> EP/EP_virsorter_collection.txt <span class="dt">\</span></span>
<span id="cb29-373"><a href="#cb29-373" aria-hidden="true" tabindex="-1"></a>                      <span class="at">-F</span> EP/EP_virsorter_annotations.txt</span>
<span id="cb29-374"><a href="#cb29-374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-375"><a href="#cb29-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-376"><a href="#cb29-376" aria-hidden="true" tabindex="-1"></a>And import the resulting files to anvi'o</span>
<span id="cb29-377"><a href="#cb29-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-380"><a href="#cb29-380" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-381"><a href="#cb29-381" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-misc-data</span> WA-virsorter_additional_info.txt </span>
<span id="cb29-382"><a href="#cb29-382" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/WA/PROFILE.db  </span>
<span id="cb29-383"><a href="#cb29-383" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">--target-data-table</span> items</span>
<span id="cb29-384"><a href="#cb29-384" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-misc-data</span> EP-virsorter_additional_info.txt </span>
<span id="cb29-385"><a href="#cb29-385" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/EP/PROFILE.db  </span>
<span id="cb29-386"><a href="#cb29-386" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">--target-data-table</span> items</span>
<span id="cb29-387"><a href="#cb29-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-388"><a href="#cb29-388" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-collection</span> WA-virsorter_collection.txt </span>
<span id="cb29-389"><a href="#cb29-389" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-c</span> 03_CONTIGS/WA-contigs.db </span>
<span id="cb29-390"><a href="#cb29-390" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/WA/PROFILE.db </span>
<span id="cb29-391"><a href="#cb29-391" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-C</span> VIRSORTER2</span>
<span id="cb29-392"><a href="#cb29-392" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-collection</span> EP-virsorter_collection.txt </span>
<span id="cb29-393"><a href="#cb29-393" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-c</span> 03_CONTIGS/EP-contigs.db </span>
<span id="cb29-394"><a href="#cb29-394" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-p</span> 06_MERGED/EP/PROFILE.db</span>
<span id="cb29-395"><a href="#cb29-395" aria-hidden="true" tabindex="-1"></a>                      <span class="ex">-C</span> VIRSORTER2</span>
<span id="cb29-396"><a href="#cb29-396" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-397"><a href="#cb29-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-398"><a href="#cb29-398" aria-hidden="true" tabindex="-1"></a>VirSorter2 annotation complete. </span>
<span id="cb29-399"><a href="#cb29-399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-400"><a href="#cb29-400" aria-hidden="true" tabindex="-1"></a><span class="fu">## Kaiju Annotation</span></span>
<span id="cb29-401"><a href="#cb29-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-402"><a href="#cb29-402" aria-hidden="true" tabindex="-1"></a>Here we use <span class="co">[</span><span class="ot">Kaiju</span><span class="co">](https://github.com/bioinformatics-centre/kaiju)</span> <span class="co">[</span><span class="ot">@menzel2016kaiju</span><span class="co">]</span> to classify gene calls. We do this against the <span class="co">[</span><span class="ot">progenomes databases</span><span class="co">](https://github.com/bioinformatics-centre/kaiju?tab=readme-ov-file#creating-the-kaiju-index)</span>, a r epresentative set of genomes from the <span class="co">[</span><span class="ot">proGenomes</span><span class="co">](http://progenomes.embl.de/)</span> database and viruses from the NCBI RefSeq database. We describe the contruction of this database <span class="co">[</span><span class="ot">here</span><span class="co">](../setup/index.html#kaiju)</span>. </span>
<span id="cb29-403"><a href="#cb29-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-404"><a href="#cb29-404" aria-hidden="true" tabindex="-1"></a>Start by grabbing gene call fasta files. </span>
<span id="cb29-405"><a href="#cb29-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-408"><a href="#cb29-408" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-409"><a href="#cb29-409" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="at">-o</span> EP_gene_calls.fna</span>
<span id="cb29-410"><a href="#cb29-410" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-get-sequences-for-gene-calls</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="at">-o</span> WA_gene_calls.fna</span>
<span id="cb29-411"><a href="#cb29-411" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-412"><a href="#cb29-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-413"><a href="#cb29-413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-416"><a href="#cb29-416" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-417"><a href="#cb29-417" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb29-418"><a href="#cb29-418" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-419"><a href="#cb29-419" aria-hidden="true" tabindex="-1"></a>      <span class="at">-f</span> kaiju_db/kaiju_db_progenomes.fmi <span class="dt">\</span></span>
<span id="cb29-420"><a href="#cb29-420" aria-hidden="true" tabindex="-1"></a>      <span class="at">-i</span> EP_gene_calls.fna <span class="dt">\ </span></span>
<span id="cb29-421"><a href="#cb29-421" aria-hidden="true" tabindex="-1"></a>      <span class="ex">-o</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-422"><a href="#cb29-422" aria-hidden="true" tabindex="-1"></a>      <span class="at">-z</span> <span class="va">$NSLOTS</span> <span class="at">-v</span></span>
<span id="cb29-423"><a href="#cb29-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-424"><a href="#cb29-424" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-425"><a href="#cb29-425" aria-hidden="true" tabindex="-1"></a>      <span class="at">-f</span> kaiju_db/kaiju_db_progenomes.fmi <span class="dt">\</span></span>
<span id="cb29-426"><a href="#cb29-426" aria-hidden="true" tabindex="-1"></a>      <span class="at">-i</span> WA_gene_calls.fna <span class="dt">\</span></span>
<span id="cb29-427"><a href="#cb29-427" aria-hidden="true" tabindex="-1"></a>      <span class="at">-o</span> WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-428"><a href="#cb29-428" aria-hidden="true" tabindex="-1"></a>      <span class="at">-z</span> <span class="va">$NSLOTS</span> <span class="at">-v</span></span>
<span id="cb29-429"><a href="#cb29-429" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-430"><a href="#cb29-430" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-addTaxonNames</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-431"><a href="#cb29-431" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-432"><a href="#cb29-432" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-i</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-433"><a href="#cb29-433" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> EP_kaiju_nr.names <span class="dt">\</span></span>
<span id="cb29-434"><a href="#cb29-434" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-r</span> superkingdom,phylum,order,class,family,genus,species</span>
<span id="cb29-435"><a href="#cb29-435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-436"><a href="#cb29-436" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju-addTaxonNames</span> <span class="at">-t</span> kaiju_db/nodes.dmp </span>
<span id="cb29-437"><a href="#cb29-437" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-438"><a href="#cb29-438" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-i</span> WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-439"><a href="#cb29-439" aria-hidden="true" tabindex="-1"></a>                    <span class="at">-o</span> WA_kaiju_nr.names <span class="dt">\ </span></span>
<span id="cb29-440"><a href="#cb29-440" aria-hidden="true" tabindex="-1"></a>                    <span class="ex">-r</span> superkingdom,phylum,order,class,family,genus,species</span>
<span id="cb29-441"><a href="#cb29-441" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span>
<span id="cb29-442"><a href="#cb29-442" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-443"><a href="#cb29-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-444"><a href="#cb29-444" aria-hidden="true" tabindex="-1"></a>Import the output to anvi'o.</span>
<span id="cb29-445"><a href="#cb29-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-448"><a href="#cb29-448" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-449"><a href="#cb29-449" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-taxonomy-for-genes</span> <span class="at">-c</span> 03_CONTIGS/EP-contigs.db <span class="dt">\</span></span>
<span id="cb29-450"><a href="#cb29-450" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-p</span> kaiju <span class="dt">\</span></span>
<span id="cb29-451"><a href="#cb29-451" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-i</span> EP_kaiju_nr.names </span>
<span id="cb29-452"><a href="#cb29-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-453"><a href="#cb29-453" aria-hidden="true" tabindex="-1"></a><span class="ex">anvi-import-taxonomy-for-genes</span> <span class="at">-c</span> 03_CONTIGS/WA-contigs.db <span class="dt">\</span></span>
<span id="cb29-454"><a href="#cb29-454" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-p</span> kaiju <span class="dt">\</span></span>
<span id="cb29-455"><a href="#cb29-455" aria-hidden="true" tabindex="-1"></a>                               <span class="at">-i</span> WA_kaiju_nr.names</span>
<span id="cb29-456"><a href="#cb29-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-457"><a href="#cb29-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-458"><a href="#cb29-458" aria-hidden="true" tabindex="-1"></a>And generate Krona plots of the data. A little dance between the Kaiju and Krona environments. </span>
<span id="cb29-459"><a href="#cb29-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-462"><a href="#cb29-462" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-463"><a href="#cb29-463" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb29-464"><a href="#cb29-464" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2krona</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-465"><a href="#cb29-465" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-466"><a href="#cb29-466" aria-hidden="true" tabindex="-1"></a>            <span class="at">-i</span> <span class="va">$KAIJU</span>/WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-467"><a href="#cb29-467" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> WA_kaiju_nr.out.krona</span>
<span id="cb29-468"><a href="#cb29-468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-469"><a href="#cb29-469" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2krona</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-470"><a href="#cb29-470" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-471"><a href="#cb29-471" aria-hidden="true" tabindex="-1"></a>            <span class="at">-i</span> EP_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-472"><a href="#cb29-472" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> EP_kaiju_nr.out.krona</span>
<span id="cb29-473"><a href="#cb29-473" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span>
<span id="cb29-474"><a href="#cb29-474" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-475"><a href="#cb29-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-478"><a href="#cb29-478" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-479"><a href="#cb29-479" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate krona</span>
<span id="cb29-480"><a href="#cb29-480" aria-hidden="true" tabindex="-1"></a><span class="ex">ktImportText</span> <span class="at">-o</span> WA_kaiju_nr.out.html WA_kaiju_nr.out.krona</span>
<span id="cb29-481"><a href="#cb29-481" aria-hidden="true" tabindex="-1"></a><span class="ex">ktImportText</span> <span class="at">-o</span> EP_kaiju_nr.out.html EP_kaiju_nr.out.krona</span>
<span id="cb29-482"><a href="#cb29-482" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate </span>
<span id="cb29-483"><a href="#cb29-483" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-484"><a href="#cb29-484" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-487"><a href="#cb29-487" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb29-488"><a href="#cb29-488" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> activate kaiju</span>
<span id="cb29-489"><a href="#cb29-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-490"><a href="#cb29-490" aria-hidden="true" tabindex="-1"></a><span class="ex">kaiju2table</span> <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-491"><a href="#cb29-491" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-492"><a href="#cb29-492" aria-hidden="true" tabindex="-1"></a>            <span class="at">-r</span> class <span class="dt">\</span></span>
<span id="cb29-493"><a href="#cb29-493" aria-hidden="true" tabindex="-1"></a>            <span class="at">-l</span> phylum,class,order,family <span class="dt">\</span></span>
<span id="cb29-494"><a href="#cb29-494" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> WA_kaiju_nr.out.summary WA_kaiju_nr.out <span class="dt">\</span></span>
<span id="cb29-495"><a href="#cb29-495" aria-hidden="true" tabindex="-1"></a>kaiju2table <span class="at">-t</span> kaiju_db/nodes.dmp <span class="dt">\</span></span>
<span id="cb29-496"><a href="#cb29-496" aria-hidden="true" tabindex="-1"></a>            <span class="at">-n</span> kaiju_db/names.dmp <span class="dt">\</span></span>
<span id="cb29-497"><a href="#cb29-497" aria-hidden="true" tabindex="-1"></a>            <span class="at">-r</span> class <span class="dt">\</span></span>
<span id="cb29-498"><a href="#cb29-498" aria-hidden="true" tabindex="-1"></a>            <span class="at">-l</span> phylum,class,order,family <span class="dt">\</span></span>
<span id="cb29-499"><a href="#cb29-499" aria-hidden="true" tabindex="-1"></a>            <span class="at">-o</span> EP_kaiju_nr.out.summary EP_kaiju_nr.out</span>
<span id="cb29-500"><a href="#cb29-500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-501"><a href="#cb29-501" aria-hidden="true" tabindex="-1"></a><span class="ex">conda</span> deactivate</span>
<span id="cb29-502"><a href="#cb29-502" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb29-503"><a href="#cb29-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-504"><a href="#cb29-504" aria-hidden="true" tabindex="-1"></a>Kaiju done.</span>
<span id="cb29-505"><a href="#cb29-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-506"><a href="#cb29-506" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_footer.qmd &gt;}}</span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../../../site_info.html">
<p>Site Information</p>
</a>
  </li>  
</ul>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/istmobiome/trans-shrimp/blob/main/workflows/mg/assembly/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/edit/main/workflows/mg/assembly/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/trans-shrimp/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>.</p>
</div>
  </div>
</footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>


</body></html>