<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.8.25">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<meta name="description" content="Workflow for running MED analysis. Workflow begins with redundant, aligned fasta file from mothur and ends with the MED analysis. A Microtable Object is produced to collate the data for downstream analysis.">
<title>5. MED Workflow – TransIsthmian Shrimp</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>

<script src="../../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../../">
<link href="../../../workflows/ssu/otu/index.html" rel="prev">
<link href="../../../assets/favicon.ico" rel="icon">
<script src="../../../site_libs/quarto-listing/list.min.js"></script>
<script src="../../../site_libs/quarto-listing/quarto-listing.js"></script>
<script src="../../../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../../../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../../../site_libs/quarto-html/axe/axe-check.js" type="module"></script>
<script src="../../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-dark-707d8167ce6003fca903bfe2be84ab7f.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="../../../site_libs/quarto-html/quarto-syntax-highlighting-7b89279ff1a6dce999919e0e67d4d9ec.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="../../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../../site_libs/bootstrap/bootstrap-dark-d7d32a8442357cea8cf74a4a848b80d6.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="../../../site_libs/bootstrap/bootstrap-e6bd33826d752a752b088f17960ea970.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light"><script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script><script>

  window.document.addEventListener("DOMContentLoaded", function (_event) {
    const listingTargetEl = window.document.querySelector('#listing-med-listing .list');
    if (!listingTargetEl) {
      // No listing discovered, do not attach.
      return; 
    }

    const options = {
      valueNames: ['listing-filename','listing-description',{ data: ['index'] },{ data: ['categories'] },{ data: ['listing-filename-sort'] }],
      
      searchColumns: ["listing-filename","listing-description"],
    };

    window['quarto-listings'] = window['quarto-listings'] || {};
    window['quarto-listings']['listing-med-listing'] = new List('listing-med-listing', options);

    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  });

  window.addEventListener('hashchange',() => {
    if (window['quarto-listing-loaded']) {
      window['quarto-listing-loaded']();
    }
  })
  </script><script src="../../../site_libs/quarto-diagram/mermaid.min.js"></script><script src="../../../site_libs/quarto-diagram/mermaid-init.js"></script><link href="../../../site_libs/quarto-diagram/mermaid.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script><script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script><script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN" && texText && texText.data) {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script><link rel="stylesheet" href="../../../assets/css/styles.css">
</head>
<body class="nav-sidebar floating nav-fixed quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="navbar navbar-expand-lg " data-bs-theme="dark"><div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a href="../../../index.html" class="navbar-brand navbar-brand-logo">
    <img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo light-content"><img src="../../../images/icon.png" alt="Istmobiome logo." class="navbar-logo dark-content"></a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
<li class="nav-item">
    <a class="nav-link active" href="../../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/portal/index.html"> 
<span class="menu-text">Data Portal</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/ssu/index.html"> 
<span class="menu-text">16S rRNA</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/mg/index.html"> 
<span class="menu-text">Metagenomics</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../../workflows/pub/index.html"> 
<span class="menu-text">Paper</span></a>
  </li>  
</ul>
<ul class="navbar-nav navbar-nav-scroll ms-auto">
<li class="nav-item compact">
    <a class="nav-link" href="https://github.com/istmobiome/shrimp-draft/"> <i class="bi bi-github" role="img" aria-label="Trans Shrimp">
</i> 
<span class="menu-text"></span></a>
  </li>  
</ul>
</div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/ssu/index.html">16S rRNA</a></li><li class="breadcrumb-item"><a href="../../../workflows/ssu/med/index.html">5. MED Workflow</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto"><div class="pt-lg-2 mt-2 text-left sidebar-header">
      <a href="../../../index.html" class="sidebar-logo-link">
      </a>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/portal/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Portal</span></a>
  </div>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/ssu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">16S rRNA</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/sampledata/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Sample Metadata</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/dada2/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. DADA2 ASV Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/otu/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">4. OTU Workflow</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/ssu/med/index.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">5. MED Workflow</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/mg/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metagenomics</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/setup/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">1. Build Databases</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/assembly/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">2. Co-Assembly &amp; Annotations</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/mg/aa-results/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">3. Assembly &amp; Annotation Results</span></a>
  </div>
</li>
      </ul>
</li>
        <li class="px-0"><hr class="sidebar-divider hi "></li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="../../../workflows/pub/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Publication Material</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/references/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workflow References</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../../../workflows/pub/archive_data/index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Submitting &amp; Archiving Data/Code</span></a>
  </div>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="1"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#workflow-input" id="toc-workflow-input" class="nav-link active" data-scroll-target="#workflow-input">Workflow Input</a></li>
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview">Overview</a></li>
  <li>
<a href="#read-processing" id="toc-read-processing" class="nav-link" data-scroll-target="#read-processing">Read Processing</a>
  <ul class="collapse">
<li><a href="#getting-started" id="toc-getting-started" class="nav-link" data-scroll-target="#getting-started">Getting Started</a></li>
  <li><a href="#remove-negative-controls" id="toc-remove-negative-controls" class="nav-link" data-scroll-target="#remove-negative-controls">Remove Negative Controls</a></li>
  <li><a href="#remove-chimeras" id="toc-remove-chimeras" class="nav-link" data-scroll-target="#remove-chimeras">Remove Chimeras</a></li>
  <li><a href="#repseq-taxonomy" id="toc-repseq-taxonomy" class="nav-link" data-scroll-target="#repseq-taxonomy">Repseq Taxonomy</a></li>
  <li><a href="#remove-contaminants" id="toc-remove-contaminants" class="nav-link" data-scroll-target="#remove-contaminants">Remove Contaminants</a></li>
  <li><a href="#track-reads-through-workflow" id="toc-track-reads-through-workflow" class="nav-link" data-scroll-target="#track-reads-through-workflow">Track Reads through Workflow</a></li>
  <li><a href="#preparing-for-analysis" id="toc-preparing-for-analysis" class="nav-link" data-scroll-target="#preparing-for-analysis">Preparing for analysis</a></li>
  </ul>
</li>
  <li><a href="#med-analysis" id="toc-med-analysis" class="nav-link" data-scroll-target="#med-analysis">MED Analysis</a></li>
  <li>
<a href="#data-set-prep" id="toc-data-set-prep" class="nav-link" data-scroll-target="#data-set-prep">Data Set Prep</a>
  <ul class="collapse">
<li>
<a href="#assign-taxonomy" id="toc-assign-taxonomy" class="nav-link" data-scroll-target="#assign-taxonomy">Assign Taxonomy</a>
  <ul class="collapse">
<li><a href="#a.-taxonomy-table" id="toc-a.-taxonomy-table" class="nav-link" data-scroll-target="#a.-taxonomy-table">A. Taxonomy Table</a></li>
  <li><a href="#b.-sequence-table" id="toc-b.-sequence-table" class="nav-link" data-scroll-target="#b.-sequence-table">B. Sequence Table</a></li>
  <li><a href="#c.-sample-table" id="toc-c.-sample-table" class="nav-link" data-scroll-target="#c.-sample-table">C. Sample Table</a></li>
  </ul>
</li>
  <li>
<a href="#experiment-level-objects" id="toc-experiment-level-objects" class="nav-link" data-scroll-target="#experiment-level-objects">Experiment-level Objects</a>
  <ul class="collapse">
<li><a href="#add-representative-sequence" id="toc-add-representative-sequence" class="nav-link" data-scroll-target="#add-representative-sequence">Add Representative Sequence</a></li>
  </ul>
</li>
  <li>
<a href="#curate-the-data-set" id="toc-curate-the-data-set" class="nav-link" data-scroll-target="#curate-the-data-set">Curate the Data Set</a>
  <ul class="collapse">
<li><a href="#remove-low-count-samples" id="toc-remove-low-count-samples" class="nav-link" data-scroll-target="#remove-low-count-samples">Remove Low-Count Samples</a></li>
  </ul>
</li>
  <li><a href="#summary" id="toc-summary" class="nav-link" data-scroll-target="#summary">Summary</a></li>
  </ul>
</li>
  <li><a href="#download-results" id="toc-download-results" class="nav-link" data-scroll-target="#download-results">Download Results</a></li>
  </ul><div class="toc-actions"><ul><li><a href="https://github.com/istmobiome/shrimp-draft/blob/main/workflows/ssu/med/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/edit/main/workflows/ssu/med/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../../../workflows/ssu/index.html">16S rRNA</a></li><li class="breadcrumb-item"><a href="../../../workflows/ssu/med/index.html">5. MED Workflow</a></li></ol></nav><div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">5. MED Workflow</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i></button></div></div>
</div>

<div>
  <div class="description">
    <p>Workflow for running MED analysis. Workflow begins with redundant, aligned fasta file from mothur and ends with the MED analysis. A Microtable Object is produced to collate the data for downstream analysis.</p>
  </div>
</div>


<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="workflow-input" class="level2"><h2 class="anchored" data-anchor-id="workflow-input">Workflow Input</h2>
<div class="callout callout-style-default callout-note no-icon callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Data &amp; Scripts
</div>
</div>
<div class="callout-body-container callout-body">
<p>Fastq sequence files, scripts, and other assets for running this workflow can be found on the <a href="../../../workflows/portal/data-med.html">MED Data Portal</a> page.</p>
<p>The Data Portal page also contains a link to the curated output of this pipeline–-feature-sample count table, taxonomy table, sample metadata, &amp; representative fasta sequences. Data is available as stand-alone text files, OR bundled in microtable and phyloseq objects. Archive also includes a table tracking sample read changes.</p>
</div>
</div>
<section id="required-packages-software" class="level4"><h4 class="anchored" data-anchor-id="required-packages-software">Required Packages &amp; Software</h4>
<p>To run this workflow you need to <a href="https://mothur.org/wiki/installation/">install mothur</a> and the <a href="https://merenlab.org/2014/08/16/installing-the-oligotyping-pipeline/">Oligotyping/MED</a>. You will need the <a href="https://www.tidyverse.org">tidyverse</a> package.</p>
</section></section><section id="overview" class="level2"><h2 class="anchored" data-anchor-id="overview">Overview</h2>
<p>With the mothur pipeline finished, we can turn our attention to <a href="https://merenlab.org/2014/11/04/med/">Minimum Entropy Decomposition (MED)</a> <span class="citation" data-cites="eren2015minimum">(<a href="#ref-eren2015minimum" role="doc-biblioref">Eren et al. 2015</a>)</span>. MED is a novel, information theory-based clustering algorithm for sensitive partitioning of high-throughput marker gene sequences.</p>
<p>(From the Meren Lab website) MED:</p>
<ul>
<li>Does not perform pairwise sequence comparison,<br>
</li>
<li>Does not rely on arbitrary sequence similarity thresholds,<br>
</li>
<li>Does not require user supervision,<br>
</li>
<li>Does not require preliminary classification or clustering results,<br>
</li>
<li>Is agnostic to sampling strategy or how long your sequences are,<br>
</li>
<li>Gives you 1 nucleotide resolution over any sequencing length with computational efficiency and minimal computational heuristics.</li>
</ul>
<p>MED needs a redundant alignment file of read data. This means <strong>all</strong> identical sequences need to be included. We again use mothur but this pipeline starts with the currated output of the <a href="../../../workflows/ssu/otu/#aligning-reads"><code>align.seqs</code></a> portion of our mothur OTU pipeline.</p>
<p>We set up our run in the same way as the mothur pipeline.</p>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
  A(Start with curated&lt;/br&gt;mothur alignment)
  B("&lt;b&gt;&lt;a href='https://mothur.org/wiki/get.groups/'&gt;get.groups&lt;/a&gt;&lt;/b&gt;&lt;br/&gt;(NC Samples)")
  A --&gt; C(Remove&lt;/br&gt;NC Samples&lt;br/&gt;files needed by mothur &lt;/br&gt;were generated in R)
  B --&gt; C
  C --&gt; D(&lt;b&gt;&lt;a href='https://mothur.org/wiki/remove.seqs/'&gt;remove.seqs&lt;/a&gt;&lt;/b&gt;)
  C --&gt; E(&lt;b&gt;&lt;a href='https://mothur.org/wiki/remove.groups/'&gt;remove.groups&lt;/a&gt;&lt;/b&gt;)
  E --&gt; H
  D --&gt; H(&lt;b&gt;&lt;a href='https://mothur.org/wiki/chimera.vsearch/'&gt;chimera.vsearch&lt;/a&gt;&lt;/b&gt;)
  H --&gt; END:::hidden
</pre>
</div>
<p></p>
</div>
</div>
</div>
<div class="cell" data-eval="true" data-layout-align="center">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
  BEGIN:::hidden --&gt; I
  I(&lt;b&gt;&lt;a href='https://mothur.org/wiki/classify.seqs/'&gt;classify.seqs&lt;/a&gt;&lt;/b&gt;)
  I --&gt; J(&lt;b&gt;&lt;a href='https://mothur.org/wiki/remove.lineage/'&gt;remove.lineage&lt;/a&gt;&lt;/b&gt;) 
  J --&gt; K(to &lt;b&gt;&lt;a href='https://merenlab.org/2014/11/04/med/'&gt;MED pipeline&lt;/a&gt;&lt;/b&gt;)

  K --&gt; END:::hidden 
</pre>
</div>
<p></p>
</div>
</div>
</div>
</section><section id="read-processing" class="level1"><h1>Read Processing</h1>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see environment variables
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<pre><code>$ export DATA=01_TRIMMED_DATA/
$ export TYPE=fastq
$ export PROC=30

$ export REF_LOC=reference_dbs
$ export TAXREF_FASTA=gsrdb.fasta
$ export TAXREF_TAX=gsrdb.tax
$ export ALIGNREF=silva.v4.fasta

$ export CONTAMINENTS=Chloroplast-Mitochondria-unknown-Eukaryota</code></pre>
</div>
</div>
</div>
<section id="getting-started" class="level2"><h2 class="anchored" data-anchor-id="getting-started">Getting Started</h2>
<p>The first thing to do is copy the output of the <code>align.seqs</code> portion of the mothur workflow to a new working directory.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb2"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>set.dir(output=pipelineFiles_med/)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Mothur's directories:
outputDir=pipelineFiles_med/</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb4"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>system(cp pipelineFiles/shrimp.trim.contigs.good.unique.good.filter.unique.fasta pipelineFiles_med/)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>system(cp pipelineFiles/shrimp.trim.contigs.good.unique.good.filter.count_table pipelineFiles_med/)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</section><section id="remove-negative-controls" class="level2"><h2 class="anchored" data-anchor-id="remove-negative-controls">Remove Negative Controls</h2>
<p>As with the OTU workflow, we remove NC samples, but in this case we skip the <code>pre.cluster</code> step. Our goal here to remove all NC samples from the dataset, as well as any reads that are predominantly found in NC samples. More on what that means in a minute. Before we do anything we need to identify all reads present in NC samples.</p>
<p>Here is what we are going to do:</p>
<ol type="1">
<li>Subset the NC samples (and associated reads) from the <code>fasta</code> and <code>count.table</code>. To do this in mothur we need all of the NC sample names collected in an <code>.accnos</code> file, which is a text file used in mothur that contains a single column of names–these can be sequences, OTU labels, or sample names. This list is used to select or remove specific items from other mothur files. That way we can process subsets of the data without having to retype long lists of names.</li>
</ol>
<p>To generate the <code>.accnos</code> file of NC samples we can use the <code>shrimp.files</code> file generated at the beginning of the mothur pipeline.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_accnos</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/shrimp.files"</span><span class="op">)</span>, </span>
<span>                                delim <span class="op">=</span> <span class="st">"\t"</span>, col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">tmp_accnos</span><span class="op">[</span>, <span class="fl">2</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">tmp_accnos</span> <span class="op">&lt;-</span> <span class="va">tmp_accnos</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">grepl</a></span><span class="op">(</span><span class="st">"Control_"</span>, <span class="va">tmp_accnos</span><span class="op">$</span><span class="va">X1</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_delim</a></span><span class="op">(</span><span class="va">tmp_accnos</span>, file <span class="op">=</span> <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/nc_samples.accnos"</span><span class="op">)</span>, </span>
<span>                   col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<ol start="2" type="1">
<li>Now we have a list of all NC sample names. The mothur command <code>get.groups</code> in conjunction with <code>accnos</code> file allows us to subset the full <code>fasta</code> and <code>count_table</code>
</li>
</ol>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb6"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>get.groups(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.fasta, count=shrimp.trim.contigs.good.unique.good.filter.count_table, accnos=nc_samples.accnos)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Selected 192842 sequences from your count file.
Selected 34262 sequences from your fasta file.

Output File names:
shrimp.trim.contigs.good.unique.good.filter.pick.count_table
shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta</code></pre>
<ol start="3" type="1">
<li>Next we rename the new files to something more informative (and shorter).</li>
</ol>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb8"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>rename.file(input=shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, new=nc.fasta)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>rename.file(input=shrimp.trim.contigs.good.unique.good.filter.pick.count_table, new=nc.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<ol start="4" type="1">
<li>And a quick summary of the NC subset.</li>
</ol>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb9"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>summary.seqs(fasta=nc.fasta, count=nc.count_table, processors=$PROC)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see negative control summary
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Start</th>
<th style="text-align: center;">End</th>
<th style="text-align: center;">NBases</th>
<th style="text-align: center;">Ambigs</th>
<th style="text-align: center;">Polymer</th>
<th style="text-align: center;">NumSeqs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Minimum</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">248</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">2.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">4822</td>
</tr>
<tr class="odd">
<td style="text-align: left;">25%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">48211</td>
</tr>
<tr class="even">
<td style="text-align: left;">Median:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">96422</td>
</tr>
<tr class="odd">
<td style="text-align: left;">75%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">144632</td>
</tr>
<tr class="even">
<td style="text-align: left;">97.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">254</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">188021</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Maximum:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">254</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">192842</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
<pre><code># of unique seqs:   34262
total # of seqs:    192842

It took 0 secs to summarize 192842 sequences.

Output File Names:
nc.summary</code></pre>
</div>
</div>
</div>
<ol start="5" type="1">
<li>Sweet. We use the command <code>list.seqs</code> to get a complete list of all repseq names in the NC subset.</li>
</ol>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb11"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>list.seqs(count=nc.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Output File Names: 
nc.accnos</code></pre>
<p>This gives us all repseq IDs in the NC samples.</p>
<ol start="6" type="1">
<li>We could simply use the <code>nc.accnos</code> file from the <code>list.seqs</code> command to remove repseqs found in negative control (NC) samples from the main dataset. This seems reasonable enough. Except mothur will remove <strong>all</strong> repseqs found in a NC sample, in other words, any repseq in the accnos file. For example, let’s consider the following scenario where we have two repseqs:</li>
</ol>
<p><code>repseq01</code> is abundant in many NC samples but not found in any other samples.<br><code>repseq02</code> on the other hand is represented by say one read in a single NC sample but very abundant in other samples.</p>
<p>It makes sense to remove <code>repseq01</code> but not necessarily <code>repseq02</code>. Essentially, for each <code>repseq</code> in the <code>nc.accnos</code> file we want to calculate:</p>
<ul>
<li>The total number of reads in NC samples.<br>
</li>
<li>The total number of reads in non-NC samples.<br>
</li>
<li>The percent of reads in the NC samples.</li>
<li>The total number of NC samples containing at least 1 read.<br>
</li>
<li>The total number of non-NC samples containing at least 1 read.<br>
</li>
<li>The percent of NC samples containing reads.</li>
</ul>
<p>Where a final data table might look something like this</p>
<table class="caption-top table">
<colgroup>
<col style="width: 16%">
<col style="width: 9%">
<col style="width: 13%">
<col style="width: 12%">
<col style="width: 12%">
<col style="width: 16%">
<col style="width: 21%">
</colgroup>
<thead><tr class="header">
<th style="text-align: left;">repseq</th>
<th style="text-align: center;">rc_nc</th>
<th style="text-align: center;">rc_samps</th>
<th style="text-align: center;">%in_nc</th>
<th style="text-align: center;">nc_samp</th>
<th style="text-align: center;">no_nc_samp</th>
<th style="text-align: center;">%_in_nc_samp</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">repseq001</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">37.5</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">33.31</td>
</tr>
<tr class="even">
<td style="text-align: left;">repseq002</td>
<td style="text-align: center;">196</td>
<td style="text-align: center;">308</td>
<td style="text-align: center;">38.9</td>
<td style="text-align: center;">17</td>
<td style="text-align: center;">38</td>
<td style="text-align: center;">30.7</td>
</tr>
<tr class="odd">
<td style="text-align: left;">repseq003</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">23</td>
<td style="text-align: center;">11.1</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">18</td>
<td style="text-align: center;">14.5</td>
</tr>
</tbody>
</table>
<p>To accomplish this we will parse out relevant data from the <code>.count_table</code> files. We got the idea on how best to do this from a <a href="https://forum.mothur.org/t/negative-control/2754">discussion on the mothur forum</a>.</p>
<p>To save space and minimize file size, mothur formats the <a href="https://mothur.org/wiki/count_file/"><code>.count_table</code></a> using a sparse format by storing only non zero sample counts. However, we need the full format which lists each repseq and its abundance counts for each sample. Using the command <code>count.seqs</code> in conjunction with the <code>.count_table</code> will return a full format table.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb13"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>count.seqs(count=nc.count_table, compress=f)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Output File Names:
nc.full.count_table</code></pre>
<p>Then we use the accnos file (<code>nc.accnos</code>)–containing repseqs found in NC samples, to generate a subset of the complete dataset containing only the reseqs found in NC samples.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb15"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>get.seqs(accnos=nc.accnos, fasta=shrimp.trim.contigs.good.unique.good.filter.unique.fasta, count=shrimp.trim.contigs.good.unique.good.filter.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb16"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>rename.file(input=shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, new=subset.fasta)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>rename.file(input=shrimp.trim.contigs.good.unique.good.filter.pick.count_table, new=subset.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>And again run <code>count.seqs</code> to get a full format table of the complete dataset, which we subsetted earlier to contain only repseqs found in NC samples.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb17"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>count.seqs(count=subset.count_table, compress=f)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Output File Names:
subset.full.count_table</code></pre>
<p>Finally we can parse out read count data from the two <code>subset.full.count_table</code> files.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">full_count_tab</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/subset.full.count_table"</span><span class="op">)</span>, </span>
<span>                                    delim <span class="op">=</span> <span class="st">"\t"</span>, col_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co"># figure out which columns to use</span></span>
<span><span class="va">control_cols</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"^Control_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">full_count_tab</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">noncontrol_cols</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">full_count_tab</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>, <span class="va">control_cols</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># now do the rowwise sums</span></span>
<span><span class="va">read_totals</span> <span class="op">&lt;-</span> <span class="va">full_count_tab</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    total_reads_nc   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">c_across</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">control_cols</span><span class="op">)</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    total_reads_non_nc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">c_across</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">noncontrol_cols</span><span class="op">)</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="va">total_reads_nc</span>, <span class="va">total_reads_non_nc</span><span class="op">)</span></span>
<span></span>
<span><span class="va">read_totals</span> <span class="op">&lt;-</span> <span class="va">read_totals</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"total_reads"</span> <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And here is what the new dataframe looks like. Three columns where the first is the repseq name, the second the total number of reads in NC samples, and the third the total number of reads in the entire dataset (this includes NC samples).</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
-------------------------------------------------------------------------------------------------
           Representative_Sequence             total_reads   total_reads_nc   total_reads_non_nc 
--------------------------------------------- ------------- ---------------- --------------------
 M06508_12_000000000-CJG44_1_1101_16846_2451      33614          19913              13701        

 M06508_12_000000000-CJG44_1_1101_9357_2876        571            342                229         

 M06508_18_000000000-CNPPR_1_1101_15534_2103     201974            2                201972       

 M06508_12_000000000-CJG44_1_1101_15015_3135      6446            703                5743        

 M06508_12_000000000-CJG44_1_1101_15072_3643      26928           2907              24021        

 M06508_9_000000000-JTBW3_1_1106_8860_16108      368528            2                368526       
-------------------------------------------------------------------------------------------------</code></pre>
</div>
</div>
<p>In total there are 34262 repseqs that were potential contaminants.</p>
<p>Now we add in a column that calculates the percent of reads in the NC samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_read_totals</span> <span class="op">&lt;-</span> <span class="va">read_totals</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>perc_reads_in_nc <span class="op">=</span> <span class="fl">100</span><span class="op">*</span><span class="op">(</span></span>
<span>    <span class="va">total_reads_nc</span> <span class="op">/</span> <span class="op">(</span><span class="va">total_reads_nc</span> <span class="op">+</span> <span class="va">total_reads_non_nc</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                .after <span class="op">=</span> <span class="st">"total_reads_non_nc"</span><span class="op">)</span></span>
<span><span class="va">tmp_read_totals</span><span class="op">$</span><span class="va">perc_reads_in_nc</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">tmp_read_totals</span><span class="op">$</span><span class="va">perc_reads_in_nc</span>, digits <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And then we calculate row sums to get the number of NC and non-NC samples containing these reads. For this we can use the <code>subset.full.count_table</code> we read in above.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">control_cols</span>     <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"^Control_"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">full_count_tab</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">noncontrol_cols</span>  <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">full_count_tab</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span><span class="op">]</span>, <span class="va">control_cols</span><span class="op">)</span></span>
<span><span class="co"># rowwise tally of non-zero columns</span></span>
<span><span class="va">samp_totals</span> <span class="op">&lt;-</span> <span class="va">full_count_tab</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span></span>
<span>    num_nc_samp     <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">c_across</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">control_cols</span><span class="op">)</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    num_non_nc_samp <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="fu">c_across</span><span class="op">(</span><span class="fu">all_of</span><span class="op">(</span><span class="va">noncontrol_cols</span><span class="op">)</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="fl">1</span>, <span class="va">num_nc_samp</span>, <span class="va">num_non_nc_samp</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Finally add a column with the total number of samples and calculate the percent of NC samples containing these reads.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">samp_totals</span><span class="op">$</span><span class="va">total_samp</span> <span class="op">&lt;-</span> <span class="va">samp_totals</span><span class="op">$</span><span class="va">num_nc_samp</span> <span class="op">+</span> <span class="va">samp_totals</span><span class="op">$</span><span class="va">num_non_nc_samp</span></span>
<span><span class="va">samp_totals</span> <span class="op">&lt;-</span> <span class="va">samp_totals</span> <span class="op">%&gt;%</span>  </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/relocate.html">relocate</a></span><span class="op">(</span><span class="st">"total_samp"</span>, .after <span class="op">=</span> <span class="st">"Representative_Sequence"</span><span class="op">)</span></span>
<span><span class="va">samp_totals</span> <span class="op">&lt;-</span> <span class="va">samp_totals</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>perc_nc_samp <span class="op">=</span> </span>
<span>                  <span class="fl">100</span><span class="op">*</span><span class="op">(</span> <span class="va">num_nc_samp</span> <span class="op">/</span> <span class="op">(</span><span class="va">num_nc_samp</span> <span class="op">+</span> <span class="va">num_non_nc_samp</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                  .after <span class="op">=</span> <span class="st">"num_non_nc_samp"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>After all of this wrangling here is a snippet of the table showing the distribution of NC reads across samples.</p>
<p>Now we remove any repseqs where:</p>
<ul>
<li>The number of reads found in NC samples accounted for more than 10% of total reads OR</li>
<li>The percent of NC samples containing the repseq was greater than 10% of total samples.</li>
</ul>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_remove</span> <span class="op">&lt;-</span> <span class="va">nc_check</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">perc_reads_in_nc</span> <span class="op">&gt;</span> <span class="fl">10</span> <span class="op">|</span> <span class="va">perc_nc_samp</span> <span class="op">&gt;</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<table class="caption-top table">
<colgroup>
<col style="width: 11%">
<col style="width: 24%">
<col style="width: 20%">
<col style="width: 21%">
<col style="width: 21%">
</colgroup>
<thead><tr class="header">
<th></th>
<th>Total rep seqs</th>
<th>NC reads</th>
<th>non NC reads</th>
<th>% NC reads</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Removed</td>
<td>32438</td>
<td>166614</td>
<td>212292</td>
<td>43.972</td>
</tr>
<tr class="even">
<td>Retained</td>
<td>1824</td>
<td>26228</td>
<td>10280853</td>
<td>0.254</td>
</tr>
</tbody>
</table>
<p>We identified a total of <strong>34262</strong> representative sequences (<code>repseqs</code>) that were present in at least 1 NC sample by at least 1 read. We removed any repseq where more than 10% of total reads were found in NC samples OR any repseq found in more than 10% of NC samples. Based on these criteria we removed <strong>32438 </strong>repseqs from the data set, which accounted for <strong>166614</strong> total reads in NC samples and <strong>212292</strong> total reads in non-NC samples. Of the total reads removed <strong>43.972%</strong> came from NC samples. Of all repseqs identified in NC samples, <strong>1824</strong> were retained because they fell below the threshold criteria. These repseqs accounted for <strong>26228</strong> reads in NC samples and <strong>10280853</strong> reads in non-NC samples. NC samples accounted for <strong>0.254%</strong> of these reads.</p>
<p>OK, now we can create a new <code>neg_control.accnos</code> containing only repseqs abundant in NC samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu">write_delim</span><span class="op">(</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">nc_remove</span><span class="op">$</span><span class="va">Representative_Sequence</span><span class="op">)</span>, </span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/nc_repseq_remove.accnos"</span><span class="op">)</span>, </span>
<span>  col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And then use this file in conjunction with the mothur command <code>remove.seqs</code>.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb26"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>remove.seqs(accnos=nc_repseq_remove.accnos, fasta=shrimp.trim.contigs.good.unique.good.filter.unique.fasta, count=shrimp.trim.contigs.good.unique.good.filter.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Removed 32438 sequences from shrimp.trim.contigs.good.unique.good.filter.unique.fasta.
Removed 378906 sequences from shrimp.trim.contigs.good.unique.good.filter.count_table.

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta
shrimp.trim.contigs.good.unique.good.filter.pick.count_table</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb28"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>count.groups(count=shrimp.trim.contigs.good.unique.good.filter.pick.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Size of smallest group: 1.

Total seqs: 35944156.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.pick.count.summary</code></pre>
<p>Before we remove the NC samples we need to check whether some NC samples were already removed. When mothur runs the <code>remove.seqs</code> command it will automatically remove any samples where the read count has fallen to zero. If mothur did remove samples and we try to remove all NC samples, we will get an error. To check we can compare the <code>count.summary</code> files before and after the previous <code>remove.seqs</code> command.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_before</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/shrimp.trim.contigs.good.unique.good.filter.count.summary"</span><span class="op">)</span>,</span>
<span>  col_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  col_select <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_after</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/shrimp.trim.contigs.good.unique.good.filter.pick.count.summary"</span><span class="op">)</span>,</span>
<span>  col_names <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  col_select <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span>
<span><span class="va">tmp_nc_lost</span> <span class="op">&lt;-</span> <span class="fu">anti_join</span><span class="op">(</span><span class="va">tmp_before</span>, <span class="va">tmp_after</span><span class="op">)</span></span>
<span><span class="va">tmp_nc_lost</span><span class="op">$</span><span class="va">X1</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>These are the samples that were removed when we ran <code>remove.seqs</code>. We need to eliminate these sample IDs from our list of NC samples to remove or mothur will throw an error.</p>
<pre><code>[1] "Control_15" "Control_18" "Control_21" "Control_5"  </code></pre>
<p>As before, we can generate a list of NC samples to use in conjunction with the <code>remove.groups</code> command to eliminate all NC samples.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb32"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">nc_to_remove</span> <span class="op">&lt;-</span> <span class="fu">semi_join</span><span class="op">(</span><span class="va">tmp_before</span>, <span class="va">tmp_after</span><span class="op">)</span></span>
<span><span class="va">nc_to_remove</span> <span class="op">&lt;-</span> <span class="va">nc_to_remove</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span></span>
<span>    <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_starts.html">str_starts</a></span><span class="op">(</span><span class="va">X1</span>, <span class="st">"Control"</span><span class="op">)</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html">write_delim</a></span><span class="op">(</span><span class="va">nc_to_remove</span>, </span>
<span>                   file <span class="op">=</span> <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"nc_screen/nc_samples_remove.accnos"</span><span class="op">)</span>, </span>
<span>                   col_names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>In total the following mothur command should remove 56 negative control samples.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb33"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>remove.groups(count=shrimp.trim.contigs.good.unique.good.filter.pick.count_table, fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.fasta, accnos=nc_samples_remove.accnos)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Removed 26228 sequences from your count file.
Removed 0 sequences from your fasta file.

Output File names: 
shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb35"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>summary.seqs(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table, processors=30)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see data set summary
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Start</th>
<th style="text-align: center;">End</th>
<th style="text-align: center;">NBases</th>
<th style="text-align: center;">Ambigs</th>
<th style="text-align: center;">Polymer</th>
<th style="text-align: center;">NumSeqs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Minimum</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">208</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">2.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">897949</td>
</tr>
<tr class="odd">
<td style="text-align: left;">25%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">8979483</td>
</tr>
<tr class="even">
<td style="text-align: left;">Median:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">17958965</td>
</tr>
<tr class="odd">
<td style="text-align: left;">75%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">26938447</td>
</tr>
<tr class="even">
<td style="text-align: left;">97.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">35019980</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Maximum:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">254</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">35917928</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
<pre><code># of unique seqs:   4146230
total # of seqs:    35917928

It took 74 secs to summarize 35917928 sequences.

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.summary</code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb37"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>count.groups(count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Size of smallest group: 49.

Total seqs: 35917928.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.pick.pick.count.summary</code></pre>
</section><section id="remove-chimeras" class="level2"><h2 class="anchored" data-anchor-id="remove-chimeras">Remove Chimeras</h2>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb39"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>chimera.vsearch(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.count_table, dereplicate=t, processors=30)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Using vsearch version v2.30.0.
Checking sequences from shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta ...
...

/******************************************/
Splitting by sample: 

...

Removing chimeras from your input files:
/******************************************/
Running command: remove.seqs(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta,
accnos=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.accnos)
Removed 710630 sequences from shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.fasta.

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.pick.fasta

/******************************************/

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.chimeras
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.accnos
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta

/******************************************/</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb41"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>summary.seqs(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, processors=30)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see data set summary
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Start</th>
<th style="text-align: center;">End</th>
<th style="text-align: center;">NBases</th>
<th style="text-align: center;">Ambigs</th>
<th style="text-align: center;">Polymer</th>
<th style="text-align: center;">NumSeqs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Minimum</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">208</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">2.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">865884</td>
</tr>
<tr class="odd">
<td style="text-align: left;">25%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">8658836</td>
</tr>
<tr class="even">
<td style="text-align: left;">Median:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">17317672</td>
</tr>
<tr class="odd">
<td style="text-align: left;">75%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">25976508</td>
</tr>
<tr class="even">
<td style="text-align: left;">97.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">33769460</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Maximum:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">254</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">34635343</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
<pre><code># of unique seqs:   3435600
total # of seqs:    34635343

It took 64 secs to summarize 34635343 sequences.

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.summary

/******************************************/</code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb43"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>count.groups(count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Size of smallest group: 49.

Total seqs: 34635343.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count.summary

/******************************************/</code></pre>
</section><section id="repseq-taxonomy" class="level2"><h2 class="anchored" data-anchor-id="repseq-taxonomy">Repseq Taxonomy</h2>
<p>The <code>classify.seqs</code> command requires properly formatted reference and taxonomy databases. For taxonomic assignment, we are using the GSR database <span class="citation" data-cites="molano2024gsr">(<a href="#ref-molano2024gsr" role="doc-biblioref">Molano, Vega-Abellaneda, and Manichanh 2024</a>)</span>. The developers of mothur maintain <a href="https://mothur.org/wiki/taxonomy_outline/">formatted versions of popular databases</a>, however the GSR-DB has not been formatted by the developers yet.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>You can visit the <a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database download</a> page to find a database suitable to your data.</p>
</div>
</div>
<p>To create a mothur formatted version GSR-DB<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>, we perform the following steps.</p>
<section id="download-a-data-base" class="level4"><h4 class="anchored" data-anchor-id="download-a-data-base">Download a data base</h4>
<p>Here we are using the <a href="https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz">GSR V4 database</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb45"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> GSR-DB_V4_cluster-1.tar.gz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>First (in the command line) we remove first line of the taxonomy file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb46"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_taxa.txt tmp0.txt</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="st">'1d'</span> tmp0.txt <span class="op">&gt;</span> tmp1.txt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Next, delete species and remove leading [a-z]__ from taxa names</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb47"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/s__.*//g'</span> tmp1.txt <span class="op">&gt;</span> tmp2.txt</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/[a-zA-Z]__//g'</span> tmp2.txt <span class="op">&gt;</span> gsrdb.tax</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_seqs.fasta gsrdb.fasta</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp<span class="pp">*</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb48"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>classify.seqs(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, reference=reference_dbs/gsrdb.fasta, taxonomy=reference_dbs/gsrdb.tax, processors=30)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Reading template taxonomy...     DONE.
Reading template probabilities...     DONE.
It took 4 seconds get probabilities.
Classifying sequences from 
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta ...
[WARNING]: M06508_12_000000000-CJG44_1_2103_6654_25682 could not be classified. 
You can use the remove.lineage command with taxon=unknown; to remove such sequences.

...

It took 839 secs to classify 3435600 sequences.

It took 1697 secs to create the summary file for 3435600 sequences.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.taxonomy
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.tax.summary

/******************************************/</code></pre>
</section></section><section id="remove-contaminants" class="level2"><h2 class="anchored" data-anchor-id="remove-contaminants">Remove Contaminants</h2>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb50"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>remove.lineage(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, taxonomy=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.taxonomy, taxon=Chloroplast-Mitochondria-unknown-Eukaryota)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Running command: 
remove.seqs(accnos=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.accnos, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table, fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta)
Removed 2160 sequences from shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.fasta.
Removed 7262 sequences from shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.count_table.

/******************************************/

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.taxonomy
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.accnos
shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.pick.fasta

/******************************************/</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb52"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>summary.seqs(fasta=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.pick.fasta, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table, processors=30)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see data set summary
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<table class="caption-top table">
<thead><tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: center;">Start</th>
<th style="text-align: center;">End</th>
<th style="text-align: center;">NBases</th>
<th style="text-align: center;">Ambigs</th>
<th style="text-align: center;">Polymer</th>
<th style="text-align: center;">NumSeqs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Minimum</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">208</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: left;">2.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">3</td>
<td style="text-align: center;">865703</td>
</tr>
<tr class="odd">
<td style="text-align: left;">25%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">8657021</td>
</tr>
<tr class="even">
<td style="text-align: left;">Median:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">17314041</td>
</tr>
<tr class="odd">
<td style="text-align: left;">75%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">25971061</td>
</tr>
<tr class="even">
<td style="text-align: left;">97.5%-tile:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">253</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">33762379</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Maximum:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">254</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">34628081</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean:</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">554</td>
<td style="text-align: center;">252</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;"></td>
</tr>
</tbody>
</table>
<pre><code># of unique seqs:   3433440
total # of seqs:    34628081

It took 62 secs to summarize 34628081 sequences.

Output File Names:
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.pick.summary

/******************************************/</code></pre>
</div>
</div>
</div>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb54"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>summary.tax(taxonomy=shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.taxonomy, count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Using shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table as input file for the count parameter.
Using shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.taxonomy as input file for the taxonomy parameter.
It took 1580 secs to create the summary file for 34628081 sequences.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.unique.pick.pick.denovo.vsearch.gsrdb.wang.pick.tax.summary

/******************************************/</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb56"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>count.groups(count=shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Size of smallest group: 49.

Total seqs: 34628081.

Output File Names: 
shrimp.trim.contigs.good.unique.good.filter.pick.pick.denovo.vsearch.pick.count.summary

/******************************************/</code></pre>
</section><section id="track-reads-through-workflow" class="level2"><h2 class="anchored" data-anchor-id="track-reads-through-workflow">Track Reads through Workflow</h2>
<p>At this point we can look at the number of reads that made it through each step of the workflow for every sample.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb58"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">read_change</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"mothur_med_pipeline_read_changes.txt"</span><span class="op">)</span>,</span>
<span>  col_names <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="preparing-for-analysis" class="level2"><h2 class="anchored" data-anchor-id="preparing-for-analysis">Preparing for analysis</h2>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb59"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>rename.file(fasta=current, count=current, taxonomy=current, prefix=final_med)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<!------------------------------------------------------------------------>
<!-------------------- Use this area to save things ---------------------->
<!------------------------------------------------------------------------>
</section></section><section id="med-analysis" class="level1"><h1>MED Analysis</h1>
<div style="text-align: center;">
<div class="cell" data-eval="true" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"></figure><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart LR
    style Z fill:transparent
    style Z font-size:1.1em
    Z(&lt;a href='https://github.com/michberr/MicrobeMiseq'&gt;mothur2oligo&lt;/a&gt;&lt;/br&gt; script)

    I(&lt;b&gt;&lt;a href='https://mothur.org/wiki/get.lineage/'&gt;get.lineage&lt;/a&gt;&lt;/b&gt;) --&gt; J(&lt;b&gt;&lt;a href='https://mothur.org/wiki/list.seqs/'&gt;list.seqs&lt;/a&gt;&lt;/b&gt;)
    J --&gt; K(&lt;b&gt;&lt;a href='https://mothur.org/wiki/get.seqs/'&gt;get.seqs&lt;/a&gt;&lt;/b&gt;)
    K --&gt; L(&lt;b&gt;&lt;a href='https://mothur.org/wiki/deunique.seqs/'&gt;deunique.seqs&lt;/a&gt;&lt;/b&gt;)
    L --&gt; END:::hidden

    BEGIN:::hidden --&gt; M
    M(renamer.pl&lt;br/&gt;header rename)
    M --&gt; N(o-trim-uninformative-columns-from-alignment)
    N --&gt; O(decompose)    
</pre>
</div>
<p></p>
</div>
</div>
</div>
</div>
<p>Now that we have the necessary files we are almost ready to run the MED pipeline. First though we need to properly format the mothur files. For this we will use a custom script called <a href="https://github.com/michberr/MicrobeMiseq"><code>mothur2oligo</code></a> written by <a href="https://github.com/michberr">Michelle Berry</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb60"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> mothur2oligo.sh</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="callout callout-style-simple callout-note callout-titled">
<div class="callout-header d-flex align-content-center collapsed" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Note</span>Expand to see mothur2oligo workflow
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>These steps are run automatically by the <code>mothur2oligo</code> script.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb61"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>get.lineage(taxonomy=final_med.taxonomy, taxon='Bacteria;-Archaea;', count=final_med.count_table)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>/******************************************/
Running command: get.seqs(accnos=final_med.pick.taxonomy.accnos, count=final_med.count_table)
Selected 34628081 sequences from final_med.count_table.

Output File Names:
final_med.pick.count_table

/******************************************/

Output File Names:
final_med.pick.taxonomy
final_med.pick.taxonomy.accnos
final_med.pick.count_table</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb63"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>list.seqs(count=current)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Using final_med.pick.count_table as input file for the count parameter.

Output File Names: 
final_med.pick.accnos</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb65"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>get.seqs(accnos=current, fasta=final_med.fasta)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Using pipelineFiles_med/final_med.pick.accnos as input file for the accnos parameter.
Selected 3433440 sequences from pipelineFiles_med/final_med.fasta.

Output File Names:
final_med.pick.fasta</code></pre>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb67"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>deunique.seqs(fasta=current, count=current)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<pre><code>Using final_med.pick.count_table as input file for the count parameter.
Using final_med.pick.fasta as input file for the fasta parameter.

Output File Names: 
final_med.pick.redundant.fasta
final_med.pick.redundant.groups</code></pre>
<p>The eventual output of the <code>mothur2oligo</code> workflow is a fasta file with reformatted headers. The script <code>renamer.pl</code> adds sample names to fasta headers and appends the extension <code>_headers-replaced.fasta</code> to a new file with these reformatted data.</p>
<pre><code>final_med.pick.redundant.fasta_headers-replaced.fasta</code></pre>
</div>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb70"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="ex">o-trim-uninformative-columns-from-alignment</span> <span class="dt">\</span></span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>        final_med.pick.redundant.renamed.fasta</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="ex">decompose</span> final_med.pick.redundant.renamed.fasta-TRIMMED <span class="dt">\</span></span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">--sample-mapping</span> med_mapping.txt <span class="dt">\</span></span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output-directory</span> MED <span class="dt">\</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">--number-of-threads</span> 24 <span class="dt">\</span></span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">--skip-gen-figures</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>To run the <code>decompose</code> command you will need this <a href="../../../share/ssu/med/med_mapping.txt">sample mapping file</a>.</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
<span class="screen-reader-only">Tip</span>MED Results
</div>
</div>
<div class="callout-body-container callout-body">
<p>Once this is finished, the MED software will output a <a href="../../../share/ssu/med/med_results/index.html">summary of the run</a>.</p>
</div>
</div>
</section><section id="data-set-prep" class="level1"><h1>Data Set Prep</h1>
<p>In this next part of the workflow our main goal is to create a <em>microtable object</em> using the R package <a href="https://joey711.github.io/phyloseq/">microeco</a> <span class="citation" data-cites="liu2021microeco">(<a href="#ref-liu2021microeco" role="doc-biblioref">Liu et al. 2021</a>)</span>. The microtable will be used to store the OTU by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</p>
<section id="assign-taxonomy" class="level2"><h2 class="anchored" data-anchor-id="assign-taxonomy">Assign Taxonomy</h2>
<p>Our first step is to classify the node representatives from the MED output. The <code>classify.seqs</code> command requires properly formatted reference and taxonomy databases. For taxonomic assignment, we are using the GSR database <span class="citation" data-cites="molano2024gsr">(<a href="#ref-molano2024gsr" role="doc-biblioref">Molano, Vega-Abellaneda, and Manichanh 2024</a>)</span>. The developers of mothur maintain <a href="https://mothur.org/wiki/taxonomy_outline/">formatted versions of popular databases</a>, however the GSR-DB has not been formatted by the developers yet.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>Here you can download an appropriate version of the <a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database</a>.</p>
</div>
</div>
<p>To create a mothur formatted version GSR-DB, we perform the following steps (we went through this process above prior to removing contaminants but will repeat here for posterity).</p>
<section id="download-a-data-base-1" class="level4"><h4 class="anchored" data-anchor-id="download-a-data-base-1">Download a data base</h4>
<p>Here we are using the <a href="https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz">GSR V4 database</a>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb71"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> GSR-DB_V4_cluster-1.tar.gz</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>First (in the command line) we remove first line of the taxonomy file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb72"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_taxa.txt tmp0.txt</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="st">'1d'</span> tmp0.txt <span class="op">&gt;</span> tmp1.txt</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Next, delete species and remove leading [a-z]__ from taxa names</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb73"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/s__.*//g'</span> tmp1.txt <span class="op">&gt;</span> tmp2.txt</span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/[a-zA-Z]__//g'</span> tmp2.txt <span class="op">&gt;</span> gsrdb.tax</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_seqs.fasta gsrdb.fasta</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>The next thing we need to do is grab the <code>node-representatives.fa.txt</code> from the MED output so that we can classify these sequences. Of course, proper formatting is required.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb74"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">--pattern</span> <span class="pp">^</span> <span class="at">--replacement</span> MED node-representatives.fa.txt <span class="op">&gt;</span> tmp1.fa</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">--pattern</span> <span class="st">"\|.*"</span> <span class="at">--replacement</span> <span class="st">''</span> tmp1.fa <span class="op">&gt;</span> med_nodes.fasta</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp1.fa</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Great, the reference database is formatted. Now we need to make a few files that mimics the normal mothur output files because the MED pipeline does not exactly generate the files we need to create a <code>microtable</code> object. First we use the <code>matrix_counts.txt</code> file from the MED analysis to create a mothur-styled <code>count.table</code>.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb75"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"med_results/matrix_counts.txt"</span><span class="op">)</span>,</span>
<span>    col_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename_with</a></span><span class="op">(</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"MED"</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span> </span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"tmp"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="va">tmp_med_counts</span>, <span class="st">"tmp"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span></span>
<span>                  <span class="fu">mutate</span><span class="op">(</span>total <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span>, .before <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span> </span>
<span>     <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"Representative_Sequence"</span><span class="op">)</span></span>
<span><span class="va">med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Now we can actually classify the representative sequences.</p>
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb76"><pre class="sourceCode default code-with-copy"><code class="sourceCode default"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>classify.seqs(fasta=med_nodes.fasta, count=med_nodes.count.table, reference=reference_dbs/gsrdb.fasta, taxonomy=reference_dbs/gsrdb.tax)</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
<p>Now we make a mothur styled <code>shared</code> file.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb77"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="fu">read_tsv</span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"med_results/matrix_counts.txt"</span><span class="op">)</span>,</span>
<span>    col_names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_n_meds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">tmp_med_counts</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename_with</a></span><span class="op">(</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="st">"MED"</span>, <span class="va">.x</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html">rename</a></span><span class="op">(</span><span class="st">"Group"</span> <span class="op">=</span> <span class="st">"MEDsamples"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_med_counts</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/add_column.html">add_column</a></span><span class="op">(</span>label <span class="op">=</span> <span class="fl">0.03</span>, .before <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/add_column.html">add_column</a></span><span class="op">(</span>numOtus <span class="op">=</span> <span class="va">tmp_n_meds</span>, .after <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">med_shared</span> <span class="op">&lt;-</span> <span class="va">tmp_med_counts</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section><section id="a.-taxonomy-table" class="level3"><h3 class="anchored" data-anchor-id="a.-taxonomy-table">A. Taxonomy Table</h3>
<p>Here is what the taxonomy table looks like in the microeco mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
---------------------------------------------------------------------
  &amp;nbsp;      Kingdom           Phylum                 Class         
---------- ------------- -------------------- -----------------------
  OTU_50    k__Bacteria   p__Proteobacteria    c__Betaproteobacteria 

 OTU_8058   k__Bacteria   p__Actinobacteria      c__Actinobacteria   

 OTU_7152   k__Bacteria   p__Verrucomicrobia    c__OPB35 soil group  
---------------------------------------------------------------------</code></pre>
</div>
</div>
<p>Our taxonomy file (below) needs a little wrangling to be properly formatted.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu">read_delim</span><span class="op">(</span></span>
<span>             <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"node_taxonomy/med_nodes.cons.taxonomy"</span><span class="op">)</span>,</span>
<span>             delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
     OTU         Size                                                                                Taxonomy                                                                              
-------------- -------- -------------------------------------------------------------------------------------------------------------------------------------------------------------------
 MED000013674   347929                     Bacteria(100);Proteobacteria(100);Gammaproteobacteria(100);Vibrionales(100);Vibrionaceae(100);Vibrionaceae_unclassified(100);                   

 MED000009391   314434         Bacteria(100);Bacteria_unclassified(100);Bacteria_unclassified(100);Bacteria_unclassified(100);Bacteria_unclassified(100);Bacteria_unclassified(100);       

 MED000010996   305834   Bacteria(100);Proteobacteria(90);Proteobacteria_unclassified(90);Proteobacteria_unclassified(90);Proteobacteria_unclassified(90);Proteobacteria_unclassified(90); 
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------</code></pre>
</div>
</div>
<p>Some fancy string manipulation…</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb81"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">tmp_tax</span>, </span>
<span>                             <span class="va">gsub</span>, </span>
<span>                             pattern <span class="op">=</span> <span class="st">"\\(\\d+\\)"</span>, </span>
<span>                             replacement <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">tmp_tax</span>, </span>
<span>                             <span class="va">gsub</span>, </span>
<span>                             pattern <span class="op">=</span> <span class="st">";$"</span>, </span>
<span>                             replacement <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu">separate_wider_delim</span><span class="op">(</span><span class="va">tmp_tax</span>, </span>
<span>                              cols <span class="op">=</span> <span class="va">Taxonomy</span>, </span>
<span>                              delim <span class="op">=</span> <span class="st">";"</span>, names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>                                <span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, </span>
<span>                                <span class="st">"Class"</span>, <span class="st">"Order"</span>, </span>
<span>                                <span class="st">"Family"</span>, <span class="st">"Genus"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">tmp_tax</span>, <span class="va">gsub</span>, </span>
<span>                           pattern <span class="op">=</span> <span class="st">"^.*_unclassified$"</span>, </span>
<span>                           replacement <span class="op">=</span> <span class="st">""</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_tax</span><span class="op">$</span><span class="va">Size</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="va">tmp_tax</span>, <span class="st">"OTU"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And we get this …</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
------------------------------------------------------------------------------
    &amp;nbsp;      Kingdom        Phylum              Class             Order    
-------------- ---------- ---------------- --------------------- -------------
 MED000013674   Bacteria   Proteobacteria   Gammaproteobacteria   Vibrionales 

 MED000009391   Bacteria                                                      

 MED000010996   Bacteria   Proteobacteria                                     

 MED000011677   Bacteria   Proteobacteria   Gammaproteobacteria               
------------------------------------------------------------------------------</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb83"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rank_prefixes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span></span>
<span>  Kingdom <span class="op">=</span> <span class="st">"k"</span>, </span>
<span>  Phylum  <span class="op">=</span> <span class="st">"p"</span>, </span>
<span>  Class   <span class="op">=</span> <span class="st">"c"</span>, </span>
<span>  Order   <span class="op">=</span> <span class="st">"o"</span>, </span>
<span>  Family  <span class="op">=</span> <span class="st">"f"</span>, </span>
<span>  Genus   <span class="op">=</span> <span class="st">"g"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_tax</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu">everything</span><span class="op">(</span><span class="op">)</span>, <span class="op">~</span><span class="fu">replace_na</span><span class="op">(</span><span class="va">.</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span> </span>
<span>  <span class="fu">mutate</span><span class="op">(</span><span class="fu">across</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">rank_prefixes</span><span class="op">)</span>, </span>
<span>                <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="va">rank_prefixes</span><span class="op">[</span><span class="fu">cur_column</span><span class="op">(</span><span class="op">)</span><span class="op">]</span>, <span class="st">"__"</span>, <span class="va">.</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span><span class="fu">tidy_taxonomy</span><span class="op">(</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>And then this. Exactly like the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
------------------------------------------------------------------------------------------
    &amp;nbsp;        Kingdom          Phylum                 Class                Order      
-------------- ------------- ------------------- ------------------------ ----------------
 MED000013674   k__Bacteria   p__Proteobacteria   c__Gammaproteobacteria   o__Vibrionales 

 MED000009391   k__Bacteria          p__                   c__                  o__       

 MED000010996   k__Bacteria   p__Proteobacteria            c__                  o__       

 MED000011677   k__Bacteria   p__Proteobacteria   c__Gammaproteobacteria        o__       
------------------------------------------------------------------------------------------</code></pre>
</div>
</div>
</section><section id="b.-sequence-table" class="level3"><h3 class="anchored" data-anchor-id="b.-sequence-table">B. Sequence Table</h3>
<p>Here is what the sequence table looks like in the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
--------------------------------------------------------------------
  &amp;nbsp;    S1   S2   S3   S4   S5   S6   S7   S9   S10   S11   S12 
---------- ---- ---- ---- ---- ---- ---- ---- ---- ----- ----- -----
 OTU_4272   1    0    1    1    0    0    1    1     0     1     1  

 OTU_236    1    4    0    2    35   5    94   0    177   14    27  

 OTU_399    9    2    2    4    4    0    3    6     0     1     2  
--------------------------------------------------------------------</code></pre>
</div>
</div>
<p>These code block will return a properly formatted sequence table.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb86"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/read_delim.html">read_delim</a></span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"node_taxonomy/med_nodes.shared"</span><span class="op">)</span>,</span>
<span>  delim <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb87"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_st</span><span class="op">$</span><span class="va">numOtus</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">tmp_st</span><span class="op">$</span><span class="va">label</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="va">tmp_st</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span><span class="op">)</span>, names_to <span class="op">=</span> <span class="st">"tmp"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tmp_st</span> <span class="op">&lt;-</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="va">tmp_st</span>, <span class="st">"tmp"</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
--------------------------------------------------------------------------
    &amp;nbsp;      EP_A_AREN_EG_8651   EP_A_AREN_EG_8654   EP_A_AREN_EG_8698 
-------------- ------------------- ------------------- -------------------
 MED000011539           0                   4                   1         

 MED000013720           7                  410                 25         

 MED000009147           0                   0                   0         

 MED000009218           0                   0                   0         
--------------------------------------------------------------------------</code></pre>
</div>
</div>
</section><section id="c.-sample-table" class="level3"><h3 class="anchored" data-anchor-id="c.-sample-table">C. Sample Table</h3>
<p>Here is what the sample table looks like in the mock data.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
----------------------------------------------------
 &amp;nbsp;   SampleID   Group   Type       Saline      
-------- ---------- ------- ------ -----------------
   S1        S1       IW      NE    Non-saline soil 

   S2        S2       IW      NE    Non-saline soil 

   S3        S3       IW      NE    Non-saline soil 
----------------------------------------------------</code></pre>
</div>
</div>
<p>No problem.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span></span>
<span>  <span class="fu">here</span><span class="op">(</span><span class="st">"working_files/ssu/sampledata"</span>, <span class="st">"sample_data.txt"</span><span class="op">)</span>,</span>
<span>  header <span class="op">=</span> <span class="cn">TRUE</span>, sep <span class="op">=</span> <span class="st">"\t"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="va">samdf</span> <span class="op">%&gt;%</span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">column_to_rownames</a></span><span class="op">(</span><span class="st">"SampleID"</span><span class="op">)</span></span>
<span><span class="va">samdf</span><span class="op">$</span><span class="va">SampleID</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">samdf</span><span class="op">)</span></span>
<span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="va">samdf</span> <span class="op">%&gt;%</span> <span class="fu">relocate</span><span class="op">(</span><span class="va">SampleID</span><span class="op">)</span></span>
<span></span>
<span><span class="va">samdf</span> <span class="op">&lt;-</span> <span class="va">samdf</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span></span>
<span>    <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_starts.html">str_starts</a></span><span class="op">(</span><span class="va">SampleID</span>, <span class="st">"Control"</span>, negate <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
-------------------------------------------------------------------------
      &amp;nbsp;             SampleID        OCEAN   SPECIES   TISSUE    ID  
------------------- ------------------- ------- --------- -------- ------
 EP_A_AREN_GL_8625   EP_A_AREN_GL_8625    EP     A_AREN      GL     8625 

 EP_A_AREN_GL_8626   EP_A_AREN_GL_8626    EP     A_AREN      GL     8626 

 EP_A_AREN_GL_8627   EP_A_AREN_GL_8627    EP     A_AREN      GL     8627 
-------------------------------------------------------------------------</code></pre>
</div>
</div>
</section></section><section id="experiment-level-objects" class="level2"><h2 class="anchored" data-anchor-id="experiment-level-objects">Experiment-level Objects</h2>
<p>In the following section we create <code>microtable</code> and <code>phyloseq</code> objects. These are single, consolidated experiment-level data objects that organize various data types, such as OTU tables (taxa abundance), sample metadata, taxonomic classifications, as well as phylogenetic trees and reference sequences, into one structured format. This unified approach facilitates reproducible, interactive analysis and visualization by bringing all related information together.</p>
<p>We begin by creating the <code>microtable</code> and then use the function <code>meco2phyloseq</code> from the <a href="https://github.com/ChiLiubio/file2meco">file2meco</a> package to create the <code>phyloseq</code> object. This way all of the underlying data is identical across the two objects.</p>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>These objects contain an OTU table (taxa abundances), sample metadata, and taxonomy table (mapping between OTUs and higher-level taxonomic classifications).</p>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb92"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">sample_info</span> <span class="op">&lt;-</span> <span class="va">samdf</span></span>
<span><span class="va">tax_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_tax</span></span>
<span><span class="va">otu_tab</span> <span class="op">&lt;-</span> <span class="va">tmp_st</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb93"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_me</span> <span class="op">&lt;-</span> <span class="va">microtable</span><span class="op">$</span><span class="fu">new</span><span class="op">(</span>sample_table <span class="op">=</span> <span class="va">sample_info</span>, </span>
<span>                         otu_table <span class="op">=</span> <span class="va">otu_tab</span>, </span>
<span>                         tax_table <span class="op">=</span> <span class="va">tax_tab</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<pre><code>microtable-class object:
sample_table have 1849 rows and 14 columns
otu_table have 436 rows and 1849 columns
tax_table have 436 rows and 6 columns</code></pre>
<section id="add-representative-sequence" class="level3"><h3 class="anchored" data-anchor-id="add-representative-sequence">Add Representative Sequence</h3>
<p>The fasta file returned by MED needs a little T.L.C. For that we use a tool called SeqKit <span class="citation" data-cites="shen2024seqkit2">(<a href="#ref-shen2024seqkit2" role="doc-biblioref">Shen, Sipos, and Zhao 2024</a>)</span> for fasta defline manipulation.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode cell-code" id="cb95"><pre class="sourceCode zsh code-with-copy"><code class="sourceCode zsh"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="pp">^</span> <span class="at">-r</span> MED node-representatives.fa.txt <span class="op">&gt;</span> tmp1.fa</span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"\|.*"</span> <span class="at">-r</span> <span class="st">''</span> tmp1.fa <span class="op">&gt;</span> tmp2.fa </span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"-"</span> <span class="at">-r</span> <span class="st">'$1'</span> <span class="at">-s</span> <span class="at">-w</span> 0 tmp2.fa <span class="op">&gt;</span> med_rep.fasta</span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp1.fa</span>
<span id="cb95-5"><a href="#cb95-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp2.fa</span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb96"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">readDNAStringSet</a></span><span class="op">(</span><span class="fu">here</span><span class="op">(</span><span class="va">work_here</span>, <span class="st">"med_rep.fasta"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">tmp_me</span><span class="op">$</span><span class="va">rep_fasta</span> <span class="op">&lt;-</span> <span class="va">rep_fasta</span></span>
<span><span class="va">tmp_me</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_me</span></span>
<span><span class="va">me_raw</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_me</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="curate-the-data-set" class="level2"><h2 class="anchored" data-anchor-id="curate-the-data-set">Curate the Data Set</h2>
<p>Pretty much the last thing to do is remove low-count samples.</p>
<section id="remove-low-count-samples" class="level3"><h3 class="anchored" data-anchor-id="remove-low-count-samples">Remove Low-Count Samples</h3>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb98"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">threshold</span> <span class="op">&lt;-</span> <span class="fl">1000</span></span>
<span><span class="va">tmp_no_low</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">me_raw</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="va">me_raw</span><span class="op">$</span><span class="va">otu_table</span> <span class="op">%&gt;%</span></span>
<span>          <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="fu">where</span><span class="op">(</span><span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">is.numeric</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&amp;&amp;</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">threshold</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span><span class="op">$</span><span class="fu">tidy_dataset</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">tmp_no_low</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>microtable-class object:
sample_table have 1799 rows and 14 columns
otu_table have 436 rows and 1799 columns
tax_table have 436 rows and 6 columns
rep_fasta have 436 sequences</code></pre>
</div>
</div>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">me_final</span> <span class="op">&lt;-</span> <span class="fu">microeco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microeco/man/clone.html">clone</a></span><span class="op">(</span><span class="va">tmp_no_low</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
<p>Lastly, we can use the package <a href="https://github.com/ChiLiubio/file2meco"><code>file2meco</code></a> to generate a <a href="https://joey711.github.io/phyloseq/">phyloseq</a> object.</p>
<div class="cell">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb101"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ps_final</span> <span class="op">&lt;-</span> <span class="fu">file2meco</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/file2meco/man/meco2phyloseq.html">meco2phyloseq</a></span><span class="op">(</span><span class="va">me_final</span><span class="op">)</span></span></code></pre></div><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></div>
</div>
</section></section><section id="summary" class="level2"><h2 class="anchored" data-anchor-id="summary">Summary</h2>
<p>Now time to summarize the data. For this we use the R package <a href="https://microbiome.github.io">miaverse</a> <span class="citation" data-cites="felix2024mia">(<a href="#ref-felix2024mia" role="doc-biblioref">Borman et al. 2024</a>)</span>.</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead><tr class="header">
<th style="text-align: left;">me_dataset</th>
<th style="text-align: right;">total_asvs</th>
<th style="text-align: right;">total_reads</th>
<th style="text-align: right;">total_samples</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">original</td>
<td style="text-align: right;">436</td>
<td style="text-align: right;">24817013</td>
<td style="text-align: right;">1849</td>
</tr>
<tr class="even">
<td style="text-align: left;">no low count samps</td>
<td style="text-align: right;">436</td>
<td style="text-align: right;">24785483</td>
<td style="text-align: right;">1799</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell" data-tbl-cap="Dataset metrics before and after curation.">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Dataset metrics before and after curation.</caption>
<thead><tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Start</th>
<th style="text-align: left;">End</th>
</tr></thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Min. no. of reads</td>
<td style="text-align: left;">1</td>
<td style="text-align: left;">1010</td>
</tr>
<tr class="even">
<td style="text-align: left;">Max. no. of reads</td>
<td style="text-align: left;">185675</td>
<td style="text-align: left;">185675</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Total no. of reads</td>
<td style="text-align: left;">24817013</td>
<td style="text-align: left;">24785483</td>
</tr>
<tr class="even">
<td style="text-align: left;">Avg. no. of reads</td>
<td style="text-align: left;">13422</td>
<td style="text-align: left;">13777</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Median no. of reads</td>
<td style="text-align: left;">10360</td>
<td style="text-align: left;">10566</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total ASVs</td>
<td style="text-align: left;">436</td>
<td style="text-align: left;">436</td>
</tr>
<tr class="odd">
<td style="text-align: left;">No.&nbsp;of singleton ASVs</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="even">
<td style="text-align: left;">% of singleton ASVs</td>
<td style="text-align: left;">NA</td>
<td style="text-align: left;">NA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sparsity</td>
<td style="text-align: left;">0.846</td>
<td style="text-align: left;">0.845</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We started off with 436 ASVs and 1849 samples. After removing low-count samples, there were 436 ASVs and 1799 samples remaining.</p>
<p>We lost a total of 110 samples after curating the dataset. This includes 60 NC samples and 50 non-NC samples.</p>
<p>Here is a list of non-NC samples that were removed.</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "EP_A_HEBE_ST_8937" "EP_A_UMBO_EG_9126" "EP_A_UTRI_MG_9071"
 [4] "EP_A_UTRI_ST_9071" "EP_E_SAMP_MD_9290" "EP_E_SAMP_MD_9293"
 [7] "EP_E_SAMP_SD_9285" "EP_E_SAMP_SD_9286" "EP_E_SAMP_SD_sed" 
[10] "WA_A_BAHA_GL_7474" "WA_A_BOUV_GL_7580" "WA_A_BOUV_ST_7585"
[13] "WA_A_CRIS_GL_9624" "WA_A_CRIS_HP_7536" "WA_A_CRIS_ST_7539"
[16] "WA_A_CRIS_ST_9471" "WA_A_FLOR_EG_7685" "WA_A_FLOR_EG_9662"
[19] "WA_A_FLOR_GL_9659" "WA_A_FLOR_HP_9659" "WA_A_FLOR_ST_9655"
[22] "WA_A_FORM_EG_7457" "WA_A_FORM_MG_7752" "WA_A_NUTT_EG_9410"
[25] "WA_A_NUTT_GL_9405" "WA_A_NUTT_ST_7430" "WA_A_NUTT_ST_9405"
[28] "WA_A_PARA_GL_9474" "WA_A_PARA_HP_9364" "WA_A_PARA_MG_9364"
[31] "WA_A_PARA_ST_7460" "WA_A_PARA_ST_7708" "WA_A_PARA_ST_9477"
[34] "WA_A_PCNS_HP_9513" "WA_A_PCNS_MG_9513" "WA_A_PCWS_EG_9438"
[37] "WA_A_PCWS_ST_9438" "WA_A_THOM_HP_7392" "WA_A_THOM_ST_7392"
[40] "WA_A_THOM_ST_9385" "WA_A_THOM_ST_9389" "WA_A_VERR_EG_9501"
[43] "WA_A_VERR_MG_9501" "WA_A_WEBS_HP_7727" "WA_A_WEBS_HP_9411"
[46] "WA_A_WEBS_MG_7612" "WA_A_WEBS_ST_9411" "WA_E_SAMP_RB_7531"
[49] "WA_E_SAMP_RB_7533" "WA_E_SAMP_RB_7535"</code></pre>
</div>
</div>
</section></section><section id="download-results" class="level1 page-columns page-full"><h1>Download Results</h1>
<p>Quick access to read changes through the pipeline and repseqs detected in negative control samples.</p>
<div id="listing-med-listing" class="column-body quarto-listing quarto-listing-container-table">
<table class="quarto-listing-table table">
<thead><tr>
<th>
File Name
</th>

<th>
Description
</th>

</tr></thead>
<tbody class="list">
<tr data-index="0" data-listing-filename-sort="reads_in_nc_samples.txt">
<td>
<a href="../../../share/ssu/med/reads_in_nc_samples.txt" class="filename listing-filename">reads_in_nc_samples.txt</a>
</td>
<td>
<span class="listing-description">Abundance table of reads detected in Negative Control samples. </span>
</td>

</tr>
<tr data-index="1" data-listing-filename-sort="mothur_med_pipeline_read_changes.txt">
<td>
<a href="../../../share/ssu/med/mothur_med_pipeline_read_changes.txt" class="filename listing-filename">mothur_med_pipeline_read_changes.txt</a>
</td>
<td>
<span class="listing-description">Read changes for all samples through mothur/MED pipeline </span>
</td>

</tr>
</tbody>
</table>
<div class="listing-no-matching d-none">No matching items</div>
</div>
<!------------------------------------------------------------------------>
<!-------------------- Use this area to save things ---------------------->
<!------------------------------------------------------------------------>
<!--------------------------------------->
<!-- These chunks are for curated data -->
<!--------------------------------------->
<!-------------------------------------------->
<!-- These chunks are for  processing data  -->
<!-------------------------------------------->
</section><div id="quarto-appendix" class="default"><section id="references" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">References</h2><div class="quarto-appendix-contents">
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-felix2024mia" class="csl-entry" role="listitem">
Borman, Tuomas, Felix G. M. Ernst, Sudarshan A. Shetty, and Leo Lahti. 2024. <em>Mia: Microbiome Analysis</em>. <a href="https://doi.org/10.18129/B9.bioc.mia">https://doi.org/10.18129/B9.bioc.mia</a>.
</div>
<div id="ref-eren2015minimum" class="csl-entry" role="listitem">
Eren, A Murat, Hilary G Morrison, Pamela J Lescault, Julie Reveillaud, Joseph H Vineis, and Mitchell L Sogin. 2015. <span>“Minimum Entropy Decomposition: Unsupervised Oligotyping for Sensitive Partitioning of High-Throughput Marker Gene Sequences.”</span> <em>The ISME Journal</em> 9 (4): 968–79. <a href="https://doi.org/10.1038/ismej.2014.195">https://doi.org/10.1038/ismej.2014.195</a>.
</div>
<div id="ref-liu2021microeco" class="csl-entry" role="listitem">
Liu, Chi, Yaoming Cui, Xiangzhen Li, and Minjie Yao. 2021. <span>“Microeco: An r Package for Data Mining in Microbial Community Ecology.”</span> <em>FEMS Microbiology Ecology</em> 97 (2): fiaa255. <a href="https://doi.org/10.1038/s41596-025-01239-4">https://doi.org/10.1038/s41596-025-01239-4</a>.
</div>
<div id="ref-molano2024gsr" class="csl-entry" role="listitem">
Molano, Leidy-Alejandra G, Sara Vega-Abellaneda, and Chaysavanh Manichanh. 2024. <span>“GSR-DB: A Manually Curated and Optimized Taxonomical Database for 16S rRNA Amplicon Analysis.”</span> <em>Msystems</em> 9 (2): e00950–23. <a href="https://doi.org/10.1128/msystems.00950-23">https://doi.org/10.1128/msystems.00950-23</a>.
</div>
<div id="ref-shen2024seqkit2" class="csl-entry" role="listitem">
Shen, Wei, Botond Sipos, and Liuyang Zhao. 2024. <span>“SeqKit2: A Swiss Army Knife for Sequence and Alignment Processing.”</span> <em>Imeta</em> 3 (3): e191. <a href="https://doi.org/10.1002/imt2.191">https://doi.org/10.1002/imt2.191</a>.
</div>
</div>
</div></section><section id="detailed-session-info" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Detailed Session Info</h2><div class="quarto-appendix-contents">
<details><summary>Expand to see Session Info</summary><div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.5.1 (2025-06-13)
 os       macOS Ventura 13.7.8
 system   x86_64, darwin20
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/Los_Angeles
 date     2025-10-03
 pandoc   3.6.3 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/x86_64/ (via rmarkdown)
 quarto   1.8.25 @ /Applications/quarto/bin/quarto

─ Packages ───────────────────────────────────────────────────────────────────
 package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
======   Devtools Session info   ===================================</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code> package                  * version   date (UTC) lib source
 Biobase                  * 2.68.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 BiocGenerics             * 0.54.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 Biostrings               * 2.76.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 downloadthis             * 0.5.0     2025-09-26 [1] Github (fmmattioni/downloadthis@18e3e5a)
 dplyr                    * 1.1.4     2023-11-17 [1] CRAN (R 4.5.0)
 fontawesome              * 0.5.3     2024-11-16 [1] CRAN (R 4.5.0)
 forcats                  * 1.0.0     2023-01-29 [1] CRAN (R 4.5.0)
 fs                       * 1.6.6     2025-04-12 [1] CRAN (R 4.5.0)
 generics                 * 0.1.4     2025-05-09 [1] CRAN (R 4.5.0)
 GenomeInfoDb             * 1.44.2    2025-08-18 [1] Bioconductor 3.21 (R 4.5.1)
 GenomicRanges            * 1.60.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 ggplot2                  * 4.0.0     2025-09-11 [1] CRAN (R 4.5.1)
 here                     * 1.0.2     2025-09-15 [1] CRAN (R 4.5.1)
 htmltools                * 0.5.8.1   2024-04-04 [1] CRAN (R 4.5.0)
 IRanges                  * 2.42.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 lubridate                * 1.9.4     2024-12-08 [1] CRAN (R 4.5.0)
 magrittr                 * 2.0.4     2025-09-12 [1] CRAN (R 4.5.1)
 MatrixGenerics           * 1.20.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 matrixStats              * 1.5.0     2025-01-07 [1] CRAN (R 4.5.0)
 mia                      * 1.15.6    2024-11-22 [1] Bioconductor 3.21 (R 4.5.0)
 microbiome               * 1.30.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 microeco                 * 1.15.0    2025-05-18 [1] CRAN (R 4.5.0)
 microViz                 * 0.12.7    2025-08-01 [1] https://david-barnett.r-universe.dev (R 4.5.1)
 MultiAssayExperiment     * 1.34.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 pander                   * 0.6.6     2025-03-01 [1] CRAN (R 4.5.0)
 phyloseq                 * 1.52.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 purrr                    * 1.1.0     2025-07-10 [1] CRAN (R 4.5.1)
 R.methodsS3              * 1.8.2     2022-06-13 [1] CRAN (R 4.5.0)
 R.oo                     * 1.27.1    2025-05-02 [1] CRAN (R 4.5.0)
 R.utils                  * 2.13.0    2025-02-24 [1] CRAN (R 4.5.0)
 reactable                * 0.4.4     2023-03-12 [1] CRAN (R 4.5.0)
 reactablefmtr            * 2.0.0     2022-03-16 [1] CRAN (R 4.5.0)
 readr                    * 2.1.5     2024-01-10 [1] CRAN (R 4.5.0)
 S4Vectors                * 0.46.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 seqinr                   * 4.2-36    2023-12-08 [1] CRAN (R 4.5.0)
 sessioninfo              * 1.2.3     2025-02-05 [1] CRAN (R 4.5.0)
 SingleCellExperiment     * 1.30.1    2025-05-05 [1] Bioconductor 3.21 (R 4.5.0)
 stringr                  * 1.5.2     2025-09-08 [1] CRAN (R 4.5.1)
 SummarizedExperiment     * 1.38.1    2025-04-28 [1] Bioconductor 3.21 (R 4.5.0)
 tibble                   * 3.3.0     2025-06-08 [1] CRAN (R 4.5.0)
 tidyr                    * 1.3.1     2024-01-24 [1] CRAN (R 4.5.0)
 tidyverse                * 2.0.0     2023-02-22 [1] CRAN (R 4.5.0)
 tinytable                * 0.13.0.10 2025-09-07 [1] https://vincentarelbundock.r-universe.dev (R 4.5.1)
 TreeSummarizedExperiment * 2.16.1    2025-05-08 [1] Bioconductor 3.21 (R 4.5.0)
 XVector                  * 0.48.0    2025-04-15 [1] Bioconductor 3.21 (R 4.5.0)
 zip                      * 2.3.3     2025-05-13 [1] CRAN (R 4.5.0)

 [1] /Library/Frameworks/R.framework/Versions/4.5-x86_64/Resources/library
 * ── Packages attached to the search path.</code></pre>
</div>
</div>
</details></div></section><section id="last-updated-on" class="level4 appendix"><h2 class="anchored quarto-appendix-heading">Last updated on</h2><div class="quarto-appendix-contents">
<p>2025-10-03</p>



<!-- -->

</div></section><section class="level4 appendix"><h2 class="quarto-appendix-heading">Collected Links</h2><div class="quarto-appendix-contents">
<ul>
<li><a href="../../../workflows/portal/data-med.html">MED Data Portal</a></li>
<li><a href="https://mothur.org/wiki/installation/">install mothur</a></li>
<li><a href="https://merenlab.org/2014/08/16/installing-the-oligotyping-pipeline/">Oligotyping/MED</a></li>
<li><a href="https://www.tidyverse.org">tidyverse</a></li>
<li><a href="https://merenlab.org/2014/11/04/med/">Minimum Entropy Decomposition (MED)</a></li>
<li><a href="../../../workflows/ssu/otu/#aligning-reads"><code>align.seqs</code></a></li>
<li><a href="https://forum.mothur.org/t/negative-control/2754">discussion on the mothur forum</a></li>
<li><a href="https://mothur.org/wiki/count_file/"><code>.count_table</code></a></li>
<li><a href="https://mothur.org/wiki/taxonomy_outline/">formatted versions of popular databases</a></li>
<li><a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database download</a></li>
<li><a href="https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz">GSR V4 database</a></li>
<li><a href="https://github.com/michberr/MicrobeMiseq"><code>mothur2oligo</code></a></li>
<li><a href="https://github.com/michberr">Michelle Berry</a></li>
<li><a href="../../../share/ssu/med/med_mapping.txt">sample mapping file</a></li>
<li><a href="../../../share/ssu/med/med_results/index.html">summary of the run</a></li>
<li><a href="https://joey711.github.io/phyloseq/">microeco</a></li>
<li><a href="https://manichanh.vhir.org/gsrdb/download_db_links2.php">GSR database</a></li>
<li><a href="https://github.com/ChiLiubio/file2meco">file2meco</a></li>
<li><a href="https://joey711.github.io/phyloseq/">phyloseq</a></li>
<li><a href="https://microbiome.github.io">miaverse</a></li>
<li><a href="https://quarto.org/">Quarto</a></li>
<li><a href="https://metacrobe.github.io/">metacrobe</a></li>
<li><a href="https://istmobiome.rbind.io/">The Istmobiome Project</a></li>
</ul></div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>
<ol>
<li id="fn1"><p>From the developers: GSR database (Greengenes, SILVA, and RDP database) is an integrated and manually curated database for bacterial and archaeal 16S amplicon taxonomy analysis. Unlike previous integration approaches, this database creation pipeline includes a taxonomy unification step to ensure consistency in taxonomical annotations. The database was validated with three mock communities and two real datasets and compared with existing 16S databases such as Greengenes, GTDB, ITGDB, SILVA, RDP, and MetaSquare. Results showed that the GSR database enhances taxonomical annotations of 16S sequences, outperforming current 16S databases at the species level. The GSR database is available for full-length 16S sequences and the most commonly used hypervariable regions: V4, V1-V3, V3-V4, and V3-V5.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol></section></div></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
      const outerScaffold = trigger.parentElement.cloneNode(true);
      const codeEl = outerScaffold.querySelector('code');
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp("https:\/\/istmobiome\.github\.io\/shrimp-draft\/");
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="../../../workflows/ssu/otu/index.html" class="pagination-link" aria-label="4. OTU Workflow">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">4. OTU Workflow</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
  </div>
</nav><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="code-copy-outer-scaffold"><div class="sourceCode" id="cb106" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb106-2"><a href="#cb106-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "5. MED Workflow"</span></span>
<span id="cb106-3"><a href="#cb106-3" aria-hidden="true" tabindex="-1"></a><span class="an">description:</span><span class="co"> |</span></span>
<span id="cb106-4"><a href="#cb106-4" aria-hidden="true" tabindex="-1"></a><span class="co">  Workflow for running MED analysis. Workflow begins with redundant, aligned fasta file from mothur and ends with the MED analysis. A Microtable Object is produced to collate the data for downstream analysis.</span></span>
<span id="cb106-5"><a href="#cb106-5" aria-hidden="true" tabindex="-1"></a><span class="an">listing:</span><span class="co"> </span></span>
<span id="cb106-6"><a href="#cb106-6" aria-hidden="true" tabindex="-1"></a><span class="co">    id: med-listing</span></span>
<span id="cb106-7"><a href="#cb106-7" aria-hidden="true" tabindex="-1"></a><span class="co">    contents: data-med.yml</span></span>
<span id="cb106-8"><a href="#cb106-8" aria-hidden="true" tabindex="-1"></a><span class="co">    type: table</span></span>
<span id="cb106-9"><a href="#cb106-9" aria-hidden="true" tabindex="-1"></a><span class="co">    sort-ui: false</span></span>
<span id="cb106-10"><a href="#cb106-10" aria-hidden="true" tabindex="-1"></a><span class="co">    filter-ui: false</span></span>
<span id="cb106-11"><a href="#cb106-11" aria-hidden="true" tabindex="-1"></a><span class="co">    fields: </span></span>
<span id="cb106-12"><a href="#cb106-12" aria-hidden="true" tabindex="-1"></a><span class="co">      - filename</span></span>
<span id="cb106-13"><a href="#cb106-13" aria-hidden="true" tabindex="-1"></a><span class="co">      - description</span></span>
<span id="cb106-14"><a href="#cb106-14" aria-hidden="true" tabindex="-1"></a><span class="co">    field-links: </span></span>
<span id="cb106-15"><a href="#cb106-15" aria-hidden="true" tabindex="-1"></a><span class="co">      - filename</span></span>
<span id="cb106-16"><a href="#cb106-16" aria-hidden="true" tabindex="-1"></a><span class="co">    field-display-names: </span></span>
<span id="cb106-17"><a href="#cb106-17" aria-hidden="true" tabindex="-1"></a><span class="co">      filename: File Name</span></span>
<span id="cb106-18"><a href="#cb106-18" aria-hidden="true" tabindex="-1"></a><span class="co">      description: Description</span></span>
<span id="cb106-19"><a href="#cb106-19" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb106-20"><a href="#cb106-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-21"><a href="#cb106-21" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_setup.qmd &gt;}}</span>
<span id="cb106-22"><a href="#cb106-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-25"><a href="#cb106-25" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-26"><a href="#cb106-26" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-27"><a href="#cb106-27" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-28"><a href="#cb106-28" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: set wd for here</span></span>
<span id="cb106-29"><a href="#cb106-29" aria-hidden="true" tabindex="-1"></a><span class="fu">remove</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb106-30"><a href="#cb106-30" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"med"</span> <span class="co"># e.g., med, dada2, etc</span></span>
<span id="cb106-31"><a href="#cb106-31" aria-hidden="true" tabindex="-1"></a>work_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"working_files/ssu/"</span>, workflow_name)</span>
<span id="cb106-32"><a href="#cb106-32" aria-hidden="true" tabindex="-1"></a>share_here <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"share/ssu/"</span>, workflow_name)</span>
<span id="cb106-33"><a href="#cb106-33" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"assets/scripts"</span>, <span class="st">"summarize_objs.R"</span>))</span>
<span id="cb106-34"><a href="#cb106-34" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"assets/scripts"</span>, <span class="st">"mia_metrics.R"</span>))</span>
<span id="cb106-35"><a href="#cb106-35" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-36"><a href="#cb106-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-37"><a href="#cb106-37" aria-hidden="true" tabindex="-1"></a><span class="fu">## Workflow Input</span></span>
<span id="cb106-38"><a href="#cb106-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-39"><a href="#cb106-39" aria-hidden="true" tabindex="-1"></a>::: {.callout-note icon=false}</span>
<span id="cb106-40"><a href="#cb106-40" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data &amp; Scripts</span></span>
<span id="cb106-41"><a href="#cb106-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-42"><a href="#cb106-42" aria-hidden="true" tabindex="-1"></a>Fastq sequence files, scripts, and other assets for running this workflow can be found on the <span class="co">[</span><span class="ot">MED Data Portal</span><span class="co">](/workflows/portal/data-med.qmd)</span> page. </span>
<span id="cb106-43"><a href="#cb106-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-44"><a href="#cb106-44" aria-hidden="true" tabindex="-1"></a>The Data Portal page also contains a link to the curated output of this pipeline–-feature-sample count table, taxonomy table, sample metadata, &amp; representative fasta sequences. Data is available as stand-alone text files, OR bundled in microtable and phyloseq objects. Archive also includes a table tracking sample read changes.</span>
<span id="cb106-45"><a href="#cb106-45" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-46"><a href="#cb106-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-47"><a href="#cb106-47" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Required Packages &amp; Software</span></span>
<span id="cb106-48"><a href="#cb106-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-49"><a href="#cb106-49" aria-hidden="true" tabindex="-1"></a>To run this workflow you need to  <span class="co">[</span><span class="ot">install mothur</span><span class="co">](https://mothur.org/wiki/installation/)</span> and the <span class="co">[</span><span class="ot">Oligotyping/MED</span><span class="co">](https://merenlab.org/2014/08/16/installing-the-oligotyping-pipeline/)</span>. You will need the <span class="co">[</span><span class="ot">tidyverse</span><span class="co">](https://www.tidyverse.org)</span> package. </span>
<span id="cb106-50"><a href="#cb106-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-51"><a href="#cb106-51" aria-hidden="true" tabindex="-1"></a><span class="fu">## Overview</span></span>
<span id="cb106-52"><a href="#cb106-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-53"><a href="#cb106-53" aria-hidden="true" tabindex="-1"></a>With the mothur pipeline finished, we can turn our attention to <span class="co">[</span><span class="ot">Minimum Entropy Decomposition (MED)</span><span class="co">](https://merenlab.org/2014/11/04/med/)</span> <span class="co">[</span><span class="ot">@eren2015minimum</span><span class="co">]</span>. MED is a novel, information theory-based clustering algorithm for sensitive partitioning of high-throughput marker gene sequences. </span>
<span id="cb106-54"><a href="#cb106-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-55"><a href="#cb106-55" aria-hidden="true" tabindex="-1"></a>(From the Meren Lab website) MED:</span>
<span id="cb106-56"><a href="#cb106-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-57"><a href="#cb106-57" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Does not perform pairwise sequence comparison,  </span>
<span id="cb106-58"><a href="#cb106-58" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Does not rely on arbitrary sequence similarity thresholds,  </span>
<span id="cb106-59"><a href="#cb106-59" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Does not require user supervision,  </span>
<span id="cb106-60"><a href="#cb106-60" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Does not require preliminary classification or clustering results,  </span>
<span id="cb106-61"><a href="#cb106-61" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Is agnostic to sampling strategy or how long your sequences are,  </span>
<span id="cb106-62"><a href="#cb106-62" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Gives you 1 nucleotide resolution over any sequencing length with computational efficiency and minimal computational heuristics.</span>
<span id="cb106-63"><a href="#cb106-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-64"><a href="#cb106-64" aria-hidden="true" tabindex="-1"></a>MED needs a redundant alignment file of read data. This means **all** identical sequences need to be included.  We again use mothur but this pipeline starts with the currated output of the <span class="co">[</span><span class="ot">`align.seqs`</span><span class="co">](/workflows/ssu/otu/#aligning-reads)</span> portion of our mothur OTU pipeline. </span>
<span id="cb106-65"><a href="#cb106-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-66"><a href="#cb106-66" aria-hidden="true" tabindex="-1"></a>We set up our run in the same way as the mothur pipeline. </span>
<span id="cb106-67"><a href="#cb106-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-68"><a href="#cb106-68" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_med_flowchart_1.qmd &gt;}}</span>
<span id="cb106-69"><a href="#cb106-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-70"><a href="#cb106-70" aria-hidden="true" tabindex="-1"></a><span class="fu"># Read Processing</span></span>
<span id="cb106-71"><a href="#cb106-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-72"><a href="#cb106-72" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_MED_Part1.qmd &gt;}}</span>
<span id="cb106-73"><a href="#cb106-73" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-74"><a href="#cb106-74" aria-hidden="true" tabindex="-1"></a><span class="fu"># MED Analysis</span></span>
<span id="cb106-75"><a href="#cb106-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-78"><a href="#cb106-78" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-79"><a href="#cb106-79" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-80"><a href="#cb106-80" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-81"><a href="#cb106-81" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false </span></span>
<span id="cb106-82"><a href="#cb106-82" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb106-83"><a href="#cb106-83" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"page_build"</span>, <span class="st">"med_part2.rdata"</span>))</span>
<span id="cb106-84"><a href="#cb106-84" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-85"><a href="#cb106-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-86"><a href="#cb106-86" aria-hidden="true" tabindex="-1"></a>::: {style="text-align: center;"}</span>
<span id="cb106-87"><a href="#cb106-87" aria-hidden="true" tabindex="-1"></a>{{&lt; include include/_med_flowchart_2.qmd &gt;}}</span>
<span id="cb106-88"><a href="#cb106-88" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-89"><a href="#cb106-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-90"><a href="#cb106-90" aria-hidden="true" tabindex="-1"></a>Now that we have the necessary files we are almost ready to run the MED pipeline. First though we need to properly format the mothur files. For this we will use a custom script called <span class="co">[</span><span class="ot">`mothur2oligo`</span><span class="co">](https://github.com/michberr/MicrobeMiseq)</span> written by <span class="co">[</span><span class="ot">Michelle Berry</span><span class="co">](https://github.com/michberr)</span>.</span>
<span id="cb106-91"><a href="#cb106-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-94"><a href="#cb106-94" aria-hidden="true" tabindex="-1"></a><span class="in">```{bash}</span></span>
<span id="cb106-95"><a href="#cb106-95" aria-hidden="true" tabindex="-1"></a><span class="fu">bash</span> mothur2oligo.sh</span>
<span id="cb106-96"><a href="#cb106-96" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-97"><a href="#cb106-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-98"><a href="#cb106-98" aria-hidden="true" tabindex="-1"></a>::: {.callout-note appearance="simple" collapse="true"}</span>
<span id="cb106-99"><a href="#cb106-99" aria-hidden="true" tabindex="-1"></a><span class="fu">### Expand to see mothur2oligo workflow</span></span>
<span id="cb106-100"><a href="#cb106-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-101"><a href="#cb106-101" aria-hidden="true" tabindex="-1"></a>These steps are run automatically by the <span class="in">`mothur2oligo`</span> script.</span>
<span id="cb106-102"><a href="#cb106-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-105"><a href="#cb106-105" aria-hidden="true" tabindex="-1"></a><span class="in">```{verbatim}</span></span>
<span id="cb106-106"><a href="#cb106-106" aria-hidden="true" tabindex="-1"></a><span class="in">get.lineage(taxonomy=final_med.taxonomy, taxon='Bacteria;-Archaea;', count=final_med.count_table)</span></span>
<span id="cb106-107"><a href="#cb106-107" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-108"><a href="#cb106-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-109"><a href="#cb106-109" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb106-110"><a href="#cb106-110" aria-hidden="true" tabindex="-1"></a><span class="in">/******************************************/</span></span>
<span id="cb106-111"><a href="#cb106-111" aria-hidden="true" tabindex="-1"></a><span class="in">Running command: get.seqs(accnos=final_med.pick.taxonomy.accnos, count=final_med.count_table)</span></span>
<span id="cb106-112"><a href="#cb106-112" aria-hidden="true" tabindex="-1"></a><span class="in">Selected 34628081 sequences from final_med.count_table.</span></span>
<span id="cb106-113"><a href="#cb106-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-114"><a href="#cb106-114" aria-hidden="true" tabindex="-1"></a><span class="in">Output File Names:</span></span>
<span id="cb106-115"><a href="#cb106-115" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.count_table</span></span>
<span id="cb106-116"><a href="#cb106-116" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-117"><a href="#cb106-117" aria-hidden="true" tabindex="-1"></a><span class="in">/******************************************/</span></span>
<span id="cb106-118"><a href="#cb106-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-119"><a href="#cb106-119" aria-hidden="true" tabindex="-1"></a><span class="in">Output File Names:</span></span>
<span id="cb106-120"><a href="#cb106-120" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.taxonomy</span></span>
<span id="cb106-121"><a href="#cb106-121" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.taxonomy.accnos</span></span>
<span id="cb106-122"><a href="#cb106-122" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.count_table</span></span>
<span id="cb106-123"><a href="#cb106-123" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-124"><a href="#cb106-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-127"><a href="#cb106-127" aria-hidden="true" tabindex="-1"></a><span class="in">```{verbatim}</span></span>
<span id="cb106-128"><a href="#cb106-128" aria-hidden="true" tabindex="-1"></a><span class="in">list.seqs(count=current)</span></span>
<span id="cb106-129"><a href="#cb106-129" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-130"><a href="#cb106-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-131"><a href="#cb106-131" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb106-132"><a href="#cb106-132" aria-hidden="true" tabindex="-1"></a><span class="in">Using final_med.pick.count_table as input file for the count parameter.</span></span>
<span id="cb106-133"><a href="#cb106-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-134"><a href="#cb106-134" aria-hidden="true" tabindex="-1"></a><span class="in">Output File Names: </span></span>
<span id="cb106-135"><a href="#cb106-135" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.accnos</span></span>
<span id="cb106-136"><a href="#cb106-136" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-137"><a href="#cb106-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-140"><a href="#cb106-140" aria-hidden="true" tabindex="-1"></a><span class="in">```{verbatim}</span></span>
<span id="cb106-141"><a href="#cb106-141" aria-hidden="true" tabindex="-1"></a><span class="in">get.seqs(accnos=current, fasta=final_med.fasta)</span></span>
<span id="cb106-142"><a href="#cb106-142" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-143"><a href="#cb106-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-144"><a href="#cb106-144" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb106-145"><a href="#cb106-145" aria-hidden="true" tabindex="-1"></a><span class="in">Using pipelineFiles_med/final_med.pick.accnos as input file for the accnos parameter.</span></span>
<span id="cb106-146"><a href="#cb106-146" aria-hidden="true" tabindex="-1"></a><span class="in">Selected 3433440 sequences from pipelineFiles_med/final_med.fasta.</span></span>
<span id="cb106-147"><a href="#cb106-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-148"><a href="#cb106-148" aria-hidden="true" tabindex="-1"></a><span class="in">Output File Names:</span></span>
<span id="cb106-149"><a href="#cb106-149" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.fasta</span></span>
<span id="cb106-150"><a href="#cb106-150" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-151"><a href="#cb106-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-154"><a href="#cb106-154" aria-hidden="true" tabindex="-1"></a><span class="in">```{verbatim}</span></span>
<span id="cb106-155"><a href="#cb106-155" aria-hidden="true" tabindex="-1"></a><span class="in">deunique.seqs(fasta=current, count=current)</span></span>
<span id="cb106-156"><a href="#cb106-156" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-157"><a href="#cb106-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-158"><a href="#cb106-158" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb106-159"><a href="#cb106-159" aria-hidden="true" tabindex="-1"></a><span class="in">Using final_med.pick.count_table as input file for the count parameter.</span></span>
<span id="cb106-160"><a href="#cb106-160" aria-hidden="true" tabindex="-1"></a><span class="in">Using final_med.pick.fasta as input file for the fasta parameter.</span></span>
<span id="cb106-161"><a href="#cb106-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-162"><a href="#cb106-162" aria-hidden="true" tabindex="-1"></a><span class="in">Output File Names: </span></span>
<span id="cb106-163"><a href="#cb106-163" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.redundant.fasta</span></span>
<span id="cb106-164"><a href="#cb106-164" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.redundant.groups</span></span>
<span id="cb106-165"><a href="#cb106-165" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-166"><a href="#cb106-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-167"><a href="#cb106-167" aria-hidden="true" tabindex="-1"></a>The eventual output of the <span class="in">`mothur2oligo`</span> workflow is a fasta file with reformatted headers. The script <span class="in">`renamer.pl`</span> adds sample names to fasta headers and appends the extension <span class="in">`_headers-replaced.fasta`</span> to a new file with these reformatted data. </span>
<span id="cb106-168"><a href="#cb106-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-169"><a href="#cb106-169" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-170"><a href="#cb106-170" aria-hidden="true" tabindex="-1"></a><span class="in">final_med.pick.redundant.fasta_headers-replaced.fasta</span></span>
<span id="cb106-171"><a href="#cb106-171" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-172"><a href="#cb106-172" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-173"><a href="#cb106-173" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-174"><a href="#cb106-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-177"><a href="#cb106-177" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-178"><a href="#cb106-178" aria-hidden="true" tabindex="-1"></a><span class="ex">o-trim-uninformative-columns-from-alignment</span> <span class="dt">\</span></span>
<span id="cb106-179"><a href="#cb106-179" aria-hidden="true" tabindex="-1"></a>        final_med.pick.redundant.renamed.fasta</span>
<span id="cb106-180"><a href="#cb106-180" aria-hidden="true" tabindex="-1"></a><span class="ex">decompose</span> final_med.pick.redundant.renamed.fasta-TRIMMED <span class="dt">\</span></span>
<span id="cb106-181"><a href="#cb106-181" aria-hidden="true" tabindex="-1"></a>        <span class="at">--sample-mapping</span> med_mapping.txt <span class="dt">\</span></span>
<span id="cb106-182"><a href="#cb106-182" aria-hidden="true" tabindex="-1"></a>        <span class="at">--output-directory</span> MED <span class="dt">\</span></span>
<span id="cb106-183"><a href="#cb106-183" aria-hidden="true" tabindex="-1"></a>        <span class="at">--number-of-threads</span> 24 <span class="dt">\</span></span>
<span id="cb106-184"><a href="#cb106-184" aria-hidden="true" tabindex="-1"></a>        <span class="at">--skip-gen-figures</span></span>
<span id="cb106-185"><a href="#cb106-185" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-186"><a href="#cb106-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-187"><a href="#cb106-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-188"><a href="#cb106-188" aria-hidden="true" tabindex="-1"></a>To run the <span class="in">`decompose`</span> command you will need this <span class="co">[</span><span class="ot">sample mapping file</span><span class="co">](/share/ssu/med/med_mapping.txt)</span>.</span>
<span id="cb106-189"><a href="#cb106-189" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-190"><a href="#cb106-190" aria-hidden="true" tabindex="-1"></a>::: {.callout-tip appearance="default"}</span>
<span id="cb106-191"><a href="#cb106-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-192"><a href="#cb106-192" aria-hidden="true" tabindex="-1"></a><span class="fu">## MED Results</span></span>
<span id="cb106-193"><a href="#cb106-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-194"><a href="#cb106-194" aria-hidden="true" tabindex="-1"></a>Once this is finished, the MED software will output a <span class="co">[</span><span class="ot">summary of the run</span><span class="co">](/share/ssu/med/med_results/index.html)</span>. </span>
<span id="cb106-195"><a href="#cb106-195" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-196"><a href="#cb106-196" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-197"><a href="#cb106-197" aria-hidden="true" tabindex="-1"></a><span class="fu"># Data Set Prep</span></span>
<span id="cb106-198"><a href="#cb106-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-199"><a href="#cb106-199" aria-hidden="true" tabindex="-1"></a>In this next part of the workflow our main goal is to create a *microtable object* using the R package <span class="co">[</span><span class="ot">microeco</span><span class="co">](https://joey711.github.io/phyloseq/)</span> <span class="co">[</span><span class="ot">@liu2021microeco</span><span class="co">]</span>. The microtable will be used to store the OTU by sample data as well the taxonomic, fasta, and sample data in a single object. More on that in a moment.</span>
<span id="cb106-200"><a href="#cb106-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-201"><a href="#cb106-201" aria-hidden="true" tabindex="-1"></a><span class="fu">## Assign Taxonomy</span></span>
<span id="cb106-202"><a href="#cb106-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-203"><a href="#cb106-203" aria-hidden="true" tabindex="-1"></a>Our first step is to classify the node representatives from the MED output. The <span class="in">`classify.seqs`</span> command requires properly formatted reference and taxonomy databases. For taxonomic assignment, we are using the GSR database <span class="co">[</span><span class="ot">@molano2024gsr</span><span class="co">]</span>. The developers of mothur maintain <span class="co">[</span><span class="ot">formatted versions of popular databases</span><span class="co">](https://mothur.org/wiki/taxonomy_outline/)</span>, however the GSR-DB has not been formatted by the developers yet.</span>
<span id="cb106-204"><a href="#cb106-204" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-205"><a href="#cb106-205" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb106-206"><a href="#cb106-206" aria-hidden="true" tabindex="-1"></a>Here you can download an appropriate version of the <span class="co">[</span><span class="ot">GSR database</span><span class="co">](https://manichanh.vhir.org/gsrdb/download_db_links2.php)</span>.</span>
<span id="cb106-207"><a href="#cb106-207" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-208"><a href="#cb106-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-209"><a href="#cb106-209" aria-hidden="true" tabindex="-1"></a>To create a mothur formatted version GSR-DB, we perform the following steps (we went through this process above prior to removing contaminants but will repeat here for posterity). </span>
<span id="cb106-210"><a href="#cb106-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-211"><a href="#cb106-211" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Download a data base</span></span>
<span id="cb106-212"><a href="#cb106-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-213"><a href="#cb106-213" aria-hidden="true" tabindex="-1"></a>Here we are using the <span class="co">[</span><span class="ot">GSR V4 database</span><span class="co">](https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz)</span>.</span>
<span id="cb106-214"><a href="#cb106-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-217"><a href="#cb106-217" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-218"><a href="#cb106-218" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-219"><a href="#cb106-219" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-220"><a href="#cb106-220" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://manichanh.vhir.org/gsrdb/GSR-DB_V4_cluster-1.tar.gz</span>
<span id="cb106-221"><a href="#cb106-221" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> GSR-DB_V4_cluster-1.tar.gz</span>
<span id="cb106-222"><a href="#cb106-222" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-223"><a href="#cb106-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-224"><a href="#cb106-224" aria-hidden="true" tabindex="-1"></a>First (in the command line) we remove first line of the taxonomy file.</span>
<span id="cb106-225"><a href="#cb106-225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-228"><a href="#cb106-228" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-229"><a href="#cb106-229" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_taxa.txt tmp0.txt</span>
<span id="cb106-230"><a href="#cb106-230" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="st">'1d'</span> tmp0.txt <span class="op">&gt;</span> tmp1.txt</span>
<span id="cb106-231"><a href="#cb106-231" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-232"><a href="#cb106-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-233"><a href="#cb106-233" aria-hidden="true" tabindex="-1"></a>Next, delete species and remove leading <span class="sc">\[</span>a-z<span class="sc">\]\_\_</span> from taxa names</span>
<span id="cb106-234"><a href="#cb106-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-237"><a href="#cb106-237" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-238"><a href="#cb106-238" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/s__.*//g'</span> tmp1.txt <span class="op">&gt;</span> tmp2.txt</span>
<span id="cb106-239"><a href="#cb106-239" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-E</span> <span class="st">'s/[a-zA-Z]__//g'</span> tmp2.txt <span class="op">&gt;</span> gsrdb.tax</span>
<span id="cb106-240"><a href="#cb106-240" aria-hidden="true" tabindex="-1"></a><span class="fu">cp</span> GSR-DB_V4_cluster-1_seqs.fasta gsrdb.fasta</span>
<span id="cb106-241"><a href="#cb106-241" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-242"><a href="#cb106-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-243"><a href="#cb106-243" aria-hidden="true" tabindex="-1"></a>The next thing we need to do is grab the <span class="in">`node-representatives.fa.txt`</span> from the MED output so that we can classify these sequences. Of course, proper formatting is required.</span>
<span id="cb106-244"><a href="#cb106-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-247"><a href="#cb106-247" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-248"><a href="#cb106-248" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">--pattern</span> <span class="pp">^</span> <span class="at">--replacement</span> MED node-representatives.fa.txt <span class="op">&gt;</span> tmp1.fa</span>
<span id="cb106-249"><a href="#cb106-249" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">--pattern</span> <span class="st">"\|.*"</span> <span class="at">--replacement</span> <span class="st">''</span> tmp1.fa <span class="op">&gt;</span> med_nodes.fasta</span>
<span id="cb106-250"><a href="#cb106-250" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp1.fa</span>
<span id="cb106-251"><a href="#cb106-251" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-252"><a href="#cb106-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-253"><a href="#cb106-253" aria-hidden="true" tabindex="-1"></a>Great, the reference database is formatted. Now we need to make a few files that mimics the normal mothur output files because the MED pipeline does not exactly generate the files we need to create a <span class="in">`microtable`</span> object. First we use the <span class="in">`matrix_counts.txt`</span> file from the MED analysis to create a mothur-styled <span class="in">`count.table`</span>.</span>
<span id="cb106-254"><a href="#cb106-254" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-257"><a href="#cb106-257" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-258"><a href="#cb106-258" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-259"><a href="#cb106-259" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-260"><a href="#cb106-260" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(</span>
<span id="cb106-261"><a href="#cb106-261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">here</span>(work_here, <span class="st">"med_results/matrix_counts.txt"</span>),</span>
<span id="cb106-262"><a href="#cb106-262" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-263"><a href="#cb106-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-264"><a href="#cb106-264" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span> </span>
<span id="cb106-265"><a href="#cb106-265" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename_with</span>( <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"MED"</span>, .x)) </span>
<span id="cb106-266"><a href="#cb106-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-267"><a href="#cb106-267" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span></span>
<span id="cb106-268"><a href="#cb106-268" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>), <span class="at">names_to =</span> <span class="st">"tmp"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-269"><a href="#cb106-269" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="fu">c</span>(<span class="dv">1</span>))</span>
<span id="cb106-270"><a href="#cb106-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-271"><a href="#cb106-271" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(tmp_med_counts, <span class="st">"tmp"</span>)</span>
<span id="cb106-272"><a href="#cb106-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-273"><a href="#cb106-273" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span></span>
<span id="cb106-274"><a href="#cb106-274" aria-hidden="true" tabindex="-1"></a>                  <span class="fu">mutate</span>(<span class="at">total =</span> <span class="fu">rowSums</span>(.), <span class="at">.before =</span> <span class="dv">1</span>)</span>
<span id="cb106-275"><a href="#cb106-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-276"><a href="#cb106-276" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span> </span>
<span id="cb106-277"><a href="#cb106-277" aria-hidden="true" tabindex="-1"></a>     tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Representative_Sequence"</span>)</span>
<span id="cb106-278"><a href="#cb106-278" aria-hidden="true" tabindex="-1"></a>med_counts <span class="ot">&lt;-</span> tmp_med_counts</span>
<span id="cb106-279"><a href="#cb106-279" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-280"><a href="#cb106-280" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-283"><a href="#cb106-283" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-284"><a href="#cb106-284" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-285"><a href="#cb106-285" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-286"><a href="#cb106-286" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(med_counts, <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.count.table"</span>), </span>
<span id="cb106-287"><a href="#cb106-287" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-288"><a href="#cb106-288" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-289"><a href="#cb106-289" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-290"><a href="#cb106-290" aria-hidden="true" tabindex="-1"></a>Now we can actually classify the representative sequences. </span>
<span id="cb106-291"><a href="#cb106-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-294"><a href="#cb106-294" aria-hidden="true" tabindex="-1"></a><span class="in">```{verbatim}</span></span>
<span id="cb106-295"><a href="#cb106-295" aria-hidden="true" tabindex="-1"></a><span class="in">classify.seqs(fasta=med_nodes.fasta, count=med_nodes.count.table, reference=reference_dbs/gsrdb.fasta, taxonomy=reference_dbs/gsrdb.tax)</span></span>
<span id="cb106-296"><a href="#cb106-296" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-297"><a href="#cb106-297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-298"><a href="#cb106-298" aria-hidden="true" tabindex="-1"></a>Now we make a mothur styled <span class="in">`shared`</span> file. </span>
<span id="cb106-299"><a href="#cb106-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-302"><a href="#cb106-302" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-303"><a href="#cb106-303" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-304"><a href="#cb106-304" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-305"><a href="#cb106-305" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(</span>
<span id="cb106-306"><a href="#cb106-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">here</span>(work_here, <span class="st">"med_results/matrix_counts.txt"</span>),</span>
<span id="cb106-307"><a href="#cb106-307" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-308"><a href="#cb106-308" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-309"><a href="#cb106-309" aria-hidden="true" tabindex="-1"></a>tmp_n_meds <span class="ot">&lt;-</span> <span class="fu">ncol</span>(tmp_med_counts) <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb106-310"><a href="#cb106-310" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span> </span>
<span id="cb106-311"><a href="#cb106-311" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename_with</span>( <span class="sc">~</span> <span class="fu">paste0</span>(<span class="st">"MED"</span>, .x)) <span class="sc">%&gt;%</span> </span>
<span id="cb106-312"><a href="#cb106-312" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">"Group"</span> <span class="ot">=</span> <span class="st">"MEDsamples"</span>)</span>
<span id="cb106-313"><a href="#cb106-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-314"><a href="#cb106-314" aria-hidden="true" tabindex="-1"></a>tmp_med_counts <span class="ot">&lt;-</span> tmp_med_counts <span class="sc">%&gt;%</span> </span>
<span id="cb106-315"><a href="#cb106-315" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">add_column</span>(<span class="at">label =</span> <span class="fl">0.03</span>, <span class="at">.before =</span> <span class="st">"Group"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb106-316"><a href="#cb106-316" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">add_column</span>(<span class="at">numOtus =</span> tmp_n_meds, <span class="at">.after =</span> <span class="st">"Group"</span>)</span>
<span id="cb106-317"><a href="#cb106-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-318"><a href="#cb106-318" aria-hidden="true" tabindex="-1"></a>med_shared <span class="ot">&lt;-</span> tmp_med_counts</span>
<span id="cb106-319"><a href="#cb106-319" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-320"><a href="#cb106-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-323"><a href="#cb106-323" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-324"><a href="#cb106-324" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-325"><a href="#cb106-325" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-326"><a href="#cb106-326" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(med_shared, <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.shared"</span>),</span>
<span id="cb106-327"><a href="#cb106-327" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-328"><a href="#cb106-328" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-329"><a href="#cb106-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-332"><a href="#cb106-332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-333"><a href="#cb106-333" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-334"><a href="#cb106-334" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-335"><a href="#cb106-335" aria-hidden="true" tabindex="-1"></a>tmp_med_tax <span class="ot">&lt;-</span> <span class="fu">read_tsv</span>(</span>
<span id="cb106-336"><a href="#cb106-336" aria-hidden="true" tabindex="-1"></a>    <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.gsrdb.wang.taxonomy"</span>),</span>
<span id="cb106-337"><a href="#cb106-337" aria-hidden="true" tabindex="-1"></a>    <span class="at">col_names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-338"><a href="#cb106-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-339"><a href="#cb106-339" aria-hidden="true" tabindex="-1"></a>tmp_1 <span class="ot">&lt;-</span> med_counts[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb106-340"><a href="#cb106-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-341"><a href="#cb106-341" aria-hidden="true" tabindex="-1"></a>med_cons_tax <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(</span>
<span id="cb106-342"><a href="#cb106-342" aria-hidden="true" tabindex="-1"></a>  tmp_1, tmp_med_tax, </span>
<span id="cb106-343"><a href="#cb106-343" aria-hidden="true" tabindex="-1"></a>  <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Representative_Sequence"</span> <span class="ot">=</span> <span class="st">"X1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb106-344"><a href="#cb106-344" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="fu">c</span>(<span class="st">"OTU"</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"Size"</span> <span class="ot">=</span> <span class="dv">2</span>, <span class="st">"Taxonomy"</span> <span class="ot">=</span> <span class="dv">3</span>))  </span>
<span id="cb106-345"><a href="#cb106-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-346"><a href="#cb106-346" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(med_cons_tax,</span>
<span id="cb106-347"><a href="#cb106-347" aria-hidden="true" tabindex="-1"></a>            <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.cons.taxonomy"</span>),</span>
<span id="cb106-348"><a href="#cb106-348" aria-hidden="true" tabindex="-1"></a>            <span class="at">row.names =</span> <span class="cn">FALSE</span>, <span class="at">quote =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-349"><a href="#cb106-349" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-350"><a href="#cb106-350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-351"><a href="#cb106-351" aria-hidden="true" tabindex="-1"></a><span class="fu">### A. Taxonomy Table</span></span>
<span id="cb106-352"><a href="#cb106-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-353"><a href="#cb106-353" aria-hidden="true" tabindex="-1"></a>Here is what the taxonomy table looks like in the microeco mock data.</span>
<span id="cb106-354"><a href="#cb106-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-357"><a href="#cb106-357" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-358"><a href="#cb106-358" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-359"><a href="#cb106-359" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-360"><a href="#cb106-360" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(taxonomy_table_16S[<span class="dv">10</span><span class="sc">:</span><span class="dv">12</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-361"><a href="#cb106-361" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-362"><a href="#cb106-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-363"><a href="#cb106-363" aria-hidden="true" tabindex="-1"></a>Our taxonomy file (below) needs a little wrangling to be properly formatted. </span>
<span id="cb106-364"><a href="#cb106-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-367"><a href="#cb106-367" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-368"><a href="#cb106-368" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-369"><a href="#cb106-369" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-370"><a href="#cb106-370" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">read_delim</span>(</span>
<span id="cb106-371"><a href="#cb106-371" aria-hidden="true" tabindex="-1"></a>             <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.cons.taxonomy"</span>),</span>
<span id="cb106-372"><a href="#cb106-372" aria-hidden="true" tabindex="-1"></a>             <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-373"><a href="#cb106-373" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-374"><a href="#cb106-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-377"><a href="#cb106-377" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-378"><a href="#cb106-378" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-379"><a href="#cb106-379" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-380"><a href="#cb106-380" aria-hidden="true" tabindex="-1"></a><span class="co">#tax1 &lt;- tmp_tax</span></span>
<span id="cb106-381"><a href="#cb106-381" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(tax1[<span class="dv">11</span><span class="sc">:</span><span class="dv">13</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>, <span class="at">split.tables =</span> <span class="cn">Inf</span>)</span>
<span id="cb106-382"><a href="#cb106-382" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-383"><a href="#cb106-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-384"><a href="#cb106-384" aria-hidden="true" tabindex="-1"></a>Some fancy string manipulation...</span>
<span id="cb106-385"><a href="#cb106-385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-388"><a href="#cb106-388" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-389"><a href="#cb106-389" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sapply</span>(tmp_tax, </span>
<span id="cb106-390"><a href="#cb106-390" aria-hidden="true" tabindex="-1"></a>                             gsub, </span>
<span id="cb106-391"><a href="#cb106-391" aria-hidden="true" tabindex="-1"></a>                             <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">(</span><span class="sc">\\</span><span class="st">d+</span><span class="sc">\\</span><span class="st">)"</span>, </span>
<span id="cb106-392"><a href="#cb106-392" aria-hidden="true" tabindex="-1"></a>                             <span class="at">replacement =</span> <span class="st">""</span>))</span>
<span id="cb106-393"><a href="#cb106-393" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sapply</span>(tmp_tax, </span>
<span id="cb106-394"><a href="#cb106-394" aria-hidden="true" tabindex="-1"></a>                             gsub, </span>
<span id="cb106-395"><a href="#cb106-395" aria-hidden="true" tabindex="-1"></a>                             <span class="at">pattern =</span> <span class="st">";$"</span>, </span>
<span id="cb106-396"><a href="#cb106-396" aria-hidden="true" tabindex="-1"></a>                             <span class="at">replacement =</span> <span class="st">""</span>))</span>
<span id="cb106-397"><a href="#cb106-397" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">separate_wider_delim</span>(tmp_tax, </span>
<span id="cb106-398"><a href="#cb106-398" aria-hidden="true" tabindex="-1"></a>                              <span class="at">cols =</span> Taxonomy, </span>
<span id="cb106-399"><a href="#cb106-399" aria-hidden="true" tabindex="-1"></a>                              <span class="at">delim =</span> <span class="st">";"</span>, <span class="at">names =</span> <span class="fu">c</span>(</span>
<span id="cb106-400"><a href="#cb106-400" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, </span>
<span id="cb106-401"><a href="#cb106-401" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Class"</span>, <span class="st">"Order"</span>, </span>
<span id="cb106-402"><a href="#cb106-402" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"Family"</span>, <span class="st">"Genus"</span>))</span>
<span id="cb106-403"><a href="#cb106-403" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">sapply</span>(tmp_tax, gsub, </span>
<span id="cb106-404"><a href="#cb106-404" aria-hidden="true" tabindex="-1"></a>                           <span class="at">pattern =</span> <span class="st">"^.*_unclassified$"</span>, </span>
<span id="cb106-405"><a href="#cb106-405" aria-hidden="true" tabindex="-1"></a>                           <span class="at">replacement =</span> <span class="st">""</span>))</span>
<span id="cb106-406"><a href="#cb106-406" aria-hidden="true" tabindex="-1"></a>tmp_tax<span class="sc">$</span>Size <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-407"><a href="#cb106-407" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(tmp_tax, <span class="st">"OTU"</span>)</span>
<span id="cb106-408"><a href="#cb106-408" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-409"><a href="#cb106-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-410"><a href="#cb106-410" aria-hidden="true" tabindex="-1"></a>And we get this ...</span>
<span id="cb106-411"><a href="#cb106-411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-414"><a href="#cb106-414" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-415"><a href="#cb106-415" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-416"><a href="#cb106-416" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-417"><a href="#cb106-417" aria-hidden="true" tabindex="-1"></a><span class="co">#tax2 &lt;- tmp_tax</span></span>
<span id="cb106-418"><a href="#cb106-418" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(tax2[<span class="dv">11</span><span class="sc">:</span><span class="dv">14</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>, <span class="at">split.tables =</span> <span class="cn">Inf</span>)</span>
<span id="cb106-419"><a href="#cb106-419" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-420"><a href="#cb106-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-423"><a href="#cb106-423" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-424"><a href="#cb106-424" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-425"><a href="#cb106-425" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-426"><a href="#cb106-426" aria-hidden="true" tabindex="-1"></a>rank_prefixes <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb106-427"><a href="#cb106-427" aria-hidden="true" tabindex="-1"></a>  <span class="at">Kingdom =</span> <span class="st">"k"</span>, </span>
<span id="cb106-428"><a href="#cb106-428" aria-hidden="true" tabindex="-1"></a>  <span class="at">Phylum  =</span> <span class="st">"p"</span>, </span>
<span id="cb106-429"><a href="#cb106-429" aria-hidden="true" tabindex="-1"></a>  <span class="at">Class   =</span> <span class="st">"c"</span>, </span>
<span id="cb106-430"><a href="#cb106-430" aria-hidden="true" tabindex="-1"></a>  <span class="at">Order   =</span> <span class="st">"o"</span>, </span>
<span id="cb106-431"><a href="#cb106-431" aria-hidden="true" tabindex="-1"></a>  <span class="at">Family  =</span> <span class="st">"f"</span>, </span>
<span id="cb106-432"><a href="#cb106-432" aria-hidden="true" tabindex="-1"></a>  <span class="at">Genus   =</span> <span class="st">"g"</span></span>
<span id="cb106-433"><a href="#cb106-433" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-434"><a href="#cb106-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-435"><a href="#cb106-435" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> tmp_tax <span class="sc">%&gt;%</span></span>
<span id="cb106-436"><a href="#cb106-436" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span><span class="fu">replace_na</span>(., <span class="st">""</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb106-437"><a href="#cb106-437" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">names</span>(rank_prefixes), </span>
<span id="cb106-438"><a href="#cb106-438" aria-hidden="true" tabindex="-1"></a>                <span class="sc">~</span> <span class="fu">paste0</span>(rank_prefixes[<span class="fu">cur_column</span>()], <span class="st">"__"</span>, .))) <span class="sc">%&gt;%</span></span>
<span id="cb106-439"><a href="#cb106-439" aria-hidden="true" tabindex="-1"></a><span class="fu">tidy_taxonomy</span>()</span>
<span id="cb106-440"><a href="#cb106-440" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-441"><a href="#cb106-441" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-442"><a href="#cb106-442" aria-hidden="true" tabindex="-1"></a>And then this. Exactly like the mock data. </span>
<span id="cb106-443"><a href="#cb106-443" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-446"><a href="#cb106-446" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-447"><a href="#cb106-447" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-448"><a href="#cb106-448" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-449"><a href="#cb106-449" aria-hidden="true" tabindex="-1"></a><span class="co">#tax3 &lt;- tmp_tax</span></span>
<span id="cb106-450"><a href="#cb106-450" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(tax3[<span class="dv">11</span><span class="sc">:</span><span class="dv">14</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>, <span class="at">split.tables =</span> <span class="cn">Inf</span>)</span>
<span id="cb106-451"><a href="#cb106-451" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-452"><a href="#cb106-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-453"><a href="#cb106-453" aria-hidden="true" tabindex="-1"></a><span class="fu">### B. Sequence Table</span></span>
<span id="cb106-454"><a href="#cb106-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-455"><a href="#cb106-455" aria-hidden="true" tabindex="-1"></a>Here is what the sequence table looks like in the mock data.</span>
<span id="cb106-456"><a href="#cb106-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-459"><a href="#cb106-459" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-460"><a href="#cb106-460" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-461"><a href="#cb106-461" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-462"><a href="#cb106-462" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(otu_table_16S[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">11</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-463"><a href="#cb106-463" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-464"><a href="#cb106-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-465"><a href="#cb106-465" aria-hidden="true" tabindex="-1"></a>These code block will return a properly formatted sequence table. </span>
<span id="cb106-466"><a href="#cb106-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-469"><a href="#cb106-469" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-470"><a href="#cb106-470" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-471"><a href="#cb106-471" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-472"><a href="#cb106-472" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(</span>
<span id="cb106-473"><a href="#cb106-473" aria-hidden="true" tabindex="-1"></a>  <span class="fu">here</span>(work_here, <span class="st">"node_taxonomy/med_nodes.shared"</span>),</span>
<span id="cb106-474"><a href="#cb106-474" aria-hidden="true" tabindex="-1"></a>  <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-475"><a href="#cb106-475" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-476"><a href="#cb106-476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-479"><a href="#cb106-479" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-480"><a href="#cb106-480" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-481"><a href="#cb106-481" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-482"><a href="#cb106-482" aria-hidden="true" tabindex="-1"></a>tmp_st<span class="sc">$</span>numOtus <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-483"><a href="#cb106-483" aria-hidden="true" tabindex="-1"></a>tmp_st<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb106-484"><a href="#cb106-484" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> tmp_st <span class="sc">%&gt;%</span></span>
<span id="cb106-485"><a href="#cb106-485" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>), <span class="at">names_to =</span> <span class="st">"tmp"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-486"><a href="#cb106-486" aria-hidden="true" tabindex="-1"></a>  tidyr<span class="sc">::</span><span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="fu">c</span>(<span class="dv">1</span>))</span>
<span id="cb106-487"><a href="#cb106-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-488"><a href="#cb106-488" aria-hidden="true" tabindex="-1"></a>tmp_st <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(tmp_st, <span class="st">"tmp"</span>)</span>
<span id="cb106-489"><a href="#cb106-489" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-490"><a href="#cb106-490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-493"><a href="#cb106-493" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-494"><a href="#cb106-494" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-495"><a href="#cb106-495" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-496"><a href="#cb106-496" aria-hidden="true" tabindex="-1"></a><span class="co"># st &lt;- tmp_st</span></span>
<span id="cb106-497"><a href="#cb106-497" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(st[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-498"><a href="#cb106-498" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-499"><a href="#cb106-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-500"><a href="#cb106-500" aria-hidden="true" tabindex="-1"></a><span class="fu">### C. Sample Table</span></span>
<span id="cb106-501"><a href="#cb106-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-502"><a href="#cb106-502" aria-hidden="true" tabindex="-1"></a>Here is what the sample table looks like in the mock data.</span>
<span id="cb106-503"><a href="#cb106-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-506"><a href="#cb106-506" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-507"><a href="#cb106-507" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-508"><a href="#cb106-508" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-509"><a href="#cb106-509" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(sample_info_16S[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>, <span class="at">split.tables =</span> <span class="cn">Inf</span>)</span>
<span id="cb106-510"><a href="#cb106-510" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-511"><a href="#cb106-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-512"><a href="#cb106-512" aria-hidden="true" tabindex="-1"></a>No problem. </span>
<span id="cb106-513"><a href="#cb106-513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-516"><a href="#cb106-516" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-517"><a href="#cb106-517" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-518"><a href="#cb106-518" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-519"><a href="#cb106-519" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb106-520"><a href="#cb106-520" aria-hidden="true" tabindex="-1"></a>  <span class="fu">here</span>(<span class="st">"working_files/ssu/sampledata"</span>, <span class="st">"sample_data.txt"</span>),</span>
<span id="cb106-521"><a href="#cb106-521" aria-hidden="true" tabindex="-1"></a>  <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-522"><a href="#cb106-522" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-523"><a href="#cb106-523" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> tibble<span class="sc">::</span><span class="fu">column_to_rownames</span>(<span class="st">"SampleID"</span>)</span>
<span id="cb106-524"><a href="#cb106-524" aria-hidden="true" tabindex="-1"></a>samdf<span class="sc">$</span>SampleID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(samdf)</span>
<span id="cb106-525"><a href="#cb106-525" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span> <span class="fu">relocate</span>(SampleID)</span>
<span id="cb106-526"><a href="#cb106-526" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-527"><a href="#cb106-527" aria-hidden="true" tabindex="-1"></a>samdf <span class="ot">&lt;-</span> samdf <span class="sc">%&gt;%</span></span>
<span id="cb106-528"><a href="#cb106-528" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(</span>
<span id="cb106-529"><a href="#cb106-529" aria-hidden="true" tabindex="-1"></a>    stringr<span class="sc">::</span><span class="fu">str_starts</span>(SampleID, <span class="st">"Control"</span>, <span class="at">negate =</span> <span class="cn">TRUE</span>))</span>
<span id="cb106-530"><a href="#cb106-530" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-531"><a href="#cb106-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-534"><a href="#cb106-534" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-535"><a href="#cb106-535" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-536"><a href="#cb106-536" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-537"><a href="#cb106-537" aria-hidden="true" tabindex="-1"></a><span class="fu">pandoc.table</span>(samdf[<span class="dv">11</span><span class="sc">:</span><span class="dv">13</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>], <span class="at">emphasize.rownames =</span> <span class="cn">FALSE</span>, <span class="at">split.tables =</span> <span class="cn">Inf</span>)</span>
<span id="cb106-538"><a href="#cb106-538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-539"><a href="#cb106-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-540"><a href="#cb106-540" aria-hidden="true" tabindex="-1"></a><span class="fu">## Experiment-level Objects</span></span>
<span id="cb106-541"><a href="#cb106-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-542"><a href="#cb106-542" aria-hidden="true" tabindex="-1"></a>In the following section we create <span class="in">`microtable`</span> and <span class="in">`phyloseq`</span> objects. These are single, consolidated experiment-level data objects that organize various data types, such as OTU tables (taxa abundance), sample metadata, taxonomic classifications, as well as phylogenetic trees and reference sequences, into one structured format. This unified approach facilitates reproducible, interactive analysis and visualization by bringing all related information together. </span>
<span id="cb106-543"><a href="#cb106-543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-544"><a href="#cb106-544" aria-hidden="true" tabindex="-1"></a>We begin by creating the <span class="in">`microtable`</span> and then use the function <span class="in">`meco2phyloseq`</span> from the <span class="co">[</span><span class="ot">file2meco</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span> package to create the <span class="in">`phyloseq`</span> object. This way all of the underlying data is identical across the two objects. </span>
<span id="cb106-545"><a href="#cb106-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-546"><a href="#cb106-546" aria-hidden="true" tabindex="-1"></a>::: callout-note</span>
<span id="cb106-547"><a href="#cb106-547" aria-hidden="true" tabindex="-1"></a>These objects contain an OTU table (taxa abundances), sample metadata, and taxonomy table (mapping between OTUs and higher-level taxonomic classifications).</span>
<span id="cb106-548"><a href="#cb106-548" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-549"><a href="#cb106-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-552"><a href="#cb106-552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-553"><a href="#cb106-553" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-554"><a href="#cb106-554" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-555"><a href="#cb106-555" aria-hidden="true" tabindex="-1"></a>sample_info <span class="ot">&lt;-</span> samdf</span>
<span id="cb106-556"><a href="#cb106-556" aria-hidden="true" tabindex="-1"></a>tax_tab <span class="ot">&lt;-</span> tmp_tax</span>
<span id="cb106-557"><a href="#cb106-557" aria-hidden="true" tabindex="-1"></a>otu_tab <span class="ot">&lt;-</span> tmp_st</span>
<span id="cb106-558"><a href="#cb106-558" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-559"><a href="#cb106-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-562"><a href="#cb106-562" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-563"><a href="#cb106-563" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-564"><a href="#cb106-564" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-565"><a href="#cb106-565" aria-hidden="true" tabindex="-1"></a>tmp_me <span class="ot">&lt;-</span> microtable<span class="sc">$</span><span class="fu">new</span>(<span class="at">sample_table =</span> sample_info, </span>
<span id="cb106-566"><a href="#cb106-566" aria-hidden="true" tabindex="-1"></a>                         <span class="at">otu_table =</span> otu_tab, </span>
<span id="cb106-567"><a href="#cb106-567" aria-hidden="true" tabindex="-1"></a>                         <span class="at">tax_table =</span> tax_tab)</span>
<span id="cb106-568"><a href="#cb106-568" aria-hidden="true" tabindex="-1"></a>tmp_me</span>
<span id="cb106-569"><a href="#cb106-569" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-570"><a href="#cb106-570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-571"><a href="#cb106-571" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb106-572"><a href="#cb106-572" aria-hidden="true" tabindex="-1"></a><span class="in">microtable-class object:</span></span>
<span id="cb106-573"><a href="#cb106-573" aria-hidden="true" tabindex="-1"></a><span class="in">sample_table have 1849 rows and 14 columns</span></span>
<span id="cb106-574"><a href="#cb106-574" aria-hidden="true" tabindex="-1"></a><span class="in">otu_table have 436 rows and 1849 columns</span></span>
<span id="cb106-575"><a href="#cb106-575" aria-hidden="true" tabindex="-1"></a><span class="in">tax_table have 436 rows and 6 columns</span></span>
<span id="cb106-576"><a href="#cb106-576" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-577"><a href="#cb106-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-578"><a href="#cb106-578" aria-hidden="true" tabindex="-1"></a><span class="fu">### Add Representative Sequence</span></span>
<span id="cb106-579"><a href="#cb106-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-580"><a href="#cb106-580" aria-hidden="true" tabindex="-1"></a>The fasta file returned by MED needs a little T.L.C. For that we use a tool called SeqKit <span class="co">[</span><span class="ot">@shen2024seqkit2</span><span class="co">]</span> for fasta defline manipulation.</span>
<span id="cb106-581"><a href="#cb106-581" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-584"><a href="#cb106-584" aria-hidden="true" tabindex="-1"></a><span class="in">```{zsh}</span></span>
<span id="cb106-585"><a href="#cb106-585" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="pp">^</span> <span class="at">-r</span> MED node-representatives.fa.txt <span class="op">&gt;</span> tmp1.fa</span>
<span id="cb106-586"><a href="#cb106-586" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"\|.*"</span> <span class="at">-r</span> <span class="st">''</span> tmp1.fa <span class="op">&gt;</span> tmp2.fa </span>
<span id="cb106-587"><a href="#cb106-587" aria-hidden="true" tabindex="-1"></a><span class="ex">seqkit</span> replace <span class="at">-p</span> <span class="st">"-"</span> <span class="at">-r</span> <span class="st">'$1'</span> <span class="at">-s</span> <span class="at">-w</span> 0 tmp2.fa <span class="op">&gt;</span> med_rep.fasta</span>
<span id="cb106-588"><a href="#cb106-588" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp1.fa</span>
<span id="cb106-589"><a href="#cb106-589" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> tmp2.fa</span>
<span id="cb106-590"><a href="#cb106-590" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-591"><a href="#cb106-591" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-594"><a href="#cb106-594" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-595"><a href="#cb106-595" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-596"><a href="#cb106-596" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-597"><a href="#cb106-597" aria-hidden="true" tabindex="-1"></a>rep_fasta <span class="ot">&lt;-</span> Biostrings<span class="sc">::</span><span class="fu">readDNAStringSet</span>(<span class="fu">here</span>(work_here, <span class="st">"med_rep.fasta"</span>))</span>
<span id="cb106-598"><a href="#cb106-598" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-599"><a href="#cb106-599" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-602"><a href="#cb106-602" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-603"><a href="#cb106-603" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-604"><a href="#cb106-604" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-605"><a href="#cb106-605" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span>rep_fasta <span class="ot">&lt;-</span> rep_fasta</span>
<span id="cb106-606"><a href="#cb106-606" aria-hidden="true" tabindex="-1"></a>tmp_me<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb106-607"><a href="#cb106-607" aria-hidden="true" tabindex="-1"></a>tmp_me</span>
<span id="cb106-608"><a href="#cb106-608" aria-hidden="true" tabindex="-1"></a>me_raw <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_me)</span>
<span id="cb106-609"><a href="#cb106-609" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-610"><a href="#cb106-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-611"><a href="#cb106-611" aria-hidden="true" tabindex="-1"></a><span class="fu">## Curate the Data Set</span></span>
<span id="cb106-612"><a href="#cb106-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-613"><a href="#cb106-613" aria-hidden="true" tabindex="-1"></a>Pretty much the last thing to do is remove low-count samples.</span>
<span id="cb106-614"><a href="#cb106-614" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-615"><a href="#cb106-615" aria-hidden="true" tabindex="-1"></a><span class="fu">### Remove Low-Count Samples</span></span>
<span id="cb106-616"><a href="#cb106-616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-619"><a href="#cb106-619" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-620"><a href="#cb106-620" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-621"><a href="#cb106-621" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-622"><a href="#cb106-622" aria-hidden="true" tabindex="-1"></a>threshold <span class="ot">&lt;-</span> <span class="dv">1000</span></span>
<span id="cb106-623"><a href="#cb106-623" aria-hidden="true" tabindex="-1"></a>tmp_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(me_raw)</span>
<span id="cb106-624"><a href="#cb106-624" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span>otu_table <span class="ot">&lt;-</span> me_raw<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span></span>
<span id="cb106-625"><a href="#cb106-625" aria-hidden="true" tabindex="-1"></a>          dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">where</span>(<span class="sc">~</span> <span class="fu">is.numeric</span>(.) <span class="sc">&amp;&amp;</span> <span class="fu">sum</span>(.) <span class="sc">&gt;=</span> threshold))</span>
<span id="cb106-626"><a href="#cb106-626" aria-hidden="true" tabindex="-1"></a>tmp_no_low<span class="sc">$</span><span class="fu">tidy_dataset</span>()</span>
<span id="cb106-627"><a href="#cb106-627" aria-hidden="true" tabindex="-1"></a>tmp_no_low</span>
<span id="cb106-628"><a href="#cb106-628" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-629"><a href="#cb106-629" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-632"><a href="#cb106-632" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-633"><a href="#cb106-633" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-634"><a href="#cb106-634" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-635"><a href="#cb106-635" aria-hidden="true" tabindex="-1"></a>me_no_low <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_low)</span>
<span id="cb106-636"><a href="#cb106-636" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-637"><a href="#cb106-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-640"><a href="#cb106-640" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-641"><a href="#cb106-641" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-642"><a href="#cb106-642" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-643"><a href="#cb106-643" aria-hidden="true" tabindex="-1"></a>me_no_low</span>
<span id="cb106-644"><a href="#cb106-644" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-645"><a href="#cb106-645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-648"><a href="#cb106-648" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-649"><a href="#cb106-649" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-650"><a href="#cb106-650" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-651"><a href="#cb106-651" aria-hidden="true" tabindex="-1"></a>me_final <span class="ot">&lt;-</span> microeco<span class="sc">::</span><span class="fu">clone</span>(tmp_no_low)</span>
<span id="cb106-652"><a href="#cb106-652" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-653"><a href="#cb106-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-654"><a href="#cb106-654" aria-hidden="true" tabindex="-1"></a>Lastly, we can use the package <span class="co">[</span><span class="ot">`file2meco`</span><span class="co">](https://github.com/ChiLiubio/file2meco)</span> to generate a <span class="co">[</span><span class="ot">phyloseq</span><span class="co">](https://joey711.github.io/phyloseq/)</span> object. </span>
<span id="cb106-655"><a href="#cb106-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-658"><a href="#cb106-658" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-659"><a href="#cb106-659" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb106-660"><a href="#cb106-660" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-661"><a href="#cb106-661" aria-hidden="true" tabindex="-1"></a>ps_final <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb106-662"><a href="#cb106-662" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-663"><a href="#cb106-663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-664"><a href="#cb106-664" aria-hidden="true" tabindex="-1"></a><span class="fu">## Summary</span></span>
<span id="cb106-665"><a href="#cb106-665" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-666"><a href="#cb106-666" aria-hidden="true" tabindex="-1"></a>Now time to summarize the data. For this we use the R package <span class="co">[</span><span class="ot">miaverse</span><span class="co">](https://microbiome.github.io)</span> <span class="co">[</span><span class="ot">@felix2024mia</span><span class="co">]</span>.</span>
<span id="cb106-667"><a href="#cb106-667" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-670"><a href="#cb106-670" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-671"><a href="#cb106-671" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-672"><a href="#cb106-672" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-673"><a href="#cb106-673" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_raw<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_raw<span class="sc">$</span>otu_table))</span>
<span id="cb106-674"><a href="#cb106-674" aria-hidden="true" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">rownames</span>(me_final<span class="sc">$</span>sample_table), <span class="fu">colnames</span>(me_final<span class="sc">$</span>otu_table))</span>
<span id="cb106-675"><a href="#cb106-675" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-676"><a href="#cb106-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-679"><a href="#cb106-679" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-680"><a href="#cb106-680" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-681"><a href="#cb106-681" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-682"><a href="#cb106-682" aria-hidden="true" tabindex="-1"></a>objs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"me_raw"</span>, <span class="st">"me_final"</span>)</span>
<span id="cb106-683"><a href="#cb106-683" aria-hidden="true" tabindex="-1"></a>pipe_summary <span class="ot">&lt;-</span> <span class="fu">summarize_objs</span>(objs)</span>
<span id="cb106-684"><a href="#cb106-684" aria-hidden="true" tabindex="-1"></a>pipe_summary<span class="sc">$</span>me_dataset <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb106-685"><a href="#cb106-685" aria-hidden="true" tabindex="-1"></a>  <span class="st">"original"</span>, <span class="st">"no low count samps"</span>)</span>
<span id="cb106-686"><a href="#cb106-686" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(pipe_summary)</span>
<span id="cb106-687"><a href="#cb106-687" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-688"><a href="#cb106-688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-691"><a href="#cb106-691" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-692"><a href="#cb106-692" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-693"><a href="#cb106-693" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-694"><a href="#cb106-694" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pipe_summary)</span>
<span id="cb106-695"><a href="#cb106-695" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-696"><a href="#cb106-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-699"><a href="#cb106-699" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-700"><a href="#cb106-700" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-701"><a href="#cb106-701" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-702"><a href="#cb106-702" aria-hidden="true" tabindex="-1"></a>tmp_raw_ps <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_raw)</span>
<span id="cb106-703"><a href="#cb106-703" aria-hidden="true" tabindex="-1"></a>tmp_final_ps <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb106-704"><a href="#cb106-704" aria-hidden="true" tabindex="-1"></a>tmp_mia_raw <span class="ot">&lt;-</span> <span class="fu">mia_metrics</span>(tmp_raw_ps)</span>
<span id="cb106-705"><a href="#cb106-705" aria-hidden="true" tabindex="-1"></a>tmp_mia_final <span class="ot">&lt;-</span> <span class="fu">mia_metrics</span>(tmp_final_ps)</span>
<span id="cb106-706"><a href="#cb106-706" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-707"><a href="#cb106-707" aria-hidden="true" tabindex="-1"></a>tmp_metrics_final <span class="ot">&lt;-</span> <span class="fu">rbind</span>(tmp_mia_raw, tmp_mia_final)</span>
<span id="cb106-708"><a href="#cb106-708" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(tmp_metrics_final) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Start"</span>, <span class="st">"End"</span>)</span>
<span id="cb106-709"><a href="#cb106-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-710"><a href="#cb106-710" aria-hidden="true" tabindex="-1"></a><span class="co"># suppose ds_metrics_final has rownames "Start" and "End"</span></span>
<span id="cb106-711"><a href="#cb106-711" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl <span class="ot">&lt;-</span> tmp_metrics_final <span class="sc">%&gt;%</span></span>
<span id="cb106-712"><a href="#cb106-712" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span>                         <span class="co"># transpose: metrics in rows</span></span>
<span id="cb106-713"><a href="#cb106-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-714"><a href="#cb106-714" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"Metric"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-715"><a href="#cb106-715" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Start =</span> Start, <span class="at">End =</span> End)</span>
<span id="cb106-716"><a href="#cb106-716" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-717"><a href="#cb106-717" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl <span class="ot">&lt;-</span> ds_metrics_tbl <span class="sc">%&gt;%</span></span>
<span id="cb106-718"><a href="#cb106-718" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb106-719"><a href="#cb106-719" aria-hidden="true" tabindex="-1"></a>    <span class="at">Start =</span> <span class="fu">ifelse</span>(Start <span class="sc">%%</span> <span class="dv">1</span> <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb106-720"><a href="#cb106-720" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(Start, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">0</span>),</span>
<span id="cb106-721"><a href="#cb106-721" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(Start, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">3</span>)),</span>
<span id="cb106-722"><a href="#cb106-722" aria-hidden="true" tabindex="-1"></a>    <span class="at">End   =</span> <span class="fu">ifelse</span>(End <span class="sc">%%</span> <span class="dv">1</span> <span class="sc">==</span> <span class="dv">0</span>,</span>
<span id="cb106-723"><a href="#cb106-723" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(End, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">0</span>),</span>
<span id="cb106-724"><a href="#cb106-724" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">formatC</span>(End, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">3</span>))</span>
<span id="cb106-725"><a href="#cb106-725" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb106-726"><a href="#cb106-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-727"><a href="#cb106-727" aria-hidden="true" tabindex="-1"></a>ds_metrics_tbl<span class="sc">$</span>Metric <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb106-728"><a href="#cb106-728" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Min. no. of reads"</span>,</span>
<span id="cb106-729"><a href="#cb106-729" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Max. no. of reads"</span>,</span>
<span id="cb106-730"><a href="#cb106-730" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Total no. of reads"</span>,</span>
<span id="cb106-731"><a href="#cb106-731" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Avg. no. of reads"</span>,</span>
<span id="cb106-732"><a href="#cb106-732" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Median no. of reads"</span>,</span>
<span id="cb106-733"><a href="#cb106-733" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Total ASVs"</span>,</span>
<span id="cb106-734"><a href="#cb106-734" aria-hidden="true" tabindex="-1"></a>  <span class="st">"No. of singleton ASVs"</span>,</span>
<span id="cb106-735"><a href="#cb106-735" aria-hidden="true" tabindex="-1"></a>  <span class="st">"% of singleton ASVs"</span>,</span>
<span id="cb106-736"><a href="#cb106-736" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Sparsity"</span></span>
<span id="cb106-737"><a href="#cb106-737" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-738"><a href="#cb106-738" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-739"><a href="#cb106-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-742"><a href="#cb106-742" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-743"><a href="#cb106-743" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-744"><a href="#cb106-744" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-745"><a href="#cb106-745" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Dataset metrics before and after curation."</span></span>
<span id="cb106-746"><a href="#cb106-746" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(ds_metrics_tbl, <span class="at">format =</span> <span class="st">"markdown"</span>)</span>
<span id="cb106-747"><a href="#cb106-747" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-748"><a href="#cb106-748" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-749"><a href="#cb106-749" aria-hidden="true" tabindex="-1"></a>We started off with <span class="in">`r nrow(me_raw$tax_table)`</span> ASVs and <span class="in">`r nrow(me_raw$sample_table)`</span> samples. After removing low-count samples, there were <span class="in">`r nrow(me_final$tax_table)`</span> ASVs and <span class="in">`r nrow(me_final$sample_table)`</span> samples remaining.</span>
<span id="cb106-750"><a href="#cb106-750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-753"><a href="#cb106-753" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-754"><a href="#cb106-754" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-755"><a href="#cb106-755" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-756"><a href="#cb106-756" aria-hidden="true" tabindex="-1"></a><span class="co"># helper functions</span></span>
<span id="cb106-757"><a href="#cb106-757" aria-hidden="true" tabindex="-1"></a>make_tables <span class="ot">&lt;-</span> <span class="cf">function</span>(obj, prefix) {</span>
<span id="cb106-758"><a href="#cb106-758" aria-hidden="true" tabindex="-1"></a>  rc <span class="ot">&lt;-</span> obj<span class="sc">$</span><span class="fu">sample_sums</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-759"><a href="#cb106-759" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-760"><a href="#cb106-760" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-761"><a href="#cb106-761" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="sc">!!</span><span class="fu">paste0</span>(prefix, <span class="st">"_rc"</span>) <span class="sc">:</span><span class="er">=</span> <span class="dv">2</span>)</span>
<span id="cb106-762"><a href="#cb106-762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-763"><a href="#cb106-763" aria-hidden="true" tabindex="-1"></a>  asv <span class="ot">&lt;-</span> obj<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span></span>
<span id="cb106-764"><a href="#cb106-764" aria-hidden="true" tabindex="-1"></a>    <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-765"><a href="#cb106-765" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-766"><a href="#cb106-766" aria-hidden="true" tabindex="-1"></a>    { <span class="fu">rowSums</span>(. <span class="sc">&gt;</span> <span class="dv">0</span>) } <span class="sc">%&gt;%</span></span>
<span id="cb106-767"><a href="#cb106-767" aria-hidden="true" tabindex="-1"></a>    <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb106-768"><a href="#cb106-768" aria-hidden="true" tabindex="-1"></a>    tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-769"><a href="#cb106-769" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="sc">!!</span><span class="fu">paste0</span>(prefix, <span class="st">"_med"</span>) <span class="sc">:</span><span class="er">=</span> <span class="dv">2</span>)</span>
<span id="cb106-770"><a href="#cb106-770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-771"><a href="#cb106-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(<span class="at">rc =</span> rc, <span class="at">asv =</span> asv)</span>
<span id="cb106-772"><a href="#cb106-772" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb106-773"><a href="#cb106-773" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-774"><a href="#cb106-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-777"><a href="#cb106-777" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-778"><a href="#cb106-778" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-779"><a href="#cb106-779" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-780"><a href="#cb106-780" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_start &lt;- microeco::clone(me_raw)</span></span>
<span id="cb106-781"><a href="#cb106-781" aria-hidden="true" tabindex="-1"></a><span class="co">#tmp_final &lt;- microeco::clone(me_final)</span></span>
<span id="cb106-782"><a href="#cb106-782" aria-hidden="true" tabindex="-1"></a><span class="co"># --- input rc file ---</span></span>
<span id="cb106-783"><a href="#cb106-783" aria-hidden="true" tabindex="-1"></a>tmp_rc <span class="ot">&lt;-</span> readr<span class="sc">::</span><span class="fu">read_delim</span>(<span class="fu">here</span>(work_here, <span class="st">"mothur_med_pipeline_read_changes.txt"</span>)) </span>
<span id="cb106-784"><a href="#cb106-784" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-785"><a href="#cb106-785" aria-hidden="true" tabindex="-1"></a><span class="co"># --- generate tables ---</span></span>
<span id="cb106-786"><a href="#cb106-786" aria-hidden="true" tabindex="-1"></a>start_tbls <span class="ot">&lt;-</span> <span class="fu">make_tables</span>(microeco<span class="sc">::</span><span class="fu">clone</span>(me_raw), <span class="st">"start"</span>)</span>
<span id="cb106-787"><a href="#cb106-787" aria-hidden="true" tabindex="-1"></a>final_tbls <span class="ot">&lt;-</span> <span class="fu">make_tables</span>(microeco<span class="sc">::</span><span class="fu">clone</span>(me_final), <span class="st">"final"</span>)</span>
<span id="cb106-788"><a href="#cb106-788" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-789"><a href="#cb106-789" aria-hidden="true" tabindex="-1"></a><span class="co"># --- summary ---</span></span>
<span id="cb106-790"><a href="#cb106-790" aria-hidden="true" tabindex="-1"></a>curate_summary <span class="ot">&lt;-</span> tmp_rc <span class="sc">%&gt;%</span></span>
<span id="cb106-791"><a href="#cb106-791" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">full_join</span>(start_tbls<span class="sc">$</span>rc,  <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-792"><a href="#cb106-792" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(start_tbls<span class="sc">$</span>asv, <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-793"><a href="#cb106-793" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(final_tbls<span class="sc">$</span>rc,  <span class="at">by =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb106-794"><a href="#cb106-794" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">left_join</span>(final_tbls<span class="sc">$</span>asv, <span class="at">by =</span> <span class="st">"SampleID"</span>)</span>
<span id="cb106-795"><a href="#cb106-795" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-796"><a href="#cb106-796" aria-hidden="true" tabindex="-1"></a>removed_samples <span class="ot">&lt;-</span> curate_summary <span class="sc">%&gt;%</span></span>
<span id="cb106-797"><a href="#cb106-797" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="fu">is.na</span>(final_rc)) <span class="sc">%&gt;%</span></span>
<span id="cb106-798"><a href="#cb106-798" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">pull</span>(SampleID)</span>
<span id="cb106-799"><a href="#cb106-799" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-800"><a href="#cb106-800" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-801"><a href="#cb106-801" aria-hidden="true" tabindex="-1"></a>We lost a total of <span class="in">`r length(removed_samples)`</span> samples after curating the dataset. This includes <span class="in">`r length(removed_samples[grepl("^Control_", removed_samples)])`</span> NC samples and <span class="in">`r length(removed_samples[!grepl("^Control_", removed_samples)])`</span> non-NC samples.</span>
<span id="cb106-802"><a href="#cb106-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-803"><a href="#cb106-803" aria-hidden="true" tabindex="-1"></a>Here is a list of non-NC samples that were removed. </span>
<span id="cb106-804"><a href="#cb106-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-807"><a href="#cb106-807" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-808"><a href="#cb106-808" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb106-809"><a href="#cb106-809" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-810"><a href="#cb106-810" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-811"><a href="#cb106-811" aria-hidden="true" tabindex="-1"></a>removed_samples[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"^Control_"</span>, removed_samples)]</span>
<span id="cb106-812"><a href="#cb106-812" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-813"><a href="#cb106-813" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-814"><a href="#cb106-814" aria-hidden="true" tabindex="-1"></a><span class="fu"># Download Results {#download-results}</span></span>
<span id="cb106-815"><a href="#cb106-815" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-816"><a href="#cb106-816" aria-hidden="true" tabindex="-1"></a>Quick access to read changes through the pipeline and repseqs detected in negative control samples.</span>
<span id="cb106-817"><a href="#cb106-817" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-818"><a href="#cb106-818" aria-hidden="true" tabindex="-1"></a>::: {#med-listing .column-body}</span>
<span id="cb106-819"><a href="#cb106-819" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-820"><a href="#cb106-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-821"><a href="#cb106-821" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!------------------------------------------------------------------------&gt;</span></span>
<span id="cb106-822"><a href="#cb106-822" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-------------------- Use this area to save things ----------------------&gt;</span></span>
<span id="cb106-823"><a href="#cb106-823" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!------------------------------------------------------------------------&gt;</span></span>
<span id="cb106-824"><a href="#cb106-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-825"><a href="#cb106-825" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!---------------------------------------&gt;</span></span>
<span id="cb106-826"><a href="#cb106-826" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- These chunks are for curated data --&gt;</span></span>
<span id="cb106-827"><a href="#cb106-827" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!---------------------------------------&gt;</span></span>
<span id="cb106-830"><a href="#cb106-830" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-831"><a href="#cb106-831" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-832"><a href="#cb106-832" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-833"><a href="#cb106-833" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"med"</span></span>
<span id="cb106-834"><a href="#cb106-834" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data"</span>)), </span>
<span id="cb106-835"><a href="#cb106-835" aria-hidden="true" tabindex="-1"></a>           <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-836"><a href="#cb106-836" aria-hidden="true" tabindex="-1"></a>           <span class="at">showWarnings =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-837"><a href="#cb106-837" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-838"><a href="#cb106-838" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-841"><a href="#cb106-841" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-842"><a href="#cb106-842" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-843"><a href="#cb106-843" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-844"><a href="#cb106-844" aria-hidden="true" tabindex="-1"></a>tmp_path <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data/"</span>))</span>
<span id="cb106-845"><a href="#cb106-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-846"><a href="#cb106-846" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb106-847"><a href="#cb106-847" aria-hidden="true" tabindex="-1"></a>  <span class="at">otu =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_otu_table.txt"</span>),</span>
<span id="cb106-848"><a href="#cb106-848" aria-hidden="true" tabindex="-1"></a>  <span class="at">tax =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_tax_table.txt"</span>),</span>
<span id="cb106-849"><a href="#cb106-849" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_sample_table.txt"</span>),</span>
<span id="cb106-850"><a href="#cb106-850" aria-hidden="true" tabindex="-1"></a>  <span class="at">rep =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_rep.fasta"</span>),</span>
<span id="cb106-851"><a href="#cb106-851" aria-hidden="true" tabindex="-1"></a>  <span class="at">me =</span> <span class="fu">paste0</span>(<span class="st">"me_"</span>, workflow_name, <span class="st">".rds"</span>),</span>
<span id="cb106-852"><a href="#cb106-852" aria-hidden="true" tabindex="-1"></a>  <span class="at">ps =</span> <span class="fu">paste0</span>(<span class="st">"ps_"</span>, workflow_name, <span class="st">".rds"</span>),</span>
<span id="cb106-853"><a href="#cb106-853" aria-hidden="true" tabindex="-1"></a>  <span class="at">counts =</span> <span class="fu">paste0</span>(workflow_name, <span class="st">"_track_read_counts.txt"</span>)</span>
<span id="cb106-854"><a href="#cb106-854" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-855"><a href="#cb106-855" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-856"><a href="#cb106-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-859"><a href="#cb106-859" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-860"><a href="#cb106-860" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-861"><a href="#cb106-861" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-862"><a href="#cb106-862" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------OTU Table--------------#</span></span>
<span id="cb106-863"><a href="#cb106-863" aria-hidden="true" tabindex="-1"></a>tmp_otu <span class="ot">&lt;-</span> me_final<span class="sc">$</span>otu_table <span class="sc">%&gt;%</span> </span>
<span id="cb106-864"><a href="#cb106-864" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="fu">paste0</span>(<span class="fu">toupper</span>(workflow_name), <span class="st">"_ID"</span>))</span>
<span id="cb106-865"><a href="#cb106-865" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_otu, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>otu), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-866"><a href="#cb106-866" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------TAX Table--------------#</span></span>
<span id="cb106-867"><a href="#cb106-867" aria-hidden="true" tabindex="-1"></a>tmp_tax <span class="ot">&lt;-</span> me_final<span class="sc">$</span>tax_table <span class="sc">%&gt;%</span> </span>
<span id="cb106-868"><a href="#cb106-868" aria-hidden="true" tabindex="-1"></a>  tibble<span class="sc">::</span><span class="fu">rownames_to_column</span>(<span class="fu">paste0</span>(<span class="fu">toupper</span>(workflow_name), <span class="st">"_ID"</span>))</span>
<span id="cb106-869"><a href="#cb106-869" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(tmp_tax, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>tax), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-870"><a href="#cb106-870" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------SAMP Table--------------#</span></span>
<span id="cb106-871"><a href="#cb106-871" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(me_final<span class="sc">$</span>sample_table, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>sample), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-872"><a href="#cb106-872" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------REP FASTA--------------#</span></span>
<span id="cb106-873"><a href="#cb106-873" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(</span>
<span id="cb106-874"><a href="#cb106-874" aria-hidden="true" tabindex="-1"></a>  <span class="at">sequences =</span> <span class="fu">as.list</span>(<span class="fu">as.character</span>(me_final<span class="sc">$</span>rep_fasta)),</span>
<span id="cb106-875"><a href="#cb106-875" aria-hidden="true" tabindex="-1"></a>  <span class="at">names =</span> <span class="fu">names</span>(me_final<span class="sc">$</span>rep_fasta),</span>
<span id="cb106-876"><a href="#cb106-876" aria-hidden="true" tabindex="-1"></a>  <span class="at">file.out =</span> <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>rep)</span>
<span id="cb106-877"><a href="#cb106-877" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-878"><a href="#cb106-878" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------        --------------#</span></span>
<span id="cb106-879"><a href="#cb106-879" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(me_final, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>me))</span>
<span id="cb106-880"><a href="#cb106-880" aria-hidden="true" tabindex="-1"></a>ps_final <span class="ot">&lt;-</span> file2meco<span class="sc">::</span><span class="fu">meco2phyloseq</span>(me_final)</span>
<span id="cb106-881"><a href="#cb106-881" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(ps_final, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>ps))</span>
<span id="cb106-882"><a href="#cb106-882" aria-hidden="true" tabindex="-1"></a><span class="co">#---------------read count changes--------------#</span></span>
<span id="cb106-883"><a href="#cb106-883" aria-hidden="true" tabindex="-1"></a><span class="fu">write_delim</span>(curate_summary, <span class="fu">paste0</span>(tmp_path, files<span class="sc">$</span>counts), <span class="at">delim =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb106-884"><a href="#cb106-884" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(<span class="st">"working_files/ssu/sampledata"</span>,  <span class="st">"all_metadata.txt"</span>), </span>
<span id="cb106-885"><a href="#cb106-885" aria-hidden="true" tabindex="-1"></a>          tmp_path, </span>
<span id="cb106-886"><a href="#cb106-886" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-887"><a href="#cb106-887" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-888"><a href="#cb106-888" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-889"><a href="#cb106-889" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-892"><a href="#cb106-892" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-893"><a href="#cb106-893" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-894"><a href="#cb106-894" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-895"><a href="#cb106-895" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_curated_data.zip"</span>)), </span>
<span id="cb106-896"><a href="#cb106-896" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_curated_data"</span>)), </span>
<span id="cb106-897"><a href="#cb106-897" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb106-898"><a href="#cb106-898" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-899"><a href="#cb106-899" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-902"><a href="#cb106-902" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-903"><a href="#cb106-903" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-904"><a href="#cb106-904" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-905"><a href="#cb106-905" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(work_here,  <span class="st">"mothur_med_pipeline_read_changes.txt"</span>), </span>
<span id="cb106-906"><a href="#cb106-906" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(share_here), </span>
<span id="cb106-907"><a href="#cb106-907" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-908"><a href="#cb106-908" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-909"><a href="#cb106-909" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-910"><a href="#cb106-910" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-913"><a href="#cb106-913" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-914"><a href="#cb106-914" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-915"><a href="#cb106-915" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-916"><a href="#cb106-916" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="st">"med_results/"</span>), </span>
<span id="cb106-917"><a href="#cb106-917" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(share_here,  <span class="st">"med_results/"</span>), </span>
<span id="cb106-918"><a href="#cb106-918" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-919"><a href="#cb106-919" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="st">"med_results.zip"</span>), </span>
<span id="cb106-920"><a href="#cb106-920" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="st">"med_results"</span>), </span>
<span id="cb106-921"><a href="#cb106-921" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb106-922"><a href="#cb106-922" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-923"><a href="#cb106-923" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-924"><a href="#cb106-924" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!--------------------------------------------&gt;</span></span>
<span id="cb106-925"><a href="#cb106-925" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!-- These chunks are for  processing data  --&gt;</span></span>
<span id="cb106-926"><a href="#cb106-926" aria-hidden="true" tabindex="-1"></a><span class="co">&lt;!--------------------------------------------&gt;</span></span>
<span id="cb106-927"><a href="#cb106-927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-930"><a href="#cb106-930" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-931"><a href="#cb106-931" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-932"><a href="#cb106-932" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-933"><a href="#cb106-933" aria-hidden="true" tabindex="-1"></a><span class="fu">dir.create</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing"</span>)), </span>
<span id="cb106-934"><a href="#cb106-934" aria-hidden="true" tabindex="-1"></a>           <span class="at">recursive =</span> <span class="cn">FALSE</span>, <span class="at">showWarnings =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-935"><a href="#cb106-935" aria-hidden="true" tabindex="-1"></a>copy_here <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing/"</span>))</span>
<span id="cb106-936"><a href="#cb106-936" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-937"><a href="#cb106-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-940"><a href="#cb106-940" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-941"><a href="#cb106-941" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-942"><a href="#cb106-942" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-943"><a href="#cb106-943" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: from sampledata (these data are common to all 16S wf)</span></span>
<span id="cb106-944"><a href="#cb106-944" aria-hidden="true" tabindex="-1"></a>tmp_sampdata_path <span class="ot">&lt;-</span> <span class="st">"working_files/ssu/sampledata"</span></span>
<span id="cb106-945"><a href="#cb106-945" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"fastq_rename_results/"</span>), </span>
<span id="cb106-946"><a href="#cb106-946" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"fastq_rename_results/"</span>), </span>
<span id="cb106-947"><a href="#cb106-947" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-948"><a href="#cb106-948" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-949"><a href="#cb106-949" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"fastq_rename_lookup/"</span>), </span>
<span id="cb106-950"><a href="#cb106-950" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"fastq_rename_lookup/"</span>), </span>
<span id="cb106-951"><a href="#cb106-951" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-952"><a href="#cb106-952" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"rename.sh"</span>), </span>
<span id="cb106-953"><a href="#cb106-953" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb106-954"><a href="#cb106-954" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-955"><a href="#cb106-955" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-956"><a href="#cb106-956" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-957"><a href="#cb106-957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-958"><a href="#cb106-958" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-961"><a href="#cb106-961" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-962"><a href="#cb106-962" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-963"><a href="#cb106-963" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-964"><a href="#cb106-964" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="st">"mothur2oligo/"</span>), </span>
<span id="cb106-965"><a href="#cb106-965" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="st">"mothur2oligo/"</span>), </span>
<span id="cb106-966"><a href="#cb106-966" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-967"><a href="#cb106-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-968"><a href="#cb106-968" aria-hidden="true" tabindex="-1"></a>fs<span class="sc">::</span><span class="fu">dir_copy</span>(<span class="at">path =</span> <span class="fu">here</span>(work_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_hydra_scripts/"</span>)), </span>
<span id="cb106-969"><a href="#cb106-969" aria-hidden="true" tabindex="-1"></a>             <span class="at">new_path =</span> <span class="fu">here</span>(copy_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_hydra_scripts/"</span>)), </span>
<span id="cb106-970"><a href="#cb106-970" aria-hidden="true" tabindex="-1"></a>             <span class="at">overwrite =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-971"><a href="#cb106-971" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-972"><a href="#cb106-972" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(work_here,  <span class="fu">paste0</span>(workflow_name, <span class="st">"_batchfile_processing/"</span>)), </span>
<span id="cb106-973"><a href="#cb106-973" aria-hidden="true" tabindex="-1"></a>          copy_here, </span>
<span id="cb106-974"><a href="#cb106-974" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-975"><a href="#cb106-975" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-976"><a href="#cb106-976" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-977"><a href="#cb106-977" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(work_here,  <span class="st">"med_mapping.txt"</span>), </span>
<span id="cb106-978"><a href="#cb106-978" aria-hidden="true" tabindex="-1"></a>          copy_here, </span>
<span id="cb106-979"><a href="#cb106-979" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-980"><a href="#cb106-980" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-981"><a href="#cb106-981" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-982"><a href="#cb106-982" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="st">"med_results.zip"</span>), </span>
<span id="cb106-983"><a href="#cb106-983" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="st">"med_results"</span>), </span>
<span id="cb106-984"><a href="#cb106-984" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb106-985"><a href="#cb106-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-986"><a href="#cb106-986" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(share_here,  <span class="st">"mothur_med_pipeline_read_changes.txt"</span>), </span>
<span id="cb106-987"><a href="#cb106-987" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb106-988"><a href="#cb106-988" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-989"><a href="#cb106-989" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-990"><a href="#cb106-990" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-991"><a href="#cb106-991" aria-hidden="true" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="fu">here</span>(tmp_sampdata_path,  <span class="st">"all_metadata.txt"</span>), </span>
<span id="cb106-992"><a href="#cb106-992" aria-hidden="true" tabindex="-1"></a>          <span class="fu">here</span>(copy_here), </span>
<span id="cb106-993"><a href="#cb106-993" aria-hidden="true" tabindex="-1"></a>          <span class="at">overwrite =</span> <span class="cn">TRUE</span>, <span class="at">recursive =</span> <span class="cn">FALSE</span>, </span>
<span id="cb106-994"><a href="#cb106-994" aria-hidden="true" tabindex="-1"></a>          <span class="at">copy.mode =</span> <span class="cn">TRUE</span>, <span class="at">copy.date =</span> <span class="cn">FALSE</span>)</span>
<span id="cb106-995"><a href="#cb106-995" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-996"><a href="#cb106-996" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-999"><a href="#cb106-999" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-1000"><a href="#cb106-1000" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-1001"><a href="#cb106-1001" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-1002"><a href="#cb106-1002" aria-hidden="true" tabindex="-1"></a>zip<span class="sc">::</span><span class="fu">zip</span>(<span class="at">zipfile =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_processing.zip"</span>)), </span>
<span id="cb106-1003"><a href="#cb106-1003" aria-hidden="true" tabindex="-1"></a>         <span class="at">files =</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_processing"</span>)), </span>
<span id="cb106-1004"><a href="#cb106-1004" aria-hidden="true" tabindex="-1"></a>         <span class="at">mode =</span> <span class="st">"cherry-pick"</span>)</span>
<span id="cb106-1005"><a href="#cb106-1005" aria-hidden="true" tabindex="-1"></a><span class="fu">dir_delete</span>(<span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name,<span class="st">"_processing"</span>)))</span>
<span id="cb106-1006"><a href="#cb106-1006" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-1007"><a href="#cb106-1007" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-1010"><a href="#cb106-1010" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-1011"><a href="#cb106-1011" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-1012"><a href="#cb106-1012" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: true</span></span>
<span id="cb106-1013"><a href="#cb106-1013" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb106-1014"><a href="#cb106-1014" aria-hidden="true" tabindex="-1"></a><span class="co">#| include: false</span></span>
<span id="cb106-1015"><a href="#cb106-1015" aria-hidden="true" tabindex="-1"></a><span class="co">#| comment: save R script</span></span>
<span id="cb106-1016"><a href="#cb106-1016" aria-hidden="true" tabindex="-1"></a>workflow_name <span class="ot">&lt;-</span> <span class="st">"med"</span></span>
<span id="cb106-1017"><a href="#cb106-1017" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">knitr.duplicate.label =</span> <span class="st">"allow"</span>)</span>
<span id="cb106-1018"><a href="#cb106-1018" aria-hidden="true" tabindex="-1"></a><span class="co"># Define sources and numbered outputs</span></span>
<span id="cb106-1019"><a href="#cb106-1019" aria-hidden="true" tabindex="-1"></a>sources <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb106-1020"><a href="#cb106-1020" aria-hidden="true" tabindex="-1"></a>  <span class="st">"include/_MED_Part1.qmd"</span>,</span>
<span id="cb106-1021"><a href="#cb106-1021" aria-hidden="true" tabindex="-1"></a>  <span class="st">"index.qmd"</span></span>
<span id="cb106-1022"><a href="#cb106-1022" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-1023"><a href="#cb106-1023" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate corresponding temporary output files</span></span>
<span id="cb106-1024"><a href="#cb106-1024" aria-hidden="true" tabindex="-1"></a>tmp_outputs <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"share/ssu"</span>, <span class="fu">paste0</span>(workflow_name, <span class="st">"_workflow"</span>, <span class="fu">seq_along</span>(sources), <span class="st">".R"</span>))</span>
<span id="cb106-1025"><a href="#cb106-1025" aria-hidden="true" tabindex="-1"></a><span class="co"># Final combined output</span></span>
<span id="cb106-1026"><a href="#cb106-1026" aria-hidden="true" tabindex="-1"></a>final_output <span class="ot">&lt;-</span> <span class="fu">here</span>(share_here, <span class="fu">paste0</span>(workflow_name, <span class="st">"_workflow.R"</span>))</span>
<span id="cb106-1027"><a href="#cb106-1027" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Extract code from qmd to temporary R scripts</span></span>
<span id="cb106-1028"><a href="#cb106-1028" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">walk2</span>(sources, tmp_outputs, <span class="sc">~</span></span>
<span id="cb106-1029"><a href="#cb106-1029" aria-hidden="true" tabindex="-1"></a>  knitr<span class="sc">::</span><span class="fu">purl</span>(.x, <span class="at">output =</span> .y, <span class="at">documentation =</span> <span class="dv">0</span>)</span>
<span id="cb106-1030"><a href="#cb106-1030" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb106-1031"><a href="#cb106-1031" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Concatenate all temporary R scripts into one, with separators</span></span>
<span id="cb106-1032"><a href="#cb106-1032" aria-hidden="true" tabindex="-1"></a><span class="fu">sink</span>(final_output)</span>
<span id="cb106-1033"><a href="#cb106-1033" aria-hidden="true" tabindex="-1"></a>purrr<span class="sc">::</span><span class="fu">walk2</span>(tmp_outputs, <span class="fu">seq_along</span>(tmp_outputs), <span class="sc">~</span>{</span>
<span id="cb106-1034"><a href="#cb106-1034" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">readLines</span>(.x), <span class="at">sep =</span> <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb106-1035"><a href="#cb106-1035" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">"</span><span class="sc">\n\n</span><span class="st">"</span>,</span>
<span id="cb106-1036"><a href="#cb106-1036" aria-hidden="true" tabindex="-1"></a>      <span class="fu">paste0</span>(<span class="st">"# -------------------- END OF SCRIPT "</span>, .y, <span class="st">" --------------------</span><span class="sc">\n\n</span><span class="st">"</span>),</span>
<span id="cb106-1037"><a href="#cb106-1037" aria-hidden="true" tabindex="-1"></a>      <span class="at">sep =</span> <span class="st">""</span>)</span>
<span id="cb106-1038"><a href="#cb106-1038" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb106-1039"><a href="#cb106-1039" aria-hidden="true" tabindex="-1"></a><span class="fu">sink</span>()</span>
<span id="cb106-1040"><a href="#cb106-1040" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 3: Clean up temporary files</span></span>
<span id="cb106-1041"><a href="#cb106-1041" aria-hidden="true" tabindex="-1"></a><span class="fu">file.remove</span>(tmp_outputs)</span>
<span id="cb106-1042"><a href="#cb106-1042" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-1043"><a href="#cb106-1043" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-1046"><a href="#cb106-1046" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb106-1047"><a href="#cb106-1047" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb106-1048"><a href="#cb106-1048" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb106-1049"><a href="#cb106-1049" aria-hidden="true" tabindex="-1"></a><span class="fu">objects</span>()</span>
<span id="cb106-1050"><a href="#cb106-1050" aria-hidden="true" tabindex="-1"></a>gdata<span class="sc">::</span><span class="fu">keep</span>(removed_samples, ds_metrics_tbl, pipe_summary, </span>
<span id="cb106-1051"><a href="#cb106-1051" aria-hidden="true" tabindex="-1"></a>            me_final, me_no_low, me_raw, curate_summary,</span>
<span id="cb106-1052"><a href="#cb106-1052" aria-hidden="true" tabindex="-1"></a>            samdf, tax1, tax2, tax3, st,</span>
<span id="cb106-1053"><a href="#cb106-1053" aria-hidden="true" tabindex="-1"></a>            <span class="at">sure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb106-1054"><a href="#cb106-1054" aria-hidden="true" tabindex="-1"></a><span class="fu">save.image</span>(<span class="fu">here</span>(<span class="st">"page_build"</span>, <span class="st">"med_part2.rdata"</span>))</span>
<span id="cb106-1055"><a href="#cb106-1055" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb106-1056"><a href="#cb106-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-1057"><a href="#cb106-1057" aria-hidden="true" tabindex="-1"></a><span class="fu">#### References {.appendix}</span></span>
<span id="cb106-1058"><a href="#cb106-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-1059"><a href="#cb106-1059" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb106-1060"><a href="#cb106-1060" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb106-1061"><a href="#cb106-1061" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb106-1062"><a href="#cb106-1062" aria-hidden="true" tabindex="-1"></a>{{&lt; include /include/_footer.qmd &gt;}}</span>
</code></pre></div><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>Site constructed using <a href="https://quarto.org/">Quarto</a>.</p>
</div>   
    <div class="nav-footer-center">
      <ul class="footer-items list-unstyled">
<li class="nav-item">
    <a class="nav-link" href="../../../site_info.html">
<p>Site Information</p>
</a>
  </li>  
</ul>
<div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/istmobiome/shrimp-draft/blob/main/workflows/ssu/med/index.qmd" target="_blank" class="toc-action"><i class="bi bi-github"></i>View source</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/edit/main/workflows/ssu/med/index.qmd" target="_blank" class="toc-action"><i class="bi empty"></i>Edit this page</a></li><li><a href="https://github.com/istmobiome/shrimp-draft/issues/new" target="_blank" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>© Copyright 2025 <a href="https://metacrobe.github.io/">metacrobe</a> &amp; <a href="https://istmobiome.rbind.io/">The Istmobiome Project</a>.</p>
</div>
  </div>
</footer>


</body></html>